# Phenome-Wide Association Study {#phewas}
<!-- ![](./img/_gwas/banner_man_standing_dna.png){width=70%} -->
```{r load-functions-packages, message = FALSE, warning = FALSE, echo = FALSE, eval = TRUE, include = FALSE}
source("scripts/functions.R")
source("scripts/colors.R")
source("scripts/packages.R")
COURSE_loc = "~/Desktop/practical" # getwd()
```

```{r include = FALSE}
if(!knitr:::is_html_output())
{
  options("width"=56)
  knitr::opts_chunk$set(tidy.opts=list(width.cutoff=56, indent = 2), tidy = TRUE)
  knitr::opts_chunk$set(fig.pos = 'H')
}
```

```{r getting_started, message = FALSE, warning = FALSE, echo = FALSE, eval = TRUE, include = FALSE}
# general setting to apply to all chunks - can be overrode per chunk.
opts_chunk$set(echo = TRUE, 
               warning = FALSE, 
               message = FALSE, 
               out.width = "85%", 
               fig.align = "center", 
               tidy = FALSE)
```
Given the large biobanks available nowadays that have also genotyped the participants and collected a vast-array of information on them, it is possible to perform a **Phe**nome-**W**ide **A**ssociation **S**tudy.

There are several options in (Table \@ref(tab:tab-phewas)).

```{r load-phewas, eval = TRUE, echo = FALSE}
library("openxlsx")
phewas <- openxlsx::read.xlsx("information.xlsx", sheet = "PheWAS")
```


```{r tab-phewas, tidy = FALSE, eval = TRUE, echo = FALSE}
library("knitr")
knitr::kable(
  phewas, 
  caption = 'PheWAS Resources.',
  # align = ,
  booktabs = FALSE
)
```


> *Assignment*
> 
> 1. Perform a PheWAS with a few of the resources and your favorite SNP from this tutorial. 
> 2. Compare the different websites. What do you notice?
> 3. How is a PheWAS informative?


## What's next?

You now know to what other traits, risk factors and diseases our trait of interest (coronary artery disease) is correlated. An obvious next question would be: is this a causal correlation? We will cover this aspect in the next Chapters with MRBase (\@ref(mr_mrbase)) and TwoSampleMR (\@ref(mr_twosamplemr)). To return to the previous Chapter on post-GWAS analyses click [here](\@ref(post-gwas)).

```{js, echo = FALSE}
title=document.getElementById('header');
title.innerHTML = '<img src="./img/headers/banner_man_standing_dna.png" alt="Phenome-Wide Association Study">' + title.innerHTML
```
