# Functional Mapping and Annotation of GWAS

## Assignment

* First, go to the FUMA website, get an account, and go through the tutorial.

* Second, you will need to use the `fwrite` function in `r` to write the concatenated results of the WTCCC1 study (remember: `wtccc1_assoc_compl`).  

<!-- fwrite(na.omit(wtccc1_assoc_compl), paste0(COURSE_loc,"/wtccc1/wtccc1_assoc_compl.txt"), sep = " ", showProgress = T, verbose = T, na = "NA", row.names = F, quote = F) -->

*Question: can you figure out the sample size of the WTCCC1 data you used? *

* Third, you will need to compress the resulting output, this will make uploading to FUMA go faster. You can use the `bash`-program `gzip` for that. If you want to know what the options are for that program, you can use `gzip -h`

* Fourth, upload via the form on the FUMA-website. Since you've done the tutorial you are familiar with its options. Select _everything_ in the tabs `Gene Mapping (positional mapping)`, `Gene Mapping (eQTL mapping)` but *not* `GTEx v6`, and `Gene Mapping (3D Chromatin Interaction mapping)` and leave the settings at `Gene types` and `MHC region` as-is, and at `MAGMA analysis` set the _MAGMA gene expression analysis_ to include all tissues, *but* `GTEx v6`. 
Don't forget to give your analysis a name.

*Questions*

1) How many lead SNPs did we find?
2) What do the results of MAGMA (the gene-based test) look like and how many genes pass the threshold of multiple testing correction?
3) How many loci were mapped?
4) How many genes were physically located and how many were mapped to these loci?
5) Do you think all loci are 'correct', _i.e._ do you 'believe' all the signals looking at the mapping results?
6) For what tissues are the signals enriched?
7) Are there any chromatin interactions discovered?


* Fifth, go back to 'My Jobs'. Select your job and perform `GENE2FUNC`.

*Questions*

1) What genes show the lowest expression across tissues?
2) And what genes the highest?
3) For what pathways are the signals enriched?
4) What molecular functions are mapped to the signals?

## Replication! 

You reached an important milestone: you recreated the work by the whole WTCCC1 in just an afternoon. Wow. Back then there wasn't much on analyses _after_ a GWAS, nowadays there are many post-GWAS  analyses methods. We will cover them in the next (Chapter \@ref(phewas)).
