# Prerequisites

## Linux, macOS, and Windows
Most programs made to execute genetic studies are developed for the Unix environment, _e.g._ Linux and macOS. Therefore, they may not work as intended on window's environment. Fortunately, Windows now allow users to install a linux subsystem within Windows 10 and you can find the detail [guide](https://docs.microsoft.com/en-us/windows/wsl/about){target="_blank"} here.  
However, we highly recommend to 1) either install a linux subsystem on your Windows computer (_e.g._ [a virtual machine with Ubuntu could work](https://blog.storagecraft.com/the-dead-simple-guide-to-installing-a-linux-virtual-machine-on-windows/){target="_blank"}), or 2) switch to macOS in combination with [homebrew](https://brew.sh){target="_blank"} as this will give you advantage of a powerful computer with a userfriendly interface and all the flexibility to use Unix-based programs for your work.

> For this practical we use a Windows laptop with Ubuntu on a VirtualMachine. Therefore every command is intended for Linux/macOS.


## Programs you need

You need few programs for this practical or for your (future) genetic epidemiology work for that matter.


-------------------------------------------------------------
Program     Link    Description
----------- ------- ----------------------------------------
PLINK       [https://www.cog-genomics.org/plink2/](https://www.cog-genomics.org/plink2/){target="_blank"} PLINK is a free, open-source genetic analysis toolset, designed to perform a range of basic data parsing and quality control, as well as basic and large-scale analyses in a computationally efficient manner.

R           [https://cran.r-project.org](https://cran.r-project.org/){target="_blank"} Progam to perform statistical analysis and visualisations.

RStudio     [https://www.rstudio.com](https://www.rstudio.com){target="_blank"} A userfriendly R-wrap-around for code editing, debugging, analyses, and visualisation

-------------------------------------------------------------

Table 1: Programs needed for genetic epidemiology.

All genetic analyses can be done in PLINK, even on your laptop, but with large datasets, e.g. UK-biobank size, it's better to switch to a [high-performance computing cluster](https://wiki.bioinformatics.umcutrecht.nl/bin/view/HPC/WebHome){target="_blank"} like we have at the Uithof.

Nowadays, a lot of people also use programs like [SNPTEST](snptest){target="_blank"}, [BOLT-LMM](https://data.broadinstitute.org/alkesgroup/BOLT-LMM/){target="_blank"}, and [GCTA](http://cnsgenomics.com/software/gcta/#Overview){target="_blank"} as alternatives to run GWAS and downstream analyses, _e.g._ heritability estimation, Fst-calculation, etc.

Mendelian randomisation can be done either with the [SMR](http://cnsgenomics.com/software/smr/#Overview){target="_blank"} or [GSMR](http://cnsgenomics.com/software/gsmr/){target="_blank"} function from GCTA or with R-packages, like `TwoSampleMR`.

## The Terminal

For all the above programs, except RStudio, you will need the `Terminal`. This comes with every major operating system. And you will (start to) make your own scripts. The benefit of using scripts is that each step is clearly layed out and easily replicable, thus allows for greater reproducibility, easier troubleshooting and can be run on high-performance computer clusters.


Download the data you need to your Desktop: LINK. 

Now open the terminal, it should be on the left in the toolbar as a little black computer-monitor-like icon. 
Mac uses can type `command + space` and type `terminal`, a terminal screen should open.

> From now on we will use little code blocks like the example to indicate a code you should type/copy-paste and hit enter. If a code is followed by a comment, it is indicated by a # - you don't need to copy-paste and execute this.

```
CODE BLOCK

CODE BLOCK # some comment here


```

You can navigate around the computer through the terminal by typing `cd <path>`; `cd` stands for "change directory" and <path> means "some_file_directory_you_want_to_go_to".

```
# For Linux/macOS Users
cd ~ # will bring you to your home directory
cd ../ # will bring you to the parent directory (up one level) 
cd XXX # will bring you to the XXX directory
```

Let's navigate to the folder you just downloaded.
```
cd ~/Desktop/practical
```

Let's check out what is inside the directory, by listing (`ls`) its contents.

```
ls -lh

# For Linux/macOS Users
ls -l # shows files as list
ls -lh # shows files as list with human readable format 
ls -lt # shows the files as list sorted by time edited
ls -lS # shows the files as list sorted by size
```
Adding the flags `-lh` will get you the contents of a directory in a list (`-l`) and make the size 'human-readable' (`-h`).

You can also count the number of files.

```
ls | wc -l
```

## Installing some R packages

I tested this VirtualMachine and everything should be fine, except some libraries weren't there. We need to install them.


To be able to install certain `r`-packages, we need to install some Linux (Ubuntu) software. Type the following:

```
sudo apt-get install libcurl4 libcurl4-openssl-dev -y

sudo apt-get install libssl-dev
```

Now close the terminal window - really making sure that the terminal-program has quit.

Open a new terminal window and open `r` by simply typing `R`. You should install the following packages, and then you're good to go!

```
install.packages(c("httr", "usethis", "data.table", "devtools", "qqman", "CMplot", "tibble", "plotly", "dplyr"))

library(c("httr", "usethis", "data.table", "devtools", "qqman", "CMplot", "tibble", "plotly", "dplyr"))
devtools::install_github("kassambara/ggpubr")
library("ggpubr")

```

All in all this may take some time, good moment to relax, review your notes, stretch your legs, or take a coffee.

