# Prerequisites
![](img/girls_behind_macbook.jpeg)
```{r prerequisites, message = FALSE, warning = FALSE, echo = FALSE, eval = TRUE}
COURSE_loc = "~/Desktop/practical"
# COURSE_loc = "/Users/swvanderlaan/git/swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics"

source(paste0(COURSE_loc, "/scripts/functions.R"))
source(paste0(COURSE_loc, "/scripts/colors.R"))
source(paste0(COURSE_loc, "/scripts/packages.R"))

```

## Linux, macOS, and Windows

Most programs made to execute genetic epidemiology studies are developed for the Unix environment, for example Linux and macOS. So, they may not work as intended in a Windows environment. Windows does allow users to install a linux subsystem within Windows 10 and you can find the detail [guide](https://docs.microsoft.com/en-us/windows/wsl/about){target="_blank"} here.  

However, I highly recommend to 1) either install a linux subsystem on your Windows computer (for example [a virtual machine with Ubuntu could work](https://blog.storagecraft.com/the-dead-simple-guide-to-installing-a-linux-virtual-machine-on-windows/){target="_blank"}), or 2) switch to macOS in combination with [homebrew](https://brew.sh){target="_blank"}. This will give you all the flexibility to use Unix-based programs for your genetic epidemiology work and at the same time you'll keep the advantage of a powerful computer with a user-friendly interface (either Windows or macOS).

> For this practical we use a Windows laptop with Ubuntu on a VirtualMachine. Therefore every command is intended for Linux/macOS, in other words Unix-systems.


## Programs you need

You need few programs for this practical, or for your (future) genetic epidemiology work for that matter (Table \@ref(tab:programs)).

```{r, eval = TRUE, echo = FALSE}
programs <- openxlsx::read.xlsx("information.xlsx", sheet = "Programs")
```


```{r programs, tidy = FALSE, eval = TRUE, echo = FALSE}
knitr::kable(
  programs, 
  caption = 'Programs needed for genetic epidemiology.',
  # align = ,
  booktabs = FALSE
)
```

<!-- Made this table with https://www.tablesgenerator.com/markdown_tables -->
<!-- | **Program** 	| **Link**                             	| **Description**                                               | -->
<!-- |-------------	|--------------------------------------	|-------------------------------------------------------------	| -->
<!-- | _PLINK_     	| https://www.cog-genomics.org/plink2/ 	| PLINK is a free, open-source genetic analysis tool set, designed to perform a range of basic data parsing and quality control, as well as basic and large-scale analyses in a computationally efficient manner. | -->
<!-- | _R_         	| https://cran.r-project.org/          	| A program to perform statistical analysis and visualizations. | -->
<!-- | _RStudio_   	| https://www.rstudio.com              	| A user-friendly R-wrap-around for code editing, debugging, analyses, and visualization. | -->
<!-- | _Homebrew_  	| https://brew.sh                      	| A great extension for Mac-users to install really useful programs that Apple didn't. | -->

<!-- **Table 2.1:** Programs needed for genetic epidemiology. -->

All genetic analyses can be done in PLINK, even on your laptop, but with large datasets, for example [UK Biobank](https://www.ukbiobank.ac.uk) size, it is better to switch to a [high-performance computing cluster](https://en.wikipedia.org/wiki/High-performance_computing){target="_blank"} like we have available at the [Utrecht Science Park](https://wiki.bioinformatics.umcutrecht.nl/bin/view/HPC/WebHome){target="_blank"}.

Nowadays, a lot of people also use programs like [SNPTEST](snptest){target="_blank"}, [BOLT-LMM](https://data.broadinstitute.org/alkesgroup/BOLT-LMM/){target="_blank"},  [GCTA](http://cnsgenomics.com/software/gcta/#Overview){target="_blank"}, or [regenie](https://rgcgithub.github.io/regenie/){target="_blank"} as alternatives to execute GWAS and downstream analyses, for example heritability estimation, Fst-calculation, and so on.

Mendelian randomization can be done either with the [SMR](http://cnsgenomics.com/software/smr/#Overview){target="_blank"} or [GSMR](http://cnsgenomics.com/software/gsmr/){target="_blank"} function from GCTA, or with R-packages, like [`TwoSampleMR`](https://mrcieu.github.io/TwoSampleMR/){target="_blank"}.


## The Terminal

For all the above programs, except RStudio, you will need the `Terminal`. This comes with every major operating system; on Windows it is called 'PowerShell', but let's not go there. And regardless, you will (have to start to) make your own scripts. The benefit of using scripts is that each step in your workflow is clearly stipulated and annotated, and it allows for greater reproducibility, easier troubleshooting, and scaling up to high-performance computer clusters.

Open the terminal, it should be on the left in the toolbar as a little black computer-monitor-like icon. 
Mac users can type `command + space` and type `terminal`, a terminal screen should open.

> From now on we will use little code blocks like the example to indicate a code you should type/copy-paste and hit enter. If a code is followed by a comment, it is indicated by a # - you don't need to copy-paste and execute this.


```
CODE BLOCK

CODE BLOCK # some comment here

```

### Download the data

First, let's start by downloading the data you need for this course to your Desktop: [LINK](). 


Alternatively, you could do this through this command. This will create a directory on your Desktop with the command `mkdir`. The `-v` flag indicates the program should be _verbose_, meaning it should tell you what it is doing.

```
mkdir -v ~/Desktop/practical/
```

```
wget "https://www.dropbox.com/sh/kumfwm7drt2flhp/AAB5n0OcUvJixI9pNiymx6-La?dl=0" -P ~/Desktop/practical/

```

### Navigating the Terminal

You can navigate around the computer through the terminal by typing `cd <path>`; `cd` stands for "change directory" and <path> means "some_file_directory_you_want_to_go_to".

```
# For Linux/macOS Users
cd ~ # will bring you to your home directory
cd ../ # will bring you to the parent directory (up one level) 
cd XXX # will bring you to the XXX directory
```


Let's navigate to the folder you just downloaded.

```
cd ~/Desktop/practical
```


Let's check out what is inside the directory, by listing (`ls`) its contents.

```
ls -lh

# For Linux/macOS Users
ls -l # shows files as list
ls -lh # shows files as list with human readable format 
ls -lt # shows the files as list sorted by time edited
ls -lS # shows the files as list sorted by size
```

Adding the flags `-lh` will get you the contents of a directory in a list (`-l`) and make the size 'human-readable' (`-h`).


You can also count the number of files.

```
ls | wc -l
```


## Installing some R packages

I tested this VirtualMachine and everything should be fine, except some libraries weren't there. We need to install them.


To be able to install certain `r`-packages, we need to install some Linux (Ubuntu) software. Type the following:

```
sudo apt-get install libcurl4 libcurl4-openssl-dev -y

sudo apt-get install libssl-dev
```

Now close the terminal window - really making sure that the terminal-program has quit.

Open a new terminal window and open `r` by simply typing `R` or `r`. You should install the following packages, and then you're good to go!

```
install.packages(c("httr", "usethis", "data.table", "devtools", "qqman", "CMplot", "tibble", "plotly", "dplyr"))
devtools::install_github("kassambara/ggpubr")
```

You should load these packages too. 

```
library("ggpubr")
library("httr")
library("usethis")
library("data.table")
library("devtools")
library("qqman")
library("CMplot")
library("tibble")
library("plotly")
library("dplyr")
```

All in all this may take some time, good moment to relax, review your notes, stretch your legs, or take a coffee.


## EIGENSOFT

Unfortunately, the preferred software **[`EIGENSOFT`](https://github.com/DReichLab/EIG){targe="_blank"}** for Principal Component Analysis (PCA) is challenging to make it work to say the least. You need to install some programs for **`EIGENSOFT`** to work, and this is not always straightforward. 

So, here's the deal. 

I will share the how-to for a macOS environment below in [EIGENSOFT] - this should work in a Linux environment too as macOS is UNIX-based. You can choose to try and make it work in the VirtualMachine you're working on now, or on your personal laptop (be it UNIX or macOS based) at home (or in the office). 

However, for you convenience I ran the PCA for mapping of this dummy dataset against two references already and shared the files in the `ref_pca`-folder. This means you can skip this section and jump straight to [Are you ready?].

### Install homebrew

You need the missing package-manager and accompanying packages that Apple didn't provide. 

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

Next, check that everything is in order.

```
brew doctor
```

### Install missing packages

Right, now that you've done that, you're ready to install `gsl` and `openblas`. 

```
brew install gsl
brew install openblas
```

You may require `llvm` if `EIGENSOFT` isn't working.

```
brew install llvm
```


### Installing EIGENSOFT

I am still sharing the code you'll need - you could try this on your personal MacBook for instance. 

```
mkdir -v $HOME/git
cd $HOME/git
git clone https://github.com/DReichLab/EIG.git
cd EIG/src
make
make install
```


## Are you ready?

Are you ready? Did you bring coffee and a good dose of energy? Let's start! We first cover some basics in Chapter \@ref(gwas-basics).
