# Phenome-Wide Association Study {#phewas}
<!-- ![](./img/_gwas/banner_man_standing_dna.png){width=70%} -->





Given the large biobanks available nowadays that have also genotyped the participants and collected a vast-array of information on them, it is possible to perform a **Phe**nome-**W**ide **A**ssociation **S**tudy.

There are several options in (Table \@ref(tab:tab-phewas)).





Table: (\#tab:tab-phewas)PheWAS Resources.

|Site                                   |Description                              |
|:--------------------------------------|:----------------------------------------|
|https://atlas.ctglab.nl                |4,155 GWAS based on UK Biobank release 2 |
|https://biobankengine.stanford.edu     |UK Biobank based                         |
|http://pheweb.sph.umich.edu            |University of Michigan EHR-based PheWAS  |
|http://pheweb.sph.umich.edu/SAIGE-UKB/ |1,403 GWAS UK Biobank based              |
|http://phewas.mrbase.org               |UK Biobank based                         |


> *Assignment*
> 
> 1. Perform a PheWAS with a few of the resources and your favorite SNP from this tutorial. 
> 2. Compare the different websites. What do you notice?
> 3. How is a PheWAS informative?


## What's next?

You now know to what other traits, risk factors and diseases our trait of interest (coronary artery disease) is correlated. An obvious next question would be: is this a causal correlation? We will cover this aspect in the next Chapters with MRBase (\@ref(mr_mrbase)) and TwoSampleMR (\@ref(mr_twosamplemr)). To return to the previous Chapter on post-GWAS analyses click [here](\@ref(post-gwas)).

<script>
title=document.getElementById('header');
title.innerHTML = '<img src="./img/headers/banner_man_standing_dna.png" alt="Phenome-Wide Association Study">' + title.innerHTML
</script>
