<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 15 Additional: regional association plots | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 15 Additional: regional association plots | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 15 Additional: regional association plots | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="epilogue.html"/>
<link rel="next" href="eigensoft.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="prerequisites.html"><a href="prerequisites.html#rstudio"><i class="fa fa-check"></i><b>2.2.1</b> RStudio<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="prerequisites.html"><a href="prerequisites.html#plink"><i class="fa fa-check"></i><b>2.2.2</b> PLINK<span></span></a></li>
<li class="chapter" data-level="2.2.3" data-path="prerequisites.html"><a href="prerequisites.html#alternatives-to-plink"><i class="fa fa-check"></i><b>2.2.3</b> Alternatives to PLINK<span></span></a></li>
<li class="chapter" data-level="2.2.4" data-path="prerequisites.html"><a href="prerequisites.html#other-programs"><i class="fa fa-check"></i><b>2.2.4</b> Other programs<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.5</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>3</b> Steps in a Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Let’s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>4.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.2</b> 1000G phase 1<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#running-pca"><i class="fa fa-check"></i><b>4.5.3</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="4.5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#pca-plotting"><i class="fa fa-check"></i><b>4.5.4</b> PCA plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>5.3.1</b> A note on allele coding<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-testing.html"><a href="gwas-testing.html"><i class="fa fa-check"></i><b>6</b> Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-testing.html"><a href="gwas-testing.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="gwas-testing.html"><a href="gwas-testing.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="gwas-testing.html"><a href="gwas-testing.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="gwas-testing.html"><a href="gwas-testing.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Let’s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>7.4</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>8.1</b> Genotyping<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>8.2</b> The data<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>8.3</b> Assignment<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>8.4</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>9</b> WTCCC1: a GWAS on coronary artery disease (CAD)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>9.1</b> Quality control<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>9.2</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>9.2.1</b> Preparing PCA<span></span></a></li>
<li class="chapter" data-level="9.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca-1"><i class="fa fa-check"></i><b>9.2.2</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="9.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>9.2.3</b> Plotting PCA<span></span></a></li>
<li class="chapter" data-level="9.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>9.2.4</b> Removing samples<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>9.3</b> Association testing<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>10</b> Post-GWAS Analyses<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>10.1</b> Clumping<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>10.2</b> Conditional analyses<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>10.3</b> Statistical finemapping<span></span></a></li>
<li class="chapter" data-level="10.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>10.4</b> FUMA: FUnctional Mapping and Annotation of GWAS<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="post-gwas.html"><a href="post-gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>10.5</b> Phenome-wide association study<span></span></a></li>
<li class="chapter" data-level="10.6" data-path="post-gwas.html"><a href="post-gwas.html#colocalization"><i class="fa fa-check"></i><b>10.6</b> Colocalization<span></span></a></li>
<li class="chapter" data-level="10.7" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>10.7</b> Genetic correlation<span></span></a></li>
<li class="chapter" data-level="10.8" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>10.8</b> Causal inference: Mendelian Randomization<span></span></a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="post-gwas.html"><a href="post-gwas.html#your-choice"><i class="fa fa-check"></i><b>10.8.1</b> Your choice<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>10.9</b> Polygenic scores<span></span></a></li>
<li class="chapter" data-level="10.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>10.10</b> What comes next<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>11</b> Functional Mapping and Annotation of GWAS<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>11.1</b> Assignment<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>11.1.1</b> Tutorial<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>11.1.2</b> Create the input<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>11.1.3</b> Run FUMA - SNP to gene<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>11.1.4</b> Run FUMA - Gene to function<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>11.2</b> Some closing thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>12</b> Phenome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>12.1</b> What’s next?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>13</b> MRBase<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>13.1</b> The end?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>14</b> Epilogue<span></span></a></li>
<li class="chapter" data-level="15" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html"><i class="fa fa-check"></i><b>15</b> Additional: regional association plots<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html#locuszoom"><i class="fa fa-check"></i><b>15.1</b> LocusZoom<span></span></a></li>
<li class="chapter" data-level="15.2" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html#racer"><i class="fa fa-check"></i><b>15.2</b> RACER<span></span></a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>16</b> Additional: EIGENSOFT<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>16.1</b> Install homebrew<span></span></a></li>
<li class="chapter" data-level="16.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>16.2</b> Install missing packages<span></span></a></li>
<li class="chapter" data-level="16.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>16.3</b> Installing EIGENSOFT<span></span></a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>17</b> Licenses and disclaimers<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>17.1</b> Copyright<span></span></a></li>
<li class="chapter" data-level="17.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>17.2</b> Disclaimer<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="add_chapter_regional_plot" class="section level1 hasAnchor" number="15">
<h1><span class="header-section-number">Chapter 15</span> Additional: regional association plots<a href="add_chapter_regional_plot.html#add_chapter_regional_plot" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>First we need to find the independent hits by <em>clumping</em> the results. We will just use the defaults, but please take a note of all the options here <a href="https://www.cog-genomics.org/plink/1.9/postproc#clump" target="_blank">https://www.cog-genomics.org/plink/1.9/postproc#clump</a></p>
<pre><code>plink --bfile gwas/gwa --clump gwas/data.assoc.logistic --clump-p1 5e-8 --clump-p2 0.05 --clump-kb 500 --clump-r2 0.05 --clump-verbose --out gwas/data.assoc.logistic</code></pre>
<p>Now you will have a list of all the <em>independent</em> SNPs, <em>i.e.</em> the genetic loci, that are associated to the trait.</p>
<pre><code>cat gwas/data.assoc.logistic.clumped</code></pre>
<div class="sourceCode" id="cb138"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb138-1"><a href="add_chapter_regional_plot.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="va">ROOTDIR</span><span class="op">=</span><span class="st">&quot;/Users/swvanderlaan/Desktop/practical&quot;</span> <span class="co"># change this to your root</span></span>
<span id="cb138-2"><a href="add_chapter_regional_plot.html#cb138-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">$ROOTDIR</span>/gwas/data.assoc.logistic.clumped</span></code></pre></div>
<pre><code>## 
##  CHR    F         SNP         BP          P    TOTAL   NSIG    S05    S01   S001  S0001
##    3    1   rs6802898   12366207   4.18e-20       50     35      4      2      1      8 
## 
##                               KB      RSQ  ALLELES    F            P 
##   (INDEX)   rs6802898          0    1.000        T    1     4.18e-20 
## 
##              rs305500       -400   0.0588    TC/CA    1       0.0476 
##              rs420014       -394   0.0552    TA/CG    1        0.015 
##              rs305494       -392   0.0681    TG/CA    1        0.025 
##              rs438129       -383   0.0721    TT/CC    1       0.0126 
##             rs7615580       -364    0.309    TC/CT    1     2.05e-08 
##              rs307560       -298    0.153    TT/CC    1     1.68e-06 
##            rs11720130       -235   0.0838    TA/CG    1      0.00218 
##             rs7616006       -124   0.0831    TA/CG    1     5.25e-05 
##             rs6775191       -119   0.0506    TG/CA    1     0.000182 
##              rs167466       -110   0.0523    TT/CC    1      0.00222 
##            rs12635120      -86.6    0.332    TG/CA    1     2.77e-07 
##             rs6798713      -85.6    0.288    TC/CT    1     2.01e-06 
##             rs2920500      -67.8    0.102    TA/CG    1     6.59e-05 
##             rs6768587      -53.1    0.295    TG/CA    1     3.36e-08 
##             rs2028760      -18.3    0.305    TA/CG    1     3.67e-08 
## 
##           RANGE: chr3:11966007..12366207
##            SPAN: 400kb
## 
## ------------------------------------------------------------------
## 
## 
##  CHR    F         SNP         BP          P    TOTAL   NSIG    S05    S01   S001  S0001
##   10    1   rs7901695  114744078   6.78e-12       32     24      2      1      1      4 
## 
##                               KB      RSQ  ALLELES    F            P 
##   (INDEX)   rs7901695          0    1.000        C    1     6.78e-12 
## 
##             rs7917983      -21.2   0.0589    CC/TT    1        0.046 
##             rs7895307      -10.1   0.0819    CG/TA    1      0.00592 
##             rs7903146       4.26    0.784    CT/TC    1     3.25e-08 
##             rs7904519       19.8    0.582    CG/TA    1     2.89e-08 
##            rs11196192       28.2    0.162    CG/TT    1       0.0268 
##            rs10885409         54    0.502    CC/TT    1     8.35e-06 
##            rs12255372       54.8    0.624    CT/TG    1     1.55e-06 
##             rs4918789       67.7     0.24    CG/TT    1     0.000248 
## 
##           RANGE: chr10:114722872..114811797
##            SPAN: 88kb
## 
## ------------------------------------------------------------------
## 
## 
##  CHR    F         SNP         BP          P    TOTAL   NSIG    S05    S01   S001  S0001
##   16    1   rs8050136   52373776   1.52e-08       23     16      1      3      2      1 
## 
##                               KB      RSQ  ALLELES    F            P 
##   (INDEX)   rs8050136          0    1.000        A    1     1.52e-08 
## 
##             rs7205986      -61.1    0.226    AG/CA    1       0.0434 
##             rs6499640      -46.6    0.258    AA/CG    1      0.00367 
##             rs1861868      -25.9     0.15    AT/CC    1       0.0063 
##             rs1075440      -25.4    0.162    AA/CG    1      0.00802 
##             rs3751812       2.18    0.994    AT/CG    1     1.63e-08 
##             rs7190492       12.5    0.258    AG/CA    1       0.0007 
##             rs8044769       22.9    0.524    AC/CT    1     0.000611 
## 
##           RANGE: chr16:52312647..52396636
##            SPAN: 83kb
## 
## ------------------------------------------------------------------</code></pre>
<p>In summary, clumping identifies three loci and now that you know them.</p>
<div id="locuszoom" class="section level2 hasAnchor" number="15.1">
<h2><span class="header-section-number">15.1</span> LocusZoom<a href="add_chapter_regional_plot.html#locuszoom" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now you are ready to visualize regions of interest using a package like <a href="http://locuszoom.org">LocusZoom</a>. Locuszoom is a great and easy tool, you’ll soon discover. It’s the fast and lazy way to get regional association plots.</p>
<p>First, let’s get what we need (<code>SNP</code> and <code>P</code>) and gzip the results.</p>
<pre><code>echo &quot;SNP P&quot; &gt; gwas/data.assoc.logistic.locuszoom
cat gwas/data.assoc.logistic | awk &#39;$5==&quot;ADD&quot;&#39; | awk &#39;{ print $2, $9 }&#39; &gt;&gt; gwas/data.assoc.logistic.locuszoom
gzip -v gwas/data.assoc.logistic.locuszoom</code></pre>
<p>You can to upload this <code>data.assoc.logistic.locuszoom.gz</code> Try to visualize each locus using the information above and by following the instructions. Choose HapMap 2, hg18, CEU as the LD-reference.</p>
<p>You should get something like below.</p>
<p><img src="img/_gwas/CEU.rs6802898.400kb.png" style="width:50.0%" /></p>
<p><img src="img/_gwas/CEU.rs7901695.400kb.png" style="width:50.0%" /></p>
<p><img src="img/_gwas/CEU.rs8050136.400kb.png" style="width:50.0%" /></p>
</div>
<div id="racer" class="section level2 hasAnchor" number="15.2">
<h2><span class="header-section-number">15.2</span> RACER<a href="add_chapter_regional_plot.html#racer" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>An alternative to create regional association plots for each of these loci is using <a href="https://github.com/oliviasabik/RACER" target="_blank"><strong>RACER</strong></a>. This packages offers substantial flexibility, and quite frankly, code-wise is easier to edit - think colorscheme, background etc. - as it is based on <a href="https://ggplot2.tidyverse.org" target="_blank"><code>ggplot2</code></a>. The creator, Olivia Sabik, also wrote a nice <a href="https://oliviasabik.github.io/RACERweb/index.html" target="_blank">vignette</a> with excellent instructions.</p>
<p>Using <strong>RACER</strong> is also a step-up to <strong>colocalization</strong> about which you read a bit more in chapter <a href="post-gwas.html#post-gwas">10</a>.</p>
<p>Right, so from the above we identified three independent hits: rs6802898, rs7901695, rs8050136. Let’s put them in a dataset with their chromosome and basepair position and create a list.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="add_chapter_regional_plot.html#cb141-1" aria-hidden="true" tabindex="-1"></a>rsID <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;rs6802898&quot;</span>, <span class="st">&quot;rs7901695&quot;</span>, <span class="st">&quot;rs8050136&quot;</span>)</span>
<span id="cb141-2"><a href="add_chapter_regional_plot.html#cb141-2" aria-hidden="true" tabindex="-1"></a>CHR <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">10</span>, <span class="dv">16</span>))</span>
<span id="cb141-3"><a href="add_chapter_regional_plot.html#cb141-3" aria-hidden="true" tabindex="-1"></a>BP <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">c</span>(<span class="dv">12366207</span>, <span class="dv">114744078</span>, <span class="dv">52373776</span>))</span>
<span id="cb141-4"><a href="add_chapter_regional_plot.html#cb141-4" aria-hidden="true" tabindex="-1"></a><span class="co"># BPend &lt;- c(12366207 + 500000, 114744078 + 500000, 52373776 + 500000)</span></span>
<span id="cb141-5"><a href="add_chapter_regional_plot.html#cb141-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-6"><a href="add_chapter_regional_plot.html#cb141-6" aria-hidden="true" tabindex="-1"></a>variant_list <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(rsID, CHR, BP)</span></code></pre></div>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="add_chapter_regional_plot.html#cb142-1" aria-hidden="true" tabindex="-1"></a>variant_list</span></code></pre></div>
<pre><code>##        rsID CHR        BP
## 1 rs6802898   3  12366207
## 2 rs7901695  10 114744078
## 3 rs8050136  16  52373776</code></pre>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="add_chapter_regional_plot.html#cb144-1" aria-hidden="true" tabindex="-1"></a>variants_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(variant_list<span class="sc">$</span>rsID)</span>
<span id="cb144-2"><a href="add_chapter_regional_plot.html#cb144-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-3"><a href="add_chapter_regional_plot.html#cb144-3" aria-hidden="true" tabindex="-1"></a>variants_of_interest</span></code></pre></div>
<pre><code>## [1] &quot;rs6802898&quot; &quot;rs7901695&quot; &quot;rs8050136&quot;</code></pre>
<p>We can now take the filtered and prepared GWAS summary statistics, <em>i.e.</em> <code>gwas_assoc_compl</code> you created earlier in chapter <a href="gwas-visuals.html#gwas-visuals">7</a>, to draw three nice regional association plots around these variants.</p>
<p>This file should contain the following:</p>
<pre><code>tibble [306,102 × 14] (S3: tbl_df/tbl/data.frame)
 $ SNP     : chr [1:306102] &quot;rs10000010&quot; &quot;rs10000023&quot; &quot;rs10000030&quot; &quot;rs1000007&quot; ...
 $ CHR     : int [1:306102] 4 4 4 2 4 4 16 4 2 4 ...
 $ BP      : int [1:306102] 21227772 95952929 103593179 237416793 21504615 157793485 24325037 33810744 235355721 77575270 ...
 $ A1      : chr [1:306102] &quot;C&quot; &quot;T&quot; &quot;A&quot; &quot;C&quot; ...
 $ A2      : chr [1:306102] &quot;T&quot; &quot;G&quot; &quot;G&quot; &quot;T&quot; ...
 $ MAF     : num [1:306102] 0.426 0.484 0.162 0.312 0.343 ...
 $ callrate: num [1:306102] 0.999 0.989 0.998 1 0.991 ...
 $ NMISS   : int [1:306102] 3996 3957 3991 4000 3963 3919 3899 3931 3927 3941 ...
 $ NCHROBS : int [1:306102] 7992 7914 7982 8000 7926 7838 7798 7862 7854 7882 ...
 $ BETA    : num [1:306102] 0.04114 -0.00985 -0.02235 0.01784 -0.07904 ...
 $ SE      : num [1:306102] 0.0457 0.0456 0.0605 0.0489 0.0471 ...
 $ OR      : num [1:306102] 1.042 0.99 0.978 1.018 0.924 ...
 $ STAT    : num [1:306102] 0.901 -0.216 -0.37 0.365 -1.677 ...
 $ P       : num [1:306102] 0.3676 0.829 0.7117 0.7152 0.0935 ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; 
 - attr(*, &quot;sorted&quot;)= chr &quot;SNP&quot;</code></pre>
<p>We could limit ourselves by limiting the region to plot by the clump-size (see above), but generally it’s fine to just ‘take the top variant ± 500kb’.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="add_chapter_regional_plot.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RACER)</span>
<span id="cb147-2"><a href="add_chapter_regional_plot.html#cb147-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-3"><a href="add_chapter_regional_plot.html#cb147-3" aria-hidden="true" tabindex="-1"></a>RANGE<span class="ot">=</span><span class="dv">500000</span></span>
<span id="cb147-4"><a href="add_chapter_regional_plot.html#cb147-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-5"><a href="add_chapter_regional_plot.html#cb147-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(VARIANT <span class="cf">in</span> variants_of_interest){</span>
<span id="cb147-6"><a href="add_chapter_regional_plot.html#cb147-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;Getting data for &quot;</span>, VARIANT,<span class="st">&quot;.</span><span class="sc">\n</span><span class="st">&quot;</span>))</span>
<span id="cb147-7"><a href="add_chapter_regional_plot.html#cb147-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-8"><a href="add_chapter_regional_plot.html#cb147-8" aria-hidden="true" tabindex="-1"></a>  tempCHR <span class="ot">&lt;-</span> <span class="fu">subset</span>(variant_list, rsID <span class="sc">==</span> VARIANT)[,<span class="dv">2</span>]</span>
<span id="cb147-9"><a href="add_chapter_regional_plot.html#cb147-9" aria-hidden="true" tabindex="-1"></a>  tempSTART <span class="ot">&lt;-</span> <span class="fu">subset</span>(variant_list, rsID <span class="sc">==</span> VARIANT)[,<span class="dv">3</span>] <span class="sc">-</span> RANGE</span>
<span id="cb147-10"><a href="add_chapter_regional_plot.html#cb147-10" aria-hidden="true" tabindex="-1"></a>  tempEND <span class="ot">&lt;-</span> <span class="fu">subset</span>(variant_list, rsID <span class="sc">==</span> VARIANT)[,<span class="dv">3</span>] <span class="sc">+</span> RANGE</span>
<span id="cb147-11"><a href="add_chapter_regional_plot.html#cb147-11" aria-hidden="true" tabindex="-1"></a>  tempVARIANTnr <span class="ot">&lt;-</span> <span class="fu">subset</span>(variant_list, rsID <span class="sc">==</span> VARIANT)[,<span class="dv">1</span>]</span>
<span id="cb147-12"><a href="add_chapter_regional_plot.html#cb147-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-13"><a href="add_chapter_regional_plot.html#cb147-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Subset required data.</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb147-14"><a href="add_chapter_regional_plot.html#cb147-14" aria-hidden="true" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_assoc_compl, </span>
<span id="cb147-15"><a href="add_chapter_regional_plot.html#cb147-15" aria-hidden="true" tabindex="-1"></a>                 CHR <span class="sc">==</span> tempCHR <span class="sc">&amp;</span> (BP <span class="sc">&gt;=</span> tempSTART <span class="sc">&amp;</span> BP <span class="sc">&lt;=</span> tempEND))</span>
<span id="cb147-16"><a href="add_chapter_regional_plot.html#cb147-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-17"><a href="add_chapter_regional_plot.html#cb147-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Formatting association data.</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb147-18"><a href="add_chapter_regional_plot.html#cb147-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make sure you have the right column numbers here!</span></span>
<span id="cb147-19"><a href="add_chapter_regional_plot.html#cb147-19" aria-hidden="true" tabindex="-1"></a>  temp_f <span class="ot">=</span> RACER<span class="sc">::</span><span class="fu">formatRACER</span>(<span class="at">assoc_data =</span> temp, <span class="at">chr_col =</span> <span class="dv">2</span>, <span class="at">pos_col =</span> <span class="dv">3</span>, <span class="at">p_col =</span> <span class="dv">14</span>)</span>
<span id="cb147-20"><a href="add_chapter_regional_plot.html#cb147-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-21"><a href="add_chapter_regional_plot.html#cb147-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Getting LD data.</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb147-22"><a href="add_chapter_regional_plot.html#cb147-22" aria-hidden="true" tabindex="-1"></a>  temp_f_ld <span class="ot">=</span> RACER<span class="sc">::</span><span class="fu">ldRACER</span>(<span class="at">assoc_data =</span> temp_f, <span class="at">rs_col =</span> <span class="dv">1</span>, <span class="at">pops =</span> <span class="st">&quot;EUR&quot;</span>, <span class="at">lead_snp =</span> VARIANT)</span>
<span id="cb147-23"><a href="add_chapter_regional_plot.html#cb147-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-24"><a href="add_chapter_regional_plot.html#cb147-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Plotting region surrounding &quot;</span>, VARIANT,<span class="st">&quot; on &quot;</span>,tempCHR,<span class="st">&quot;:&quot;</span>,tempSTART,<span class="st">&quot;-&quot;</span>,tempEND,<span class="st">&quot;.</span><span class="sc">\n</span><span class="st">&quot;</span>))</span>
<span id="cb147-25"><a href="add_chapter_regional_plot.html#cb147-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># source(paste0(PROJECT_loc, &quot;/scripts/functions.R&quot;))</span></span>
<span id="cb147-26"><a href="add_chapter_regional_plot.html#cb147-26" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">singlePlotRACER</span>(<span class="at">assoc_data =</span> temp_f_ld, </span>
<span id="cb147-27"><a href="add_chapter_regional_plot.html#cb147-27" aria-hidden="true" tabindex="-1"></a>                               <span class="at">chr =</span> tempCHR, <span class="at">build =</span> <span class="st">&quot;hg19&quot;</span>, </span>
<span id="cb147-28"><a href="add_chapter_regional_plot.html#cb147-28" aria-hidden="true" tabindex="-1"></a>                               <span class="at">plotby =</span> <span class="st">&quot;snp&quot;</span>, <span class="at">snp_plot =</span> VARIANT,</span>
<span id="cb147-29"><a href="add_chapter_regional_plot.html#cb147-29" aria-hidden="true" tabindex="-1"></a>                               <span class="at">label_lead =</span> <span class="cn">TRUE</span>)</span>
<span id="cb147-30"><a href="add_chapter_regional_plot.html#cb147-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-31"><a href="add_chapter_regional_plot.html#cb147-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p1)</span>
<span id="cb147-32"><a href="add_chapter_regional_plot.html#cb147-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;Saving image for &quot;</span>, VARIANT,<span class="st">&quot;.</span><span class="sc">\n</span><span class="st">&quot;</span>))</span>
<span id="cb147-33"><a href="add_chapter_regional_plot.html#cb147-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/gwas/&quot;</span>, tempVARIANTnr, <span class="st">&quot;.&quot;</span>,VARIANT,<span class="st">&quot;.regional_assoc.png&quot;</span>), <span class="at">plot =</span> p1)</span>
<span id="cb147-34"><a href="add_chapter_regional_plot.html#cb147-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ggsave(filename = paste0(COURSE_loc, &quot;/gwas/&quot;, tempVARIANTnr, &quot;.&quot;,VARIANT,&quot;.regional_assoc.pdf&quot;), plot = p1)</span></span>
<span id="cb147-35"><a href="add_chapter_regional_plot.html#cb147-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ggsave(filename = paste0(COURSE_loc, &quot;/gwas/&quot;, tempVARIANTnr, &quot;.&quot;,VARIANT,&quot;.regional_assoc.eps&quot;), plot = p1)</span></span>
<span id="cb147-36"><a href="add_chapter_regional_plot.html#cb147-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-37"><a href="add_chapter_regional_plot.html#cb147-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(temp, p1,</span>
<span id="cb147-38"><a href="add_chapter_regional_plot.html#cb147-38" aria-hidden="true" tabindex="-1"></a>     temp_f, temp_f_ld,</span>
<span id="cb147-39"><a href="add_chapter_regional_plot.html#cb147-39" aria-hidden="true" tabindex="-1"></a>     tempCHR, tempSTART, tempEND,</span>
<span id="cb147-40"><a href="add_chapter_regional_plot.html#cb147-40" aria-hidden="true" tabindex="-1"></a>     VARIANT, tempVARIANTnr)</span>
<span id="cb147-41"><a href="add_chapter_regional_plot.html#cb147-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-42"><a href="add_chapter_regional_plot.html#cb147-42" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
This should produce the figure similar to these below.
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-racer"></span>
<img src="img/rs6802898.rs6802898.regional_assoc.png" alt="Regional association plots." width="909" /><img src="img/rs7901695.rs7901695.regional_assoc.png" alt="Regional association plots." width="909" /><img src="img/rs8050136.rs8050136.regional_assoc.png" alt="Regional association plots." width="909" />
<p class="caption">
Figure 15.1: Regional association plots.
</p>
</div>
<blockquote>
<p>Please note that we are letting RACER plot the variants on genome build 37 (hg19), but we actually provide hg18-based chromosome and basepair positions. The proper thing to do is to <a href="https://genome.sph.umich.edu/wiki/LiftOver" target="_blank">liftOver</a> our coordinates to match those in hg19. Here we didn’t, because I just wanted to show you the ‘how it’s done’. In a future edition I will make sure we will liftOver first.</p>
</blockquote>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="epilogue.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="eigensoft.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": ["twitter", "linkedin", "whatsapp"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics/edit/master/add_chapter_regional_plot.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
