<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>21 Answers to the questions | A Practical Primer in Human Complex Genetics for CoCalc</title>
  <meta name="description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="21 Answers to the questions | A Practical Primer in Human Complex Genetics for CoCalc" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/img/cover/cover_2024.png" />
  <meta property="og:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="21 Answers to the questions | A Practical Primer in Human Complex Genetics for CoCalc" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="twitter:image" content="/img/cover/cover_2024.png" />

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  <link rel="apple-touch-icon-precomposed" sizes="180x180" href="img/cover/iconified/apple-touch-icon-180x180.png" />
  <link rel="shortcut icon" href="img/cover/iconified/favicon.ico" type="image/x-icon" />
<link rel="prev" href="epilogue.html"/>
<link rel="next" href="license.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a></li>
<li class="chapter" data-level="4" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html"><i class="fa fa-check"></i><b>4</b> Getting started - programs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#plink"><i class="fa fa-check"></i><b>4.2</b> PLINK</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#alternatives-to-plink"><i class="fa fa-check"></i><b>4.2.1</b> Alternatives to <code>PLINK</code></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#other-programs"><i class="fa fa-check"></i><b>4.3</b> Other programs</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html"><i class="fa fa-check"></i><b>5</b> Getting started - CoCalc</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#starting-the-course-on-cocalc"><i class="fa fa-check"></i><b>5.1</b> Starting the course on CoCalc</a></li>
<li class="chapter" data-level="5.2" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#but-wait-what-is-cocalc-anyway"><i class="fa fa-check"></i><b>5.2</b> But wait, what is CoCalc anyway?</a></li>
<li class="chapter" data-level="5.3" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#okay-and-whats-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.3</b> Okay, and what’s a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.4" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-does-a-jupyter-notebook-work"><i class="fa fa-check"></i><b>5.4</b> How does a Jupyter Notebook work?</a></li>
<li class="chapter" data-level="5.5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#and-what-is-the-force-behind-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.5</b> And what is the Force behind a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.6" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-to-use-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.6</b> How to use a Jupyter Notebook</a></li>
<li class="chapter" data-level="5.7" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#trouble-shooting-juptyer-notebook"><i class="fa fa-check"></i><b>5.7</b> Trouble shooting Juptyer Notebook</a></li>
<li class="chapter" data-level="5.8" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#cocalc-and-this-practical-primer"><i class="fa fa-check"></i><b>5.8</b> CoCalc and this Practical Primer</a></li>
<li class="chapter" data-level="5.9" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-jupyter-notebook-with-a-r-kernel"><i class="fa fa-check"></i><b>5.9</b> Beginning your own Jupyter Notebook with a <code>R</code> kernel</a></li>
<li class="chapter" data-level="5.10" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-terminal-for-bash-code"><i class="fa fa-check"></i><b>5.10</b> Beginning your own Terminal for <code>bash</code> code</a></li>
<li class="chapter" data-level="5.11" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#this-book-and-cocalc"><i class="fa fa-check"></i><b>5.11</b> This book and CoCalc</a></li>
<li class="chapter" data-level="5.12" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#are-you-ready"><i class="fa fa-check"></i><b>5.12</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html"><i class="fa fa-check"></i><b>6</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#setting-the-stage"><i class="fa fa-check"></i><b>6.1.1</b> Setting the stage</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#converting-the-data"><i class="fa fa-check"></i><b>6.2</b> Converting the data</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#quality-control"><i class="fa fa-check"></i><b>6.3</b> Quality control</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>6.4</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html"><i class="fa fa-check"></i><b>7</b> Sample QC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sex"><i class="fa fa-check"></i><b>7.1</b> Sex</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sample-call-rates"><i class="fa fa-check"></i><b>7.2</b> Sample call rates</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>7.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="7.4" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#relatedness"><i class="fa fa-check"></i><b>7.4</b> Relatedness</a></li>
<li class="chapter" data-level="7.5" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#ancestral-background"><i class="fa fa-check"></i><b>7.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#g-phase-1"><i class="fa fa-check"></i><b>7.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="7.5.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#principal-component-analysis"><i class="fa fa-check"></i><b>7.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="7.5.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>7.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#removing-samples"><i class="fa fa-check"></i><b>7.6</b> Removing samples</a></li>
<li class="chapter" data-level="7.7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#the-next-step"><i class="fa fa-check"></i><b>7.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html"><i class="fa fa-check"></i><b>8</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="8.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#snp-call-rates"><i class="fa fa-check"></i><b>8.1</b> SNP call rates</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>8.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#allele-frequencies"><i class="fa fa-check"></i><b>8.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>8.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>8.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="8.5" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#final-snp-qc"><i class="fa fa-check"></i><b>8.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="8.6" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-milestone"><i class="fa fa-check"></i><b>8.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html"><i class="fa fa-check"></i><b>9</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#exploring-the-data"><i class="fa fa-check"></i><b>9.1</b> Exploring the data</a></li>
<li class="chapter" data-level="9.2" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#genetic-models"><i class="fa fa-check"></i><b>9.2</b> Genetic models</a></li>
<li class="chapter" data-level="9.3" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#logistic-regression"><i class="fa fa-check"></i><b>9.3</b> Logistic regression</a></li>
<li class="chapter" data-level="9.4" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#lets-get-visual"><i class="fa fa-check"></i><b>9.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html"><i class="fa fa-check"></i><b>10</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#qq-plots"><i class="fa fa-check"></i><b>10.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="10.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#manhattan-plots"><i class="fa fa-check"></i><b>10.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="10.2" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#other-plots"><i class="fa fa-check"></i><b>10.2</b> Other plots</a></li>
<li class="chapter" data-level="10.3" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#interactive-plots"><i class="fa fa-check"></i><b>10.3</b> Interactive plots</a></li>
<li class="chapter" data-level="10.4" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#getting-up-close"><i class="fa fa-check"></i><b>10.4</b> Getting up close</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html"><i class="fa fa-check"></i><b>11</b> Regional association plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#locuszoom"><i class="fa fa-check"></i><b>11.1</b> LocusZoom</a></li>
<li class="chapter" data-level="11.2" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#racer"><i class="fa fa-check"></i><b>11.2</b> RACER</a></li>
<li class="chapter" data-level="11.3" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#stop-playing-around"><i class="fa fa-check"></i><b>11.3</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>12</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="12.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>12.1</b> Genotyping</a></li>
<li class="chapter" data-level="12.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>12.2</b> The data</a></li>
<li class="chapter" data-level="12.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>12.3</b> Assignment</a></li>
<li class="chapter" data-level="12.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>12.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html"><i class="fa fa-check"></i><b>13</b> WTCCC1: quality control</a>
<ul>
<li class="chapter" data-level="13.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#quality-control-1"><i class="fa fa-check"></i><b>13.1</b> Quality control</a></li>
<li class="chapter" data-level="13.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#ancestral-background-1"><i class="fa fa-check"></i><b>13.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#preparing-pca"><i class="fa fa-check"></i><b>13.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="13.2.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#running-pca"><i class="fa fa-check"></i><b>13.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="13.2.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#plotting-pca"><i class="fa fa-check"></i><b>13.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="13.2.4" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#removing-samples-1"><i class="fa fa-check"></i><b>13.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#summary"><i class="fa fa-check"></i><b>13.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html"><i class="fa fa-check"></i><b>14</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#association-testing"><i class="fa fa-check"></i><b>14.1</b> Association testing</a></li>
<li class="chapter" data-level="14.2" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#independent-hits"><i class="fa fa-check"></i><b>14.2</b> Independent hits</a></li>
<li class="chapter" data-level="14.3" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#replication"><i class="fa fa-check"></i><b>14.3</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>15</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="15.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>15.1</b> Clumping</a></li>
<li class="chapter" data-level="15.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>15.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="15.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>15.3</b> Statistical finemapping</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="post-gwas.html"><a href="post-gwas.html#finemap"><i class="fa fa-check"></i><b>15.3.1</b> FINEMAP</a></li>
<li class="chapter" data-level="15.3.2" data-path="post-gwas.html"><a href="post-gwas.html#locuszoom-1"><i class="fa fa-check"></i><b>15.3.2</b> LocusZoom</a></li>
<li class="chapter" data-level="15.3.3" data-path="post-gwas.html"><a href="post-gwas.html#susie"><i class="fa fa-check"></i><b>15.3.3</b> SuSIE</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>15.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="15.5" data-path="post-gwas.html"><a href="post-gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>15.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="15.6" data-path="post-gwas.html"><a href="post-gwas.html#colocalization"><i class="fa fa-check"></i><b>15.6</b> Colocalization</a></li>
<li class="chapter" data-level="15.7" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>15.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="15.8" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>15.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="post-gwas.html"><a href="post-gwas.html#mrbase"><i class="fa fa-check"></i><b>15.8.1</b> MRBase</a></li>
<li class="chapter" data-level="15.8.2" data-path="post-gwas.html"><a href="post-gwas.html#twosamplemr"><i class="fa fa-check"></i><b>15.8.2</b> TwoSampleMR</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>15.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="15.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>15.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>16</b> Functional Mapping and Annotation of GWAS</a>
<ul>
<li class="chapter" data-level="16.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>16.1</b> Assignment</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>16.1.1</b> Tutorial</a></li>
<li class="chapter" data-level="16.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>16.1.2</b> Create the input</a></li>
<li class="chapter" data-level="16.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>16.1.3</b> Run FUMA - SNP to gene</a></li>
<li class="chapter" data-level="16.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>16.1.4</b> Run FUMA - Gene to function</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>16.2</b> Some closing thoughts</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>17</b> Phenome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="17.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>17.1</b> What’s next?</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>18</b> MRBase</a>
<ul>
<li class="chapter" data-level="18.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>18.1</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html"><i class="fa fa-check"></i><b>19</b> Two Sample Mendelian Randomisation</a>
<ul>
<li class="chapter" data-level="19.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#loading-the-package"><i class="fa fa-check"></i><b>19.1</b> Loading the package</a></li>
<li class="chapter" data-level="19.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#available-outcome"><i class="fa fa-check"></i><b>19.2</b> Available outcome</a></li>
<li class="chapter" data-level="19.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#instruments"><i class="fa fa-check"></i><b>19.3</b> Instruments</a></li>
<li class="chapter" data-level="19.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#clumping-1"><i class="fa fa-check"></i><b>19.4</b> Clumping</a></li>
<li class="chapter" data-level="19.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#outcomes"><i class="fa fa-check"></i><b>19.5</b> Outcomes</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#harmonisation"><i class="fa fa-check"></i><b>19.5.1</b> Harmonisation</a></li>
<li class="chapter" data-level="19.5.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#mendelian-randomisation"><i class="fa fa-check"></i><b>19.5.2</b> Mendelian randomisation</a></li>
<li class="chapter" data-level="19.5.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#heterogeneity-pleiotropy"><i class="fa fa-check"></i><b>19.5.3</b> Heterogeneity &amp; Pleiotropy</a></li>
<li class="chapter" data-level="19.5.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#leave-one-out"><i class="fa fa-check"></i><b>19.5.4</b> Leave-one-out</a></li>
<li class="chapter" data-level="19.5.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#visualisation"><i class="fa fa-check"></i><b>19.5.5</b> Visualisation</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#the-end-1"><i class="fa fa-check"></i><b>19.6</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>20</b> Epilogue</a>
<ul>
<li class="chapter" data-level="20.1" data-path="epilogue.html"><a href="epilogue.html#course-lecturers"><i class="fa fa-check"></i><b>20.1</b> 2024 Course Lecturers</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="answers.html"><a href="answers.html"><i class="fa fa-check"></i><b>21</b> Answers to the questions</a>
<ul>
<li class="chapter" data-level="21.1" data-path="answers.html"><a href="answers.html#chapter-3.1-gwas-basics"><i class="fa fa-check"></i><b>21.1</b> Chapter 3.1: GWAS Basics</a></li>
<li class="chapter" data-level="21.2" data-path="answers.html"><a href="answers.html#chapter-3.4-gwas-basics---association-testing"><i class="fa fa-check"></i><b>21.2</b> Chapter 3.4: GWAS Basics - Association testing</a></li>
<li class="chapter" data-level="21.3" data-path="answers.html"><a href="answers.html#chapter-3.5-gwas-visualization"><i class="fa fa-check"></i><b>21.3</b> Chapter 3.5: GWAS visualization</a></li>
<li class="chapter" data-level="21.4" data-path="answers.html"><a href="answers.html#chapter-3.6-regional-association-plots"><i class="fa fa-check"></i><b>21.4</b> Chapter 3.6: Regional association plots</a></li>
<li class="chapter" data-level="21.5" data-path="answers.html"><a href="answers.html#chapter-4.2-wtccc1"><i class="fa fa-check"></i><b>21.5</b> Chapter 4.2: WTCCC1</a></li>
<li class="chapter" data-level="21.6" data-path="answers.html"><a href="answers.html#chapter-4.3-wtccc1---association-testing"><i class="fa fa-check"></i><b>21.6</b> Chapter 4.3: WTCCC1 - Association testing</a></li>
<li class="chapter" data-level="21.7" data-path="answers.html"><a href="answers.html#chapter-5.4-fuma"><i class="fa fa-check"></i><b>21.7</b> Chapter 5.4: FUMA</a></li>
<li class="chapter" data-level="21.8" data-path="answers.html"><a href="answers.html#chapter-5.5-phewas"><i class="fa fa-check"></i><b>21.8</b> Chapter 5.5: PheWAS</a></li>
<li class="chapter" data-level="21.9" data-path="answers.html"><a href="answers.html#chapter-5.7-twosample-mr"><i class="fa fa-check"></i><b>21.9</b> Chapter 5.7: TwoSample MR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>22</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="22.1" data-path="license.html"><a href="license.html#images-and-data-used"><i class="fa fa-check"></i><b>22.1</b> Images and data used</a></li>
<li class="chapter" data-level="22.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>22.2</b> Disclaimer</a></li>
<li class="chapter" data-level="22.3" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>22.3</b> Copyright</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="colophon.html"><a href="colophon.html"><i class="fa fa-check"></i><b>23</b> Colophon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://vanderlaanand.science" target="blank">&copy; 1979-2024. Sander W. van der Laan.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics for CoCalc</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="answers" class="section level1 hasAnchor" number="21">
<h1><span class="header-section-number">21</span> Answers to the questions<a href="answers.html#answers" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- ![](./img/_headers/banner_man_standing_dna.png){width=100%} -->
<p>Throughout the book you encountered questions related to some steps and output that you generated. Here you can review the answers for each of these questions - of course, for some different answers are possible.</p>
<div id="chapter-3.1-gwas-basics" class="section level2 hasAnchor" number="21.1">
<h2><span class="header-section-number">21.1</span> Chapter 3.1: GWAS Basics<a href="answers.html#chapter-3.1-gwas-basics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Question: Can you think off what the ‘11,440 heterozygous haploid genotypes present’ represent?</p>
</blockquote>
<p>These are the variants on the X-chromosome. Males only have 1 copy.</p>
<blockquote>
<p>Question: Can you think of other scenarios in which you may want to extend the check on differential missingness beyond a check between cases and controls?</p>
</blockquote>
<p>For instance ‘genotyping platform’, or ‘hospital of inclusion’, if you think this might influence the genotyping experiment technically.</p>
</div>
<div id="chapter-3.4-gwas-basics---association-testing" class="section level2 hasAnchor" number="21.2">
<h2><span class="header-section-number">21.2</span> Chapter 3.4: GWAS Basics - Association testing<a href="answers.html#chapter-3.4-gwas-basics---association-testing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Question: Load the data using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<pre><code>library(&quot;data.table&quot;)
gwas_HWE &lt;- data.table::fread(&quot;dummy_project/gwa.hwe&quot;)
gwas_FRQ &lt;- data.table::fread(&quot;dummy_project/gwa.frq&quot;)
gwas_IMISS &lt;- data.table::fread(&quot;dummy_project/gwa.imiss&quot;)
gwas_LMISS &lt;- data.table::fread(&quot;dummy_project/gwa.lmiss&quot;)

# you&#39;ll need to calculate the -log10(P) for the HWE p-values
gwas_HWE$logP &lt;- -log10(gwas_HWE$P)

# you&#39;ll need to calculate the sample call rate
gwas_IMISS$callrate &lt;- 1 - gwas_IMISS$F_MISS

# you&#39;ll need to calculate the SNP call rate
gwas_LMISS$callrate &lt;- 1 - gwas_LMISS$F_MISS</code></pre>
<blockquote>
<p>Question: Plot the per-stratum HWE p-values using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<pre><code>library(&quot;ggpubr&quot;)
ggpubr::gghistogram(gwas_HWE, x = &quot;logP&quot;,
                    add = &quot;mean&quot;,
                    add.params = list(color = &quot;#595A5C&quot;, linetype = &quot;dashed&quot;, size = 1),
                    rug = TRUE,
                    # color = &quot;#1290D9&quot;, fill = &quot;#1290D9&quot;,
                    color = &quot;TEST&quot;, fill = &quot;TEST&quot;,
                    palette = &quot;lancet&quot;,
                    facet.by = &quot;TEST&quot;,
                    bins = 50,
                    xlab = &quot;HWE -log10(P)&quot;) +
  ggplot2::geom_vline(xintercept = 5, linetype = &quot;dashed&quot;,
                      color = &quot;#E55738&quot;, size = 1)
ggplot2::ggsave(&quot;dummy_project/gwas-hwe.png&quot;,
       plot = last_plot())</code></pre>
<blockquote>
<p>Question: Plot the allele frequencies using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<pre><code>ggpubr::gghistogram(gwas_FRQ, x = &quot;MAF&quot;,
                    add = &quot;mean&quot;, add.params = list(color = &quot;#595A5C&quot;, linetype = &quot;dashed&quot;, size = 1),
                    rug = TRUE,
                    color = &quot;#1290D9&quot;, fill = &quot;#1290D9&quot;,
                    xlab = &quot;minor allele frequency&quot;) +
  ggplot2::geom_vline(xintercept = 0.05, linetype = &quot;dashed&quot;,
                      color = &quot;#E55738&quot;, size = 1)
ggplot2::ggsave(&quot;dummy_project/gwas-freq.png&quot;,
       plot = last_plot())</code></pre>
<blockquote>
<p>Question: Plot the per-sample call rates using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<pre><code>ggpubr::gghistogram(gwas_IMISS, x = &quot;callrate&quot;,
                    add = &quot;mean&quot;, add.params = list(color = &quot;#595A5C&quot;, linetype = &quot;dashed&quot;, size = 1),
                    rug = TRUE, bins = 50,
                    color = &quot;#1290D9&quot;, fill = &quot;#1290D9&quot;,
                    xlab = &quot;per sample call rate&quot;) +
  ggplot2::geom_vline(xintercept = 0.95, linetype = &quot;dashed&quot;,
                      color = &quot;#E55738&quot;, size = 1)
ggplot2::ggsave(&quot;data/dummy_project/gwas-sample-call-rate.png&quot;,
       plot = last_plot())</code></pre>
<blockquote>
<p>Question: Plot the per-SNP call rates using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<pre><code>ggpubr::gghistogram(gwas_LMISS, x = &quot;callrate&quot;,
                    add = &quot;mean&quot;, add.params = list(color = &quot;#595A5C&quot;, linetype = &quot;dashed&quot;, size = 1),
                    rug = TRUE, bins = 50,
                    color = &quot;#1290D9&quot;, fill = &quot;#1290D9&quot;,
                    xlab = &quot;per SNP call rate&quot;) +
  ggplot2::geom_vline(xintercept = 0.95, linetype = &quot;dashed&quot;,
                      color = &quot;#E55738&quot;, size = 1)
ggplot2::ggsave(&quot;data/dummy_project/gwas-snp-call-rate.png&quot;,
       plot = last_plot())</code></pre>
<blockquote>
<p>Question: Do you know which model, <em>i.e.</em> <code>TEST</code> is most commonly used and reported? And why is that, do think?</p>
</blockquote>
<p>The additive model. It is the most powerful and robust model. It is also the most commonly used model in GWAS, because most variants are ‘behaving’ according an additive effect.
<!-- in the future version add a reference to this --></p>
<blockquote>
<p>Question: How come there are more lines in this file than there are variants?</p>
</blockquote>
<p>Because the results for different models are reported, so there are 3x more lines than there are variants.</p>
<blockquote>
<p>Question: Can you write down the mathematical relation between <em>beta</em> and <em>OR</em>?</p>
</blockquote>
<p>The odds ratio is the exponentiated beta. So, <code>OR = exp(beta)</code>.</p>
</div>
<div id="chapter-3.5-gwas-visualization" class="section level2 hasAnchor" number="21.3">
<h2><span class="header-section-number">21.3</span> Chapter 3.5: GWAS visualization<a href="answers.html#chapter-3.5-gwas-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Question: Try to figure out how to get the number of variants per chromosomes. Why do the number of variants per chrosome (approximately) correlate with the chromosome number?</p>
</blockquote>
<pre><code>df &lt;- as.data.frame(table(gwas_assoc_compl$CHR))
names(df)[names(df) == &quot;Var1&quot;] &lt;- &quot;Chr&quot;
View(df)</code></pre>
<blockquote>
<p>Question: Where are the data for chromosome X, Y and MT?</p>
</blockquote>
<p>The data for chromosome X, Y and MT are not present in the dataset. In this case it is a dummy dataset, so it wasn’t even generated. In real-life, a lot of the older generation genotyping platforms didn’t have chromosome Y or MT, and only few variants on chromosome X. Nowadays, with the newer platforms, chromosome X and Y are included, as well as MT. A downside still remains: the reference datasets for imputation do not include the Y and the MT chromosome.</p>
<blockquote>
<p>Question: What do the grey spots on the density plot indicate?</p>
</blockquote>
<p>These are the chromosomal regions without any (or with very low) variant coverage, as they are notoriously hard to sequence. Therefore companies cannot design probes against these regions and they are not included in the genotyping arrays. But what are these regions: usually the ends of chromosomes and the centromeres.</p>
</div>
<div id="chapter-3.6-regional-association-plots" class="section level2 hasAnchor" number="21.4">
<h2><span class="header-section-number">21.4</span> Chapter 3.6: Regional association plots<a href="answers.html#chapter-3.6-regional-association-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Question: How many independent hits did you find?</p>
</blockquote>
<p>Two.</p>
</div>
<div id="chapter-4.2-wtccc1" class="section level2 hasAnchor" number="21.5">
<h2><span class="header-section-number">21.5</span> Chapter 4.2: WTCCC1<a href="answers.html#chapter-4.2-wtccc1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Question: What do you notice in the ‘per sample call rate’ graph? Can you think of a reason why this is? And how would you deal with this?</p>
</blockquote>
<p>There are two bumps in the histogram. This could be due to the genotyping platform used, or due to the case/control status. The two bumps could represent two different genotyping platforms. If this is the case, you could split the data and analyze them separately or correct for this somehow. If it is due to the case/control status, you could remove the samples with a low call rate. It could also be that the cases and the controls were genotype separately, this causes a batch effect which is evident in a slightly different per sample call rate.</p>
<blockquote>
<p>Question: Do you have any thoughts on that? Do you agree with the filters I set below? How would you do it differently and why?</p>
</blockquote>
<p>These were my settings <code>--maf 0.01 --geno 0.05 --hwe 0.00001</code>.</p>
<p>You could have set the call rate filter to 0.97 or even 0.98; it’s a bit arbitrary, but anything between 0.95 and 0.97 is a common threshold.
The HWE filter is set to 0.00001, which is relatively strict. Some would set it to 0.0001 or even 0.000001. Again, it’s a bit arbitrary, of course you don’t want to kill any signals - if all is well there should be slight differences between cases and controls. So an alternative could also be to only filter the HWE in the cases.
The MAF filter is set to 0.01, which is strict. You could have set it to 0.05. It’s also dependent on the size of your GWAS. Suppose you have 1000 samples, then 1% of a 1000 is only 10 cases. This maybe to few cases for a GWAS, so you might want the threshold to be at 5%. If you are dealing with the UK Biobank dataset, which has 500,000 samples, then 1% is 5000 cases, which is a lot. So you could set the threshold to 1% or even go lower.</p>
</div>
<div id="chapter-4.3-wtccc1---association-testing" class="section level2 hasAnchor" number="21.6">
<h2><span class="header-section-number">21.6</span> Chapter 4.3: WTCCC1 - Association testing<a href="answers.html#chapter-4.3-wtccc1---association-testing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Question: How many independent hits did you find?</p>
</blockquote>
<p>There are two hits. But, the one on chromosome 22 is probably a false positive because <code>PLINK</code> couldn’t find any correlated variants to the top hit (rs9617611).</p>
</div>
<div id="chapter-5.4-fuma" class="section level2 hasAnchor" number="21.7">
<h2><span class="header-section-number">21.7</span> Chapter 5.4: FUMA<a href="answers.html#chapter-5.4-fuma" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The results from this practical primer using the WTCCC1 data are published publicly under ID <code>471422</code> on the <a href="http://fuma.ctglab.nl" target="_blank">FUMA</a> website.</p>
<blockquote>
<p>Question: can you figure out the sample size of the WTCCC1 data you used?</p>
</blockquote>
<p>According to the <code>PLINK</code> log there are 3,268 individuals.</p>
<pre><code>PLINK v1.90b7.2 64-bit (11 Dec 2023)
Options in effect:
  --bfile wtccc1/wtccc1_clean
  --exclude wtccc1/fail-ancestry-QC.txt
  --make-bed
  --out wtccc1/wtccc1_qc

Start time: Mon Apr  8 17:35:51 2024

Random number seed: 1712612151
16384 MB RAM detected; reserving 8192 MB for main workspace.
379407 variants loaded from .bim file.
3268 people (2160 males, 1108 females) loaded from .fam.
3268 phenotype values loaded from .fam.
--exclude: 379407 variants remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 3268 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.997173.
379407 variants and 3268 people pass filters and QC.
Among remaining phenotypes, 1871 are cases and 1397 are controls.
--make-bed to wtccc1/wtccc1_qc.bed + wtccc1/wtccc1_qc.bim +
wtccc1/wtccc1_qc.fam ... done.

End time: Mon Apr  8 17:35:52 2024</code></pre>
<blockquote>
<p>Questions
1. How many lead SNPs did we find?</p>
</blockquote>
<p>The same two we found through <code>PLINK</code>.</p>
<p>You can find the Manhatton plot that <code>FUMA</code> generated under ‘Genome-wide plots’ on the <code>FUMA</code>-website (below).</p>
<p><img src="img/gwas_wtccc/manhattan_FUMA_jobs471422.png" /></p>
<p>You can find more about the two loci under ’Summary of results.</p>
<p><img src="img/gwas_wtccc/lociPlot_FUMA_jobs471422.png" /></p>
<p>Under the ‘Results’ heading on the <code>FUMA</code> website you can find much more, such as the main ‘Genomic risk loci’, but also the ‘lead SNPs’ and the ‘Ind. Sig. SNPs’ (the independent significant SNPs. The ‘lead SNPs’ are the same as we found through <em>clumping</em> with <code>PLINK</code>. The ‘Ind. Sig. SNPs’ are the SNPs that are independent (from the other) and significant.</p>
<p><img src="img/gwas_wtccc/FUMA_genomic_risk_loci.png" /></p>
<p><img src="img/gwas_wtccc/FUMA_Ind_Sig_SNPs.png" /></p>
<p><img src="img/gwas_wtccc/FUMA_lead_SNPs.png" /></p>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>What do the results of MAGMA (the gene-based test) look like and how many genes pass the threshold of multiple testing correction?</li>
</ol>
</blockquote>
<p>You can find the results of the gene-based test under ‘Genome-wide plots’.</p>
<p>The region at chromosome 9 is significant, and you can see that a few genes reach the significance threshold. The latter is determined by the Bonferroni correction based on the number of genes tested.</p>
<p><img src="img/gwas_wtccc/geneManhattan_FUMA_jobs471422.png" /></p>
<blockquote>
<ol start="3" style="list-style-type: decimal">
<li>How many loci were mapped?</li>
</ol>
</blockquote>
<p>Two.</p>
<blockquote>
<ol start="4" style="list-style-type: decimal">
<li>How many genes were physically located and how many were mapped to these loci?</li>
</ol>
</blockquote>
<p>There are 12 genes mapped to the loci.</p>
<blockquote>
<ol start="5" style="list-style-type: decimal">
<li>Do you think all loci are ‘correct’, <em>i.e.</em> do you ‘believe’ all the signals looking at the mapping results? Why?</li>
</ol>
</blockquote>
<p>The results from chromosome may be a bit dubious, as there is only 1 variant reaching genome-wide significance. The results from chromosome 9 are more convincing, as there are multiple variants reaching genome-wide significance.
All the genes are in physical proximity and they are protein coding. Some have an expression quantitative trait locus (eQTL), but other than that there is no additional information.</p>
<blockquote>
<ol start="6" style="list-style-type: decimal">
<li>For what tissues are the signals enriched?</li>
</ol>
</blockquote>
<p>There is no enrichment in any tissue. However, most enrichment is found in the ovaries - which is strange in the context of coronary artery disease; unless it is related to a specific process. For instance, the ovaries may be very well perfused and the genes may be involved in angiogenesis. The next ‘best’ tissue are the blood vessels, which is more plausible in the context of CAD.</p>
<p><img src="img/gwas_wtccc/magma_exp_gtex_v8_ts_general_avg_log2TPM_FUMA_jobs471422.png" /></p>
<blockquote>
<ol start="7" style="list-style-type: decimal">
<li>Are there any chromatin interactions discovered?</li>
</ol>
</blockquote>
<p>Yes, for both chromosome 9 and 22. The ‘circos’ plots only show genes mapped as eQTL and/or by chromatin interactions. If the gene is mapped only by chromatin interactions or only by eQTLs, it is colored orange or green, respectively. When the gene is mapped by both, it is colored red.</p>
<p><img src="img/gwas_wtccc/circos_chr9.png" /></p>
<p><img src="img/gwas_wtccc/circos_chr22.png" /></p>
<blockquote>
<p>Questions</p>
<ol style="list-style-type: decimal">
<li>What genes show the lowest expression across tissues?</li>
<li>And what genes the highest?</li>
</ol>
</blockquote>
<p>You can answer these questions when you run <code>GENE2FUNCTION</code>. You can then review which of the mapped genes is expressed most in which tissue, and look at their role in relation to pathways. This may be informative for overall conclusions.</p>
<p><em>ATP6V1E1</em> is most expressed across all included tissues. <em>CECR6</em> is among the least expressed genes.</p>
<p><img src="img/gwas_wtccc/expHeat_FUMA_jobs.png" /></p>
<blockquote>
<ol start="3" style="list-style-type: decimal">
<li>For what pathways are the signals enriched?</li>
</ol>
</blockquote>
<p><em>CDKN2A</em> and <em>CDKN2B</em> are most enriched in the ‘Cell cycle’ pathway.</p>
<p><img src="img/gwas_wtccc/BioCarta_FUMA_jobs471422.png" /></p>
<blockquote>
<ol start="4" style="list-style-type: decimal">
<li>What molecular functions are mapped to the signals?</li>
</ol>
</blockquote>
<p><img src="img/gwas_wtccc/GO_mf_FUMA_jobs471422.png" /></p>
<p>Again <em>CDKN2A</em> and <em>CDKN2B</em> are most enriched and involved in a cyclin dependent protein serine-threonine kinase inhibitor activity.</p>
</div>
<div id="chapter-5.5-phewas" class="section level2 hasAnchor" number="21.8">
<h2><span class="header-section-number">21.8</span> Chapter 5.5: PheWAS<a href="answers.html#chapter-5.5-phewas" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p><em>Assignment</em></p>
<ol style="list-style-type: decimal">
<li>Perform a PheWAS with a few of the resources and your favorite SNP from this tutorial.</li>
<li>Compare the different websites. What do you notice?</li>
</ol>
</blockquote>
<p>Some websites report different phenotypes, but most of them report the same. Some website use the subsets of the same datasets, or they use different datasets altogether. The results are similar across different websites and datasets. Usually these datasets are FinnGen, UKBiobank and likewise large biobanks..</p>
<blockquote>
<ol start="3" style="list-style-type: decimal">
<li>How is a PheWAS informative?</li>
</ol>
</blockquote>
<p>It can tell you something about the pleiotropic effects of a variant or gene. It can also tell you something about the underlying biology of a variant. A variant associated to CAD could also be associated to T2D, which could be due to the fact that the variant is involved in glucose metabolism.</p>
</div>
<div id="chapter-5.7-twosample-mr" class="section level2 hasAnchor" number="21.9">
<h2><span class="header-section-number">21.9</span> Chapter 5.7: TwoSample MR<a href="answers.html#chapter-5.7-twosample-mr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Question: What is the causal effect of T2D on CAD? Can you describe this using the knowledge from the lecture and the results in the table.
Question: What do the results of the heterogeneity and pleiotropy tests tell you?
Question: What do the results of the single SNP and leave-one-out analyses tell you?
Question: What do you see in the scatter plot? And what can you conclude from this?
Question: What do you see in the forest plot? And what can you conclude from this?
Question: What does a leave-one-out ploy tell you? And what can you conclude from this?
Question: Why would yoy use a funnel plot? And what can it tell you?
Question: What are your main conclusions from the MR analysis?</p>
</blockquote>
<script>
title=document.getElementById('header');
title.innerHTML = '<img src="img/headers/banner_man_standing_dna.png" alt="Answers to questions">' + title.innerHTML
</script>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="epilogue.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="license.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans-serif",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"always_allow_html": true,
"split_by": "chapter",
"toolbar": {
"position": "fixed"
},
"info": true,
"code_folding": "show"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
