<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>22 Colophon | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="22 Colophon | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="22 Colophon | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="license.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#cocalc-vs.-standalone"><i class="fa fa-check"></i><b>3.1</b> CoCalc vs. Standalone</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="getting-started-programs.html"><a href="getting-started-programs.html"><i class="fa fa-check"></i><b>4</b> Getting started - programs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting-started-programs.html"><a href="getting-started-programs.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started-programs.html"><a href="getting-started-programs.html#plink"><i class="fa fa-check"></i><b>4.2</b> PLINK</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started-programs.html"><a href="getting-started-programs.html#alternatives-to-plink"><i class="fa fa-check"></i><b>4.2.1</b> Alternatives to <code>PLINK</code></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started-programs.html"><a href="getting-started-programs.html#other-programs"><i class="fa fa-check"></i><b>4.3</b> Other programs</a></li>
<li class="chapter" data-level="4.4" data-path="getting-started-programs.html"><a href="getting-started-programs.html#did-you-choose"><i class="fa fa-check"></i><b>4.4</b> Did you choose?</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html"><i class="fa fa-check"></i><b>5</b> Getting started - Standalone</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#the-data-you-need"><i class="fa fa-check"></i><b>5.1</b> The data you need</a></li>
<li class="chapter" data-level="5.2" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#terminal"><i class="fa fa-check"></i><b>5.2</b> Terminal</a></li>
<li class="chapter" data-level="5.3" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#navigating-the-terminal"><i class="fa fa-check"></i><b>5.3</b> Navigating the Terminal</a></li>
<li class="chapter" data-level="5.4" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#installing-the-software"><i class="fa fa-check"></i><b>5.4</b> Installing the software</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#brew"><i class="fa fa-check"></i><b>5.4.1</b> brew</a></li>
<li class="chapter" data-level="5.4.2" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#plink-1"><i class="fa fa-check"></i><b>5.4.2</b> PLINK</a></li>
<li class="chapter" data-level="5.4.3" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>5.4.3</b> Installing R and RStudio</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="getting-started-standalone.html"><a href="getting-started-standalone.html#are-you-ready"><i class="fa fa-check"></i><b>5.5</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html"><i class="fa fa-check"></i><b>6</b> Getting started - CoCalc</a>
<ul>
<li class="chapter" data-level="6.1" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#starting-the-course-on-cocalc"><i class="fa fa-check"></i><b>6.1</b> Starting the course on CoCalc</a></li>
<li class="chapter" data-level="6.2" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-notebook"><i class="fa fa-check"></i><b>6.2</b> Beginning your own notebook</a></li>
<li class="chapter" data-level="6.3" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#are-you-ready-1"><i class="fa fa-check"></i><b>6.3</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>7</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>7.1</b> Converting datasets</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>7.2</b> Quality control</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>7.3</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>8</b> Sample QC</a>
<ul>
<li class="chapter" data-level="8.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>8.1</b> Sex</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>8.2</b> Sample call rates</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>8.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="8.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>8.4</b> Relatedness</a></li>
<li class="chapter" data-level="8.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>8.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>8.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="8.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="8.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>8.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>8.6</b> Removing samples</a></li>
<li class="chapter" data-level="8.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>8.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>9</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>9.1</b> SNP call rates</a></li>
<li class="chapter" data-level="9.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>9.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="9.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>9.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>9.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>9.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="9.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>9.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="9.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>9.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="gwas_testing.html"><a href="gwas_testing.html"><i class="fa fa-check"></i><b>10</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="10.1" data-path="gwas_testing.html"><a href="gwas_testing.html#exploring-the-data"><i class="fa fa-check"></i><b>10.1</b> Exploring the data</a></li>
<li class="chapter" data-level="10.2" data-path="gwas_testing.html"><a href="gwas_testing.html#genetic-models"><i class="fa fa-check"></i><b>10.2</b> Genetic models</a></li>
<li class="chapter" data-level="10.3" data-path="gwas_testing.html"><a href="gwas_testing.html#logistic-regression"><i class="fa fa-check"></i><b>10.3</b> Logistic regression</a></li>
<li class="chapter" data-level="10.4" data-path="gwas_testing.html"><a href="gwas_testing.html#lets-get-visual"><i class="fa fa-check"></i><b>10.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>11</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>11.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="11.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>11.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="11.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>11.2</b> Other plots</a></li>
<li class="chapter" data-level="11.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>11.3</b> Interactive plots</a></li>
<li class="chapter" data-level="11.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>11.4</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html"><i class="fa fa-check"></i><b>12</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="12.1" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#genotyping"><i class="fa fa-check"></i><b>12.1</b> Genotyping</a></li>
<li class="chapter" data-level="12.2" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#the-data"><i class="fa fa-check"></i><b>12.2</b> The data</a></li>
<li class="chapter" data-level="12.3" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#assignment"><i class="fa fa-check"></i><b>12.3</b> Assignment</a></li>
<li class="chapter" data-level="12.4" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#there-you-go"><i class="fa fa-check"></i><b>12.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>13</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="13.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>13.1</b> Quality control</a></li>
<li class="chapter" data-level="13.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>13.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>13.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="13.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca"><i class="fa fa-check"></i><b>13.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="13.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>13.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="13.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>13.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>13.3</b> Association testing</a></li>
<li class="chapter" data-level="13.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>13.4</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="post_gwas.html"><a href="post_gwas.html"><i class="fa fa-check"></i><b>14</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="14.1" data-path="post_gwas.html"><a href="post_gwas.html#clumping"><i class="fa fa-check"></i><b>14.1</b> Clumping</a></li>
<li class="chapter" data-level="14.2" data-path="post_gwas.html"><a href="post_gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>14.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="14.3" data-path="post_gwas.html"><a href="post_gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>14.3</b> Statistical finemapping</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="post_gwas.html"><a href="post_gwas.html#finemap"><i class="fa fa-check"></i><b>14.3.1</b> FINEMAP</a></li>
<li class="chapter" data-level="14.3.2" data-path="post_gwas.html"><a href="post_gwas.html#locuszoom"><i class="fa fa-check"></i><b>14.3.2</b> LocusZoom</a></li>
<li class="chapter" data-level="14.3.3" data-path="post_gwas.html"><a href="post_gwas.html#susie"><i class="fa fa-check"></i><b>14.3.3</b> SuSIE</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="post_gwas.html"><a href="post_gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>14.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="14.5" data-path="post_gwas.html"><a href="post_gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>14.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="14.6" data-path="post_gwas.html"><a href="post_gwas.html#colocalization"><i class="fa fa-check"></i><b>14.6</b> Colocalization</a></li>
<li class="chapter" data-level="14.7" data-path="post_gwas.html"><a href="post_gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>14.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="14.8" data-path="post_gwas.html"><a href="post_gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>14.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="14.8.1" data-path="post_gwas.html"><a href="post_gwas.html#mrbase"><i class="fa fa-check"></i><b>14.8.1</b> MRBase</a></li>
<li class="chapter" data-level="14.8.2" data-path="post_gwas.html"><a href="post_gwas.html#twosamplemr"><i class="fa fa-check"></i><b>14.8.2</b> TwoSampleMR</a></li>
<li class="chapter" data-level="14.8.3" data-path="post_gwas.html"><a href="post_gwas.html#your-choice"><i class="fa fa-check"></i><b>14.8.3</b> Your choice</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="post_gwas.html"><a href="post_gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>14.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="14.10" data-path="post_gwas.html"><a href="post_gwas.html#what-comes-next"><i class="fa fa-check"></i><b>14.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>15</b> Functional Mapping and Annotation of GWAS</a>
<ul>
<li class="chapter" data-level="15.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>15.1</b> Assignment</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>15.1.1</b> Tutorial</a></li>
<li class="chapter" data-level="15.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>15.1.2</b> Create the input</a></li>
<li class="chapter" data-level="15.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>15.1.3</b> Run FUMA - SNP to gene</a></li>
<li class="chapter" data-level="15.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>15.1.4</b> Run FUMA - Gene to function</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>15.2</b> Some closing thoughts</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>16</b> Phenome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="16.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>16.1</b> What’s next?</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>17</b> MRBase</a>
<ul>
<li class="chapter" data-level="17.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>17.1</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>18</b> Epilogue</a></li>
<li class="chapter" data-level="19" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>19</b> Additional: EIGENSOFT</a>
<ul>
<li class="chapter" data-level="19.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>19.1</b> Install homebrew</a></li>
<li class="chapter" data-level="19.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>19.2</b> Install missing packages</a></li>
<li class="chapter" data-level="19.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>19.3</b> Installing EIGENSOFT</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="answers.html"><a href="answers.html"><i class="fa fa-check"></i><b>20</b> Answers to the questions</a>
<ul>
<li class="chapter" data-level="20.1" data-path="answers.html"><a href="answers.html#chapter-3.1-gwas-basics"><i class="fa fa-check"></i><b>20.1</b> Chapter 3.1: GWAS Basics</a></li>
<li class="chapter" data-level="20.2" data-path="answers.html"><a href="answers.html#chapter-3.4-gwas-basics---association-testing"><i class="fa fa-check"></i><b>20.2</b> Chapter 3.4: GWAS Basics - Association testing</a></li>
<li class="chapter" data-level="20.3" data-path="answers.html"><a href="answers.html#chapter-3.5-gwas-visualization"><i class="fa fa-check"></i><b>20.3</b> Chapter 3.5: GWAS visualization</a></li>
<li class="chapter" data-level="20.4" data-path="answers.html"><a href="answers.html#chapter-5.4-fuma"><i class="fa fa-check"></i><b>20.4</b> Chapter 5.4: FUMA</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>21</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="21.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>21.1</b> Copyright</a></li>
<li class="chapter" data-level="21.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>21.2</b> Disclaimer</a></li>
<li class="chapter" data-level="21.3" data-path="license.html"><a href="license.html#images-and-data-used"><i class="fa fa-check"></i><b>21.3</b> Images and data used</a></li>
<li class="chapter" data-level="21.4" data-path="license.html"><a href="license.html#copyright-1"><i class="fa fa-check"></i><b>21.4</b> Copyright</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="colophon.html"><a href="colophon.html"><i class="fa fa-check"></i><b>22</b> Colophon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://vanderlaanand.science" target="blank">&copy; 1979-2024. Sander W. van der Laan.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="colophon" class="section level1 hasAnchor" number="22">
<h1><span class="header-section-number">22</span> Colophon<a href="colophon.html#colophon" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The 2022 and 2024 editions of this book were produce in RStudio and with the <code>bookdown</code> package. Below a listing of installed programs and libraries, the operating system, and their specific versions.</p>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.3.3 (2024-02-29)
##  os       macOS Sonoma 14.5
##  system   x86_64, darwin20
##  ui       X11
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       America/New_York
##  date     2024-04-04
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package           * version date (UTC) lib source
##  askpass             1.2.0   2023-09-03 [2] CRAN (R 4.3.0)
##  bookdown          * 0.38.1  2024-03-26 [2] Github (rstudio/bookdown@50a1c1e)
##  bslib               0.6.2   2024-03-22 [2] CRAN (R 4.3.2)
##  cachem              1.0.8   2023-05-01 [2] CRAN (R 4.3.0)
##  chromote            0.2.0   2024-02-12 [1] CRAN (R 4.3.2)
##  cli                 3.6.2   2023-12-11 [2] CRAN (R 4.3.0)
##  colorspace          2.1-0   2023-01-23 [2] CRAN (R 4.3.0)
##  crayon              1.5.2   2022-09-29 [2] CRAN (R 4.3.0)
##  crul                1.4.0   2023-05-17 [2] CRAN (R 4.3.0)
##  curl                5.2.1   2024-03-01 [2] CRAN (R 4.3.2)
##  data.table          1.15.4  2024-03-30 [1] CRAN (R 4.3.2)
##  digest              0.6.35  2024-03-11 [2] CRAN (R 4.3.2)
##  evaluate            0.23    2023-11-01 [2] CRAN (R 4.3.0)
##  fastmap             1.1.1   2023-02-24 [2] CRAN (R 4.3.0)
##  flextable         * 0.9.5   2024-03-06 [1] CRAN (R 4.3.2)
##  fontBitstreamVera   0.1.1   2017-02-01 [2] CRAN (R 4.3.0)
##  fontLiberation      0.1.0   2016-10-15 [2] CRAN (R 4.3.0)
##  fontquiver          0.2.1   2017-02-01 [2] CRAN (R 4.3.0)
##  formatR           * 1.14    2023-01-17 [2] CRAN (R 4.3.0)
##  gdtools             0.3.7   2024-03-05 [2] CRAN (R 4.3.2)
##  gfonts              0.2.0   2023-01-08 [2] CRAN (R 4.3.0)
##  glue                1.7.0   2024-01-09 [2] CRAN (R 4.3.0)
##  htmltools           0.5.8   2024-03-25 [2] CRAN (R 4.3.2)
##  httpcode            0.3.0   2020-04-10 [2] CRAN (R 4.3.0)
##  httpuv              1.6.15  2024-03-26 [2] CRAN (R 4.3.2)
##  jquerylib           0.1.4   2021-04-26 [2] CRAN (R 4.3.0)
##  jsonlite            1.8.8   2023-12-04 [2] CRAN (R 4.3.0)
##  kableExtra        * 1.4.0   2024-01-24 [1] CRAN (R 4.3.2)
##  knitr             * 1.45    2023-10-30 [1] CRAN (R 4.3.0)
##  later               1.3.2   2023-12-06 [2] CRAN (R 4.3.0)
##  lifecycle           1.0.4   2023-11-07 [2] CRAN (R 4.3.0)
##  magrittr            2.0.3   2022-03-30 [2] CRAN (R 4.3.0)
##  mime                0.12    2021-09-28 [2] CRAN (R 4.3.0)
##  munsell             0.5.1   2024-04-01 [1] CRAN (R 4.3.2)
##  officer             0.6.5   2024-02-24 [2] CRAN (R 4.3.2)
##  openssl             2.1.1   2023-09-25 [2] CRAN (R 4.3.0)
##  processx            3.8.4   2024-03-16 [2] CRAN (R 4.3.2)
##  promises            1.2.1   2023-08-10 [2] CRAN (R 4.3.0)
##  ps                  1.7.6   2024-01-18 [2] CRAN (R 4.3.0)
##  R6                  2.5.1   2021-08-19 [2] CRAN (R 4.3.0)
##  ragg                1.3.0   2024-03-13 [2] CRAN (R 4.3.2)
##  Rcpp                1.0.12  2024-01-09 [2] CRAN (R 4.3.0)
##  rlang               1.1.3   2024-01-10 [2] CRAN (R 4.3.0)
##  rmarkdown         * 2.26.1  2024-03-26 [2] Github (rstudio/rmarkdown@ee69d59)
##  rstudioapi          0.16.0  2024-03-24 [2] CRAN (R 4.3.2)
##  sass                0.4.9   2024-03-15 [2] CRAN (R 4.3.2)
##  scales              1.3.0   2023-11-28 [2] CRAN (R 4.3.0)
##  sessioninfo         1.2.2   2021-12-06 [2] CRAN (R 4.3.0)
##  shiny               1.8.1   2024-03-26 [2] CRAN (R 4.3.2)
##  stringi             1.8.3   2023-12-11 [2] CRAN (R 4.3.0)
##  stringr             1.5.1   2023-11-14 [2] CRAN (R 4.3.0)
##  svglite             2.1.3   2023-12-08 [1] CRAN (R 4.3.0)
##  systemfonts         1.0.6   2024-03-07 [2] CRAN (R 4.3.2)
##  textshaping         0.3.7   2023-10-09 [2] CRAN (R 4.3.0)
##  tinytex           * 0.50    2024-03-16 [2] CRAN (R 4.3.2)
##  uuid                1.2-0   2024-01-14 [2] CRAN (R 4.3.0)
##  viridisLite         0.4.2   2023-05-02 [2] CRAN (R 4.3.0)
##  webshot           * 0.5.5   2023-06-26 [1] CRAN (R 4.3.0)
##  webshot2          * 0.1.1   2023-08-11 [1] CRAN (R 4.3.0)
##  websocket           1.4.1   2021-08-18 [1] CRAN (R 4.3.0)
##  xfun                0.43    2024-03-25 [2] CRAN (R 4.3.2)
##  xml2                1.3.6   2023-12-04 [2] CRAN (R 4.3.0)
##  xtable              1.8-4   2019-04-21 [2] CRAN (R 4.3.0)
##  yaml                2.3.8   2023-12-11 [2] CRAN (R 4.3.0)
##  zip                 2.3.1   2024-01-27 [2] CRAN (R 4.3.2)
## 
##  [1] /Users/slaan3/Library/R/x86_64/4.3/library
##  [2] /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────</code></pre>
<!-- ```{js, echo = FALSE} -->
<!-- title=document.getElementById('header'); -->
<!-- title.innerHTML = '<img src="img/_headers/banner_man_standing_dna.png" alt="Colofon">' + title.innerHTML -->
<!-- ``` -->
<!-- example: https://yearbookdiscoveries.com/wp-content/uploads/2014/05/Writing_a_Yearbook_Colophon.pdf -->
<!-- SPECIAL THANKS: (The staff wrote a few paragraphs about the year and mentioned a variety -->
<!-- of people who were instrumental in the success of their yearbook.) -->
<!-- COVER & ENDSHEETS: The 2013 Pinnacle cover is a four-color lithograph. An iridescent foil -->
<!-- covers a portion of the theme design. The endsheets are standard stock paper. The theme -->
<!-- concept was created and expanded by the editorial team and members of the 2013 Pinnacle -->
<!-- staff. Cover and endsheets were designed by Pinnacle co-editors-in-chief Regan Brown and -->
<!-- Ellena Sullivan, with inspiration provided by an early design from co-reference editor -->
<!-- Amanda Farrer. -->
<!-- TYPE & COLOR TREATMENT: Body copy throughout the book is set in Frutiger Light -->
<!-- Condensed (8.5 pt.) Captions are set in Frutiger Light Condensed (7.5 pt.) Headline -->
<!-- treatments are designed with variations of AHJ Nashville, Arno Pro and Frutiger. Photo -->
<!-- credits and spread credits appear in Frutiger Italic (6 pt.) -->
<!-- For consistency, a color palette was chosen. In addition to the traditional black, the -->
<!-- following colors appear throughout the publication: Pantone 151C, Pantone 3005C, Panton -->
<!-- 2985C, Pantone 115C, Pantone 376C, Pantone 363C, Pantone 266C, Pantone 185C and -->
<!-- Pantone Cool Grey 7C. -->
<!-- PUBLISHING: Volume 107 of the Pinnacle was designed and produced by the 2013 Pinnacle -->
<!-- staff. The 456-page, all-color Pinnacle is printed on 80 lb. gloss paper by Herff Jones -->
<!-- Publishing Co. in Kansas City, MO. Approximately 3,000 copies were pre-ordered for $52. -->
<!-- Any extra copies were sold for $60. A 48-page supplement was included in this price. The -->
<!-- publication was created using Adobe CS5.5 software on 42 Macintosh desktop and laptop -->
<!-- computers. -->
<!-- PHOTOGRAPHY: Pinnacle staff photographers shot digital photos using four Nikon D70s, -->
<!-- two Nikon D80s and one Nikon D40. Sport group photos were shot by Prestige Portraits, -->
<!-- and club and group photos were shot by both Prestige Portraits and Pinnacle yearbook staff -->
<!-- photographers. Some submitted photos appear throughout the book as well. -->
<!-- EDITORS’ NOTE: (A special note from the co-editors-in-chief was included here. The yearbook -->
<!-- staff photo with names and staff positions was included on the spread with the colophon.) -->

</div>
            </section>

          </div>
        </div>
      </div>
<a href="license.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"info": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
