<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>23 Colophon | A Practical Primer in Human Complex Genetics for CoCalc</title>
  <meta name="description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="23 Colophon | A Practical Primer in Human Complex Genetics for CoCalc" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/img/cover/cover_2024.png" />
  <meta property="og:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="23 Colophon | A Practical Primer in Human Complex Genetics for CoCalc" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="twitter:image" content="/img/cover/cover_2024.png" />

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  <link rel="apple-touch-icon-precomposed" sizes="180x180" href="img/cover/iconified/apple-touch-icon-180x180.png" />
  <link rel="shortcut icon" href="img/cover/iconified/favicon.ico" type="image/x-icon" />
<link rel="prev" href="license.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a></li>
<li class="chapter" data-level="4" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html"><i class="fa fa-check"></i><b>4</b> Getting started - programs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#plink"><i class="fa fa-check"></i><b>4.2</b> PLINK</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#alternatives-to-plink"><i class="fa fa-check"></i><b>4.2.1</b> Alternatives to <code>PLINK</code></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#other-programs"><i class="fa fa-check"></i><b>4.3</b> Other programs</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html"><i class="fa fa-check"></i><b>5</b> Getting started - CoCalc</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#starting-the-course-on-cocalc"><i class="fa fa-check"></i><b>5.1</b> Starting the course on CoCalc</a></li>
<li class="chapter" data-level="5.2" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#but-wait-what-is-cocalc-anyway"><i class="fa fa-check"></i><b>5.2</b> But wait, what is CoCalc anyway?</a></li>
<li class="chapter" data-level="5.3" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#okay-and-whats-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.3</b> Okay, and what’s a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.4" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-does-a-jupyter-notebook-work"><i class="fa fa-check"></i><b>5.4</b> How does a Jupyter Notebook work?</a></li>
<li class="chapter" data-level="5.5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#and-what-is-the-force-behind-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.5</b> And what is the Force behind a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.6" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-to-use-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.6</b> How to use a Jupyter Notebook</a></li>
<li class="chapter" data-level="5.7" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#trouble-shooting-juptyer-notebook"><i class="fa fa-check"></i><b>5.7</b> Trouble shooting Juptyer Notebook</a></li>
<li class="chapter" data-level="5.8" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#cocalc-and-this-practical-primer"><i class="fa fa-check"></i><b>5.8</b> CoCalc and this Practical Primer</a></li>
<li class="chapter" data-level="5.9" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-jupyter-notebook-with-a-r-kernel"><i class="fa fa-check"></i><b>5.9</b> Beginning your own Jupyter Notebook with a <code>R</code> kernel</a></li>
<li class="chapter" data-level="5.10" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-terminal-for-bash-code"><i class="fa fa-check"></i><b>5.10</b> Beginning your own Terminal for <code>bash</code> code</a></li>
<li class="chapter" data-level="5.11" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#this-book-and-cocalc"><i class="fa fa-check"></i><b>5.11</b> This book and CoCalc</a></li>
<li class="chapter" data-level="5.12" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#are-you-ready"><i class="fa fa-check"></i><b>5.12</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html"><i class="fa fa-check"></i><b>6</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#setting-the-stage"><i class="fa fa-check"></i><b>6.1.1</b> Setting the stage</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#converting-the-data"><i class="fa fa-check"></i><b>6.2</b> Converting the data</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#quality-control"><i class="fa fa-check"></i><b>6.3</b> Quality control</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>6.4</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html"><i class="fa fa-check"></i><b>7</b> Sample QC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sex"><i class="fa fa-check"></i><b>7.1</b> Sex</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sample-call-rates"><i class="fa fa-check"></i><b>7.2</b> Sample call rates</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>7.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="7.4" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#relatedness"><i class="fa fa-check"></i><b>7.4</b> Relatedness</a></li>
<li class="chapter" data-level="7.5" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#ancestral-background"><i class="fa fa-check"></i><b>7.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#g-phase-1"><i class="fa fa-check"></i><b>7.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="7.5.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#principal-component-analysis"><i class="fa fa-check"></i><b>7.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="7.5.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>7.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#removing-samples"><i class="fa fa-check"></i><b>7.6</b> Removing samples</a></li>
<li class="chapter" data-level="7.7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#the-next-step"><i class="fa fa-check"></i><b>7.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html"><i class="fa fa-check"></i><b>8</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="8.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#snp-call-rates"><i class="fa fa-check"></i><b>8.1</b> SNP call rates</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>8.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#allele-frequencies"><i class="fa fa-check"></i><b>8.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>8.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>8.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="8.5" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#final-snp-qc"><i class="fa fa-check"></i><b>8.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="8.6" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-milestone"><i class="fa fa-check"></i><b>8.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html"><i class="fa fa-check"></i><b>9</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#exploring-the-data"><i class="fa fa-check"></i><b>9.1</b> Exploring the data</a></li>
<li class="chapter" data-level="9.2" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#genetic-models"><i class="fa fa-check"></i><b>9.2</b> Genetic models</a></li>
<li class="chapter" data-level="9.3" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#logistic-regression"><i class="fa fa-check"></i><b>9.3</b> Logistic regression</a></li>
<li class="chapter" data-level="9.4" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#lets-get-visual"><i class="fa fa-check"></i><b>9.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html"><i class="fa fa-check"></i><b>10</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#qq-plots"><i class="fa fa-check"></i><b>10.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="10.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#manhattan-plots"><i class="fa fa-check"></i><b>10.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="10.2" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#other-plots"><i class="fa fa-check"></i><b>10.2</b> Other plots</a></li>
<li class="chapter" data-level="10.3" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#interactive-plots"><i class="fa fa-check"></i><b>10.3</b> Interactive plots</a></li>
<li class="chapter" data-level="10.4" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#getting-up-close"><i class="fa fa-check"></i><b>10.4</b> Getting up close</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html"><i class="fa fa-check"></i><b>11</b> Regional association plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#locuszoom"><i class="fa fa-check"></i><b>11.1</b> LocusZoom</a></li>
<li class="chapter" data-level="11.2" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#racer"><i class="fa fa-check"></i><b>11.2</b> RACER</a></li>
<li class="chapter" data-level="11.3" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#stop-playing-around"><i class="fa fa-check"></i><b>11.3</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>12</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="12.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>12.1</b> Genotyping</a></li>
<li class="chapter" data-level="12.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>12.2</b> The data</a></li>
<li class="chapter" data-level="12.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>12.3</b> Assignment</a></li>
<li class="chapter" data-level="12.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>12.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html"><i class="fa fa-check"></i><b>13</b> WTCCC1: quality control</a>
<ul>
<li class="chapter" data-level="13.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#quality-control-1"><i class="fa fa-check"></i><b>13.1</b> Quality control</a></li>
<li class="chapter" data-level="13.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#ancestral-background-1"><i class="fa fa-check"></i><b>13.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#preparing-pca"><i class="fa fa-check"></i><b>13.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="13.2.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#running-pca"><i class="fa fa-check"></i><b>13.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="13.2.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#plotting-pca"><i class="fa fa-check"></i><b>13.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="13.2.4" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#removing-samples-1"><i class="fa fa-check"></i><b>13.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#summary"><i class="fa fa-check"></i><b>13.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html"><i class="fa fa-check"></i><b>14</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#association-testing"><i class="fa fa-check"></i><b>14.1</b> Association testing</a></li>
<li class="chapter" data-level="14.2" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#independent-hits"><i class="fa fa-check"></i><b>14.2</b> Independent hits</a></li>
<li class="chapter" data-level="14.3" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#replication"><i class="fa fa-check"></i><b>14.3</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>15</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="15.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>15.1</b> Clumping</a></li>
<li class="chapter" data-level="15.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>15.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="15.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>15.3</b> Statistical finemapping</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="post-gwas.html"><a href="post-gwas.html#finemap"><i class="fa fa-check"></i><b>15.3.1</b> FINEMAP</a></li>
<li class="chapter" data-level="15.3.2" data-path="post-gwas.html"><a href="post-gwas.html#locuszoom-1"><i class="fa fa-check"></i><b>15.3.2</b> LocusZoom</a></li>
<li class="chapter" data-level="15.3.3" data-path="post-gwas.html"><a href="post-gwas.html#susie"><i class="fa fa-check"></i><b>15.3.3</b> SuSIE</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>15.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="15.5" data-path="post-gwas.html"><a href="post-gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>15.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="15.6" data-path="post-gwas.html"><a href="post-gwas.html#colocalization"><i class="fa fa-check"></i><b>15.6</b> Colocalization</a></li>
<li class="chapter" data-level="15.7" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>15.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="15.8" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>15.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="post-gwas.html"><a href="post-gwas.html#mrbase"><i class="fa fa-check"></i><b>15.8.1</b> MRBase</a></li>
<li class="chapter" data-level="15.8.2" data-path="post-gwas.html"><a href="post-gwas.html#twosamplemr"><i class="fa fa-check"></i><b>15.8.2</b> TwoSampleMR</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>15.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="15.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>15.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>16</b> Functional Mapping and Annotation of GWAS</a>
<ul>
<li class="chapter" data-level="16.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>16.1</b> Assignment</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>16.1.1</b> Tutorial</a></li>
<li class="chapter" data-level="16.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>16.1.2</b> Create the input</a></li>
<li class="chapter" data-level="16.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>16.1.3</b> Run FUMA - SNP to gene</a></li>
<li class="chapter" data-level="16.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>16.1.4</b> Run FUMA - Gene to function</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>16.2</b> Some closing thoughts</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>17</b> Phenome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="17.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>17.1</b> What’s next?</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>18</b> MRBase</a>
<ul>
<li class="chapter" data-level="18.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>18.1</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html"><i class="fa fa-check"></i><b>19</b> Two Sample Mendelian Randomisation</a>
<ul>
<li class="chapter" data-level="19.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#loading-the-package"><i class="fa fa-check"></i><b>19.1</b> Loading the package</a></li>
<li class="chapter" data-level="19.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#available-outcome"><i class="fa fa-check"></i><b>19.2</b> Available outcome</a></li>
<li class="chapter" data-level="19.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#instruments"><i class="fa fa-check"></i><b>19.3</b> Instruments</a></li>
<li class="chapter" data-level="19.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#clumping-1"><i class="fa fa-check"></i><b>19.4</b> Clumping</a></li>
<li class="chapter" data-level="19.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#outcomes"><i class="fa fa-check"></i><b>19.5</b> Outcomes</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#harmonisation"><i class="fa fa-check"></i><b>19.5.1</b> Harmonisation</a></li>
<li class="chapter" data-level="19.5.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#mendelian-randomisation"><i class="fa fa-check"></i><b>19.5.2</b> Mendelian randomisation</a></li>
<li class="chapter" data-level="19.5.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#heterogeneity-pleiotropy"><i class="fa fa-check"></i><b>19.5.3</b> Heterogeneity &amp; Pleiotropy</a></li>
<li class="chapter" data-level="19.5.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#leave-one-out"><i class="fa fa-check"></i><b>19.5.4</b> Leave-one-out</a></li>
<li class="chapter" data-level="19.5.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#visualisation"><i class="fa fa-check"></i><b>19.5.5</b> Visualisation</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#the-end-1"><i class="fa fa-check"></i><b>19.6</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>20</b> Epilogue</a>
<ul>
<li class="chapter" data-level="20.1" data-path="epilogue.html"><a href="epilogue.html#course-lecturers"><i class="fa fa-check"></i><b>20.1</b> 2024 Course Lecturers</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="answers.html"><a href="answers.html"><i class="fa fa-check"></i><b>21</b> Answers to the questions</a>
<ul>
<li class="chapter" data-level="21.1" data-path="answers.html"><a href="answers.html#chapter-3.1-gwas-basics"><i class="fa fa-check"></i><b>21.1</b> Chapter 3.1: GWAS Basics</a></li>
<li class="chapter" data-level="21.2" data-path="answers.html"><a href="answers.html#chapter-3.4-gwas-basics---association-testing"><i class="fa fa-check"></i><b>21.2</b> Chapter 3.4: GWAS Basics - Association testing</a></li>
<li class="chapter" data-level="21.3" data-path="answers.html"><a href="answers.html#chapter-3.5-gwas-visualization"><i class="fa fa-check"></i><b>21.3</b> Chapter 3.5: GWAS visualization</a></li>
<li class="chapter" data-level="21.4" data-path="answers.html"><a href="answers.html#chapter-3.6-regional-association-plots"><i class="fa fa-check"></i><b>21.4</b> Chapter 3.6: Regional association plots</a></li>
<li class="chapter" data-level="21.5" data-path="answers.html"><a href="answers.html#chapter-4.2-wtccc1"><i class="fa fa-check"></i><b>21.5</b> Chapter 4.2: WTCCC1</a></li>
<li class="chapter" data-level="21.6" data-path="answers.html"><a href="answers.html#chapter-4.3-wtccc1---association-testing"><i class="fa fa-check"></i><b>21.6</b> Chapter 4.3: WTCCC1 - Association testing</a></li>
<li class="chapter" data-level="21.7" data-path="answers.html"><a href="answers.html#chapter-5.4-fuma"><i class="fa fa-check"></i><b>21.7</b> Chapter 5.4: FUMA</a></li>
<li class="chapter" data-level="21.8" data-path="answers.html"><a href="answers.html#chapter-5.5-phewas"><i class="fa fa-check"></i><b>21.8</b> Chapter 5.5: PheWAS</a></li>
<li class="chapter" data-level="21.9" data-path="answers.html"><a href="answers.html#chapter-5.7-twosample-mr"><i class="fa fa-check"></i><b>21.9</b> Chapter 5.7: TwoSample MR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>22</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="22.1" data-path="license.html"><a href="license.html#images-and-data-used"><i class="fa fa-check"></i><b>22.1</b> Images and data used</a></li>
<li class="chapter" data-level="22.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>22.2</b> Disclaimer</a></li>
<li class="chapter" data-level="22.3" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>22.3</b> Copyright</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="colophon.html"><a href="colophon.html"><i class="fa fa-check"></i><b>23</b> Colophon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://vanderlaanand.science" target="blank">&copy; 1979-2024. Sander W. van der Laan.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics for CoCalc</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="colophon" class="section level1 hasAnchor" number="23">
<h1><span class="header-section-number">23</span> Colophon<a href="colophon.html#colophon" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The 2022 and 2024 editions of this book were produce in RStudio and with the <code>bookdown</code> package. Below a listing of installed programs and libraries, the operating system, and their specific versions.</p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="colophon.html#cb173-1" tabindex="-1"></a>sessioninfo<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.3.3 (2024-02-29)
##  os       macOS Sonoma 14.5
##  system   x86_64, darwin20
##  ui       X11
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       America/New_York
##  date     2024-04-10
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package           * version     date (UTC) lib source
##  abind               1.4-5       2016-07-21 [2] CRAN (R 4.3.0)
##  annotate          * 1.80.0      2023-10-24 [2] Bioconductor
##  AnnotationDbi     * 1.64.1      2023-11-03 [2] Bioconductor
##  askpass             1.2.0       2023-09-03 [2] CRAN (R 4.3.0)
##  backports           1.4.1       2021-12-13 [2] CRAN (R 4.3.0)
##  Biobase           * 2.62.0      2023-10-24 [2] Bioconductor
##  BiocGenerics      * 0.48.1      2023-11-01 [2] Bioconductor
##  Biostrings          2.70.3      2024-03-13 [2] Bioconductor 3.18 (R 4.3.3)
##  bit                 4.0.5       2022-11-15 [2] CRAN (R 4.3.0)
##  bit64               4.0.5       2020-08-30 [2] CRAN (R 4.3.0)
##  bitops              1.0-7       2021-04-24 [2] CRAN (R 4.3.0)
##  blob                1.2.4       2023-03-17 [2] CRAN (R 4.3.0)
##  bookdown          * 0.38.1      2024-03-26 [2] Github (rstudio/bookdown@50a1c1e)
##  broom               1.0.5       2023-06-09 [2] CRAN (R 4.3.0)
##  bslib               0.6.2       2024-03-22 [2] CRAN (R 4.3.2)
##  cachem              1.0.8       2023-05-01 [2] CRAN (R 4.3.0)
##  calibrate           1.7.7       2020-06-19 [2] CRAN (R 4.3.0)
##  car                 3.1-2       2023-03-30 [2] CRAN (R 4.3.0)
##  carData             3.0-5       2022-01-06 [2] CRAN (R 4.3.0)
##  chromote            0.2.0       2024-02-12 [1] CRAN (R 4.3.2)
##  cli                 3.6.2       2023-12-11 [2] CRAN (R 4.3.0)
##  CMplot            * 4.5.1       2024-01-19 [2] CRAN (R 4.3.0)
##  colorspace          2.1-0       2023-01-23 [2] CRAN (R 4.3.0)
##  crayon              1.5.2       2022-09-29 [2] CRAN (R 4.3.0)
##  crul                1.4.0       2023-05-17 [2] CRAN (R 4.3.0)
##  curl                5.2.1       2024-03-01 [2] CRAN (R 4.3.2)
##  data.table        * 1.15.4      2024-03-30 [1] CRAN (R 4.3.2)
##  DBI                 1.2.2       2024-02-16 [2] CRAN (R 4.3.2)
##  devtools          * 2.4.5       2022-10-11 [2] CRAN (R 4.3.0)
##  digest              0.6.35      2024-03-11 [2] CRAN (R 4.3.2)
##  dplyr             * 1.1.4       2023-11-17 [2] CRAN (R 4.3.0)
##  DT                * 0.33        2024-04-04 [1] CRAN (R 4.3.2)
##  ellipsis            0.3.2       2021-04-29 [2] CRAN (R 4.3.0)
##  evaluate            0.23        2023-11-01 [2] CRAN (R 4.3.0)
##  fansi               1.0.6       2023-12-08 [2] CRAN (R 4.3.0)
##  fastmap             1.1.1       2023-02-24 [2] CRAN (R 4.3.0)
##  flextable         * 0.9.5       2024-03-06 [1] CRAN (R 4.3.2)
##  fontBitstreamVera   0.1.1       2017-02-01 [2] CRAN (R 4.3.0)
##  fontLiberation      0.1.0       2016-10-15 [2] CRAN (R 4.3.0)
##  fontquiver          0.2.1       2017-02-01 [2] CRAN (R 4.3.0)
##  forcats           * 1.0.0       2023-01-29 [2] CRAN (R 4.3.0)
##  foreign             0.8-86      2023-11-28 [2] CRAN (R 4.3.3)
##  formatR           * 1.14        2023-01-17 [2] CRAN (R 4.3.0)
##  fs                  1.6.3       2023-07-20 [2] CRAN (R 4.3.0)
##  gdtools             0.3.7       2024-03-05 [2] CRAN (R 4.3.2)
##  geneplotter       * 1.80.0      2023-10-24 [2] Bioconductor
##  generics            0.1.3       2022-07-05 [2] CRAN (R 4.3.0)
##  GenomeInfoDb        1.38.8      2024-03-15 [2] Bioconductor 3.18 (R 4.3.3)
##  GenomeInfoDbData    1.2.11      2024-03-26 [2] Bioconductor
##  gfonts              0.2.0       2023-01-08 [2] CRAN (R 4.3.0)
##  ggplot2           * 3.5.0       2024-02-23 [2] CRAN (R 4.3.2)
##  ggpubr            * 0.6.0.999   2024-03-26 [2] Github (kassambara/ggpubr@6aeb4f7)
##  ggsci             * 3.0.3       2024-03-25 [2] CRAN (R 4.3.2)
##  ggsignif            0.6.4       2022-10-13 [2] CRAN (R 4.3.0)
##  ggthemes          * 5.1.0       2024-02-10 [2] CRAN (R 4.3.2)
##  glue                1.7.0       2024-01-09 [2] CRAN (R 4.3.0)
##  gtable              0.3.4       2023-08-21 [2] CRAN (R 4.3.0)
##  hms                 1.1.3       2023-03-21 [2] CRAN (R 4.3.0)
##  htmltools           0.5.8       2024-03-25 [2] CRAN (R 4.3.2)
##  htmlwidgets         1.6.4       2023-12-06 [2] CRAN (R 4.3.0)
##  httpcode            0.3.0       2020-04-10 [2] CRAN (R 4.3.0)
##  httpuv              1.6.15      2024-03-26 [2] CRAN (R 4.3.2)
##  httr                1.4.7       2023-08-15 [2] CRAN (R 4.3.0)
##  ieugwasr          * 0.2.2-9000  2024-04-05 [1] Github (MRCIEU/ieugwasr@52e03a7)
##  IRanges           * 2.36.0      2023-10-24 [2] Bioconductor
##  jquerylib           0.1.4       2021-04-26 [2] CRAN (R 4.3.0)
##  jsonlite            1.8.8       2023-12-04 [2] CRAN (R 4.3.0)
##  kableExtra        * 1.4.0       2024-01-24 [1] CRAN (R 4.3.2)
##  KEGGREST            1.42.0      2023-10-24 [2] Bioconductor
##  knitr             * 1.45        2023-10-30 [1] CRAN (R 4.3.0)
##  later               1.3.2       2023-12-06 [2] CRAN (R 4.3.0)
##  lattice           * 0.22-6      2024-03-20 [2] CRAN (R 4.3.2)
##  lazyeval            0.2.2       2019-03-15 [2] CRAN (R 4.3.0)
##  lifecycle           1.0.4       2023-11-07 [2] CRAN (R 4.3.0)
##  lubridate         * 1.9.3       2023-09-27 [2] CRAN (R 4.3.0)
##  magrittr            2.0.3       2022-03-30 [2] CRAN (R 4.3.0)
##  MASS                7.3-60.0.1  2024-01-13 [2] CRAN (R 4.3.3)
##  Matrix              1.6-5       2024-01-11 [2] CRAN (R 4.3.3)
##  memoise             2.0.1       2021-11-26 [2] CRAN (R 4.3.0)
##  mime                0.12        2021-09-28 [2] CRAN (R 4.3.0)
##  miniUI              0.1.1.1     2018-05-18 [2] CRAN (R 4.3.0)
##  mitools             2.4         2019-04-26 [1] CRAN (R 4.3.0)
##  MRInstruments     * 0.3.2       2024-03-27 [1] Github (MRCIEU/MRInstruments@efa2ca0)
##  munsell             0.5.1       2024-04-01 [1] CRAN (R 4.3.2)
##  officer             0.6.5       2024-02-24 [2] CRAN (R 4.3.2)
##  openssl             2.1.1       2023-09-25 [2] CRAN (R 4.3.0)
##  openxlsx          * 4.2.5.2     2023-02-06 [2] CRAN (R 4.3.0)
##  pillar              1.9.0       2023-03-22 [2] CRAN (R 4.3.0)
##  pkgbuild            1.4.4       2024-03-17 [2] CRAN (R 4.3.2)
##  pkgconfig           2.0.3       2019-09-22 [2] CRAN (R 4.3.0)
##  pkgload             1.3.4       2024-01-16 [2] CRAN (R 4.3.0)
##  plotly            * 4.10.4      2024-01-13 [2] CRAN (R 4.3.0)
##  png                 0.1-8       2022-11-29 [2] CRAN (R 4.3.0)
##  processx            3.8.4       2024-03-16 [2] CRAN (R 4.3.2)
##  profvis             0.3.8       2023-05-02 [2] CRAN (R 4.3.0)
##  promises            1.2.1       2023-08-10 [2] CRAN (R 4.3.0)
##  ps                  1.7.6       2024-01-18 [2] CRAN (R 4.3.0)
##  purrr             * 1.0.2       2023-08-10 [2] CRAN (R 4.3.0)
##  qqman             * 0.1.9       2023-08-23 [2] CRAN (R 4.3.0)
##  R6                  2.5.1       2021-08-19 [2] CRAN (R 4.3.0)
##  RACER             * 1.0.0       2024-03-26 [2] Github (oliviasabik/RACER@1394c9d)
##  ragg                1.3.0       2024-03-13 [2] CRAN (R 4.3.2)
##  RColorBrewer        1.1-3       2022-04-03 [2] CRAN (R 4.3.0)
##  Rcpp                1.0.12      2024-01-09 [2] CRAN (R 4.3.0)
##  RCurl               1.98-1.14   2024-01-09 [2] CRAN (R 4.3.0)
##  readr             * 2.1.5       2024-01-10 [2] CRAN (R 4.3.0)
##  remotes           * 2.5.0       2024-03-17 [2] CRAN (R 4.3.2)
##  rlang               1.1.3       2024-01-10 [2] CRAN (R 4.3.0)
##  rmarkdown         * 2.26.1      2024-03-26 [2] Github (rstudio/rmarkdown@ee69d59)
##  RNHANES           * 1.1.1       2024-03-27 [1] Github (silentspringinstitute/RNHANES@860fc94)
##  RSQLite             2.3.5       2024-01-21 [2] CRAN (R 4.3.0)
##  rstatix             0.7.2       2023-02-01 [2] CRAN (R 4.3.0)
##  rstudioapi          0.16.0      2024-03-24 [2] CRAN (R 4.3.2)
##  rvest               1.0.4       2024-02-12 [2] CRAN (R 4.3.2)
##  S4Vectors         * 0.40.2      2023-11-23 [2] Bioconductor
##  sass                0.4.9       2024-03-15 [2] CRAN (R 4.3.2)
##  scales              1.3.0       2023-11-28 [2] CRAN (R 4.3.0)
##  sessioninfo         1.2.2       2021-12-06 [2] CRAN (R 4.3.0)
##  shiny               1.8.1       2024-03-26 [2] CRAN (R 4.3.2)
##  stringi             1.8.3       2023-12-11 [2] CRAN (R 4.3.0)
##  stringr           * 1.5.1       2023-11-14 [2] CRAN (R 4.3.0)
##  survey              4.4-2       2024-03-20 [1] CRAN (R 4.3.2)
##  survival            3.5-8       2024-02-14 [2] CRAN (R 4.3.3)
##  svglite             2.1.3       2023-12-08 [1] CRAN (R 4.3.0)
##  systemfonts         1.0.6       2024-03-07 [2] CRAN (R 4.3.2)
##  textshaping         0.3.7       2023-10-09 [2] CRAN (R 4.3.0)
##  tibble            * 3.2.1       2023-03-20 [2] CRAN (R 4.3.0)
##  tidyr             * 1.3.1       2024-01-24 [2] CRAN (R 4.3.2)
##  tidyselect          1.2.1       2024-03-11 [2] CRAN (R 4.3.2)
##  tidyverse         * 2.0.0       2023-02-22 [2] CRAN (R 4.3.0)
##  timechange          0.3.0       2024-01-18 [2] CRAN (R 4.3.0)
##  tinytex           * 0.50        2024-03-16 [2] CRAN (R 4.3.2)
##  TwoSampleMR       * 0.5.11      2024-03-26 [2] Github (MRCIEU/TwoSampleMR@ebfeb95)
##  tzdb                0.4.0       2023-05-12 [2] CRAN (R 4.3.0)
##  urlchecker          1.0.1       2021-11-30 [2] CRAN (R 4.3.0)
##  usethis           * 2.2.3       2024-02-19 [2] CRAN (R 4.3.2)
##  utf8                1.2.4       2023-10-22 [2] CRAN (R 4.3.0)
##  uuid                1.2-0       2024-01-14 [2] CRAN (R 4.3.0)
##  vctrs               0.6.5       2023-12-01 [2] CRAN (R 4.3.0)
##  viridisLite         0.4.2       2023-05-02 [2] CRAN (R 4.3.0)
##  webshot           * 0.5.5       2023-06-26 [1] CRAN (R 4.3.0)
##  webshot2          * 0.1.1       2023-08-11 [1] CRAN (R 4.3.0)
##  websocket           1.4.1       2021-08-18 [1] CRAN (R 4.3.0)
##  withr               3.0.0       2024-01-16 [2] CRAN (R 4.3.0)
##  xfun                0.43        2024-03-25 [2] CRAN (R 4.3.2)
##  XML               * 3.99-0.16.1 2024-01-22 [2] CRAN (R 4.3.2)
##  xml2              * 1.3.6       2023-12-04 [2] CRAN (R 4.3.0)
##  xtable              1.8-4       2019-04-21 [2] CRAN (R 4.3.0)
##  XVector             0.42.0      2023-10-24 [2] Bioconductor
##  yaml                2.3.8       2023-12-11 [2] CRAN (R 4.3.0)
##  zip                 2.3.1       2024-01-27 [2] CRAN (R 4.3.2)
##  zlibbioc            1.48.2      2024-03-13 [2] Bioconductor 3.18 (R 4.3.3)
## 
##  [1] /Users/slaan3/Library/R/x86_64/4.3/library
##  [2] /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────</code></pre>
<script>
title=document.getElementById('header');
title.innerHTML = '<img src="img/headers/banner_man_standing_dna.png" alt="Colofon">' + title.innerHTML
</script>
<!-- example: https://yearbookdiscoveries.com/wp-content/uploads/2014/05/Writing_a_Yearbook_Colophon.pdf -->
<!-- SPECIAL THANKS: (The staff wrote a few paragraphs about the year and mentioned a variety -->
<!-- of people who were instrumental in the success of their yearbook.) -->
<!-- COVER & ENDSHEETS: The 2013 Pinnacle cover is a four-color lithograph. An iridescent foil -->
<!-- covers a portion of the theme design. The endsheets are standard stock paper. The theme -->
<!-- concept was created and expanded by the editorial team and members of the 2013 Pinnacle -->
<!-- staff. Cover and endsheets were designed by Pinnacle co-editors-in-chief Regan Brown and -->
<!-- Ellena Sullivan, with inspiration provided by an early design from co-reference editor -->
<!-- Amanda Farrer. -->
<!-- TYPE & COLOR TREATMENT: Body copy throughout the book is set in Frutiger Light -->
<!-- Condensed (8.5 pt.) Captions are set in Frutiger Light Condensed (7.5 pt.) Headline -->
<!-- treatments are designed with variations of AHJ Nashville, Arno Pro and Frutiger. Photo -->
<!-- credits and spread credits appear in Frutiger Italic (6 pt.) -->
<!-- For consistency, a color palette was chosen. In addition to the traditional black, the -->
<!-- following colors appear throughout the publication: Pantone 151C, Pantone 3005C, Panton -->
<!-- 2985C, Pantone 115C, Pantone 376C, Pantone 363C, Pantone 266C, Pantone 185C and -->
<!-- Pantone Cool Grey 7C. -->
<!-- PUBLISHING: Volume 107 of the Pinnacle was designed and produced by the 2013 Pinnacle -->
<!-- staff. The 456-page, all-color Pinnacle is printed on 80 lb. gloss paper by Herff Jones -->
<!-- Publishing Co. in Kansas City, MO. Approximately 3,000 copies were pre-ordered for $52. -->
<!-- Any extra copies were sold for $60. A 48-page supplement was included in this price. The -->
<!-- publication was created using Adobe CS5.5 software on 42 Macintosh desktop and laptop -->
<!-- computers. -->
<!-- PHOTOGRAPHY: Pinnacle staff photographers shot digital photos using four Nikon D70s, -->
<!-- two Nikon D80s and one Nikon D40. Sport group photos were shot by Prestige Portraits, -->
<!-- and club and group photos were shot by both Prestige Portraits and Pinnacle yearbook staff -->
<!-- photographers. Some submitted photos appear throughout the book as well. -->
<!-- EDITORS’ NOTE: (A special note from the co-editors-in-chief was included here. The yearbook -->
<!-- staff photo with names and staff positions was included on the spread with the colophon.) -->

</div>
            </section>

          </div>
        </div>
      </div>
<a href="license.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans-serif",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"always_allow_html": true,
"split_by": "chapter",
"toolbar": {
"position": "fixed"
},
"info": true,
"code_folding": "show"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
