<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Getting started - programs | A Practical Primer in Human Complex Genetics for CoCalc</title>
  <meta name="description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Getting started - programs | A Practical Primer in Human Complex Genetics for CoCalc" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/img/cover/cover_2024.png" />
  <meta property="og:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Getting started - programs | A Practical Primer in Human Complex Genetics for CoCalc" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="twitter:image" content="/img/cover/cover_2024.png" />

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  <link rel="apple-touch-icon-precomposed" sizes="180x180" href="img/cover/iconified/apple-touch-icon-180x180.png" />
  <link rel="shortcut icon" href="img/cover/iconified/favicon.ico" type="image/x-icon" />
<link rel="prev" href="getting-started.html"/>
<link rel="next" href="getting-started-cocalc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a></li>
<li class="chapter" data-level="4" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html"><i class="fa fa-check"></i><b>4</b> Getting started - programs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#plink"><i class="fa fa-check"></i><b>4.2</b> PLINK</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#alternatives-to-plink"><i class="fa fa-check"></i><b>4.2.1</b> Alternatives to <code>PLINK</code></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#other-programs"><i class="fa fa-check"></i><b>4.3</b> Other programs</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html"><i class="fa fa-check"></i><b>5</b> Getting started - CoCalc</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#starting-the-course-on-cocalc"><i class="fa fa-check"></i><b>5.1</b> Starting the course on CoCalc</a></li>
<li class="chapter" data-level="5.2" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#but-wait-what-is-cocalc-anyway"><i class="fa fa-check"></i><b>5.2</b> But wait, what is CoCalc anyway?</a></li>
<li class="chapter" data-level="5.3" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#okay-and-whats-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.3</b> Okay, and what’s a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.4" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-does-a-jupyter-notebook-work"><i class="fa fa-check"></i><b>5.4</b> How does a Jupyter Notebook work?</a></li>
<li class="chapter" data-level="5.5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#and-what-is-the-force-behind-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.5</b> And what is the Force behind a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.6" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-to-use-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.6</b> How to use a Jupyter Notebook</a></li>
<li class="chapter" data-level="5.7" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#trouble-shooting-juptyer-notebook"><i class="fa fa-check"></i><b>5.7</b> Trouble shooting Juptyer Notebook</a></li>
<li class="chapter" data-level="5.8" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#cocalc-and-this-practical-primer"><i class="fa fa-check"></i><b>5.8</b> CoCalc and this Practical Primer</a></li>
<li class="chapter" data-level="5.9" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-jupyter-notebook-with-a-r-kernel"><i class="fa fa-check"></i><b>5.9</b> Beginning your own Jupyter Notebook with a <code>R</code> kernel</a></li>
<li class="chapter" data-level="5.10" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-terminal-for-bash-code"><i class="fa fa-check"></i><b>5.10</b> Beginning your own Terminal for <code>bash</code> code</a></li>
<li class="chapter" data-level="5.11" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#this-book-and-cocalc"><i class="fa fa-check"></i><b>5.11</b> This book and CoCalc</a></li>
<li class="chapter" data-level="5.12" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#are-you-ready"><i class="fa fa-check"></i><b>5.12</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html"><i class="fa fa-check"></i><b>6</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#setting-the-stage"><i class="fa fa-check"></i><b>6.1.1</b> Setting the stage</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#converting-the-data"><i class="fa fa-check"></i><b>6.2</b> Converting the data</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#quality-control"><i class="fa fa-check"></i><b>6.3</b> Quality control</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>6.4</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html"><i class="fa fa-check"></i><b>7</b> Sample QC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sex"><i class="fa fa-check"></i><b>7.1</b> Sex</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sample-call-rates"><i class="fa fa-check"></i><b>7.2</b> Sample call rates</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>7.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="7.4" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#relatedness"><i class="fa fa-check"></i><b>7.4</b> Relatedness</a></li>
<li class="chapter" data-level="7.5" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#ancestral-background"><i class="fa fa-check"></i><b>7.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#g-phase-1"><i class="fa fa-check"></i><b>7.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="7.5.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#principal-component-analysis"><i class="fa fa-check"></i><b>7.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="7.5.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>7.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#removing-samples"><i class="fa fa-check"></i><b>7.6</b> Removing samples</a></li>
<li class="chapter" data-level="7.7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#the-next-step"><i class="fa fa-check"></i><b>7.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html"><i class="fa fa-check"></i><b>8</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="8.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#snp-call-rates"><i class="fa fa-check"></i><b>8.1</b> SNP call rates</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>8.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#allele-frequencies"><i class="fa fa-check"></i><b>8.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>8.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>8.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="8.5" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#final-snp-qc"><i class="fa fa-check"></i><b>8.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="8.6" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-milestone"><i class="fa fa-check"></i><b>8.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html"><i class="fa fa-check"></i><b>9</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#exploring-the-data"><i class="fa fa-check"></i><b>9.1</b> Exploring the data</a></li>
<li class="chapter" data-level="9.2" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#genetic-models"><i class="fa fa-check"></i><b>9.2</b> Genetic models</a></li>
<li class="chapter" data-level="9.3" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#logistic-regression"><i class="fa fa-check"></i><b>9.3</b> Logistic regression</a></li>
<li class="chapter" data-level="9.4" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#lets-get-visual"><i class="fa fa-check"></i><b>9.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html"><i class="fa fa-check"></i><b>10</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#qq-plots"><i class="fa fa-check"></i><b>10.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="10.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#manhattan-plots"><i class="fa fa-check"></i><b>10.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="10.2" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#other-plots"><i class="fa fa-check"></i><b>10.2</b> Other plots</a></li>
<li class="chapter" data-level="10.3" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#interactive-plots"><i class="fa fa-check"></i><b>10.3</b> Interactive plots</a></li>
<li class="chapter" data-level="10.4" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#getting-up-close"><i class="fa fa-check"></i><b>10.4</b> Getting up close</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html"><i class="fa fa-check"></i><b>11</b> Regional association plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#locuszoom"><i class="fa fa-check"></i><b>11.1</b> LocusZoom</a></li>
<li class="chapter" data-level="11.2" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#racer"><i class="fa fa-check"></i><b>11.2</b> RACER</a></li>
<li class="chapter" data-level="11.3" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#stop-playing-around"><i class="fa fa-check"></i><b>11.3</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>12</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="12.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>12.1</b> Genotyping</a></li>
<li class="chapter" data-level="12.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>12.2</b> The data</a></li>
<li class="chapter" data-level="12.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>12.3</b> Assignment</a></li>
<li class="chapter" data-level="12.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>12.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html"><i class="fa fa-check"></i><b>13</b> WTCCC1: quality control</a>
<ul>
<li class="chapter" data-level="13.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#quality-control-1"><i class="fa fa-check"></i><b>13.1</b> Quality control</a></li>
<li class="chapter" data-level="13.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#ancestral-background-1"><i class="fa fa-check"></i><b>13.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#preparing-pca"><i class="fa fa-check"></i><b>13.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="13.2.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#running-pca"><i class="fa fa-check"></i><b>13.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="13.2.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#plotting-pca"><i class="fa fa-check"></i><b>13.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="13.2.4" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#removing-samples-1"><i class="fa fa-check"></i><b>13.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#summary"><i class="fa fa-check"></i><b>13.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html"><i class="fa fa-check"></i><b>14</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#association-testing"><i class="fa fa-check"></i><b>14.1</b> Association testing</a></li>
<li class="chapter" data-level="14.2" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#independent-hits"><i class="fa fa-check"></i><b>14.2</b> Independent hits</a></li>
<li class="chapter" data-level="14.3" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#replication"><i class="fa fa-check"></i><b>14.3</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>15</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="15.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>15.1</b> Clumping</a></li>
<li class="chapter" data-level="15.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>15.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="15.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>15.3</b> Statistical finemapping</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="post-gwas.html"><a href="post-gwas.html#finemap"><i class="fa fa-check"></i><b>15.3.1</b> FINEMAP</a></li>
<li class="chapter" data-level="15.3.2" data-path="post-gwas.html"><a href="post-gwas.html#locuszoom-1"><i class="fa fa-check"></i><b>15.3.2</b> LocusZoom</a></li>
<li class="chapter" data-level="15.3.3" data-path="post-gwas.html"><a href="post-gwas.html#susie"><i class="fa fa-check"></i><b>15.3.3</b> SuSIE</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>15.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="15.5" data-path="post-gwas.html"><a href="post-gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>15.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="15.6" data-path="post-gwas.html"><a href="post-gwas.html#colocalization"><i class="fa fa-check"></i><b>15.6</b> Colocalization</a></li>
<li class="chapter" data-level="15.7" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>15.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="15.8" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>15.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="post-gwas.html"><a href="post-gwas.html#mrbase"><i class="fa fa-check"></i><b>15.8.1</b> MRBase</a></li>
<li class="chapter" data-level="15.8.2" data-path="post-gwas.html"><a href="post-gwas.html#twosamplemr"><i class="fa fa-check"></i><b>15.8.2</b> TwoSampleMR</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>15.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="15.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>15.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>16</b> Functional Mapping and Annotation of GWAS</a>
<ul>
<li class="chapter" data-level="16.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>16.1</b> Assignment</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>16.1.1</b> Tutorial</a></li>
<li class="chapter" data-level="16.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>16.1.2</b> Create the input</a></li>
<li class="chapter" data-level="16.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>16.1.3</b> Run FUMA - SNP to gene</a></li>
<li class="chapter" data-level="16.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>16.1.4</b> Run FUMA - Gene to function</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>16.2</b> Some closing thoughts</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>17</b> Phenome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="17.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>17.1</b> What’s next?</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>18</b> MRBase</a>
<ul>
<li class="chapter" data-level="18.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>18.1</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html"><i class="fa fa-check"></i><b>19</b> Two Sample Mendelian Randomisation</a>
<ul>
<li class="chapter" data-level="19.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#loading-the-package"><i class="fa fa-check"></i><b>19.1</b> Loading the package</a></li>
<li class="chapter" data-level="19.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#available-outcome"><i class="fa fa-check"></i><b>19.2</b> Available outcome</a></li>
<li class="chapter" data-level="19.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#instruments"><i class="fa fa-check"></i><b>19.3</b> Instruments</a></li>
<li class="chapter" data-level="19.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#clumping-1"><i class="fa fa-check"></i><b>19.4</b> Clumping</a></li>
<li class="chapter" data-level="19.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#outcomes"><i class="fa fa-check"></i><b>19.5</b> Outcomes</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#harmonisation"><i class="fa fa-check"></i><b>19.5.1</b> Harmonisation</a></li>
<li class="chapter" data-level="19.5.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#mendelian-randomisation"><i class="fa fa-check"></i><b>19.5.2</b> Mendelian randomisation</a></li>
<li class="chapter" data-level="19.5.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#heterogeneity-pleiotropy"><i class="fa fa-check"></i><b>19.5.3</b> Heterogeneity &amp; Pleiotropy</a></li>
<li class="chapter" data-level="19.5.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#leave-one-out"><i class="fa fa-check"></i><b>19.5.4</b> Leave-one-out</a></li>
<li class="chapter" data-level="19.5.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#visualisation"><i class="fa fa-check"></i><b>19.5.5</b> Visualisation</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#the-end-1"><i class="fa fa-check"></i><b>19.6</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>20</b> Epilogue</a>
<ul>
<li class="chapter" data-level="20.1" data-path="epilogue.html"><a href="epilogue.html#course-lecturers"><i class="fa fa-check"></i><b>20.1</b> 2024 Course Lecturers</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="answers.html"><a href="answers.html"><i class="fa fa-check"></i><b>21</b> Answers to the questions</a>
<ul>
<li class="chapter" data-level="21.1" data-path="answers.html"><a href="answers.html#chapter-3.1-gwas-basics"><i class="fa fa-check"></i><b>21.1</b> Chapter 3.1: GWAS Basics</a></li>
<li class="chapter" data-level="21.2" data-path="answers.html"><a href="answers.html#chapter-3.4-gwas-basics---association-testing"><i class="fa fa-check"></i><b>21.2</b> Chapter 3.4: GWAS Basics - Association testing</a></li>
<li class="chapter" data-level="21.3" data-path="answers.html"><a href="answers.html#chapter-3.5-gwas-visualization"><i class="fa fa-check"></i><b>21.3</b> Chapter 3.5: GWAS visualization</a></li>
<li class="chapter" data-level="21.4" data-path="answers.html"><a href="answers.html#chapter-3.6-regional-association-plots"><i class="fa fa-check"></i><b>21.4</b> Chapter 3.6: Regional association plots</a></li>
<li class="chapter" data-level="21.5" data-path="answers.html"><a href="answers.html#chapter-4.2-wtccc1"><i class="fa fa-check"></i><b>21.5</b> Chapter 4.2: WTCCC1</a></li>
<li class="chapter" data-level="21.6" data-path="answers.html"><a href="answers.html#chapter-4.3-wtccc1---association-testing"><i class="fa fa-check"></i><b>21.6</b> Chapter 4.3: WTCCC1 - Association testing</a></li>
<li class="chapter" data-level="21.7" data-path="answers.html"><a href="answers.html#chapter-5.4-fuma"><i class="fa fa-check"></i><b>21.7</b> Chapter 5.4: FUMA</a></li>
<li class="chapter" data-level="21.8" data-path="answers.html"><a href="answers.html#chapter-5.5-phewas"><i class="fa fa-check"></i><b>21.8</b> Chapter 5.5: PheWAS</a></li>
<li class="chapter" data-level="21.9" data-path="answers.html"><a href="answers.html#chapter-5.7-twosample-mr"><i class="fa fa-check"></i><b>21.9</b> Chapter 5.7: TwoSample MR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>22</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="22.1" data-path="license.html"><a href="license.html#images-and-data-used"><i class="fa fa-check"></i><b>22.1</b> Images and data used</a></li>
<li class="chapter" data-level="22.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>22.2</b> Disclaimer</a></li>
<li class="chapter" data-level="22.3" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>22.3</b> Copyright</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="colophon.html"><a href="colophon.html"><i class="fa fa-check"></i><b>23</b> Colophon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://vanderlaanand.science" target="blank">&copy; 1979-2024. Sander W. van der Laan.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics for CoCalc</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="getting-started-programs-cocalc" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Getting started - programs<a href="getting-started-programs-cocalc.html#getting-started-programs-cocalc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- ![](./img/headers/women_behind_macbook.png){width=100%} -->
<p>We’ll use a few programs throughout this practical. You’ll probably need these for your (future) genetic epidemiology work too (Table <a href="getting-started-programs-cocalc.html#tab:programs">4.1</a>).</p>
<div class="tabwid"><style>.cl-03061060{table-layout:auto;}.cl-02ff2750{font-family:'Helvetica';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-0301f0a2{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-030208da{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-030208db{background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-030208e4{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing='true' class='cl-03061060'>

<caption style="display:table-caption;">
<span id="tab:programs">Table 4.1: </span><span>Programs needed for genetic epidemiology.</span>
</caption>

<thead><tr style="overflow-wrap:break-word;"><th class="cl-030208da"><p class="cl-0301f0a2"><span class="cl-02ff2750">Program</span></p></th><th class="cl-030208da"><p class="cl-0301f0a2"><span class="cl-02ff2750">Link</span></p></th><th class="cl-030208da"><p class="cl-0301f0a2"><span class="cl-02ff2750">Description</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">PLINK</span></p></td><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">https://www.cog-genomics.org/plink2/</span></p></td><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">PLINK is a free, open-source genetic analysis tool set, designed to perform a range of basic data parsing and quality control, as well as basic and large-scale analyses in a computationally efficient manner.</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">R</span></p></td><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">https://cran.r-project.org/</span></p></td><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">A program to perform statistical analysis and visualizations.</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">RStudio</span></p></td><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">https://www.rstudio.com</span></p></td><td class="cl-030208db"><p class="cl-0301f0a2"><span class="cl-02ff2750">A user-friendly R-wrap-around for code editing, debugging, analyses, and visualization.</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-030208e4"><p class="cl-0301f0a2"><span class="cl-02ff2750">Homebrew</span></p></td><td class="cl-030208e4"><p class="cl-0301f0a2"><span class="cl-02ff2750">https://brew.sh</span></p></td><td class="cl-030208e4"><p class="cl-0301f0a2"><span class="cl-02ff2750">A great extension for Mac-users to install really useful programs that Apple didn't.</span></p></td></tr></tbody></table></div>
<!-- This solution works for EPUB - but it makes figures, therefore I would have to adjust all the references.  -->
<!-- ```{r, tab.id = "programscreate", tab.cap = "Programs needed for genetic epidemiology.", eval = TRUE, echo = FALSE, message = FALSE, warning = FALSE, include = FALSE} -->
<!-- set_flextable_defaults(post_process_pdf = TRUE, post_process_html = TRUE, -->
<!--                        table.layout = "autofit") -->
<!-- ft <- flextable(programs) -->
<!-- save_as_image(ft, path = "./img/programs.png") -->
<!-- ``` -->
<!-- ```{r, tab.id = "programs", tab.cap = "Programs needed for genetic epidemiology.", eval = TRUE, echo = FALSE, message = FALSE, warning = FALSE} -->
<!-- knitr::include_graphics("./img/programs.png") -->
<!-- ``` -->
<div id="rstudio" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> RStudio<a href="getting-started-programs-cocalc.html#rstudio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>RStudio</strong> is a very user-friendly interface around <code>R</code> that makes your <code>R</code>-scripting-life a lot easier. You should get used to that. <strong>RStudio</strong> comes with <code>R</code> so you don’t have to worry about that.</p>
</div>
<div id="plink" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> PLINK<a href="getting-started-programs-cocalc.html#plink" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Right, onto <code>PLINK</code>.</p>
<p>All genetic analyses can be done in PLINK, even on your laptop, but with large datasets, for example <a href="https://www.ukbiobank.ac.uk" target="_blank">UK Biobank</a> size, it is better to switch to a <a href="https://en.wikipedia.org/wiki/High-performance_computing" target="_blank">high-performance computing cluster (HPC)</a> like we have available at the <a href="https://wiki.bioinformatics.umcutrecht.nl/bin/view/HPC/WebHome" target="_blank">Utrecht Science Park</a>. The original PLINK v1.07 can be found <a href="https://zzz.bwh.harvard.edu/plink/index.shtml" target="_blank">here</a>, but nowadays we are using a newer, faster version: <strong>PLINK v1.9</strong> which can be found <a href="https://www.cog-genomics.org/plink2" target="_blank">here</a>. It still says ‘PLINK 1.90 beta’ (Figure <a href="getting-started-programs-cocalc.html#fig:plinkprogram">4.1</a>), but you can consider this version stable and save to work with, but as you can see, some functions are not supported anymore.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:plinkprogram"></span>
<img src="img/plink.png" alt="The PLINK v1.9 website." width="85%" />
<p class="caption">
Figure 4.1: The PLINK v1.9 website.
</p>
</div>
<div id="alternatives-to-plink" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Alternatives to <code>PLINK</code><a href="getting-started-programs-cocalc.html#alternatives-to-plink" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Nowadays, a lot of people also use programs like <a href="snptest" target="_blank">SNPTEST</a>, <a href="https://data.broadinstitute.org/alkesgroup/BOLT-LMM/" target="_blank">BOLT-LMM</a>, <a href="http://cnsgenomics.com/software/gcta/#Overview" target="_blank">GCTA</a>, or <a href="https://rgcgithub.github.io/regenie/" target="_blank">regenie</a> as alternatives to execute GWAS. These programs were designed with specific use-cases in mind, for instance really large biobank data including hundreds of thousands individuals, better control for population stratification, the ability to estimate trait heritability or Fst, and so on.</p>
</div>
</div>
<div id="other-programs" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Other programs<a href="getting-started-programs-cocalc.html#other-programs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mendelian randomization can be done either with the <a href="http://cnsgenomics.com/software/smr/#Overview" target="_blank">SMR</a> or <a href="http://cnsgenomics.com/software/gsmr/" target="_blank">GSMR</a> function from GCTA, or with R-packages, like <a href="https://mrcieu.github.io/TwoSampleMR/" target="_blank"><code>TwoSampleMR</code></a>.</p>
<p>As you are following the <strong>Genetic Epidemiology</strong> course, the next thing we’ll cover are the <strong>CoCalc</strong>-instructions in Chapter <a href="getting-started-cocalc.html#getting-started-cocalc">5</a>.</p>
<script>
title=document.getElementById('header');
title.innerHTML = '<img src="img/headers/women_behind_macbook.png" alt="Getting started">' + title.innerHTML
</script>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="getting-started.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="getting-started-cocalc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans-serif",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"always_allow_html": true,
"split_by": "chapter",
"toolbar": {
"position": "fixed"
},
"info": true,
"code_folding": "show"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
