<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Genome-Wide Association Study | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Genome-Wide Association Study | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Genome-Wide Association Study | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas_basics_snp_qc.html"/>
<link rel="next" href="epilogue.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#the-programs-we-use"><i class="fa fa-check"></i><b>3.1</b> The programs we use</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#cocalc"><i class="fa fa-check"></i><b>3.2</b> CoCalc</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#standalone"><i class="fa fa-check"></i><b>3.3</b> Standalone</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="getting-started.html"><a href="getting-started.html#terminal"><i class="fa fa-check"></i><b>3.3.1</b> Terminal</a></li>
<li class="chapter" data-level="3.3.2" data-path="getting-started.html"><a href="getting-started.html#the-data-you-need"><i class="fa fa-check"></i><b>3.3.2</b> The data you need</a></li>
<li class="chapter" data-level="3.3.3" data-path="getting-started.html"><a href="getting-started.html#navigating-the-terminal"><i class="fa fa-check"></i><b>3.3.3</b> Navigating the Terminal</a></li>
<li class="chapter" data-level="3.3.4" data-path="getting-started.html"><a href="getting-started.html#installing-the-software"><i class="fa fa-check"></i><b>3.3.4</b> Installing the software</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="getting-started.html"><a href="getting-started.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>3.4</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="3.5" data-path="getting-started.html"><a href="getting-started.html#are-you-ready"><i class="fa fa-check"></i><b>3.5</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>4</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>4.1</b> Converting datasets</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>4.2</b> Quality control</a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>4.3</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>5</b> Sample QC</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>5.1</b> Sex</a></li>
<li class="chapter" data-level="5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>5.2</b> Sample call rates</a></li>
<li class="chapter" data-level="5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>5.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>5.4</b> Relatedness</a></li>
<li class="chapter" data-level="5.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>5.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>5.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#principal-component-analysis"><i class="fa fa-check"></i><b>5.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="5.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>5.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>5.6</b> Removing samples</a></li>
<li class="chapter" data-level="5.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>5.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>6</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>6.1</b> SNP call rates</a></li>
<li class="chapter" data-level="6.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>6.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="6.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>6.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>6.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>6.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="6.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>6.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="6.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>6.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-testing.html"><a href="gwas-testing.html"><i class="fa fa-check"></i><b>7</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-testing.html"><a href="gwas-testing.html#exploring-the-data"><i class="fa fa-check"></i><b>7.1</b> Exploring the data</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-testing.html"><a href="gwas-testing.html#genetic-models"><i class="fa fa-check"></i><b>7.2</b> Genetic models</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-testing.html"><a href="gwas-testing.html#logistic-regression"><i class="fa fa-check"></i><b>7.3</b> Logistic regression</a></li>
<li class="chapter" data-level="7.4" data-path="gwas-testing.html"><a href="gwas-testing.html#lets-get-visual"><i class="fa fa-check"></i><b>7.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>8</b> Epilogue</a></li>
<li class="chapter" data-level="9" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>9</b> Additional: EIGENSOFT</a>
<ul>
<li class="chapter" data-level="9.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>9.1</b> Install homebrew</a></li>
<li class="chapter" data-level="9.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>9.2</b> Install missing packages</a></li>
<li class="chapter" data-level="9.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>9.3</b> Installing EIGENSOFT</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>10</b> Things to do</a>
<ul>
<li class="chapter" data-level="10.1" data-path="todo.html"><a href="todo.html#moscow"><i class="fa fa-check"></i><b>10.1</b> MoSCoW</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="todo.html"><a href="todo.html#contents"><i class="fa fa-check"></i><b>10.1.1</b> Contents</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="todo.html"><a href="todo.html#book-fixes"><i class="fa fa-check"></i><b>10.2</b> Book fixes</a></li>
<li class="chapter" data-level="10.3" data-path="todo.html"><a href="todo.html#useful-links-for-me-mostly"><i class="fa fa-check"></i><b>10.3</b> Useful links (for me mostly)</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>11</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="11.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>11.1</b> Copyright</a></li>
<li class="chapter" data-level="11.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>11.2</b> Disclaimer</a></li>
<li class="chapter" data-level="11.3" data-path="license.html"><a href="license.html#images-used"><i class="fa fa-check"></i><b>11.3</b> Images used</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="colofon.html"><a href="colofon.html"><i class="fa fa-check"></i><b>12</b> Colofon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-testing" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Genome-Wide Association Study<a href="gwas-testing.html#gwas-testing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="img/_gwas/interactive_plot.png" style="width:70.0%" /></p>
<p>Now that you have learned how to perform QC, you can easily run a GWAS and execute some downstream visualization and analyses. Let’s do this with a dummy dataset.</p>
<div id="exploring-the-data" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Exploring the data<a href="gwas-testing.html#exploring-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Even though someone says that the QC was done, it is still wise and good practice to run some of the commands above to get a ‘feeling’ about the data. So let’s do this.</p>
<pre><code>plink --bfile gwas/gwa --freq --out gwas/gwa</code></pre>
<pre><code>plink --bfile gwas/gwa --missing --out gwas/gwa</code></pre>
<pre><code>plink --bfile gwas/gwa --hardy --out gwas/gwa</code></pre>
<p>Let’s visualize the results. First we should load in all the results.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="gwas-testing.html#cb84-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb84-2"><a href="gwas-testing.html#cb84-2" tabindex="-1"></a></span>
<span id="cb84-3"><a href="gwas-testing.html#cb84-3" tabindex="-1"></a>COURSE_loc <span class="ot">=</span> <span class="st">&quot;~/Desktop/practical&quot;</span> <span class="co"># getwd()</span></span>
<span id="cb84-4"><a href="gwas-testing.html#cb84-4" tabindex="-1"></a></span>
<span id="cb84-5"><a href="gwas-testing.html#cb84-5" tabindex="-1"></a>gwas_HWE <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/gwas/gwa.hwe&quot;</span>))</span>
<span id="cb84-6"><a href="gwas-testing.html#cb84-6" tabindex="-1"></a>gwas_FRQ <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/gwas/gwa.frq&quot;</span>))</span>
<span id="cb84-7"><a href="gwas-testing.html#cb84-7" tabindex="-1"></a>gwas_IMISS <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/gwas/gwa.imiss&quot;</span>))</span>
<span id="cb84-8"><a href="gwas-testing.html#cb84-8" tabindex="-1"></a>gwas_LMISS <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/gwas/gwa.lmiss&quot;</span>))</span></code></pre></div>
<p>We can plot the per-stratum HWE p-values.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="gwas-testing.html#cb85-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb85-2"><a href="gwas-testing.html#cb85-2" tabindex="-1"></a></span>
<span id="cb85-3"><a href="gwas-testing.html#cb85-3" tabindex="-1"></a>gwas_HWE<span class="sc">$</span>logP <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(gwas_HWE<span class="sc">$</span>P)</span>
<span id="cb85-4"><a href="gwas-testing.html#cb85-4" tabindex="-1"></a></span>
<span id="cb85-5"><a href="gwas-testing.html#cb85-5" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(gwas_HWE, <span class="at">x =</span> <span class="st">&quot;logP&quot;</span>,</span>
<span id="cb85-6"><a href="gwas-testing.html#cb85-6" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>,</span>
<span id="cb85-7"><a href="gwas-testing.html#cb85-7" tabindex="-1"></a>                    <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb85-8"><a href="gwas-testing.html#cb85-8" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb85-9"><a href="gwas-testing.html#cb85-9" tabindex="-1"></a>                    <span class="co"># color = &quot;#1290D9&quot;, fill = &quot;#1290D9&quot;,</span></span>
<span id="cb85-10"><a href="gwas-testing.html#cb85-10" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;TEST&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;TEST&quot;</span>,</span>
<span id="cb85-11"><a href="gwas-testing.html#cb85-11" tabindex="-1"></a>                    <span class="at">palette =</span> <span class="st">&quot;lancet&quot;</span>,</span>
<span id="cb85-12"><a href="gwas-testing.html#cb85-12" tabindex="-1"></a>                    <span class="at">facet.by =</span> <span class="st">&quot;TEST&quot;</span>,</span>
<span id="cb85-13"><a href="gwas-testing.html#cb85-13" tabindex="-1"></a>                    <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb85-14"><a href="gwas-testing.html#cb85-14" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;HWE -log10(P)&quot;</span>) <span class="sc">+</span></span>
<span id="cb85-15"><a href="gwas-testing.html#cb85-15" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">5</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb85-16"><a href="gwas-testing.html#cb85-16" tabindex="-1"></a>                      <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb85-17"><a href="gwas-testing.html#cb85-17" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/dummy_project/gwas-hwe.png&quot;</span>),</span>
<span id="cb85-18"><a href="gwas-testing.html#cb85-18" tabindex="-1"></a>       <span class="at">plot =</span> <span class="fu">last_plot</span>())</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-hwe-gwas"></span>
<img src="img/_gwas_dummy/show-hwe-gwas.png" alt="Per stratum HWE p-values." width="672" />
<p class="caption">
Figure 7.1: Per stratum HWE p-values.
</p>
</div>
<p>We will want to see what the distribution of allele frequencies looks like.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="gwas-testing.html#cb86-1" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(gwas_FRQ, <span class="at">x =</span> <span class="st">&quot;MAF&quot;</span>,</span>
<span id="cb86-2"><a href="gwas-testing.html#cb86-2" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb86-3"><a href="gwas-testing.html#cb86-3" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb86-4"><a href="gwas-testing.html#cb86-4" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb86-5"><a href="gwas-testing.html#cb86-5" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;minor allele frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb86-6"><a href="gwas-testing.html#cb86-6" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.05</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb86-7"><a href="gwas-testing.html#cb86-7" tabindex="-1"></a>                      <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb86-8"><a href="gwas-testing.html#cb86-8" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/dummy_project/gwas-freq.png&quot;</span>),</span>
<span id="cb86-9"><a href="gwas-testing.html#cb86-9" tabindex="-1"></a>       <span class="at">plot =</span> <span class="fu">last_plot</span>())</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-freq-gwas"></span>
<img src="img/_gwas_dummy/show-freq-gwas.png" alt="Minor allele frequencies." width="672" />
<p class="caption">
Figure 7.2: Minor allele frequencies.
</p>
</div>
<p>We will want to identify samples that have poor call rates.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="gwas-testing.html#cb87-1" tabindex="-1"></a>gwas_IMISS<span class="sc">$</span>callrate <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> gwas_IMISS<span class="sc">$</span>F_MISS</span>
<span id="cb87-2"><a href="gwas-testing.html#cb87-2" tabindex="-1"></a></span>
<span id="cb87-3"><a href="gwas-testing.html#cb87-3" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(gwas_IMISS, <span class="at">x =</span> <span class="st">&quot;callrate&quot;</span>,</span>
<span id="cb87-4"><a href="gwas-testing.html#cb87-4" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb87-5"><a href="gwas-testing.html#cb87-5" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>, <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb87-6"><a href="gwas-testing.html#cb87-6" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb87-7"><a href="gwas-testing.html#cb87-7" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;per sample call rate&quot;</span>) <span class="sc">+</span></span>
<span id="cb87-8"><a href="gwas-testing.html#cb87-8" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb87-9"><a href="gwas-testing.html#cb87-9" tabindex="-1"></a>                      <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb87-10"><a href="gwas-testing.html#cb87-10" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/dummy_project/gwas-sample-call-rate.png&quot;</span>),</span>
<span id="cb87-11"><a href="gwas-testing.html#cb87-11" tabindex="-1"></a>       <span class="at">plot =</span> <span class="fu">last_plot</span>())</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-sample-callrate-gwas"></span>
<img src="img/_gwas_dummy/show-sample-callrate-gwas.png" alt="Per sample call rates." width="672" />
<p class="caption">
Figure 7.3: Per sample call rates.
</p>
</div>
<p>We also need to know what the per SNP call rates are.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="gwas-testing.html#cb88-1" tabindex="-1"></a>gwas_LMISS<span class="sc">$</span>callrate <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> gwas_LMISS<span class="sc">$</span>F_MISS</span>
<span id="cb88-2"><a href="gwas-testing.html#cb88-2" tabindex="-1"></a></span>
<span id="cb88-3"><a href="gwas-testing.html#cb88-3" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(gwas_LMISS, <span class="at">x =</span> <span class="st">&quot;callrate&quot;</span>,</span>
<span id="cb88-4"><a href="gwas-testing.html#cb88-4" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb88-5"><a href="gwas-testing.html#cb88-5" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>, <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb88-6"><a href="gwas-testing.html#cb88-6" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb88-7"><a href="gwas-testing.html#cb88-7" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;per SNP call rate&quot;</span>) <span class="sc">+</span></span>
<span id="cb88-8"><a href="gwas-testing.html#cb88-8" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb88-9"><a href="gwas-testing.html#cb88-9" tabindex="-1"></a>                      <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb88-10"><a href="gwas-testing.html#cb88-10" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/dummy_project/gwas-snp-call-rate.png&quot;</span>),</span>
<span id="cb88-11"><a href="gwas-testing.html#cb88-11" tabindex="-1"></a>       <span class="at">plot =</span> <span class="fu">last_plot</span>())</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-snp-callrate-gwas"></span>
<img src="img/_gwas_dummy/show-snp-callrate-gwas.png" alt="Per SNP call rates." width="672" />
<p class="caption">
Figure 7.4: Per SNP call rates.
</p>
</div>
</div>
<div id="genetic-models" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Genetic models<a href="gwas-testing.html#genetic-models" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A simple chi-square test of association can be done.</p>
<pre><code>plink --bfile gwas/gwa --model --out gwas/data</code></pre>
<p><em>Genotypic</em>, <em>dominant</em> and <em>recessive</em> tests will not be conducted if any one of the cells in the table of case-control by genotype counts contains less than five observations. This is because the chi-square approximation may not be reliable when cell counts are small. For SNPs with MAFs &lt; 5%, a sample of more than 2,000 cases and controls would be required to meet this threshold and more than 50,000 would be required for SNPs with MAF &lt; 1%.</p>
<p>You can change this default behaviour by adding the flag <code>--cell</code>, <em>e.g.</em>, we could lower the threshold to 3.</p>
<pre><code>plink --bfile gwas/gwa --model --cell 3 --out gwas/data</code></pre>
<p>Let’s review the contents of the results.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="gwas-testing.html#cb91-1" tabindex="-1"></a>gwas_model <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/gwas/data.model&quot;</span>))</span>
<span id="cb91-2"><a href="gwas-testing.html#cb91-2" tabindex="-1"></a></span>
<span id="cb91-3"><a href="gwas-testing.html#cb91-3" tabindex="-1"></a><span class="fu">dim</span>(gwas_model)</span>
<span id="cb91-4"><a href="gwas-testing.html#cb91-4" tabindex="-1"></a></span>
<span id="cb91-5"><a href="gwas-testing.html#cb91-5" tabindex="-1"></a>N_SNPS <span class="ot">=</span> <span class="fu">length</span>(gwas_model<span class="sc">$</span>SNP)</span>
<span id="cb91-6"><a href="gwas-testing.html#cb91-6" tabindex="-1"></a></span>
<span id="cb91-7"><a href="gwas-testing.html#cb91-7" tabindex="-1"></a>gwas_model[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<p>It contains 1,530,510 rows, one for each SNP, and each type of test (<em>genotypic</em>, <em>trend</em>, <em>allelic</em>, <em>dominant</em>, and <em>recessive</em>) and the following columns:</p>
<ul>
<li>chromosome [CHR],</li>
<li>the SNP identifier [SNP],</li>
<li>the minor allele [A1] (remember, <code>PLINK</code> always codes the A1-allele as the minor allele!),</li>
<li>the major allele [A2],</li>
<li>the test performed [TEST]:
<ul>
<li>GENO (genotypic association);</li>
<li>TREND (Cochran-Armitage trend);</li>
<li>ALLELIC (allelic as- sociation);</li>
<li>DOM (dominant model); and</li>
<li>REC (recessive model)],</li>
</ul></li>
<li>the cell frequency counts for cases [AFF],</li>
<li>the cell frequency counts for controls [UNAFF],</li>
<li>the chi-square test statistic [CHISQ],</li>
<li>the degrees of freedom for the test [DF],</li>
<li>and the asymptotic P value [P] of association.</li>
</ul>
</div>
<div id="logistic-regression" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Logistic regression<a href="gwas-testing.html#logistic-regression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can also perform a test of association using logistic regression. In this case we might want to correct for covariates/confounding factors, for example age, sex, ancestral background, i.e. principal components, and other study specific covariates (e.g. hospital of inclusion, genotyping centre etc.). In that case each of these P values is adjusted for the effect of the covariates.</p>
<p>When running a regression analysis, be it linear or logistic, PLINK assumes a multiplicative model. By default, when at least one male and one female is present, sex (male = 1, female = 0) is automatically added as a covariate on X chromosome SNPs, and nowhere else. The <code>sex</code> flag causes it to be added everywhere, while <code>no-x-sex</code> excludes it.</p>
<pre><code>plink --bfile gwas/gwa --logistic sex --covar gwas/gwa.covar --out gwas/data</code></pre>
<p>Let’s examine the results</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="gwas-testing.html#cb93-1" tabindex="-1"></a>gwas_assoc <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/gwas/data.assoc.logistic&quot;</span>))</span>
<span id="cb93-2"><a href="gwas-testing.html#cb93-2" tabindex="-1"></a></span>
<span id="cb93-3"><a href="gwas-testing.html#cb93-3" tabindex="-1"></a><span class="fu">dim</span>(gwas_assoc)</span>
<span id="cb93-4"><a href="gwas-testing.html#cb93-4" tabindex="-1"></a></span>
<span id="cb93-5"><a href="gwas-testing.html#cb93-5" tabindex="-1"></a>gwas_assoc[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>]</span></code></pre></div>
<p>If no model option is specified, the first row for each SNP corresponds to results for a multiplicative test of association. The C &gt;= 0 subsequent rows for each SNP correspond to separate tests of significance for each of the C covariates included in the regression model. We can remove the covariate-specific lines from the main report by adding the <code>hide-covar</code> flag.</p>
<p>The columns in the association results are:
- the chromosome [CHR],
- the SNP identifier [SNP],
- the base-pair location [BP],
- the minor allele [A1],
- the test performed [TEST]: ADD (multiplicative model or genotypic model testing additivity),
- GENO_2DF (genotypic model),
- DOMDEV (genotypic model testing deviation from additivity),
- DOM (dominant model), or
- REC (recessive model)],
- the number of missing individuals included [NMISS],
- the OR relative to the A1, <em>i.e.</em> minor allele,
- the coefficient z-statistic [STAT], and
- the asymptotic P-value [P] of association.</p>
<p>We need to calculate the standard error and confidence interval from the z-statistic. We can modify the effect size (OR) to output the beta by adding the <code>beta</code> flag.</p>
</div>
<div id="lets-get-visual" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Let’s get visual<a href="gwas-testing.html#lets-get-visual" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Looking at numbers is important, but it won’t give you a perfect overview. We should turn to visualizing our results in Chapter <a href="#gwas-visuals"><strong>??</strong></a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas_basics_snp_qc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="epilogue.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics03_4_gwas_basics_association_testing.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
