<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Genome-Wide Association Study | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Genome-Wide Association Study | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Genome-Wide Association Study | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas_basics_snp_qc.html"/>
<link rel="next" href="gwas-visuals.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#your-computer"><i class="fa fa-check"></i><b>3.1</b> Your computer</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#cocalc-vs.-standalone"><i class="fa fa-check"></i><b>3.2</b> CoCalc vs. Standalone</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#the-programs-we-use"><i class="fa fa-check"></i><b>3.3</b> The programs we use</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="getting-started.html"><a href="getting-started.html#rstudio"><i class="fa fa-check"></i><b>3.3.1</b> RStudio</a></li>
<li class="chapter" data-level="3.3.2" data-path="getting-started.html"><a href="getting-started.html#plink"><i class="fa fa-check"></i><b>3.3.2</b> PLINK</a></li>
<li class="chapter" data-level="3.3.3" data-path="getting-started.html"><a href="getting-started.html#alternatives-to-plink"><i class="fa fa-check"></i><b>3.3.3</b> Alternatives to <code>PLINK</code></a></li>
<li class="chapter" data-level="3.3.4" data-path="getting-started.html"><a href="getting-started.html#other-programs"><i class="fa fa-check"></i><b>3.3.4</b> Other programs</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="getting-started.html"><a href="getting-started.html#cocalc"><i class="fa fa-check"></i><b>3.4</b> CoCalc</a></li>
<li class="chapter" data-level="3.5" data-path="getting-started.html"><a href="getting-started.html#standalone"><i class="fa fa-check"></i><b>3.5</b> Standalone</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="getting-started.html"><a href="getting-started.html#the-data-you-need"><i class="fa fa-check"></i><b>3.5.1</b> The data you need</a></li>
<li class="chapter" data-level="3.5.2" data-path="getting-started.html"><a href="getting-started.html#terminal"><i class="fa fa-check"></i><b>3.5.2</b> Terminal</a></li>
<li class="chapter" data-level="3.5.3" data-path="getting-started.html"><a href="getting-started.html#navigating-the-terminal"><i class="fa fa-check"></i><b>3.5.3</b> Navigating the Terminal</a></li>
<li class="chapter" data-level="3.5.4" data-path="getting-started.html"><a href="getting-started.html#installing-the-software"><i class="fa fa-check"></i><b>3.5.4</b> Installing the software</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="getting-started.html"><a href="getting-started.html#are-you-ready"><i class="fa fa-check"></i><b>3.6</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>4</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>4.1</b> Converting datasets</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>4.2</b> Quality control</a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>4.3</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>5</b> Sample QC</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>5.1</b> Sex</a></li>
<li class="chapter" data-level="5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>5.2</b> Sample call rates</a></li>
<li class="chapter" data-level="5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>5.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>5.4</b> Relatedness</a></li>
<li class="chapter" data-level="5.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>5.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>5.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#principal-component-analysis"><i class="fa fa-check"></i><b>5.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="5.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>5.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>5.6</b> Removing samples</a></li>
<li class="chapter" data-level="5.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>5.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>6</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>6.1</b> SNP call rates</a></li>
<li class="chapter" data-level="6.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>6.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="6.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>6.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>6.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>6.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="6.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>6.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="6.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>6.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas_testing.html"><a href="gwas_testing.html"><i class="fa fa-check"></i><b>7</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas_testing.html"><a href="gwas_testing.html#exploring-the-data"><i class="fa fa-check"></i><b>7.1</b> Exploring the data</a></li>
<li class="chapter" data-level="7.2" data-path="gwas_testing.html"><a href="gwas_testing.html#genetic-models"><i class="fa fa-check"></i><b>7.2</b> Genetic models</a></li>
<li class="chapter" data-level="7.3" data-path="gwas_testing.html"><a href="gwas_testing.html#logistic-regression"><i class="fa fa-check"></i><b>7.3</b> Logistic regression</a></li>
<li class="chapter" data-level="7.4" data-path="gwas_testing.html"><a href="gwas_testing.html#lets-get-visual"><i class="fa fa-check"></i><b>7.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>8</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="8.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>8.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="8.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>8.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>8.2</b> Other plots</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>8.3</b> Interactive plots</a></li>
<li class="chapter" data-level="8.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>8.4</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html"><i class="fa fa-check"></i><b>9</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#genotyping"><i class="fa fa-check"></i><b>9.1</b> Genotyping</a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#the-data"><i class="fa fa-check"></i><b>9.2</b> The data</a></li>
<li class="chapter" data-level="9.3" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#assignment"><i class="fa fa-check"></i><b>9.3</b> Assignment</a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#there-you-go"><i class="fa fa-check"></i><b>9.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>10</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>10.1</b> Quality control</a></li>
<li class="chapter" data-level="10.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>10.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>10.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="10.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca"><i class="fa fa-check"></i><b>10.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="10.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>10.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="10.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>10.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>10.3</b> Association testing</a></li>
<li class="chapter" data-level="10.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>10.4</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="post_gwas.html"><a href="post_gwas.html"><i class="fa fa-check"></i><b>11</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="11.1" data-path="post_gwas.html"><a href="post_gwas.html#clumping"><i class="fa fa-check"></i><b>11.1</b> Clumping</a></li>
<li class="chapter" data-level="11.2" data-path="post_gwas.html"><a href="post_gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>11.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="11.3" data-path="post_gwas.html"><a href="post_gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>11.3</b> Statistical finemapping</a></li>
<li class="chapter" data-level="11.4" data-path="post_gwas.html"><a href="post_gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>11.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="11.5" data-path="post_gwas.html"><a href="post_gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>11.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="11.6" data-path="post_gwas.html"><a href="post_gwas.html#colocalization"><i class="fa fa-check"></i><b>11.6</b> Colocalization</a></li>
<li class="chapter" data-level="11.7" data-path="post_gwas.html"><a href="post_gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>11.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="11.8" data-path="post_gwas.html"><a href="post_gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>11.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="post_gwas.html"><a href="post_gwas.html#your-choice"><i class="fa fa-check"></i><b>11.8.1</b> Your choice</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="post_gwas.html"><a href="post_gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>11.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="11.10" data-path="post_gwas.html"><a href="post_gwas.html#what-comes-next"><i class="fa fa-check"></i><b>11.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="conditional_analysis.html"><a href="conditional_analysis.html"><i class="fa fa-check"></i><b>12</b> Conditional analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="conditional_analysis.html"><a href="conditional_analysis.html#section-1"><i class="fa fa-check"></i><b>12.1</b> Section 1</a></li>
<li class="chapter" data-level="12.2" data-path="conditional_analysis.html"><a href="conditional_analysis.html#section-2"><i class="fa fa-check"></i><b>12.2</b> Section 2</a></li>
<li class="chapter" data-level="12.3" data-path="conditional_analysis.html"><a href="conditional_analysis.html#section-3"><i class="fa fa-check"></i><b>12.3</b> Section 3</a></li>
<li class="chapter" data-level="12.4" data-path="conditional_analysis.html"><a href="conditional_analysis.html#replication-1"><i class="fa fa-check"></i><b>12.4</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="finemapping.html"><a href="finemapping.html"><i class="fa fa-check"></i><b>13</b> Statistical finemapping</a>
<ul>
<li class="chapter" data-level="13.1" data-path="finemapping.html"><a href="finemapping.html#section-1-1"><i class="fa fa-check"></i><b>13.1</b> Section 1</a></li>
<li class="chapter" data-level="13.2" data-path="finemapping.html"><a href="finemapping.html#section-2-1"><i class="fa fa-check"></i><b>13.2</b> Section 2</a></li>
<li class="chapter" data-level="13.3" data-path="finemapping.html"><a href="finemapping.html#section-3-1"><i class="fa fa-check"></i><b>13.3</b> Section 3</a></li>
<li class="chapter" data-level="13.4" data-path="finemapping.html"><a href="finemapping.html#replication-2"><i class="fa fa-check"></i><b>13.4</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>14</b> Functional Mapping and Annotation of GWAS</a>
<ul>
<li class="chapter" data-level="14.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>14.1</b> Assignment</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>14.1.1</b> Tutorial</a></li>
<li class="chapter" data-level="14.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>14.1.2</b> Create the input</a></li>
<li class="chapter" data-level="14.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>14.1.3</b> Run FUMA - SNP to gene</a></li>
<li class="chapter" data-level="14.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>14.1.4</b> Run FUMA - Gene to function</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>14.2</b> Some closing thoughts</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>15</b> Phenome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="15.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>15.1</b> What’s next?</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>16</b> MRBase</a>
<ul>
<li class="chapter" data-level="16.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>16.1</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i>Summary</a>
<ul>
<li class="chapter" data-level="16.2" data-path="summary.html"><a href="summary.html#section-1-2"><i class="fa fa-check"></i><b>16.2</b> Section 1</a></li>
<li class="chapter" data-level="16.3" data-path="summary.html"><a href="summary.html#section-2-2"><i class="fa fa-check"></i><b>16.3</b> Section 2</a></li>
<li class="chapter" data-level="16.4" data-path="summary.html"><a href="summary.html#section-3-2"><i class="fa fa-check"></i><b>16.4</b> Section 3</a></li>
<li class="chapter" data-level="16.5" data-path="summary.html"><a href="summary.html#replication-3"><i class="fa fa-check"></i><b>16.5</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>17</b> Epilogue</a></li>
<li class="chapter" data-level="18" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>18</b> Additional: EIGENSOFT</a>
<ul>
<li class="chapter" data-level="18.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>18.1</b> Install homebrew</a></li>
<li class="chapter" data-level="18.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>18.2</b> Install missing packages</a></li>
<li class="chapter" data-level="18.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>18.3</b> Installing EIGENSOFT</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="answers.html"><a href="answers.html"><i class="fa fa-check"></i><b>19</b> Answers to the questions</a>
<ul>
<li class="chapter" data-level="19.1" data-path="answers.html"><a href="answers.html#chapter-3.1-gwas-basics"><i class="fa fa-check"></i><b>19.1</b> Chapter 3.1: GWAS Basics</a></li>
<li class="chapter" data-level="19.2" data-path="answers.html"><a href="answers.html#chapter-3.4-gwas-basics---association-testing"><i class="fa fa-check"></i><b>19.2</b> Chapter 3.4: GWAS Basics - Association testing</a></li>
<li class="chapter" data-level="19.3" data-path="answers.html"><a href="answers.html#chapter-3.5-gwas-visualization"><i class="fa fa-check"></i><b>19.3</b> Chapter 3.5: GWAS visualization</a></li>
<li class="chapter" data-level="19.4" data-path="answers.html"><a href="answers.html#chapter-5.4-fuma"><i class="fa fa-check"></i><b>19.4</b> Chapter 5.4: FUMA</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>20</b> Things to do</a>
<ul>
<li class="chapter" data-level="20.1" data-path="todo.html"><a href="todo.html#moscow"><i class="fa fa-check"></i><b>20.1</b> MoSCoW</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="todo.html"><a href="todo.html#contents"><i class="fa fa-check"></i><b>20.1.1</b> Contents</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="todo.html"><a href="todo.html#book-fixes"><i class="fa fa-check"></i><b>20.2</b> Book fixes</a></li>
<li class="chapter" data-level="20.3" data-path="todo.html"><a href="todo.html#useful-links"><i class="fa fa-check"></i><b>20.3</b> Useful links</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>21</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="21.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>21.1</b> Copyright</a></li>
<li class="chapter" data-level="21.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>21.2</b> Disclaimer</a></li>
<li class="chapter" data-level="21.3" data-path="license.html"><a href="license.html#images-and-data-used"><i class="fa fa-check"></i><b>21.3</b> Images and data used</a></li>
<li class="chapter" data-level="21.4" data-path="license.html"><a href="license.html#copyright-1"><i class="fa fa-check"></i><b>21.4</b> Copyright</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="colophon.html"><a href="colophon.html"><i class="fa fa-check"></i><b>22</b> Colophon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://vanderlaanand.science" target="blank">&copy; 1979-2024. Sander W. van der Laan.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas_testing" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Genome-Wide Association Study<a href="gwas_testing.html#gwas_testing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- ![](./img/_gwas/interactive_plot.png){width=70%} -->
<p>Now that you have learned how to perform QC, you can easily run a GWAS and execute some downstream visualization and analyses. Let’s do this with a dummy dataset.</p>
<div id="exploring-the-data" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Exploring the data<a href="gwas_testing.html#exploring-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Even though someone says that the QC was done, it is still wise and good practice to run some of the commands above to get a ‘feeling’ about the data. So let’s do this.</p>
<pre><code>plink --bfile gwas/gwa --freq --out dummy_project/gwa</code></pre>
<pre><code>plink --bfile gwas/gwa --missing --out dummy_project/gwa</code></pre>
<pre><code>plink --bfile gwas/gwa --hardy --out dummy_project/gwa</code></pre>
<p>Let’s visualize the results. First we should load in all the results.</p>
<blockquote>
<p>Question: Load the data using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<p>We can plot the per-stratum HWE p-values.</p>
<blockquote>
<p>Question: Plot the per-stratum HWE p-values using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-2"></span>
<img src="img/_gwas_dummy/show-hwe-gwas.png" alt="Per stratum HWE p-values." width="85%" />
<p class="caption">
Figure 7.1: Per stratum HWE p-values.
</p>
</div>
<p>We will want to see what the distribution of allele frequencies looks like.</p>
<blockquote>
<p>Question: Plot the allele frequencies using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-3"></span>
<img src="img/_gwas_dummy/show-freq-gwas.png" alt="Minor allele frequencies." width="85%" />
<p class="caption">
Figure 7.2: Minor allele frequencies.
</p>
</div>
<p>We will want to identify samples that have poor call rates.</p>
<blockquote>
<p>Question: Plot the per-sample call rates using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="img/_gwas_dummy/show-sample-callrate-gwas.png" alt="Per sample call rates." width="85%" />
<p class="caption">
Figure 7.3: Per sample call rates.
</p>
</div>
<p>We also need to know what the per SNP call rates are.</p>
<blockquote>
<p>Question: Plot the per-SNP call rates using R. [Hint: use and adapt the examples from the previous chapters.]</p>
</blockquote>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="img/_gwas_dummy/show-snp-callrate-gwas.png" alt="Per SNP call rates." width="85%" />
<p class="caption">
Figure 7.4: Per SNP call rates.
</p>
</div>
</div>
<div id="genetic-models" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Genetic models<a href="gwas_testing.html#genetic-models" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A simple chi-square test of association can be done.</p>
<pre><code>plink --bfile gwas/gwa --model --out gwas/data</code></pre>
<p><em>Genotypic</em>, <em>dominant</em> and <em>recessive</em> tests will not be conducted if any one of the cells in the table of case-control by genotype counts contains less than five observations. This is because the chi-square approximation may not be reliable when cell counts are small. For SNPs with MAFs &lt; 5%, a sample of more than 2,000 cases and controls would be required to meet this threshold and more than 50,000 would be required for SNPs with MAF &lt; 1%.</p>
<p>You can change this default behaviour by adding the flag <code>--cell</code>, <em>e.g.</em>, we could lower the threshold to 3.</p>
<pre><code>plink --bfile gwas/gwa --model --cell 3 --out gwas/data</code></pre>
<p>Let’s review the contents of the results.</p>
<p>It contains 1,530,510 rows, one for each SNP, and each type of test (<em>genotypic</em>, <em>trend</em>, <em>allelic</em>, <em>dominant</em>, and <em>recessive</em>) and the following columns:</p>
<ul>
<li>chromosome [<code>CHR</code>],</li>
<li>the SNP identifier [<code>SNP</code>],</li>
<li>the minor allele [<code>A1</code>] (remember, <code>PLINK</code> always codes the A1-allele as the minor allele!),</li>
<li>the major allele [<code>A2</code>],</li>
<li>the test performed [<code>TEST</code>]:
<ul>
<li><code>GENO</code> (genotypic association);</li>
<li><code>TREND</code> (Cochran-Armitage trend);</li>
<li><code>ALLELIC</code> (allelic as- sociation);</li>
<li><code>DOM</code> (dominant model); and</li>
<li><code>REC</code> (recessive model)],</li>
</ul></li>
<li>the cell frequency counts for cases [<code>AFF</code>],</li>
<li>the cell frequency counts for controls [<code>UNAFF</code>],</li>
<li>the chi-square test statistic [<code>CHISQ</code>],</li>
<li>the degrees of freedom for the test [<code>DF</code>],</li>
<li>and the asymptotic P value [<code>P</code>] of association.</li>
</ul>
<blockquote>
<p>Question: Do you know which model, <em>i.e.</em> <code>TEST</code> is most commonly used and reported? And why is that, do think?</p>
</blockquote>
</div>
<div id="logistic-regression" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Logistic regression<a href="gwas_testing.html#logistic-regression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can also perform a test of association using logistic regression. In this case we might want to correct for covariates/confounding factors, for example age, sex, ancestral background, i.e. principal components, and other study specific covariates (e.g. hospital of inclusion, genotyping centre etc.). In that case each of these P values is adjusted for the effect of the covariates.</p>
<p>When running a regression analysis, be it linear or logistic, PLINK assumes a multiplicative model. By default, when at least one male and one female is present, sex (male = 1, female = 0) is automatically added as a covariate on X chromosome SNPs, and nowhere else. The <code>sex</code> flag causes it to be added everywhere, while <code>no-x-sex</code> excludes it.</p>
<pre><code>plink --bfile gwas/gwa --logistic sex --covar gwas/gwa.covar --out gwas/data</code></pre>
<p>Let’s examine the results.</p>
<pre><code>## [1] 918306      9</code></pre>
<pre><code>##      CHR       SNP      BP     A1   TEST NMISS     OR     STAT      P
##    &lt;int&gt;    &lt;char&gt;   &lt;int&gt; &lt;char&gt; &lt;char&gt; &lt;int&gt;  &lt;num&gt;    &lt;num&gt;  &lt;num&gt;
## 1:     1 rs3934834  995669      T    ADD  3818 1.0290  0.38120 0.7031
## 2:     1 rs3934834  995669      T    AGE  3818 1.0020  1.11800 0.2635
## 3:     1 rs3934834  995669      T    SEX  3818 1.0120  0.19090 0.8486
## 4:     1 rs3737728 1011278      A    ADD  3982 1.0190  0.38670 0.6990
## 5:     1 rs3737728 1011278      A    AGE  3982 1.0020  1.09800 0.2721
## 6:     1 rs3737728 1011278      A    SEX  3982 1.0060  0.09898 0.9212
## 7:     1 rs6687776 1020428      T    ADD  3915 0.9692 -0.33330 0.7389
## 8:     1 rs6687776 1020428      T    AGE  3915 1.0020  1.04000 0.2984
## 9:     1 rs6687776 1020428      T    SEX  3915 1.0150  0.23690 0.8127</code></pre>
<blockquote>
<p>Question: How come there are more lines in this file than there are variants?</p>
</blockquote>
<p>If no model option is specified, the first row for each SNP corresponds to results for a multiplicative test of association. The C &gt;= 0 subsequent rows for each SNP correspond to separate tests of significance for each of the C covariates included in the regression model. We can remove the covariate-specific lines from the main report by adding the <code>hide-covar</code> flag.</p>
<p>The columns in the association results are:</p>
<ul>
<li>the chromosome [<code>CHR</code>],</li>
<li>the SNP identifier [<code>SNP</code>],</li>
<li>the base-pair location [<code>BP</code>],</li>
<li>the minor allele [<code>A1</code>],</li>
<li>the test performed [<code>TEST</code>]: ADD (multiplicative model or genotypic model testing additivity),
<ul>
<li><code>GENO_2DF</code> (genotypic model),</li>
<li><code>DOMDEV</code> (genotypic model testing deviation from additivity),</li>
<li><code>DOM</code> (dominant model), or</li>
<li><code>REC</code> (recessive model)],</li>
</ul></li>
<li>the number of missing individuals included [<code>NMISS</code>],</li>
<li>the <code>OR</code> relative to the A1, <em>i.e.</em> minor allele,</li>
<li>the coefficient z-statistic [<code>STAT</code>], and</li>
<li>the asymptotic P-value [<code>P</code>] of association.</li>
</ul>
<p>We need to calculate the standard error and confidence interval from the z-statistic. We can modify the effect size (<code>OR</code>) to output the <em>beta</em> by adding the <code>beta</code> flag.</p>
<blockquote>
<p>Question: Can you write down the mathematical relation between <em>beta</em> and <em>OR</em>?</p>
</blockquote>
</div>
<div id="lets-get-visual" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Let’s get visual<a href="gwas_testing.html#lets-get-visual" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Looking at numbers is important, but it won’t give you a perfect overview. We should turn to visualizing our results in Chapter <a href="gwas-visuals.html#gwas-visuals">8</a>.</p>
<!-- ```{js, echo = FALSE} -->
<!-- title=document.getElementById('header'); -->
<!-- title.innerHTML = '<img src="img/_headers/interactive_plot.png" alt="GWAS basics: association testing">' + title.innerHTML -->
<!-- ``` -->

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas_basics_snp_qc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gwas-visuals.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"info": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
