<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Per-SNP QC | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Per-SNP QC | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Per-SNP QC | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas-basics-sample-qc.html"/>
<link rel="next" href="gwas-testing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="prerequisites.html"><a href="prerequisites.html#rstudio"><i class="fa fa-check"></i><b>2.2.1</b> RStudio<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="prerequisites.html"><a href="prerequisites.html#plink"><i class="fa fa-check"></i><b>2.2.2</b> PLINK<span></span></a></li>
<li class="chapter" data-level="2.2.3" data-path="prerequisites.html"><a href="prerequisites.html#alternatives-to-plink"><i class="fa fa-check"></i><b>2.2.3</b> Alternatives to PLINK<span></span></a></li>
<li class="chapter" data-level="2.2.4" data-path="prerequisites.html"><a href="prerequisites.html#other-programs"><i class="fa fa-check"></i><b>2.2.4</b> Other programs<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.5</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>3</b> Steps in a Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Let’s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>4.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.2</b> 1000G phase 1<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#running-pca"><i class="fa fa-check"></i><b>4.5.3</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="4.5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#pca-plotting"><i class="fa fa-check"></i><b>4.5.4</b> PCA plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>5.3.1</b> A note on allele coding<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-testing.html"><a href="gwas-testing.html"><i class="fa fa-check"></i><b>6</b> Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-testing.html"><a href="gwas-testing.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="gwas-testing.html"><a href="gwas-testing.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="gwas-testing.html"><a href="gwas-testing.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="gwas-testing.html"><a href="gwas-testing.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Let’s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>7.4</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>8.1</b> Genotyping<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>8.2</b> The data<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>8.3</b> Assignment<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>8.4</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>9</b> WTCCC1: a GWAS on coronary artery disease (CAD)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>9.1</b> Quality control<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>9.2</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>9.2.1</b> Preparing PCA<span></span></a></li>
<li class="chapter" data-level="9.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca-1"><i class="fa fa-check"></i><b>9.2.2</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="9.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>9.2.3</b> Plotting PCA<span></span></a></li>
<li class="chapter" data-level="9.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>9.2.4</b> Removing samples<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>9.3</b> Association testing<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>10</b> Post-GWAS Analyses<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>10.1</b> Clumping<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>10.2</b> Conditional analyses<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>10.3</b> Statistical finemapping<span></span></a></li>
<li class="chapter" data-level="10.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>10.4</b> FUMA: FUnctional Mapping and Annotation of GWAS<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>10.5</b> PheWAS<span></span></a></li>
<li class="chapter" data-level="10.6" data-path="post-gwas.html"><a href="post-gwas.html#colocalization"><i class="fa fa-check"></i><b>10.6</b> Colocalization<span></span></a></li>
<li class="chapter" data-level="10.7" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>10.7</b> Genetic correlation<span></span></a></li>
<li class="chapter" data-level="10.8" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>10.8</b> Causal inference: Mendelian Randomization<span></span></a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="post-gwas.html"><a href="post-gwas.html#your-choice"><i class="fa fa-check"></i><b>10.8.1</b> Your choice<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>10.9</b> Polygenic scores<span></span></a></li>
<li class="chapter" data-level="10.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>10.10</b> What comes next<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>11</b> Functional Mapping and Annotation of GWAS<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>11.1</b> Assignment<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>11.1.1</b> Tutorial<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>11.1.2</b> Create the input<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>11.1.3</b> Run FUMA - SNP to gene<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>11.1.4</b> Run FUMA - Gene to function<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>11.2</b> Some closing thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>12</b> Phenome-Wide Association Study (PheWAS)<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>12.1</b> What’s next?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>13</b> MRBase<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>13.1</b> The end?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>14</b> Epilogue<span></span></a></li>
<li class="chapter" data-level="15" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html"><i class="fa fa-check"></i><b>15</b> Additional: regional association plots<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html#locuszoom"><i class="fa fa-check"></i><b>15.1</b> LocusZoom<span></span></a></li>
<li class="chapter" data-level="15.2" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html#racer"><i class="fa fa-check"></i><b>15.2</b> RACER<span></span></a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>16</b> Additional: EIGENSOFT<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>16.1</b> Install homebrew<span></span></a></li>
<li class="chapter" data-level="16.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>16.2</b> Install missing packages<span></span></a></li>
<li class="chapter" data-level="16.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>16.3</b> Installing EIGENSOFT<span></span></a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>17</b> Things to do<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="todo.html"><a href="todo.html#moscow"><i class="fa fa-check"></i><b>17.1</b> MoSCoW<span></span></a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="todo.html"><a href="todo.html#contents"><i class="fa fa-check"></i><b>17.1.1</b> Contents<span></span></a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="todo.html"><a href="todo.html#book-fixes"><i class="fa fa-check"></i><b>17.2</b> Book fixes<span></span></a></li>
<li class="chapter" data-level="17.3" data-path="todo.html"><a href="todo.html#useful-links"><i class="fa fa-check"></i><b>17.3</b> Useful links<span></span></a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>18</b> Licenses and disclaimers<span></span></a>
<ul>
<li class="chapter" data-level="18.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>18.1</b> Copyright<span></span></a></li>
<li class="chapter" data-level="18.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>18.2</b> Disclaimer<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas_basics_snp_qc" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Per-SNP QC<a href="gwas_basics_snp_qc.html#gwas_basics_snp_qc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="img/gwas_snp_qc.png" style="width:70.0%" /></p>
<p>Now that we removed samples, we can focus on low-quality variants.</p>
<div id="snp-call-rates" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> SNP call rates<a href="gwas_basics_snp_qc.html#snp-call-rates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We start by calculating the missing genotype rate for each SNP, in other words the per-SNP call rate.</p>
<pre><code>plink --bfile rawdata/clean_inds_data --missing --out rawdata/clean_inds_data</code></pre>
<p>Let’s visualize the results to identify a threshold for extreme genotype failure rate. We chose a callrate threshold of 3%, but it’s arbitrary and depending on the dataset and the number of samples (Figure <a href="gwas_basics_snp_qc.html#fig:show-snp-callrate">5.1</a>).</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="gwas_basics_snp_qc.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb78-2"><a href="gwas_basics_snp_qc.html#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="gwas_basics_snp_qc.html#cb78-3" aria-hidden="true" tabindex="-1"></a>COURSE_loc <span class="ot">=</span> <span class="st">&quot;~/Desktop/practical&quot;</span> <span class="co"># getwd()</span></span>
<span id="cb78-4"><a href="gwas_basics_snp_qc.html#cb78-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-5"><a href="gwas_basics_snp_qc.html#cb78-5" aria-hidden="true" tabindex="-1"></a>clean_LMISS <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/rawdata/clean_inds_data.lmiss&quot;</span>))</span>
<span id="cb78-6"><a href="gwas_basics_snp_qc.html#cb78-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-7"><a href="gwas_basics_snp_qc.html#cb78-7" aria-hidden="true" tabindex="-1"></a>clean_LMISS<span class="sc">$</span>callrate <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> clean_LMISS<span class="sc">$</span>F_MISS</span></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="gwas_basics_snp_qc.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb79-2"><a href="gwas_basics_snp_qc.html#cb79-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-3"><a href="gwas_basics_snp_qc.html#cb79-3" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(clean_LMISS, <span class="at">x =</span> <span class="st">&quot;callrate&quot;</span>,</span>
<span id="cb79-4"><a href="gwas_basics_snp_qc.html#cb79-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb79-5"><a href="gwas_basics_snp_qc.html#cb79-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>, <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb79-6"><a href="gwas_basics_snp_qc.html#cb79-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb79-7"><a href="gwas_basics_snp_qc.html#cb79-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;per SNP call rate&quot;</span>) <span class="sc">+</span></span>
<span id="cb79-8"><a href="gwas_basics_snp_qc.html#cb79-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb79-9"><a href="gwas_basics_snp_qc.html#cb79-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-snp-callrate"></span>
<img src="img/show-snp-callrate.png" alt="Per SNP call rate." width="672" />
<p class="caption">
Figure 5.1: Per SNP call rate.
</p>
</div>
</div>
<div id="differential-snp-call-rates" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Differential SNP call rates<a href="gwas_basics_snp_qc.html#differential-snp-call-rates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There could also be differences in genotype call rates between cases and controls. It is very important to check for this because these differences could lead to spurious associations. We can test all markers for differences in call rate between cases and controls, or based on</p>
<pre><code>plink --bfile rawdata/clean_inds_data --test-missing --out rawdata/clean_inds_data
</code></pre>
<p>Let’s collect all the SNPs ith a significantly different (P &lt; 0.00001) missing data rate between cases and controls.</p>
<pre><code>cat rawdata/clean_inds_data.missing | awk &#39;$5 &lt; 0.00001&#39; | awk &#39;{ print $2 }&#39; &gt; rawdata/fail-diffmiss-qc.txt</code></pre>
</div>
<div id="allele-frequencies" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Allele frequencies<a href="gwas_basics_snp_qc.html#allele-frequencies" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We should also get an idea on what the allele frequencies are in our dataset. Low frequent SNPs should probably be excluded, as these are uninformative when monomorphic (allele frequency = 0), or they may lead to spurious associations.</p>
<pre><code>plink --bfile rawdata/clean_inds_data --freq --out rawdata/clean_inds_data</code></pre>
<p>Let’s also plot these data. You can view the result below, and try it yourself (Figure <a href="gwas_basics_snp_qc.html#fig:show-freq">5.2</a>).</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="gwas_basics_snp_qc.html#cb83-1" aria-hidden="true" tabindex="-1"></a>clean_FREQ <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/rawdata/clean_inds_data.frq&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="gwas_basics_snp_qc.html#cb84-1" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(clean_FREQ, <span class="at">x =</span> <span class="st">&quot;MAF&quot;</span>,</span>
<span id="cb84-2"><a href="gwas_basics_snp_qc.html#cb84-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb84-3"><a href="gwas_basics_snp_qc.html#cb84-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb84-4"><a href="gwas_basics_snp_qc.html#cb84-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb84-5"><a href="gwas_basics_snp_qc.html#cb84-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;minor allele frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb84-6"><a href="gwas_basics_snp_qc.html#cb84-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.05</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb84-7"><a href="gwas_basics_snp_qc.html#cb84-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-freq"></span>
<img src="img/show-freq.png" alt="Minor allele frequency." width="672" />
<p class="caption">
Figure 5.2: Minor allele frequency.
</p>
</div>
<div id="a-note-on-allele-coding" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> A note on allele coding<a href="gwas_basics_snp_qc.html#a-note-on-allele-coding" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Oh, one more thing about alleles.</p>
<p><code>PLINK</code> codes alleles as follows:</p>
<p>A1 = minor allele, the least frequent allele
A2 = major allele, the most frequent allele</p>
<p>And when you use <code>PLINK</code> the flag <code>--freq</code> or <code>--maf</code> is always relative to the A1-allele, as is the odds ratio (OR) or effect size (beta).</p>
<p>However, <code>SNPTEST</code> makes use of the so-called OXFORD-format, this codes alleles as follows:</p>
<p>A = the ‘other’ allele
B = the ‘coded’ allele</p>
<p>When you use <code>SNPTEST</code> it will report the allele frequency as <code>CAF</code>, in other words the <em>coded allele frequency</em>, and the effect size (beta) is always relative to the B-allele. This means, <code>CAF</code> <em>could</em> be the <code>MAF</code>, or <em>minor allele frequency</em>, but this is <strong>not</strong> a given.</p>
<p>In other words, always make sure what the allele-coding of a given program, be it <code>PLINK</code>, <code>SNPTEST</code>, <code>GCTA</code>, et cetera, is! I cannot stress this enough. Ask yourself: ‘what is the allele frequency referring to?’, ‘the effect size is relative to…?’.</p>
<p>Right, let’s continue.</p>
</div>
</div>
<div id="hardy-weinberg-equilibrium" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Hardy-Weinberg Equilibrium<a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Because we are performing a case-control genome-wide association study, we probably expect some differences in Hardy-Weinberg Equilibrium (HWE), but extreme deviations are probably indicative of genotyping errors.</p>
<pre><code>plink --bfile rawdata/clean_inds_data --hardy --out rawdata/clean_inds_data</code></pre>
<p>Let’s also plot these data. You can view the result below, and type over the code to do it yourself.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="gwas_basics_snp_qc.html#cb86-1" aria-hidden="true" tabindex="-1"></a>clean_HWE <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/rawdata/clean_inds_data.hwe&quot;</span>))</span>
<span id="cb86-2"><a href="gwas_basics_snp_qc.html#cb86-2" aria-hidden="true" tabindex="-1"></a>clean_HWE<span class="sc">$</span>logP <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(clean_HWE<span class="sc">$</span>P)</span></code></pre></div>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="gwas_basics_snp_qc.html#cb87-1" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(clean_HWE, <span class="at">x =</span> <span class="st">&quot;logP&quot;</span>,</span>
<span id="cb87-2"><a href="gwas_basics_snp_qc.html#cb87-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>,</span>
<span id="cb87-3"><a href="gwas_basics_snp_qc.html#cb87-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb87-4"><a href="gwas_basics_snp_qc.html#cb87-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb87-5"><a href="gwas_basics_snp_qc.html#cb87-5" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># color = &quot;#1290D9&quot;, fill = &quot;#1290D9&quot;,</span></span>
<span id="cb87-6"><a href="gwas_basics_snp_qc.html#cb87-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;TEST&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;TEST&quot;</span>,</span>
<span id="cb87-7"><a href="gwas_basics_snp_qc.html#cb87-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">palette =</span> <span class="st">&quot;lancet&quot;</span>,</span>
<span id="cb87-8"><a href="gwas_basics_snp_qc.html#cb87-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">facet.by =</span> <span class="st">&quot;TEST&quot;</span>,</span>
<span id="cb87-9"><a href="gwas_basics_snp_qc.html#cb87-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb87-10"><a href="gwas_basics_snp_qc.html#cb87-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;HWE -log10(P)&quot;</span>) <span class="sc">+</span></span>
<span id="cb87-11"><a href="gwas_basics_snp_qc.html#cb87-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">5</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb87-12"><a href="gwas_basics_snp_qc.html#cb87-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-hwe"></span>
<img src="img/show-hwe.png" alt="Hardy-Weinberg Equilibrium p-values per stratum." width="672" />
<p class="caption">
Figure 5.3: Hardy-Weinberg Equilibrium p-values per stratum.
</p>
</div>
</div>
<div id="final-snp-qc" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Final SNP QC<a href="gwas_basics_snp_qc.html#final-snp-qc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are ready to perform the final QC. After inspectig the graphs we will filter on a MAF &lt; 0.01, call rate &lt; 0.05, and HWE &lt; 0.00001, in addition those SNPs that failed the differential call rate test will be removed.</p>
<pre><code>plink --bfile rawdata/clean_inds_data --exclude rawdata/fail-diffmiss-qc.txt --maf 0.01 --geno 0.05 --hwe 0.00001 --make-bed --out rawdata/cleandata</code></pre>
</div>
<div id="a-milestone" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> A Milestone<a href="gwas_basics_snp_qc.html#a-milestone" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Congratulations. You reached a very important milestone. Now that you filtered samples and SNPs, we can finally start the association analyses in Chapter <a href="gwas-testing.html#gwas-testing">6</a>.</p>
<p>Before we move on, you could clean up our mess a bit.</p>
<pre><code>rm -v rawdata/rawdata.* rawdata/rawdata_1kg_phase1_raw_no_atcg*</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas-basics-sample-qc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gwas-testing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": ["twitter", "linkedin", "whatsapp"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics/edit/master/gwas_basics_snp_qc.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
