<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 GWAS visualisation | A primer in Human Cardiovascular Genetics</title>
  <meta name="description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 GWAS visualisation | A primer in Human Cardiovascular Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 GWAS visualisation | A primer in Human Cardiovascular Genetics" />
  
  <meta name="twitter:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genome-wide-association-study.html"/>
<link rel="next" href="the-welcome-trust-case-control-consortium-wtccc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A primer in Human Cardiovascular Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.5</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>3</b> Basics of a Genome-Wide Association Study (GWAS)<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Let’s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sample-qc.html"><a href="sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="sample-qc.html"><a href="sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="sample-qc.html"><a href="sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>4.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="sample-qc.html"><a href="sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="sample-qc.html"><a href="sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="sample-qc.html"><a href="sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="sample-qc.html"><a href="sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="sample-qc.html"><a href="sample-qc.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.5.2</b> Principal Component Analysis<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="sample-qc.html"><a href="sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.3</b> 1000G phase 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="sample-qc.html"><a href="sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="sample-qc.html"><a href="sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="per-snp-qc.html"><a href="per-snp-qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="per-snp-qc.html"><a href="per-snp-qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="per-snp-qc.html"><a href="per-snp-qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="per-snp-qc.html"><a href="per-snp-qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="per-snp-qc.html"><a href="per-snp-qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="per-snp-qc.html"><a href="per-snp-qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="per-snp-qc.html"><a href="per-snp-qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html"><i class="fa fa-check"></i><b>6</b> Genome-wide association study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Let’s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#regional-association-plots"><i class="fa fa-check"></i><b>7.4</b> Regional association plots<span></span></a></li>
<li class="chapter" data-level="7.5" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#stop-playing-around"><i class="fa fa-check"></i><b>7.5</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium (WTCCC)<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#genotyping"><i class="fa fa-check"></i><b>8.1</b> Genotyping<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#the-data"><i class="fa fa-check"></i><b>8.2</b> The data<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#assignment"><i class="fa fa-check"></i><b>8.3</b> Assignment<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#there-you-go"><i class="fa fa-check"></i><b>8.4</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html"><i class="fa fa-check"></i><b>9</b> Mendelian Randomization (MR)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-1"><i class="fa fa-check"></i><b>9.1</b> Section 1<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-2"><i class="fa fa-check"></i><b>9.2</b> Section 2<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-3"><i class="fa fa-check"></i><b>9.3</b> Section 3<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>10</b> Epilogue<span></span></a></li>
<li class="chapter" data-level="11" data-path="licenses-and-disclaimers.html"><a href="licenses-and-disclaimers.html"><i class="fa fa-check"></i><b>11</b> Licenses and disclaimers<span></span></a></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A primer in Human Cardiovascular Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-visualisation" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> GWAS visualisation<a href="gwas-visualisation.html#gwas-visualisation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Data visualization is key, not only for presentation but also to inspect the results.</p>
<div id="qq-plots" class="section level3 hasAnchor" number="7.0.1">
<h3><span class="header-section-number">7.0.1</span> QQ plots<a href="gwas-visualisation.html#qq-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We should create <em>quantile-quantile (QQ) plots</em> to compare the observed association test statistics with their expected values under the null hypothesis of no association and so assess the number, magnitude and quality of true associations.</p>
<p>First, we will add the standard error, call rate, A2, and allele frequencies.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="gwas-visualisation.html#cb99-1" aria-hidden="true" tabindex="-1"></a>gwas_assoc_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_assoc, TEST <span class="sc">==</span> <span class="st">&quot;ADD&quot;</span>)</span>
<span id="cb99-2"><a href="gwas-visualisation.html#cb99-2" aria-hidden="true" tabindex="-1"></a>gwas_assoc_sub<span class="sc">$</span>TEST <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb99-3"><a href="gwas-visualisation.html#cb99-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-4"><a href="gwas-visualisation.html#cb99-4" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_FRQ, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;A2&quot;</span>, <span class="st">&quot;MAF&quot;</span>, <span class="st">&quot;NCHROBS&quot;</span>))</span>
<span id="cb99-5"><a href="gwas-visualisation.html#cb99-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-6"><a href="gwas-visualisation.html#cb99-6" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrq <span class="ot">&lt;-</span> <span class="fu">merge</span>(gwas_assoc_sub, temp, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>)</span>
<span id="cb99-7"><a href="gwas-visualisation.html#cb99-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-8"><a href="gwas-visualisation.html#cb99-8" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_LMISS, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;callrate&quot;</span>))</span>
<span id="cb99-9"><a href="gwas-visualisation.html#cb99-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-10"><a href="gwas-visualisation.html#cb99-10" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss <span class="ot">&lt;-</span> <span class="fu">merge</span>(gwas_assoc_subfrq, temp, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>)</span>
<span id="cb99-11"><a href="gwas-visualisation.html#cb99-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gwas_assoc_subfrqlmiss)</span></code></pre></div>
<pre><code>##           SNP CHR        BP A1 NMISS     OR    STAT       P A2    MAF NCHROBS
## 1: rs10000010   4  21227772  C  3996 1.0420  0.9010 0.36760  T 0.4258    7992
## 2: rs10000023   4  95952929  T  3957 0.9902 -0.2160 0.82900  G 0.4841    7914
## 3: rs10000030   4 103593179  A  3991 0.9779 -0.3696 0.71170  G 0.1616    7982
## 4:  rs1000007   2 237416793  C  4000 1.0180  0.3649 0.71520  T 0.3122    8000
## 5: rs10000092   4  21504615  C  3963 0.9240 -1.6770 0.09354  T 0.3430    7926
## 6: rs10000121   4 157793485  G  3919 0.9665 -0.7525 0.45170  A 0.4532    7838
##    callrate
## 1:  0.99900
## 2:  0.98925
## 3:  0.99775
## 4:  1.00000
## 5:  0.99075
## 6:  0.97975</code></pre>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="gwas-visualisation.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remember:</span></span>
<span id="cb101-2"><a href="gwas-visualisation.html#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - that z = beta/se</span></span>
<span id="cb101-3"><a href="gwas-visualisation.html#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - beta = log(OR), because log is the natural log in r</span></span>
<span id="cb101-4"><a href="gwas-visualisation.html#cb101-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-5"><a href="gwas-visualisation.html#cb101-5" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss<span class="sc">$</span>BETA <span class="ot">=</span> <span class="fu">log</span>(gwas_assoc_subfrqlmiss<span class="sc">$</span>OR)</span>
<span id="cb101-6"><a href="gwas-visualisation.html#cb101-6" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss<span class="sc">$</span>SE <span class="ot">=</span> gwas_assoc_subfrqlmiss<span class="sc">$</span>BETA<span class="sc">/</span>gwas_assoc_subfrqlmiss<span class="sc">$</span>STAT</span>
<span id="cb101-7"><a href="gwas-visualisation.html#cb101-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-8"><a href="gwas-visualisation.html#cb101-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-9"><a href="gwas-visualisation.html#cb101-9" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss_tib <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">as_tibble</span>(gwas_assoc_subfrqlmiss)</span>
<span id="cb101-10"><a href="gwas-visualisation.html#cb101-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-11"><a href="gwas-visualisation.html#cb101-11" aria-hidden="true" tabindex="-1"></a>col_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb101-12"><a href="gwas-visualisation.html#cb101-12" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;A1&quot;</span>, <span class="st">&quot;A2&quot;</span>, <span class="st">&quot;MAF&quot;</span>, <span class="st">&quot;callrate&quot;</span>, <span class="st">&quot;NMISS&quot;</span>, <span class="st">&quot;NCHROBS&quot;</span>,</span>
<span id="cb101-13"><a href="gwas-visualisation.html#cb101-13" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;BETA&quot;</span>, <span class="st">&quot;SE&quot;</span>, <span class="st">&quot;OR&quot;</span>, <span class="st">&quot;STAT&quot;</span>, <span class="st">&quot;P&quot;</span>)</span>
<span id="cb101-14"><a href="gwas-visualisation.html#cb101-14" aria-hidden="true" tabindex="-1"></a>gwas_assoc_compl <span class="ot">&lt;-</span> gwas_assoc_subfrqlmiss_tib[, col_order]</span>
<span id="cb101-15"><a href="gwas-visualisation.html#cb101-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-16"><a href="gwas-visualisation.html#cb101-16" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gwas_assoc_compl)</span></code></pre></div>
<pre><code>## [1] 306102     14</code></pre>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="gwas-visualisation.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gwas_assoc_compl)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 14
##   SNP        CHR     BP A1    A2      MAF callrate NMISS NCHROBS     BETA     SE
##   &lt;chr&gt;    &lt;int&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;int&gt;   &lt;int&gt;    &lt;dbl&gt;  &lt;dbl&gt;
## 1 rs10000…     4 2.12e7 C     T     0.426    0.999  3996    7992  0.0411  0.0457
## 2 rs10000…     4 9.60e7 T     G     0.484    0.989  3957    7914 -0.00985 0.0456
## 3 rs10000…     4 1.04e8 A     G     0.162    0.998  3991    7982 -0.0223  0.0605
## 4 rs10000…     2 2.37e8 C     T     0.312    1      4000    8000  0.0178  0.0489
## 5 rs10000…     4 2.15e7 C     T     0.343    0.991  3963    7926 -0.0790  0.0471
## 6 rs10000…     4 1.58e8 G     A     0.453    0.980  3919    7838 -0.0341  0.0453
## # … with 3 more variables: OR &lt;dbl&gt;, STAT &lt;dbl&gt;, P &lt;dbl&gt;</code></pre>
<p>Let’s list the number of SNPs per chromosome.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="gwas-visualisation.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of SNPs per chromosome</span></span>
<span id="cb105-2"><a href="gwas-visualisation.html#cb105-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">table</span>(gwas_assoc_compl<span class="sc">$</span>CHR))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">23173</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="right">25206</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="right">21402</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">19008</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="right">19157</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="right">20672</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="right">16581</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="right">18089</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="right">15709</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="right">15536</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="right">14564</td>
</tr>
<tr class="even">
<td align="left">12</td>
<td align="right">14889</td>
</tr>
<tr class="odd">
<td align="left">13</td>
<td align="right">11524</td>
</tr>
<tr class="even">
<td align="left">14</td>
<td align="right">9822</td>
</tr>
<tr class="odd">
<td align="left">15</td>
<td align="right">8838</td>
</tr>
<tr class="even">
<td align="left">16</td>
<td align="right">8920</td>
</tr>
<tr class="odd">
<td align="left">17</td>
<td align="right">8262</td>
</tr>
<tr class="even">
<td align="left">18</td>
<td align="right">10356</td>
</tr>
<tr class="odd">
<td align="left">19</td>
<td align="right">5820</td>
</tr>
<tr class="even">
<td align="left">20</td>
<td align="right">7792</td>
</tr>
<tr class="odd">
<td align="left">21</td>
<td align="right">5412</td>
</tr>
<tr class="even">
<td align="left">22</td>
<td align="right">5370</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="gwas-visualisation.html#cb106-1" aria-hidden="true" tabindex="-1"></a>gwas_threshold <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">5e-8</span>)</span>
<span id="cb106-2"><a href="gwas-visualisation.html#cb106-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-3"><a href="gwas-visualisation.html#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qq</span>(gwas_assoc_compl<span class="sc">$</span>P, <span class="at">main =</span> <span class="st">&quot;QQ plot of GWAS&quot;</span>,</span>
<span id="cb106-4"><a href="gwas-visualisation.html#cb106-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">7</span>),</span>
<span id="cb106-5"><a href="gwas-visualisation.html#cb106-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>),</span>
<span id="cb106-6"><a href="gwas-visualisation.html#cb106-6" aria-hidden="true" tabindex="-1"></a>   <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col =</span> uithof_color[<span class="dv">16</span>], <span class="at">cex =</span> <span class="fl">1.5</span>, <span class="at">las =</span> <span class="dv">1</span>, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>)</span>
<span id="cb106-7"><a href="gwas-visualisation.html#cb106-7" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> gwas_threshold,</span>
<span id="cb106-8"><a href="gwas-visualisation.html#cb106-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> uithof_color[<span class="dv">25</span>], <span class="at">lty =</span> <span class="st">&quot;dashed&quot;</span>)</span></code></pre></div>
<p><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
<div id="manhattan-plots" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Manhattan plots<a href="gwas-visualisation.html#manhattan-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We also need to create a <em>Manhattan plot</em> to display the association test P-values as a function of chromosomal location and thus provide a visual summary of association test results that draw immediate attention to any regions of significance.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="gwas-visualisation.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(gwas_assoc_compl, <span class="at">main =</span> <span class="st">&quot;Manhattan Plot&quot;</span>,</span>
<span id="cb107-2"><a href="gwas-visualisation.html#cb107-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>),</span>
<span id="cb107-3"><a href="gwas-visualisation.html#cb107-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">cex =</span> <span class="fl">0.6</span>, <span class="at">cex.axis =</span> <span class="fl">0.9</span>,</span>
<span id="cb107-4"><a href="gwas-visualisation.html#cb107-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#1290D9&quot;</span>, <span class="st">&quot;#49A01D&quot;</span>))</span></code></pre></div>
<p><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="gwas-visualisation.html#cb108-1" aria-hidden="true" tabindex="-1"></a>gwas_assoc_complsub <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_assoc_compl, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;BP&quot;</span>, <span class="st">&quot;P&quot;</span>))</span></code></pre></div>
</div>
<div id="other-plots" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Other plots<a href="gwas-visualisation.html#other-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is also informative to plot the density per chromosome. We can use the <code>CMplot</code> for that which you can find <a href="https://github.com/YinLiLin/R-CMplot" target="_blank">here</a>. For now we just make these graphs ‘quick-n-dirty’, you can further prettify them, but you easily loose track of time, so maybe carry on.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="gwas-visualisation.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">CMplot</span>(gwas_assoc_complsub,</span>
<span id="cb109-2"><a href="gwas-visualisation.html#cb109-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot.type =</span> <span class="st">&quot;d&quot;</span>,</span>
<span id="cb109-3"><a href="gwas-visualisation.html#cb109-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">bin.size =</span> <span class="fl">1e6</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;darkgreen&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb109-4"><a href="gwas-visualisation.html#cb109-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">file =</span> <span class="st">&quot;jpg&quot;</span>, <span class="at">memo =</span> <span class="st">&quot;&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">file.output =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##  SNP-Density Plotting.</code></pre>
<p><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="gwas-visualisation.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">CMplot</span>(gwas_assoc_complsub,</span>
<span id="cb111-2"><a href="gwas-visualisation.html#cb111-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot.type =</span> <span class="st">&quot;m&quot;</span>, <span class="at">LOG10 =</span> <span class="cn">TRUE</span>, <span class="at">ylim =</span> <span class="cn">NULL</span>,</span>
<span id="cb111-3"><a href="gwas-visualisation.html#cb111-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">threshold =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>,<span class="fl">1e-4</span>), <span class="at">threshold.lty =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">threshold.lwd =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">threshold.col =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb111-4"><a href="gwas-visualisation.html#cb111-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">amplify =</span> <span class="cn">TRUE</span>,</span>
<span id="cb111-5"><a href="gwas-visualisation.html#cb111-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">bin.size =</span> <span class="fl">1e6</span>, <span class="at">chr.den.col =</span> <span class="fu">c</span>(<span class="st">&quot;darkgreen&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb111-6"><a href="gwas-visualisation.html#cb111-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">signal.col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>), <span class="at">signal.cex =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">signal.pch =</span> <span class="fu">c</span>(<span class="dv">19</span>,<span class="dv">19</span>),</span>
<span id="cb111-7"><a href="gwas-visualisation.html#cb111-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">file =</span> <span class="st">&quot;jpg&quot;</span>, <span class="at">memo =</span> <span class="st">&quot;&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">file.output =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##  Rectangular-Manhattan Plotting P.</code></pre>
<p><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-20-2.png" width="672" /></p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="gwas-visualisation.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">CMplot</span>(gwas_assoc_complsub,</span>
<span id="cb113-2"><a href="gwas-visualisation.html#cb113-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot.type =</span> <span class="st">&quot;b&quot;</span>, <span class="at">LOG10 =</span> <span class="cn">TRUE</span>, <span class="at">ylim =</span> <span class="cn">NULL</span>,</span>
<span id="cb113-3"><a href="gwas-visualisation.html#cb113-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">threshold =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>,<span class="fl">1e-4</span>), <span class="at">threshold.lty =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">threshold.lwd =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">threshold.col =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb113-4"><a href="gwas-visualisation.html#cb113-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">amplify =</span> <span class="cn">TRUE</span>,</span>
<span id="cb113-5"><a href="gwas-visualisation.html#cb113-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">bin.size =</span> <span class="fl">1e6</span>, <span class="at">chr.den.col =</span> <span class="fu">c</span>(<span class="st">&quot;darkgreen&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb113-6"><a href="gwas-visualisation.html#cb113-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">signal.col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>), <span class="at">signal.cex =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">signal.pch =</span> <span class="fu">c</span>(<span class="dv">19</span>,<span class="dv">19</span>),</span>
<span id="cb113-7"><a href="gwas-visualisation.html#cb113-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">file =</span> <span class="st">&quot;jpg&quot;</span>, <span class="at">memo =</span> <span class="st">&quot;&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">file.output =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##  SNP-Density Plotting.</code></pre>
<p><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-20-3.png" width="672" /><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-20-4.png" width="672" /></p>
<pre><code>##  Circular-Manhattan Plotting P.
##  Rectangular-Manhattan Plotting P.</code></pre>
<p><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-20-5.png" width="672" /></p>
<pre><code>##  QQ Plotting P.</code></pre>
<p><img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/unnamed-chunk-20-6.png" width="672" /></p>
</div>
<div id="interactive-plots" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Interactive plots<a href="gwas-visualisation.html#interactive-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can also make an <a href="https://r-graph-gallery.com/101_Manhattan_plot.html" target="_blank">interactive version</a> of the Manhattan - just because you can. The code below shows you how.</p>
<pre><code>library(plotly)
library(dplyr)

# Prepare the dataset (as an example we use the data (gwasResults) from the `qqman`-package)
don &lt;- gwasResults %&gt;%

  # Compute chromosome size
  group_by(CHR) %&gt;%
  summarise(chr_len=max(BP)) %&gt;%

  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %&gt;%
  select(-chr_len) %&gt;%

  # Add this info to the initial dataset
  left_join(gwasResults, ., by=c(&quot;CHR&quot;=&quot;CHR&quot;)) %&gt;%

  # Add a cumulative position of each SNP
  arrange(CHR, BP) %&gt;%
  mutate( BPcum=BP+tot) %&gt;%

  # Add highlight and annotation information
  mutate( is_highlight=ifelse(SNP %in% snpsOfInterest, &quot;yes&quot;, &quot;no&quot;)) %&gt;%

  # Filter SNP to make the plot lighter
  filter(-log10(P)&gt;0.5)

# Prepare X axis
axisdf &lt;- don %&gt;% group_by(CHR) %&gt;% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

# Prepare text description for each SNP:
don$text &lt;- paste(&quot;SNP: &quot;, don$SNP, &quot;\nPosition: &quot;, don$BP, &quot;\nChromosome: &quot;, don$CHR, &quot;\nLOD score:&quot;, -log10(don$P) %&gt;% round(2), &quot;\nWhat else do you wanna know&quot;, sep=&quot;&quot;)

# Make the plot
p &lt;- ggplot(don, aes(x=BPcum, y=-log10(P), text=text)) +

    # Show all points
    geom_point( aes(color=as.factor(CHR)), alpha=0.8, size=1.3) +
    scale_color_manual(values = rep(c(&quot;grey&quot;, &quot;skyblue&quot;), 22 )) +

    # custom X axis:
    scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
    scale_y_continuous(expand = c(0, 0), ylim = c(0,9) ) +     # remove space between plot area and x axis

    # Add highlighted points
    geom_point(data=subset(don, is_highlight==&quot;yes&quot;), color=&quot;orange&quot;, size=2) +

    # Custom the theme:
    theme_bw() +
    theme(
      legend.position=&quot;none&quot;,
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
ggplotly(p, tooltip=&quot;text&quot;)</code></pre>
<p>It will produce something like this.</p>
<p><img src="img/_gwas/interactive%20plot.png" /></p>
<p>Again, this is an example with dummy data - you can try to do it for our GWAS, but careful with the time. You can also choose to carry on.</p>
</div>
<div id="regional-association-plots" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Regional association plots<a href="gwas-visualisation.html#regional-association-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can further visualise regions of interest using a package like <a href="http://locuszoom.org">LocusZoom</a>. But first we need to find the independent hits by <em>clumping</em> the results. We will just use the defaults, but please take a note of all the options here <a href="https://www.cog-genomics.org/plink/1.9/postproc#clump" target="_blank">https://www.cog-genomics.org/plink/1.9/postproc#clump</a></p>
<pre><code>plink --bfile gwas/gwa --clump gwas/data.assoc.logistic --clump-p1 5e-8 --clump-p2 0.05 --clump-kb 500 --clump-r2 0.05 --clump-verbose --out gwas/data.assoc.logistic</code></pre>
<p>Now you will have a list of all the <em>independent</em> SNPs, <em>i.e.</em> the genetic loci, that are associated to the trait.</p>
<pre><code>cat gwas/data.assoc.logistic.clumped</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb120-1"><a href="gwas-visualisation.html#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="va">ROOTDIR</span><span class="op">=</span><span class="st">&quot;/Users/swvanderlaan/Desktop/practical&quot;</span> <span class="co"># change this to your root</span></span>
<span id="cb120-2"><a href="gwas-visualisation.html#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">$ROOTDIR</span>/gwas/data.assoc.logistic.clumped</span></code></pre></div>
<pre><code>## 
##  CHR    F         SNP         BP          P    TOTAL   NSIG    S05    S01   S001  S0001
##    3    1   rs6802898   12366207   4.18e-20       50     35      4      2      1      8 
## 
##                               KB      RSQ  ALLELES    F            P 
##   (INDEX)   rs6802898          0    1.000        T    1     4.18e-20 
## 
##              rs305500       -400   0.0588    TC/CA    1       0.0476 
##              rs420014       -394   0.0552    TA/CG    1        0.015 
##              rs305494       -392   0.0681    TG/CA    1        0.025 
##              rs438129       -383   0.0721    TT/CC    1       0.0126 
##             rs7615580       -364    0.309    TC/CT    1     2.05e-08 
##              rs307560       -298    0.153    TT/CC    1     1.68e-06 
##            rs11720130       -235   0.0838    TA/CG    1      0.00218 
##             rs7616006       -124   0.0831    TA/CG    1     5.25e-05 
##             rs6775191       -119   0.0506    TG/CA    1     0.000182 
##              rs167466       -110   0.0523    TT/CC    1      0.00222 
##            rs12635120      -86.6    0.332    TG/CA    1     2.77e-07 
##             rs6798713      -85.6    0.288    TC/CT    1     2.01e-06 
##             rs2920500      -67.8    0.102    TA/CG    1     6.59e-05 
##             rs6768587      -53.1    0.295    TG/CA    1     3.36e-08 
##             rs2028760      -18.3    0.305    TA/CG    1     3.67e-08 
## 
##           RANGE: chr3:11966007..12366207
##            SPAN: 400kb
## 
## ------------------------------------------------------------------
## 
## 
##  CHR    F         SNP         BP          P    TOTAL   NSIG    S05    S01   S001  S0001
##   10    1   rs7901695  114744078   6.78e-12       32     24      2      1      1      4 
## 
##                               KB      RSQ  ALLELES    F            P 
##   (INDEX)   rs7901695          0    1.000        C    1     6.78e-12 
## 
##             rs7917983      -21.2   0.0589    CC/TT    1        0.046 
##             rs7895307      -10.1   0.0819    CG/TA    1      0.00592 
##             rs7903146       4.26    0.784    CT/TC    1     3.25e-08 
##             rs7904519       19.8    0.582    CG/TA    1     2.89e-08 
##            rs11196192       28.2    0.162    CG/TT    1       0.0268 
##            rs10885409         54    0.502    CC/TT    1     8.35e-06 
##            rs12255372       54.8    0.624    CT/TG    1     1.55e-06 
##             rs4918789       67.7     0.24    CG/TT    1     0.000248 
## 
##           RANGE: chr10:114722872..114811797
##            SPAN: 88kb
## 
## ------------------------------------------------------------------
## 
## 
##  CHR    F         SNP         BP          P    TOTAL   NSIG    S05    S01   S001  S0001
##   16    1   rs8050136   52373776   1.52e-08       23     16      1      3      2      1 
## 
##                               KB      RSQ  ALLELES    F            P 
##   (INDEX)   rs8050136          0    1.000        A    1     1.52e-08 
## 
##             rs7205986      -61.1    0.226    AG/CA    1       0.0434 
##             rs6499640      -46.6    0.258    AA/CG    1      0.00367 
##             rs1861868      -25.9     0.15    AT/CC    1       0.0063 
##             rs1075440      -25.4    0.162    AA/CG    1      0.00802 
##             rs3751812       2.18    0.994    AT/CG    1     1.63e-08 
##             rs7190492       12.5    0.258    AG/CA    1       0.0007 
##             rs8044769       22.9    0.524    AC/CT    1     0.000611 
## 
##           RANGE: chr16:52312647..52396636
##            SPAN: 83kb
## 
## ------------------------------------------------------------------</code></pre>
<p>Clumping identifies three loci and now that you know them, you can visualize them using LocusZoom. First, let’s get what we need (<code>SNP</code> and <code>P</code>) and gzip the results.</p>
<pre><code>echo &quot;SNP P&quot; &gt; gwas/data.assoc.logistic.locuszoom
cat gwas/data.assoc.logistic | awk &#39;$5==&quot;ADD&quot;&#39; | awk &#39;{ print $2, $9 }&#39; &gt;&gt; gwas/data.assoc.logistic.locuszoom
gzip -v gwas/data.assoc.logistic.locuszoom</code></pre>
<p>Now you are ready to upload this <code>data.assoc.logistic.locuszoom.gz</code> file to the site: <a href="http://locuszoom.org" class="uri">http://locuszoom.org</a>. Try to visualize each locus using the information above and by following the instructions. Choose HapMap 2, hg18, CEU as the LD-reference.</p>
<p>You should get something like below.</p>
<p><img src="img/_gwas/CEU.rs6802898.400kb.png" style="width:50.0%" /></p>
<p><img src="img/_gwas/CEU.rs7901695.400kb.png" style="width:50.0%" /></p>
<p><img src="img/_gwas/CEU.rs8050136.400kb.png" style="width:50.0%" /></p>
<p>You will encounter the above three types of visualizations in any high-quality GWAS paper, because each is so critically informative. Usually, analysts of large-scale meta-analyses of GWAS will also stratify the QQ-plots based on the imputation quality (if your GWAS was imputed), call rate, and allele frequency.</p>
</div>
<div id="stop-playing-around" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Stop playing around<a href="gwas-visualisation.html#stop-playing-around" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Alright. It’s time to stop playing around and do a quick recap of what you’ve learned.</p>
<ol style="list-style-type: decimal">
<li>You learned how to convert datasets.</li>
<li>You learned how to execute sample QC and create diagnostic graphics</li>
<li>You learned how to do the same for SNP QC</li>
<li>You learned how to execute an association study given a dataset, covariates, and different assumptions regarding the genetic model.</li>
<li>You learned how to visualize results and played around with different visuals.</li>
</ol>
<p>You should be ready for the real stuff. And if not, the next chapter will help you get ready: Chapter @ref(wtccc1_intro).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genome-wide-association-study.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="the-welcome-trust-case-control-consortium-wtccc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics/edit/master/gwas_basics_results_visualization.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["A primer in Human Cardiovascular Genetics.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
