<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 GWAS visualisation | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 GWAS visualisation | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 GWAS visualisation | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas-testing.html"/>
<link rel="next" href="wtccc1-intro.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.5</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>3</b> Steps in a Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Let’s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>4.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.2</b> 1000G phase 1<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#running-pca"><i class="fa fa-check"></i><b>4.5.3</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="4.5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#pca-plotting"><i class="fa fa-check"></i><b>4.5.4</b> PCA plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>5.3.1</b> A note on allele coding<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-testing.html"><a href="gwas-testing.html"><i class="fa fa-check"></i><b>6</b> Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-testing.html"><a href="gwas-testing.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="gwas-testing.html"><a href="gwas-testing.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="gwas-testing.html"><a href="gwas-testing.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="gwas-testing.html"><a href="gwas-testing.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Let’s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>7.4</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>8.1</b> Genotyping<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>8.2</b> The data<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>8.3</b> Assignment<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>8.4</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>9</b> WTCCC1: a GWAS on coronary artery disease (CAD)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>9.1</b> Quality control<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>9.2</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>9.2.1</b> Preparing PCA<span></span></a></li>
<li class="chapter" data-level="9.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca-1"><i class="fa fa-check"></i><b>9.2.2</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="9.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>9.2.3</b> Plotting PCA<span></span></a></li>
<li class="chapter" data-level="9.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>9.2.4</b> Removing samples<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>9.3</b> Association testing<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>10</b> Post-GWAS Analyses<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>10.1</b> Clumping<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>10.2</b> Conditional analyses<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>10.3</b> Statistical finemapping<span></span></a></li>
<li class="chapter" data-level="10.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>10.4</b> FUMA: FUnctional Mapping and Annotation of GWAS<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>10.5</b> PheWAS<span></span></a></li>
<li class="chapter" data-level="10.6" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>10.6</b> Genetic correlation<span></span></a></li>
<li class="chapter" data-level="10.7" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>10.7</b> Causal inference: Mendelian Randomization<span></span></a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="post-gwas.html"><a href="post-gwas.html#your-choice"><i class="fa fa-check"></i><b>10.7.1</b> Your choice<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>10.8</b> Polygenic scores<span></span></a></li>
<li class="chapter" data-level="10.9" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>10.9</b> What comes next<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>11</b> Functional Mapping and Annotation of GWAS<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>11.1</b> Assignment<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>11.1.1</b> Tutorial<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>11.1.2</b> Create the input<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>11.1.3</b> Run FUMA - SNP to gene<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>11.1.4</b> Run FUMA - Gene to function<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>11.2</b> Some closing thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>12</b> Phenome-Wide Association Study (PheWAS)<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>12.1</b> What’s next?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>13</b> MRBase<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>13.1</b> The end?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>14</b> Epilogue<span></span></a></li>
<li class="chapter" data-level="15" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html"><i class="fa fa-check"></i><b>15</b> Additional: regional association plots<span></span></a></li>
<li class="chapter" data-level="16" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>16</b> Additional: EIGENSOFT<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>16.1</b> Install homebrew<span></span></a></li>
<li class="chapter" data-level="16.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>16.2</b> Install missing packages<span></span></a></li>
<li class="chapter" data-level="16.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>16.3</b> Installing EIGENSOFT<span></span></a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>17</b> Things to do<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="todo.html"><a href="todo.html#moscow"><i class="fa fa-check"></i><b>17.1</b> MoSCoW<span></span></a></li>
<li class="chapter" data-level="17.2" data-path="todo.html"><a href="todo.html#book-fixes"><i class="fa fa-check"></i><b>17.2</b> Book fixes<span></span></a></li>
<li class="chapter" data-level="17.3" data-path="todo.html"><a href="todo.html#useful-links"><i class="fa fa-check"></i><b>17.3</b> Useful links<span></span></a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>18</b> Licenses and disclaimers<span></span></a>
<ul>
<li class="chapter" data-level="18.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>18.1</b> Copyright<span></span></a></li>
<li class="chapter" data-level="18.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>18.2</b> Disclaimer<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-visuals" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> GWAS visualisation<a href="gwas-visuals.html#gwas-visuals" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Data visualization is key, not only for presentation but also to inspect the results.</p>
<div id="qq-plots" class="section level3 hasAnchor" number="7.0.1">
<h3><span class="header-section-number">7.0.1</span> QQ plots<a href="gwas-visuals.html#qq-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We should create <em>quantile-quantile (QQ) plots</em> to compare the observed association test statistics with their expected values under the null hypothesis of no association and so assess the number, magnitude and quality of true associations.</p>
<p>First, we will add the standard error, call rate, A2, and allele frequencies.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="gwas-visuals.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb104-2"><a href="gwas-visuals.html#cb104-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-3"><a href="gwas-visuals.html#cb104-3" aria-hidden="true" tabindex="-1"></a>COURSE_loc <span class="ot">=</span> <span class="st">&quot;~/Desktop/practical&quot;</span> <span class="co"># getwd()</span></span>
<span id="cb104-4"><a href="gwas-visuals.html#cb104-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-5"><a href="gwas-visuals.html#cb104-5" aria-hidden="true" tabindex="-1"></a>gwas_assoc_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_assoc, TEST <span class="sc">==</span> <span class="st">&quot;ADD&quot;</span>)</span>
<span id="cb104-6"><a href="gwas-visuals.html#cb104-6" aria-hidden="true" tabindex="-1"></a>gwas_assoc_sub<span class="sc">$</span>TEST <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb104-7"><a href="gwas-visuals.html#cb104-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-8"><a href="gwas-visuals.html#cb104-8" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_FRQ, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;A2&quot;</span>, <span class="st">&quot;MAF&quot;</span>, <span class="st">&quot;NCHROBS&quot;</span>))</span>
<span id="cb104-9"><a href="gwas-visuals.html#cb104-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-10"><a href="gwas-visuals.html#cb104-10" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrq <span class="ot">&lt;-</span> <span class="fu">merge</span>(gwas_assoc_sub, temp, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>)</span>
<span id="cb104-11"><a href="gwas-visuals.html#cb104-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-12"><a href="gwas-visuals.html#cb104-12" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_LMISS, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;callrate&quot;</span>))</span>
<span id="cb104-13"><a href="gwas-visuals.html#cb104-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-14"><a href="gwas-visuals.html#cb104-14" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss <span class="ot">&lt;-</span> <span class="fu">merge</span>(gwas_assoc_subfrq, temp, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>)</span>
<span id="cb104-15"><a href="gwas-visuals.html#cb104-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gwas_assoc_subfrqlmiss)</span>
<span id="cb104-16"><a href="gwas-visuals.html#cb104-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Remember:</span></span>
<span id="cb104-17"><a href="gwas-visuals.html#cb104-17" aria-hidden="true" tabindex="-1"></a><span class="co"># - that z = beta/se</span></span>
<span id="cb104-18"><a href="gwas-visuals.html#cb104-18" aria-hidden="true" tabindex="-1"></a><span class="co"># - beta = log(OR), because log is the natural log in r</span></span>
<span id="cb104-19"><a href="gwas-visuals.html#cb104-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-20"><a href="gwas-visuals.html#cb104-20" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss<span class="sc">$</span>BETA <span class="ot">=</span> <span class="fu">log</span>(gwas_assoc_subfrqlmiss<span class="sc">$</span>OR)</span>
<span id="cb104-21"><a href="gwas-visuals.html#cb104-21" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss<span class="sc">$</span>SE <span class="ot">=</span> gwas_assoc_subfrqlmiss<span class="sc">$</span>BETA<span class="sc">/</span>gwas_assoc_subfrqlmiss<span class="sc">$</span>STAT</span>
<span id="cb104-22"><a href="gwas-visuals.html#cb104-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-23"><a href="gwas-visuals.html#cb104-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-24"><a href="gwas-visuals.html#cb104-24" aria-hidden="true" tabindex="-1"></a>gwas_assoc_subfrqlmiss_tib <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">as_tibble</span>(gwas_assoc_subfrqlmiss)</span>
<span id="cb104-25"><a href="gwas-visuals.html#cb104-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-26"><a href="gwas-visuals.html#cb104-26" aria-hidden="true" tabindex="-1"></a>col_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb104-27"><a href="gwas-visuals.html#cb104-27" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;A1&quot;</span>, <span class="st">&quot;A2&quot;</span>, <span class="st">&quot;MAF&quot;</span>, <span class="st">&quot;callrate&quot;</span>, <span class="st">&quot;NMISS&quot;</span>, <span class="st">&quot;NCHROBS&quot;</span>,</span>
<span id="cb104-28"><a href="gwas-visuals.html#cb104-28" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;BETA&quot;</span>, <span class="st">&quot;SE&quot;</span>, <span class="st">&quot;OR&quot;</span>, <span class="st">&quot;STAT&quot;</span>, <span class="st">&quot;P&quot;</span>)</span>
<span id="cb104-29"><a href="gwas-visuals.html#cb104-29" aria-hidden="true" tabindex="-1"></a>gwas_assoc_compl <span class="ot">&lt;-</span> gwas_assoc_subfrqlmiss_tib[, col_order]</span>
<span id="cb104-30"><a href="gwas-visuals.html#cb104-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-31"><a href="gwas-visuals.html#cb104-31" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gwas_assoc_compl)</span>
<span id="cb104-32"><a href="gwas-visuals.html#cb104-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-33"><a href="gwas-visuals.html#cb104-33" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gwas_assoc_compl)</span></code></pre></div>
<p>Let’s list the number of SNPs per chromosome. This gives a pretty good idea about the per-chromosome coverage. And it’s a sanity check: did the whole analysis run properly (we expect 22 chromosomes)?</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="gwas-visuals.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;knitr&quot;</span>)</span>
<span id="cb105-2"><a href="gwas-visuals.html#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="gwas-visuals.html#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of SNPs per chromosome</span></span>
<span id="cb105-4"><a href="gwas-visuals.html#cb105-4" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">table</span>(gwas_assoc_compl<span class="sc">$</span>CHR))</span></code></pre></div>
<p>Let’s plot the QQ plot to diagnose our GWAS.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="gwas-visuals.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;qqman&quot;</span>)</span>
<span id="cb106-2"><a href="gwas-visuals.html#cb106-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-3"><a href="gwas-visuals.html#cb106-3" aria-hidden="true" tabindex="-1"></a>gwas_threshold <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">5e-8</span>)</span>
<span id="cb106-4"><a href="gwas-visuals.html#cb106-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-5"><a href="gwas-visuals.html#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="fu">qq</span>(gwas_assoc_compl<span class="sc">$</span>P, <span class="at">main =</span> <span class="st">&quot;QQ plot of GWAS&quot;</span>,</span>
<span id="cb106-6"><a href="gwas-visuals.html#cb106-6" aria-hidden="true" tabindex="-1"></a>   <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">7</span>),</span>
<span id="cb106-7"><a href="gwas-visuals.html#cb106-7" aria-hidden="true" tabindex="-1"></a>   <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>),</span>
<span id="cb106-8"><a href="gwas-visuals.html#cb106-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col =</span> uithof_color[<span class="dv">16</span>], <span class="at">cex =</span> <span class="fl">1.5</span>, <span class="at">las =</span> <span class="dv">1</span>, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>)</span>
<span id="cb106-9"><a href="gwas-visuals.html#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> gwas_threshold,</span>
<span id="cb106-10"><a href="gwas-visuals.html#cb106-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> uithof_color[<span class="dv">25</span>], <span class="at">lty =</span> <span class="st">&quot;dashed&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-qq"></span>
<img src="img/_gwas/show-qq.png" alt="A QQ plot." width="672" />
<p class="caption">
Figure 7.1: A QQ plot.
</p>
</div>
</div>
<div id="manhattan-plots" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Manhattan plots<a href="gwas-visuals.html#manhattan-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We also need to create a <em>Manhattan plot</em> to display the association test P-values as a function of chromosomal location and thus provide a visual summary of association test results that draw immediate attention to any regions of significance (Figure <a href="gwas-visuals.html#fig:show-manhattan">7.2</a>).</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="gwas-visuals.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(gwas_assoc_compl, <span class="at">main =</span> <span class="st">&quot;Manhattan Plot&quot;</span>,</span>
<span id="cb107-2"><a href="gwas-visuals.html#cb107-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>),</span>
<span id="cb107-3"><a href="gwas-visuals.html#cb107-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">cex =</span> <span class="fl">0.6</span>, <span class="at">cex.axis =</span> <span class="fl">0.9</span>,</span>
<span id="cb107-4"><a href="gwas-visuals.html#cb107-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#1290D9&quot;</span>, <span class="st">&quot;#49A01D&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-manhattan"></span>
<img src="img/_gwas/show-manhattan.png" alt="A manhattan plot." width="672" />
<p class="caption">
Figure 7.2: A manhattan plot.
</p>
</div>
</div>
<div id="other-plots" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Other plots<a href="gwas-visuals.html#other-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is also informative to plot the density per chromosome. We can use the <code>CMplot</code> for that which you can find <a href="https://github.com/YinLiLin/R-CMplot" target="_blank">here</a>. For now we just make these graphs ‘quick-n-dirty’, you can further prettify them, but you easily loose track of time, so maybe carry on.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="gwas-visuals.html#cb108-1" aria-hidden="true" tabindex="-1"></a>gwas_assoc_complsub <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_assoc_compl, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;BP&quot;</span>, <span class="st">&quot;P&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="gwas-visuals.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;CMplot&quot;</span>)</span>
<span id="cb109-2"><a href="gwas-visuals.html#cb109-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-3"><a href="gwas-visuals.html#cb109-3" aria-hidden="true" tabindex="-1"></a><span class="fu">CMplot</span>(gwas_assoc_complsub,</span>
<span id="cb109-4"><a href="gwas-visuals.html#cb109-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot.type =</span> <span class="st">&quot;b&quot;</span>, <span class="at">LOG10 =</span> <span class="cn">TRUE</span>, <span class="at">ylim =</span> <span class="cn">NULL</span>,</span>
<span id="cb109-5"><a href="gwas-visuals.html#cb109-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">threshold =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>,<span class="fl">1e-4</span>), <span class="at">threshold.lty =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">threshold.lwd =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">threshold.col =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb109-6"><a href="gwas-visuals.html#cb109-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">amplify =</span> <span class="cn">TRUE</span>,</span>
<span id="cb109-7"><a href="gwas-visuals.html#cb109-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">bin.size =</span> <span class="fl">1e6</span>, <span class="at">chr.den.col =</span> <span class="fu">c</span>(<span class="st">&quot;darkgreen&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb109-8"><a href="gwas-visuals.html#cb109-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">signal.col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>), <span class="at">signal.cex =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">signal.pch =</span> <span class="fu">c</span>(<span class="dv">19</span>,<span class="dv">19</span>),</span>
<span id="cb109-9"><a href="gwas-visuals.html#cb109-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">file =</span> <span class="st">&quot;jpg&quot;</span>, <span class="at">memo =</span> <span class="st">&quot;&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">file.output =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<blockquote>
<p>What do the grey spots on the density plot indicate?</p>
</blockquote>
This would lead to the following graphs.
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-cmplot-all-density"></span>
<img src="img/_gwas/show-cmplot-all-density.png" alt="SNP density of the association results." width="672" />
<p class="caption">
Figure 7.3: SNP density of the association results.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-cmplot-all-qq"></span>
<img src="img/_gwas/show-cmplot-all-qq.png" alt="A QQ plot including a 95% confidence interval (blue area) and genome-wide significant hits (red)." width="672" />
<p class="caption">
Figure 7.4: A QQ plot including a 95% confidence interval (blue area) and genome-wide significant hits (red).
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-cmplot-all-manhattan"></span>
<img src="img/_gwas/show-cmplot-all-manhattan.png" alt="A regular manhattan plot. Colored by chromosome, suggestive hits are green, genome-wide hits are red. The bottom graph shows the per-chromosome SNP density." width="672" />
<p class="caption">
Figure 7.5: A regular manhattan plot. Colored by chromosome, suggestive hits are green, genome-wide hits are red. The bottom graph shows the per-chromosome SNP density.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-cmplot-all-circular"></span>
<img src="img/_gwas/show-cmplot-all-circular.png" alt="A circular manhattan." width="672" />
<p class="caption">
Figure 7.6: A circular manhattan.
</p>
</div>
</div>
<div id="interactive-plots" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Interactive plots<a href="gwas-visuals.html#interactive-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can also make an <a href="https://r-graph-gallery.com/101_Manhattan_plot.html" target="_blank">interactive version</a> of the Manhattan - just because you can. The code below shows you how.</p>
<pre><code>library(plotly)
library(dplyr)

# Prepare the dataset (as an example we use the data (gwasResults) from the `qqman`-package)
don &lt;- gwasResults %&gt;%

  # Compute chromosome size
  group_by(CHR) %&gt;%
  summarise(chr_len=max(BP)) %&gt;%

  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %&gt;%
  select(-chr_len) %&gt;%

  # Add this info to the initial dataset
  left_join(gwasResults, ., by=c(&quot;CHR&quot;=&quot;CHR&quot;)) %&gt;%

  # Add a cumulative position of each SNP
  arrange(CHR, BP) %&gt;%
  mutate( BPcum=BP+tot) %&gt;%

  # Add highlight and annotation information
  mutate( is_highlight=ifelse(SNP %in% snpsOfInterest, &quot;yes&quot;, &quot;no&quot;)) %&gt;%

  # Filter SNP to make the plot lighter
  filter(-log10(P)&gt;0.5)

# Prepare X axis
axisdf &lt;- don %&gt;% group_by(CHR) %&gt;% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

# Prepare text description for each SNP:
don$text &lt;- paste(&quot;SNP: &quot;, don$SNP, &quot;\nPosition: &quot;, don$BP, &quot;\nChromosome: &quot;, don$CHR, &quot;\nLOD score:&quot;, -log10(don$P) %&gt;% round(2), &quot;\nWhat else do you wanna know&quot;, sep=&quot;&quot;)

# Make the plot
p &lt;- ggplot(don, aes(x=BPcum, y=-log10(P), text=text)) +

    # Show all points
    geom_point( aes(color=as.factor(CHR)), alpha=0.8, size=1.3) +
    scale_color_manual(values = rep(c(&quot;grey&quot;, &quot;skyblue&quot;), 22 )) +

    # custom X axis:
    scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
    scale_y_continuous(expand = c(0, 0), ylim = c(0,9) ) +     # remove space between plot area and x axis

    # Add highlighted points
    geom_point(data=subset(don, is_highlight==&quot;yes&quot;), color=&quot;orange&quot;, size=2) +

    # Custom the theme:
    theme_bw() +
    theme(
      legend.position=&quot;none&quot;,
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
ggplotly(p, tooltip=&quot;text&quot;)</code></pre>
<p>It will produce something like this.</p>
<p><img src="img/_gwas/interactive%20plot.png" /></p>
<p>Again, this is an example with dummy data - you can try to do it for our GWAS, but careful with the time. You can also choose to carry on.</p>
<p>You will encounter the above types of visualizations in any high-quality GWAS paper, because each is so critically informative. Usually, analysts of large-scale meta-analyses of GWAS will also stratify the QQ-plots based on the imputation quality (if your GWAS was imputed), call rate, and allele frequency (although that is rarely shared in publications, not even in supplemental material).</p>
</div>
<div id="stop-playing-around" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Stop playing around<a href="gwas-visuals.html#stop-playing-around" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Alright. It’s time to stop playing around and do a quick recap of what you’ve learned.</p>
<ol style="list-style-type: decimal">
<li>You learned how to convert datasets.</li>
<li>You learned how to execute sample QC and create diagnostic graphics</li>
<li>You learned how to do the same for SNP QC</li>
<li>You learned how to execute an association study given a dataset, covariates, and different assumptions regarding the genetic model.</li>
<li>You learned how to visualize results and played around with different visuals.</li>
</ol>
<p>You should be ready for the real stuff. And if not, the next chapter will help you get ready: Chapter <a href="wtccc1-intro.html#wtccc1-intro">8</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas-testing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="wtccc1-intro.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["twitter", "linkedin", "whatsapp"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics/edit/master/gwas_basics_results_visualization.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
