<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A primer in Human Cardiovascular Genetics</title>
  <meta name="description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="A primer in Human Cardiovascular Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A primer in Human Cardiovascular Genetics" />
  
  <meta name="twitter:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="prerequisites.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A primer in Human Cardiovascular Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.5</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>3</b> Basics of a Genome-Wide Association Study (GWAS)<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Letâ€™s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sample-qc.html"><a href="sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="sample-qc.html"><a href="sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="sample-qc.html"><a href="sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>4.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="sample-qc.html"><a href="sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="sample-qc.html"><a href="sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="sample-qc.html"><a href="sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="sample-qc.html"><a href="sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="sample-qc.html"><a href="sample-qc.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.5.2</b> Principal Component Analysis<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="sample-qc.html"><a href="sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.3</b> 1000G phase 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="sample-qc.html"><a href="sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="sample-qc.html"><a href="sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="per-snp-qc.html"><a href="per-snp-qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="per-snp-qc.html"><a href="per-snp-qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="per-snp-qc.html"><a href="per-snp-qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="per-snp-qc.html"><a href="per-snp-qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="per-snp-qc.html"><a href="per-snp-qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="per-snp-qc.html"><a href="per-snp-qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="per-snp-qc.html"><a href="per-snp-qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html"><i class="fa fa-check"></i><b>6</b> Genome-wide association study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Letâ€™s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#regional-association-plots"><i class="fa fa-check"></i><b>7.4</b> Regional association plots<span></span></a></li>
<li class="chapter" data-level="7.5" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#stop-playing-around"><i class="fa fa-check"></i><b>7.5</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium (WTCCC)<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#genotyping"><i class="fa fa-check"></i><b>8.1</b> Genotyping<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#the-data"><i class="fa fa-check"></i><b>8.2</b> The data<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#assignment"><i class="fa fa-check"></i><b>8.3</b> Assignment<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#there-you-go"><i class="fa fa-check"></i><b>8.4</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html"><i class="fa fa-check"></i><b>9</b> Mendelian Randomization (MR)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-1"><i class="fa fa-check"></i><b>9.1</b> Section 1<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-2"><i class="fa fa-check"></i><b>9.2</b> Section 2<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-3"><i class="fa fa-check"></i><b>9.3</b> Section 3<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>10</b> Epilogue<span></span></a></li>
<li class="chapter" data-level="11" data-path="licenses-and-disclaimers.html"><a href="licenses-and-disclaimers.html"><i class="fa fa-check"></i><b>11</b> Licenses and disclaimers<span></span></a></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A primer in Human Cardiovascular Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">A primer in Human Cardiovascular Genetics</h1>
<p class="author"><em><a href="https://swvanderlaan.github.iio">dr. Sander W. van der Laan</a> <a href="https://www.twitter.com/swvanderlaan"><img src="img/twitter_circle_blue.png" style="width:2.5%" /></a> <a href="mailto:s.w.vanderlaan@gmail.com"><img src="img/email_circle_blue.png" style="width:2.5%" /></a></em></p>
<p class="date"><em>Version 1.0.0 | last update: 2022-03-28</em></p>
</div>
<div id="about-this-primer" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> About this primer<a href="index.html#about-this-primer" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="img/banner_man_standing_dna.png" /></p>
<div id="introduction" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Welcome to the <em>A primer in Human Cardiovascular Genetics</em> as part of the <strong>Genetic Epidemiology</strong> course. In the next few days we will use this <a href="https://cjvanlissa.github.io/gitbook-demo/">GitBook</a> to perform quality control (QC), executing a genome-wide association study (GWAS), annotating the GWAS results, and performing further downstream analyses. We will use data from the first release of the <a href="https://www.wtccc.org.uk/ccc1/overview.html" target="_blank"><em>Welcome Trust Case-Control Consortium (WTCCC)</em></a> and focus on coronary artery disease (CAD).</p>
<p>Unfortunately, during this course there is no time to perform <a href="https://www.nature.com/articles/nrg2796">imputation</a>, but I will provide some pointers during the course as to how to do this with minimal coding/scripting experience. Likewise, this practical does not cover the aspects of meta-analyses of GWAS. But rest assured, I will add chapters on these subjects to a future version.</p>
</div>
<div id="background-reading" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Background reading<a href="index.html#background-reading" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Part of this is based on four great Nature Protocols from the <a href="https://www.well.ox.ac.uk/research/research-groups/zondervan-group" target="_blank">Zondervan group</a> at the Wellcome Center Human Genetics.</p>
<ol style="list-style-type: decimal">
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/17947991" target="_blank">Zondervan KT <em>et al.</em> <em>Designing candidate gene and genome-wide case-control association studies.</em> Nat Protoc 2007.</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/19390530" target="_blank">Pettersson FH <em>et al.</em> <em>Marker selection for genetic case-control association studies.</em> Nat Protoc 2009.</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/21085122" target="_blank">Anderson CA <em>et al.</em> <em>Data QC in genetic case-control association studies.</em> Nat Protoc 2010.</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/21293453" target="_blank">Clarke GM <em>et al.</em> <em>Basic statistical analysis in genetic case-control studies.</em> Nat Protoc 2011.</a></li>
</ol>
<p>An update on the community standards of QC for GWAS can be found here:</p>
<ol style="list-style-type: decimal">
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/20718045" target="_blank">Laurie CC <em>et al.</em> <em>Quality control and quality assurance in genotypic data for genome-wide association studies.</em> Genet Epidemiol 2010.</a></li>
</ol>
<p>With respect to imputation you should also get familiar with the following two works:</p>
<ol style="list-style-type: decimal">
<li><a href="https://doi.org/10.1038/nrg2796" target="_blank">Marchini, J. and Howie, B. <em>Genotype imputation for genome-wide association studies.</em> Nat Rev Genet 2010</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/18852200" target="_blank">de Bakker PIW <em>et al.</em> <em>Practical aspects of imputation-driven meta-analysis of genome-wide association studies.</em> Hum Mol Genet 2008.</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/24762786" target="_blank">Winkler TW <em>et al.</em> <em>Quality control and conduct of genome-wide association meta-analyses.</em> Nat Protoc 2014.</a></li>
</ol>
</div>
<div id="meet-the-team" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Meet the Team<a href="index.html#meet-the-team" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We work with a team of enthusiastic lecturers with experience in bioinformatics, GWAS, genetic analyses, Mendelian randomization, and epidemiology. This year the team consists of:</p>
<p><img src="img/sander_vander_laan.jpg" style="width:15.0%" /></br>
Sander W. van der Laan</br>
<em>Assistant professor</em></br>
Course coordinator</br>
<a href="mailto:s.w.vanderlaan-2@umcutrecht.nl" class="email">s.w.vanderlaan-2@umcutrecht.nl</a> | <a href="http://www.twitter.com/swvanderlaan" target="_blank">swvanderlaan</a></p>
<p><img src="img/charlotte_onland.jpg" style="width:15.0%" /></br>
Charlotte Onland-Moret</br>
<em>Associate Professor</em></br >
<a href="mailto:N.C.Onland@umcutrecht.nl" class="email">N.C.Onland@umcutrecht.nl</a> | <a href="http://www.twitter.com/nconland" target="_blank">nconland</a></p>
<p><img src="img/jessica_van_setten.jpg" style="width:15.0%" /></br>
Jessica van Setten</br>
<em>Assistant professor</em></br>
<a href="mailto:j.vansetten@umcutrecht.nl" class="email">j.vansetten@umcutrecht.nl</a> | <a href="http://www.twitter.com/j_vansetten" target="_blank">j_vansetten</a></p>
<!-- ![](img/_team/arjan_boltjes.jpg){width=15%} Arjan Boltjes </br>_Postdoctoral researcher_</br><a.boltjes@umcutrecht.nl> | [arjanboltjes](http://www.twitter.com/arjanboltjes){target="_blank"} -->
<!-- ![](img/_team/joanna_bon_berg.jpg){width=15%} Joanna von Berg </br>_PhD student_</br><j.vonberg-24@umcutrecht.nl> | [JoannavonBerg](http://www.twitter.com/JoannavonBerg){target="_blank"} -->
</div>
<div id="final-thoughts" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Final thoughts<a href="index.html#final-thoughts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I can imagine this seems overwhelming, but trust me, youâ€™ll be okay. Just follow this practical, but also work on the questions asked during the lectures and in this practical. Youâ€™ll learn by doing and at the end of the day, you can execute a GWAS independently.</p>
<p><strong>Ready to start?</strong></p>
<p>Your first point of action is to prepare your system for this course in Chapter <a href="prerequisites.html#prerequisites">2</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="prerequisites.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics/edit/master/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["A primer in Human Cardiovascular Genetics.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
