<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="somebackgroundreading.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#the-programs-we-use"><i class="fa fa-check"></i><b>3.1</b> The programs we use</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#cocalc"><i class="fa fa-check"></i><b>3.2</b> CoCalc</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#standalone"><i class="fa fa-check"></i><b>3.3</b> Standalone</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="getting-started.html"><a href="getting-started.html#terminal"><i class="fa fa-check"></i><b>3.3.1</b> Terminal</a></li>
<li class="chapter" data-level="3.3.2" data-path="getting-started.html"><a href="getting-started.html#the-data-you-need"><i class="fa fa-check"></i><b>3.3.2</b> The data you need</a></li>
<li class="chapter" data-level="3.3.3" data-path="getting-started.html"><a href="getting-started.html#navigating-the-terminal"><i class="fa fa-check"></i><b>3.3.3</b> Navigating the Terminal</a></li>
<li class="chapter" data-level="3.3.4" data-path="getting-started.html"><a href="getting-started.html#installing-the-software"><i class="fa fa-check"></i><b>3.3.4</b> Installing the software</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="getting-started.html"><a href="getting-started.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>3.4</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="3.5" data-path="getting-started.html"><a href="getting-started.html#are-you-ready"><i class="fa fa-check"></i><b>3.5</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>4</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>4.1</b> Converting datasets</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>4.2</b> Quality control</a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>4.3</b> Letâ€™s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>5</b> Sample QC</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>5.1</b> Sex</a></li>
<li class="chapter" data-level="5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>5.2</b> Sample call rates</a></li>
<li class="chapter" data-level="5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>5.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>5.4</b> Relatedness</a></li>
<li class="chapter" data-level="5.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>5.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>5.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#principal-component-analysis"><i class="fa fa-check"></i><b>5.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="5.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>5.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>5.6</b> Removing samples</a></li>
<li class="chapter" data-level="5.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>5.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>6</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>6.1</b> SNP call rates</a></li>
<li class="chapter" data-level="6.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>6.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="6.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>6.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>6.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>6.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="6.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>6.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="6.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>6.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-testing.html"><a href="gwas-testing.html"><i class="fa fa-check"></i><b>7</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-testing.html"><a href="gwas-testing.html#exploring-the-data"><i class="fa fa-check"></i><b>7.1</b> Exploring the data</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-testing.html"><a href="gwas-testing.html#genetic-models"><i class="fa fa-check"></i><b>7.2</b> Genetic models</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-testing.html"><a href="gwas-testing.html#logistic-regression"><i class="fa fa-check"></i><b>7.3</b> Logistic regression</a></li>
<li class="chapter" data-level="7.4" data-path="gwas-testing.html"><a href="gwas-testing.html#lets-get-visual"><i class="fa fa-check"></i><b>7.4</b> Letâ€™s get visual</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>8</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="8.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>8.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="8.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>8.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>8.2</b> Other plots</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>8.3</b> Interactive plots</a></li>
<li class="chapter" data-level="8.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>8.4</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>9</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>9.1</b> Genotyping</a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>9.2</b> The data</a></li>
<li class="chapter" data-level="9.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>9.3</b> Assignment</a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>9.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>10</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>10.1</b> Quality control</a></li>
<li class="chapter" data-level="10.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>10.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>10.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="10.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca"><i class="fa fa-check"></i><b>10.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="10.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>10.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="10.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>10.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>10.3</b> Association testing</a></li>
<li class="chapter" data-level="10.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>10.4</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>11</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="11.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>11.1</b> Clumping</a></li>
<li class="chapter" data-level="11.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>11.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="11.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>11.3</b> Statistical finemapping</a></li>
<li class="chapter" data-level="11.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>11.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="11.5" data-path="post-gwas.html"><a href="post-gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>11.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="11.6" data-path="post-gwas.html"><a href="post-gwas.html#colocalization"><i class="fa fa-check"></i><b>11.6</b> Colocalization</a></li>
<li class="chapter" data-level="11.7" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>11.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="11.8" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>11.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="post-gwas.html"><a href="post-gwas.html#your-choice"><i class="fa fa-check"></i><b>11.8.1</b> Your choice</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>11.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="11.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>11.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>12</b> Epilogue</a></li>
<li class="chapter" data-level="13" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>13</b> Additional: EIGENSOFT</a>
<ul>
<li class="chapter" data-level="13.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>13.1</b> Install homebrew</a></li>
<li class="chapter" data-level="13.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>13.2</b> Install missing packages</a></li>
<li class="chapter" data-level="13.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>13.3</b> Installing EIGENSOFT</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>14</b> Things to do</a>
<ul>
<li class="chapter" data-level="14.1" data-path="todo.html"><a href="todo.html#moscow"><i class="fa fa-check"></i><b>14.1</b> MoSCoW</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="todo.html"><a href="todo.html#contents"><i class="fa fa-check"></i><b>14.1.1</b> Contents</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="todo.html"><a href="todo.html#book-fixes"><i class="fa fa-check"></i><b>14.2</b> Book fixes</a></li>
<li class="chapter" data-level="14.3" data-path="todo.html"><a href="todo.html#useful-links-for-me-mostly"><i class="fa fa-check"></i><b>14.3</b> Useful links (for me mostly)</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>15</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>15.1</b> Copyright</a></li>
<li class="chapter" data-level="15.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>15.2</b> Disclaimer</a></li>
<li class="chapter" data-level="15.3" data-path="license.html"><a href="license.html#images-used"><i class="fa fa-check"></i><b>15.3</b> Images used</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="colofon.html"><a href="colofon.html"><i class="fa fa-check"></i><b>16</b> Colofon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">A Practical Primer in Human Complex Genetics</h1>
<h2 class="subtitle"><em>with a use-case in cardiovascular disease</em></h2>
<p class="author"><em><a href="https://vanderlaanand.science">dr. Sander W. van der Laan</a> <a href="https://www.twitter.com/swvanderlaan"><img src="img/_logo/twitter_circle_blue.png" style="width:2.0%" /></a> <a href="mailto:s.w.vanderlaan@gmail.com"><img src="img/_logo/email_circle_blue.png" style="width:2.0%" /></a></em></p>
<p class="date"><em>Version 2.0.2 (2024-03-26)</em></p>
</div>
<div id="about-this-primer" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> About this primer<a href="index.html#about-this-primer" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="img/_headers/banner_man_standing_dna.png" style="width:100.0%" /></p>
<p>Ever since the first genome-wide association study (GWAS) on <a href="https://doi.org/10.1126/science.1109557" target="_blank">age-related macular degeneration</a>, and the promise of personalized medicine in the wake of the Human Genome Project, large-scale genetic association studies hold significant sway in contemporary health research and <a href="http://dx.doi.org/10.1038/nrd.2017.262" target="_blank">drive drug-development pipelines</a>. In the past 2 decades, researchers delved into GWAS, aiming to unveil genetic variations linked to both human traits, such as the color of your eyes, and rare and common complex diseases. These findings serve as crucial keys to unravel the intricate mechanisms underlying diseases, shedding light on whether the correlations identified in observational studies between risk factors and diseases are truly causal.</p>
<p>These studies have ushered in an exciting era where many researchers thrive on developing new methods and bioinformatic tools to parse ever-growing large datasets collected large population-based biobanks. However, the analyses of these data are challenging and it can be daunting to see the forest for tree among the many tools and their various functions. Enter <em>A Practical Primer in Human Complex Genetics</em>. This <a href="https://cjvanlissa.github.io/gitbook-demo/" target="_blank">GitBook</a> was originally written back in 2022 for the <strong>Genetic Epidemiology</strong> course organized by the <a href="https://epidemiology-education.nl" target="_blank">Master Epidemiology</a> of Utrecht University. This practical guide will teach you how to design a GWAS, perform quality control (QC), execute the actual analyses, annotate the GWAS results, and perform further downstream post-GWAS analyses. Throughout the book youâ€™ll work with â€˜dummyâ€™, that is fake, data, but in the end, we will use real-world data from the first release of the <a href="https://www.wtccc.org.uk/ccc1/overview.html" target="_blank"><em>Welcome Trust Case-Control Consortium (WTCCC)</em></a> focusing on coronary artery disease (CAD).</p>
<p>A major component of modern-day GWAS is <a href="https://www.nature.com/articles/nrg2796" target="_blank">genetic imputation</a>, but for practical reasons it is not part of this book. However, I will provide some pointers as to how to go about do this with minimal coding or scripting experience. Likewise, the courses does not cover the aspects of meta-analyses of GWAS, but some excellent resources exist to which I will direct. As this practical primer evolves, these and other topics may find their place in this book.
I should also point out that emphasis of this book is on it being a <em>practical primer</em>. It is intended to provide some practical guidance to doing GWAS, and while theory is important, I will not cover this. Again, some very useful and excellent work exists to which I will point you, but I really want you to learn - and understand the theory - by <em>doing</em>.</p>
<p>So, although originally crafted as a companion for the course, this practical guide stands on its own as a comprehensive resource for diving into all facets of doing a GWAS â€” save for experimental follow-up, of course ðŸ˜‰.</p>
<p>I can imagine this seems overwhelming, but trust me, youâ€™ll be okay. Just follow this practical. Youâ€™ll learn by doing and at the end of the day, you can execute a GWAS independently.</p>
<p><strong>Ready to start?</strong></p>
<p>Your first point of action is to prepare your system for this course in Chapter <a href="somebackgroundreading.html#somebackgroundreading">2</a>.</p>

</div>
            </section>

          </div>
        </div>
      </div>

<a href="somebackgroundreading.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_Practical_Primer_in_Human_Complex_Geneticsindex.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
