<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 15 Two Sample Mendelian Randomisation | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 15 Two Sample Mendelian Randomisation | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 15 Two Sample Mendelian Randomisation | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="the-end.html"/>
<link rel="next" href="the-end-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A primer in Human Cardiovascular Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#eigensoft"><i class="fa fa-check"></i><b>2.5</b> EIGENSOFT<span></span></a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="prerequisites.html"><a href="prerequisites.html#install-homebrew"><i class="fa fa-check"></i><b>2.5.1</b> Install homebrew<span></span></a></li>
<li class="chapter" data-level="2.5.2" data-path="prerequisites.html"><a href="prerequisites.html#install-missing-packages"><i class="fa fa-check"></i><b>2.5.2</b> Install missing packages<span></span></a></li>
<li class="chapter" data-level="2.5.3" data-path="prerequisites.html"><a href="prerequisites.html#installing-eigensoft"><i class="fa fa-check"></i><b>2.5.3</b> Installing EIGENSOFT<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.6</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>3</b> Steps in a Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Let’s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>4.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.2</b> 1000G phase 1<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#running-pca"><i class="fa fa-check"></i><b>4.5.3</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="4.5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#pca-plotting"><i class="fa fa-check"></i><b>4.5.4</b> PCA plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>5.3.1</b> A note on allele coding<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="gwas-basics-snp-qc.html"><a href="gwas-basics-snp-qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-testing.html"><a href="gwas-testing.html"><i class="fa fa-check"></i><b>6</b> Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-testing.html"><a href="gwas-testing.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="gwas-testing.html"><a href="gwas-testing.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="gwas-testing.html"><a href="gwas-testing.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="gwas-testing.html"><a href="gwas-testing.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Let’s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>7.4</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>8.1</b> Genotyping<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>8.2</b> The data<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>8.3</b> Assignment<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>8.4</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>9</b> WTCCC1: a GWAS on coronary artery disease (CAD)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>9.1</b> Quality control<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>9.2</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>9.2.1</b> Preparing PCA<span></span></a></li>
<li class="chapter" data-level="9.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca-1"><i class="fa fa-check"></i><b>9.2.2</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="9.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>9.2.3</b> Plotting PCA<span></span></a></li>
<li class="chapter" data-level="9.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>9.2.4</b> Removing samples<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>9.3</b> Association testing<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>10</b> Post-GWAS Analyses<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>10.1</b> Clumping<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>10.2</b> Conditional analyses<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>10.3</b> Statistical finemapping<span></span></a></li>
<li class="chapter" data-level="10.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>10.4</b> FUMA: FUnctional Mapping and Annotation of GWAS<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>10.5</b> PheWAS<span></span></a></li>
<li class="chapter" data-level="10.6" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>10.6</b> Genetic correlation<span></span></a></li>
<li class="chapter" data-level="10.7" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>10.7</b> Causal inference: Mendelian Randomization<span></span></a></li>
<li class="chapter" data-level="10.8" data-path="post-gwas.html"><a href="post-gwas.html#your-choice"><i class="fa fa-check"></i><b>10.8</b> Your choice<span></span></a></li>
<li class="chapter" data-level="10.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>10.9</b> Polygenic scores<span></span></a></li>
<li class="chapter" data-level="10.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>10.10</b> What comes next<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>11</b> Functional Mapping and Annotation of GWAS<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>11.1</b> Assignment<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>11.1.1</b> Tutorial<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>11.1.2</b> Create the input<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>11.1.3</b> Run FUMA - SNP to gene<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>11.1.4</b> Run FUMA - Gene to function<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>11.2</b> Some closing thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>12</b> Phenome-Wide Association Study (PheWAS)<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>12.1</b> What’s next?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>13</b> MRBase<span></span></a></li>
<li class="chapter" data-level="14" data-path="the-end.html"><a href="the-end.html"><i class="fa fa-check"></i><b>14</b> The end?<span></span></a></li>
<li class="chapter" data-level="15" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html"><i class="fa fa-check"></i><b>15</b> Two Sample Mendelian Randomisation<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#loading-the-package"><i class="fa fa-check"></i><b>15.1</b> Loading the package<span></span></a></li>
<li class="chapter" data-level="15.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#available-outcome"><i class="fa fa-check"></i><b>15.2</b> Available outcome<span></span></a></li>
<li class="chapter" data-level="15.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#instruments"><i class="fa fa-check"></i><b>15.3</b> Instruments<span></span></a></li>
<li class="chapter" data-level="15.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#clumping-1"><i class="fa fa-check"></i><b>15.4</b> Clumping<span></span></a></li>
<li class="chapter" data-level="15.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#outcomes"><i class="fa fa-check"></i><b>15.5</b> Outcomes<span></span></a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#harmonisation"><i class="fa fa-check"></i><b>15.5.1</b> Harmonisation<span></span></a></li>
<li class="chapter" data-level="15.5.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#mendelian-randomisation"><i class="fa fa-check"></i><b>15.5.2</b> Mendelian randomisation<span></span></a></li>
<li class="chapter" data-level="15.5.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#heterogeneity-pleiotropy"><i class="fa fa-check"></i><b>15.5.3</b> Heterogeneity &amp; Pleiotropy<span></span></a></li>
<li class="chapter" data-level="15.5.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#leave-one-out"><i class="fa fa-check"></i><b>15.5.4</b> Leave-one-out<span></span></a></li>
<li class="chapter" data-level="15.5.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#visualisation"><i class="fa fa-check"></i><b>15.5.5</b> Visualisation<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="the-end-1.html"><a href="the-end-1.html"><i class="fa fa-check"></i><b>16</b> The end?<span></span></a></li>
<li class="chapter" data-level="17" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>17</b> Epilogue {-}<span></span></a></li>
<li class="chapter" data-level="18" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html"><i class="fa fa-check"></i><b>18</b> Regional association plots<span></span></a></li>
<li class="chapter" data-level="19" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>19</b> Things to do<span></span></a>
<ul>
<li class="chapter" data-level="19.1" data-path="todo.html"><a href="todo.html#useful-links"><i class="fa fa-check"></i><b>19.1</b> Useful links<span></span></a></li>
</ul></li>
<li><a href="licenses-and-disclaimers.html#licenses-and-disclaimers">Licenses and disclaimers<span></span></a>
<ul>
<li class="chapter" data-level="19.2" data-path="licenses-and-disclaimers.html"><a href="licenses-and-disclaimers.html"><i class="fa fa-check"></i><b>19.2</b> Copyright<span></span></a></li>
<li class="chapter" data-level="19.3" data-path="licenses-and-disclaimers.html"><a href="licenses-and-disclaimers.html#disclaimer"><i class="fa fa-check"></i><b>19.3</b> Disclaimer<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mr_twosamplemr" class="section level1 hasAnchor" number="15">
<h1><span class="header-section-number">Chapter 15</span> Two Sample Mendelian Randomisation<a href="mr_twosamplemr.html#mr_twosamplemr" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>So you took the hard way. You want to get your hands dirty yourself. Your choice.</p>
<p>As I wrote, we will use the <code>TwoSampleMR</code> package to assess the causal effect of type 2 diabetes on coronary artery disease and ischemic stroke.</p>
<p>Briefly, the workflow of a Two-Sample MR is as follows (see also diagram below):</p>
<ol style="list-style-type: decimal">
<li>Select instruments for the exposure (perform LD clumping if necessary)</li>
<li>Extract the instruments from the MR Base GWAS database for the outcomes of interest</li>
<li>Harmonise the effect sizes for the instruments on the exposures and the outcomes to be each for the same reference allele</li>
<li>Perform MR analysis, sensitivity analyses, create plots, compile reports</li>
</ol>
<p><img src="img/twosamplemr.png" /></p>
<p>You need R 3.5.0 or higher to be able to install the packages you need.</p>
<pre><code>install.packages(&quot;remotes&quot;)
remotes::install_github(&quot;MRCIEU/TwoSampleMR&quot;) # to run TwoSampleMR
library(&quot;TwoSampleMR&quot;)</code></pre>
<!-- ``` -->
<!-- devtools::install_github("MRCIEU/MRInstruments") -->
<!-- library(MRInstruments) -->
<!-- ``` -->
<div id="loading-the-package" class="section level2 hasAnchor" number="15.1">
<h2><span class="header-section-number">15.1</span> Loading the package<a href="mr_twosamplemr.html#loading-the-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>TwoSampleMR</code> is free use. It used to require server-authentication using a gmail-account, now it doesn’t anymore. Should run into an issue, please refer to <a href="https://mrcieu.github.io/TwoSampleMR/articles/introduction.html" target="_blank">this page</a>.</p>
<p>First we load the package.</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="mr_twosamplemr.html#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;TwoSampleMR&quot;</span>)</span></code></pre></div>
<pre><code>## TwoSampleMR version 0.5.6 
## [&gt;] New: Option to use non-European LD reference panels for clumping etc
## [&gt;] Some studies temporarily quarantined to verify effect allele
## [&gt;] See news(package=&#39;TwoSampleMR&#39;) and https://gwas.mrcieu.ac.uk for further details</code></pre>
<pre><code>## 
## Attaching package: &#39;TwoSampleMR&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:remotes&#39;:
## 
##     add_metadata</code></pre>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="mr_twosamplemr.html#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;MRInstruments&quot;</span>)</span></code></pre></div>
</div>
<div id="available-outcome" class="section level2 hasAnchor" number="15.2">
<h2><span class="header-section-number">15.2</span> Available outcome<a href="mr_twosamplemr.html#available-outcome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are ready to load in the available outcomes.</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="mr_twosamplemr.html#cb165-1" aria-hidden="true" tabindex="-1"></a>ao_mr <span class="ot">&lt;-</span> <span class="fu">available_outcomes</span>()</span></code></pre></div>
<pre><code>## API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
</div>
<div id="instruments" class="section level2 hasAnchor" number="15.3">
<h2><span class="header-section-number">15.3</span> Instruments<a href="mr_twosamplemr.html#instruments" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>MRInstruments</code>-package, which comes with <code>TwoSampleMR</code>, includes several types of instruments:</p>
<ul>
<li><em>gwas_catalog</em>, which includes data from GWAS collected by the GWAS Catalog</li>
<li><em>metab_qtls</em>, all known quantitative trait loci (meQTLs) associated to circulating metabolites</li>
<li><em>proteomic_qtls</em>, all known protein QTLs (pQTLs) associated to circulating protein levels</li>
<li><em>gtex_eqtl</em>, all known expression QTLs (eQTLs) associated to gene expression in 53 different tissues from GTEx Portal</li>
<li><em>aries_mqtl</em>, all known methylation QTLs (mQTLs) associated to DNA methylation in blood from ARIES</li>
</ul>
<p>Let’s select T2D associated instruments.</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="mr_twosamplemr.html#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;gwas_catalog&quot;</span>)</span>
<span id="cb167-2"><a href="mr_twosamplemr.html#cb167-2" aria-hidden="true" tabindex="-1"></a>t2d_gwas <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_catalog, <span class="fu">grepl</span>(<span class="st">&quot;Mahajan&quot;</span>, Author) <span class="sc">&amp;</span> PubmedID <span class="sc">==</span> <span class="st">&quot;24509480&quot;</span> <span class="sc">&amp;</span> Phenotype <span class="sc">==</span> <span class="st">&quot;Type 2 diabetes&quot;</span>)</span>
<span id="cb167-3"><a href="mr_twosamplemr.html#cb167-3" aria-hidden="true" tabindex="-1"></a>t2d_exp_dat <span class="ot">&lt;-</span> <span class="fu">format_data</span>(t2d_gwas)</span></code></pre></div>
<pre><code>## Warning in format_data(t2d_gwas): other_allele column has some values that are
## not A/C/T/G or an indel comprising only these characters or D/I. These SNPs will
## be excluded</code></pre>
<pre><code>## Warning in format_data(t2d_gwas): The following SNP(s) are missing required information for the MR tests and will be excluded
## rs12427353
## rs3923113
## rs4430796
## rs516946
## rs6808574
## rs7612463
## rs7756992
## rs7903146</code></pre>
</div>
<div id="clumping-1" class="section level2 hasAnchor" number="15.4">
<h2><span class="header-section-number">15.4</span> Clumping<a href="mr_twosamplemr.html#clumping-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Like with identifying independent GWAS hits, it is important for TwoSampleMR to ensure that the instruments for the exposure are independent. We can use MR Base to perform clumping. The European samples from the 1000 genomes project are used to estimate LD between SNPs. You can provide a list of SNP IDs, the SNPs will be extracted from 1000 genomes data, LD calculated between them, and amongst those SNPs that have LD R-square above the specified threshold only the SNP with the lowest P-value will be retained. To do this, use the following command.</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="mr_twosamplemr.html#cb170-1" aria-hidden="true" tabindex="-1"></a>t2d_exp_dat <span class="ot">&lt;-</span> <span class="fu">clump_data</span>(t2d_exp_dat)</span></code></pre></div>
<pre><code>## Please look at vignettes for options on running this locally if you need to run many instances of this command.</code></pre>
<pre><code>## Clumping 5cQr3c, 62 variants, using EUR population reference</code></pre>
<pre><code>## Removing 8 of 62 variants due to LD with other variants or absence from LD reference panel</code></pre>
</div>
<div id="outcomes" class="section level2 hasAnchor" number="15.5">
<h2><span class="header-section-number">15.5</span> Outcomes<a href="mr_twosamplemr.html#outcomes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Right, so now we are ready to select the proper outcome, CAD. There are multiple datasets available in MRBase, the database used by <code>TwoSampleMR</code>. Let’s review them all and select one with the largest overlap with our exposure SNPs.</p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="mr_twosamplemr.html#cb174-1" aria-hidden="true" tabindex="-1"></a>ao_twosamplemr <span class="ot">&lt;-</span> <span class="fu">available_outcomes</span>()</span>
<span id="cb174-2"><a href="mr_twosamplemr.html#cb174-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-3"><a href="mr_twosamplemr.html#cb174-3" aria-hidden="true" tabindex="-1"></a>ao_twosamplemr[<span class="fu">grepl</span>(<span class="st">&quot;heart disease&quot;</span>, ao_twosamplemr<span class="sc">$</span>trait), ]</span></code></pre></div>
<pre><code>## # A tibble: 27 × 23
##    id      trait note  ncase group_name  year consortium author sex   population
##    &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;int&gt; &lt;chr&gt;      &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;     
##  1 finn-b… Majo… I9_C… 21012 public      2021 NA         NA     Male… European  
##  2 ieu-a-8 Coro… &lt;NA&gt;  22233 public      2011 CARDIoGRAM Schun… Male… European  
##  3 ukb-e-… I25 … NA     1205 public      2020 NA         Pan-U… Male… South Asi…
##  4 ukb-b-… Diag… 4120…  5738 public      2018 MRC-IEU    Ben E… Male… European  
##  5 ieu-a-9 Coro… &lt;NA&gt;  63746 public      2013 CARDIoGRA… Delou… Male… Mixed     
##  6 ieu-a-6 Coro… &lt;NA&gt;  15420 public      2011 C4D        Peden  Male… Mixed     
##  7 ieu-a-7 Coro… &lt;NA&gt;  60801 public      2015 CARDIoGRA… Nikpay Male… Mixed     
##  8 ukb-d-… Majo… NA    10157 public      2018 NA         Neale… Male… European  
##  9 ukb-a-… Diag… NA     8755 public      2017 Neale Lab  Neale  Male… European  
## 10 ukb-b-… Diag… 4120…  5861 public      2018 MRC-IEU    Ben E… Male… European  
## # … with 17 more rows, and 13 more variables: unit &lt;chr&gt;, nsnp &lt;int&gt;,
## #   sample_size &lt;int&gt;, build &lt;chr&gt;, ncontrol &lt;int&gt;, subcategory &lt;chr&gt;,
## #   category &lt;chr&gt;, ontology &lt;chr&gt;, doi &lt;chr&gt;, mr &lt;int&gt;, priority &lt;int&gt;,
## #   pmid &lt;int&gt;, sd &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="mr_twosamplemr.html#cb176-1" aria-hidden="true" tabindex="-1"></a>chd_out_dat <span class="ot">&lt;-</span> <span class="fu">extract_outcome_data</span>(</span>
<span id="cb176-2"><a href="mr_twosamplemr.html#cb176-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">snps =</span> t2d_exp_dat<span class="sc">$</span>SNP,</span>
<span id="cb176-3"><a href="mr_twosamplemr.html#cb176-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">outcomes =</span> <span class="st">&quot;UKB-a:534&quot;</span></span>
<span id="cb176-4"><a href="mr_twosamplemr.html#cb176-4" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Deprecated IDs being used? Detected numeric IDs. Trying to fix, but please note the changes below for future.</code></pre>
<pre><code>## UKB-a:534  -&gt;  ukb-a-534</code></pre>
<pre><code>## Extracting data for 54 SNP(s) from 1 GWAS(s)</code></pre>
<div id="harmonisation" class="section level3 hasAnchor" number="15.5.1">
<h3><span class="header-section-number">15.5.1</span> Harmonisation<a href="mr_twosamplemr.html#harmonisation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is important to make sure that the effect allele associated to T2D of any given SNP is the same as the effect allele associated CAD. In other words, we need to harmonise the data.</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="mr_twosamplemr.html#cb180-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">harmonise_data</span>(</span>
<span id="cb180-2"><a href="mr_twosamplemr.html#cb180-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">exposure_dat =</span> t2d_exp_dat,</span>
<span id="cb180-3"><a href="mr_twosamplemr.html#cb180-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">outcome_dat =</span> chd_out_dat</span>
<span id="cb180-4"><a href="mr_twosamplemr.html#cb180-4" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Harmonising Type 2 diabetes (5cQr3c) and Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534 (ukb-a-534)</code></pre>
<pre><code>## Removing the following SNPs for being palindromic with intermediate allele frequencies:
## rs1727313, rs243088</code></pre>
<!-- ### Drop duplicate datasets -->
<!-- After data harmonisation, users may find that their dataset contains duplicate exposure-outcome summary sets. This can arise, for example, when a GWAS consortium has released multiple results from separate GWAS analyses for the same trait.  -->
<!-- The best thing to do, is to prune the datasets so that only the exposure-outcome combination with the highested expected power is retained.  -->
<!-- In our case we only have 1 dataset for the exposure and 1 for the outcome. But if you don't, you can use the command below. -->
<!-- ```{r MR_powerprune} -->
<!-- dat <- power_prune(dat,  -->
<!--                    method = 1, -->
<!--                    dist.outcome = "binary") -->
<!-- ``` -->
</div>
<div id="mendelian-randomisation" class="section level3 hasAnchor" number="15.5.2">
<h3><span class="header-section-number">15.5.2</span> Mendelian randomisation<a href="mr_twosamplemr.html#mendelian-randomisation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that the data is harmonised, we are ready to perform a MR and test for a causal relation of T2D with CAD.</p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="mr_twosamplemr.html#cb183-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">mr</span>(dat)</span></code></pre></div>
<pre><code>## Analysing &#39;5cQr3c&#39; on &#39;ukb-a-534&#39;</code></pre>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="mr_twosamplemr.html#cb185-1" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(res, <span class="at">caption =</span> <span class="st">&quot;MR of type 2 diabetes and coronary artery disease&quot;</span>, <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-9574c5d91c687314cc46" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9574c5d91c687314cc46">{"x":{"filter":"none","vertical":false,"caption":"<caption>MR of type 2 diabetes and coronary artery disease<\/caption>","data":[["5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c"],["ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534"],["Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534"],["Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes"],["MR Egger","Weighted median","Inverse variance weighted","Simple mode","Weighted mode"],[51,51,51,51,51],[0.0040962806998675,0.00346293004986617,0.00378199956131184,0.00237387116230173,0.00308809967775673],[0.00145154542394796,0.0009650083747574,0.00069381149793921,0.00168145357304108,0.00109256304445388],[0.00687557724320252,0.000332589301640957,5.00739395245472e-08,0.164203040437951,0.00674876356902608]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>id.exposure<\/th>\n      <th>id.outcome<\/th>\n      <th>outcome<\/th>\n      <th>exposure<\/th>\n      <th>method<\/th>\n      <th>nsnp<\/th>\n      <th>b<\/th>\n      <th>se<\/th>\n      <th>pval<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[5,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Several tests are applied. You can change the method(s) applied by using the <code>method_list</code> flag. Below a list of available methods.</p>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="mr_twosamplemr.html#cb186-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">mr_method_list</span>(), <span class="at">caption =</span> <span class="st">&quot;Available MR methods&quot;</span>, row.names</span>
<span id="cb186-2"><a href="mr_twosamplemr.html#cb186-2" aria-hidden="true" tabindex="-1"></a> <span class="ot">=</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<table>
<caption>(#tab:MR_methods)Available MR methods</caption>
<colgroup>
<col width="15%" />
<col width="30%" />
<col width="4%" />
<col width="31%" />
<col width="7%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">obj</th>
<th align="left">name</th>
<th align="left">PubmedID</th>
<th align="left">Description</th>
<th align="left">use_by_default</th>
<th align="left">heterogeneity_test</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">mr_wald_ratio</td>
<td align="left">Wald ratio</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_two_sample_ml</td>
<td align="left">Maximum likelihood</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">mr_egger_regression</td>
<td align="left">MR Egger</td>
<td align="left">26050253</td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">mr_egger_regression_bootstrap</td>
<td align="left">MR Egger (bootstrap)</td>
<td align="left">26050253</td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_simple_median</td>
<td align="left">Simple median</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_weighted_median</td>
<td align="left">Weighted median</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_penalised_weighted_median</td>
<td align="left">Penalised weighted median</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_ivw</td>
<td align="left">Inverse variance weighted</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">mr_ivw_radial</td>
<td align="left">IVW radial</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">mr_ivw_mre</td>
<td align="left">Inverse variance weighted (multiplicative random effects)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_ivw_fe</td>
<td align="left">Inverse variance weighted (fixed effects)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_simple_mode</td>
<td align="left">Simple mode</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_weighted_mode</td>
<td align="left">Weighted mode</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_weighted_mode_nome</td>
<td align="left">Weighted mode (NOME)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_simple_mode_nome</td>
<td align="left">Simple mode (NOME)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_raps</td>
<td align="left">Robust adjusted profile score (RAPS)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_sign</td>
<td align="left">Sign concordance test</td>
<td align="left"></td>
<td align="left">Tests for concordance of signs between exposure and outcome</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_uwr</td>
<td align="left">Unweighted regression</td>
<td align="left"></td>
<td align="left">Doesn’t use any weights</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
</div>
<div id="heterogeneity-pleiotropy" class="section level3 hasAnchor" number="15.5.3">
<h3><span class="header-section-number">15.5.3</span> Heterogeneity &amp; Pleiotropy<a href="mr_twosamplemr.html#heterogeneity-pleiotropy" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Many genetic variants are associated to multiple traits, thus <em>pleiotropy</em> is common and widespread in the human genome. This poses as an issue to tease apart causal relations in Mendelian randomisation.
<em>Vertical pleiotropy</em> arises because a SNP influences one trait, which in turn influences another. We can use MR to estimate the causal influence between the traits.
<em>Horizontal pleiotropy</em> arises due to SNPs influencing two traits through independent pathways<span class="citation">(<a href="#ref-hemani2018" role="doc-biblioref">Hemani G. 2018</a>)</span>.</p>
<p>Further, we should test for study heterogeneity.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="mr_twosamplemr.html#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mr_heterogeneity</span>(dat)</span></code></pre></div>
<pre><code>##   id.exposure id.outcome
## 1      5cQr3c  ukb-a-534
## 2      5cQr3c  ukb-a-534
##                                                                       outcome
## 1 Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534
## 2 Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534
##          exposure                    method       Q Q_df     Q_pval
## 1 Type 2 diabetes                  MR Egger 67.8423   49 0.03851066
## 2 Type 2 diabetes Inverse variance weighted 67.9269   50 0.04653850</code></pre>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="mr_twosamplemr.html#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mr_pleiotropy_test</span>(dat)</span></code></pre></div>
<pre><code>##   id.exposure id.outcome
## 1      5cQr3c  ukb-a-534
##                                                                       outcome
## 1 Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534
##          exposure egger_intercept           se      pval
## 1 Type 2 diabetes   -3.691553e-05 0.0001493361 0.8057881</code></pre>
</div>
<div id="leave-one-out" class="section level3 hasAnchor" number="15.5.4">
<h3><span class="header-section-number">15.5.4</span> Leave-one-out<a href="mr_twosamplemr.html#leave-one-out" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is also key to inspect the causal effects per SNP and perform a leave-one-out analysis.</p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="mr_twosamplemr.html#cb191-1" aria-hidden="true" tabindex="-1"></a>res_single <span class="ot">&lt;-</span> <span class="fu">mr_singlesnp</span>(dat)</span>
<span id="cb191-2"><a href="mr_twosamplemr.html#cb191-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(res_single, <span class="at">caption =</span> <span class="st">&quot;Single SNP MR analysis&quot;</span>, <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-dd7b07bfb1cdd559fdc2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-dd7b07bfb1cdd559fdc2">{"x":{"filter":"none","vertical":false,"caption":"<caption>Single SNP MR analysis<\/caption>","data":[["Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes"],["Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534"],["5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c"],["ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534"],[337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199],["rs10190052","rs10507349","rs10811661","rs10830963","rs10842994","rs1111875","rs11257655","rs11717195","rs12571751","rs12970134","rs1359790","rs1535500","rs1552224","rs1561927","rs163184","rs17106184","rs17168486","rs17791513","rs1801282","rs2028299","rs2075423","rs2261181","rs2284219","rs2796441","rs2812533","rs2943640","rs3132524","rs319598","rs3802177","rs3923113","rs4273712","rs4402960","rs4458523","rs4812829","rs516946","rs5215","rs6813195","rs6937795","rs702634","rs7041847","rs7163757","rs7178572","rs7612463","rs7756992","rs7845219","rs7903146","rs8108269","rs849135","rs9472138","rs9502570","rs9936385","All - Inverse variance weighted","All - MR Egger"],[0.00701799416202893,0.00655383826975928,0.0032408350892764,0.000923086296036809,0.0132415932401823,0.000521922483047612,0.0227294459882807,0.00475734462491583,0.00382238113071166,0.0097587946955077,0.00188419537523369,0.00753271012929167,0.000218214012213546,-0.00445053817463408,0.00622236406157904,0.00270150576285364,0.00174584464814024,0.0131753502362296,0.000106917527642952,-0.00757716971740318,0.0153577704063199,0.00327862136652149,-0.000762592228031708,-0.000608779526773151,0.00793871902729296,0.0160116835351498,0.00601207693584697,0.019328096472994,0.00279852436111297,0.00820303799022285,-0.0100238851072858,0.00889339758733801,0.0015936141947431,-0.00139712072487797,0.00479166025011818,-0.00609572054996241,-0.00463567013560736,-0.0156349528271823,0.00566168151457529,-0.00899568606647525,0.00688144008373845,0.00154161116129059,-0.0056426711302416,-0.000473416317399841,0.0103112040622781,0.00419464028941228,-0.00207747156927059,0.00799828390794882,0.00231476793272022,0.00332820720764808,0.00022432524169289,0.00378199956131184,0.0040962806998675],[0.00754302681936538,0.00786160260675129,0.00307085596081619,0.00412692541929851,0.00561624435452754,0.00280200026806157,0.00813813087060201,0.00517295100060516,0.00448147278061054,0.00565046929693701,0.00446313290850278,0.00313906423687325,0.00433565916953703,0.00749983162667476,0.00448682219258265,0.00694968786503058,0.00417252395282911,0.00419645625284877,0.00398265718646862,0.0109305253661389,0.00532257612361878,0.00441457335537668,0.00819046229506983,0.00575426806154244,0.00772489566063794,0.00469312847338956,0.00653035509822515,0.00800696219515401,0.00280718222363613,0.010241629172579,0.00893587893014616,0.00339824942393599,0.00456026346887642,0.00778818394819008,0.00474792934950969,0.00521995277181108,0.00573092558895946,0.0100488738809949,0.00716739018193746,0.00789337372718422,0.00668104161436683,0.00552847250659484,0.0064604326763851,0.00239038656571218,0.00500915780645575,0.00125974138091273,0.00736392724378928,0.00339747169004004,0.00725899993011813,0.00740176903748324,0.00322757846971237,0.00069381149793921,0.00145154542394796],[0.352166625332494,0.404477290801146,0.291264201888315,0.823010906771706,0.0183870130432801,0.852234741319233,0.00522287055184732,0.357751623020355,0.393698303155212,0.0841538631865469,0.672901827497513,0.0164099744440318,0.959859364909994,0.552901169838809,0.165500102867962,0.697480827165237,0.675644065032581,0.00169157172620101,0.978582741404563,0.488176612266362,0.00390914469722915,0.457674614400211,0.925818279467187,0.915744074984518,0.30410048503052,0.000645522029569791,0.357240745044448,0.0157823041800362,0.318805373317494,0.423160303857641,0.261965756259697,0.00886926653544695,0.726746543044201,0.857631643523976,0.312873682204741,0.242898292200127,0.418580254740515,0.119734004519823,0.429573132679199,0.254432003196038,0.303012308725295,0.78036041282406,0.38243419182087,0.843005859623255,0.0395449809145371,0.000869199213526549,0.777855619767605,0.0185632665749813,0.749815626840599,0.652962657845079,0.944589527069326,5.00739395245472e-08,0.00687557724320252]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>exposure<\/th>\n      <th>outcome<\/th>\n      <th>id.exposure<\/th>\n      <th>id.outcome<\/th>\n      <th>samplesize<\/th>\n      <th>SNP<\/th>\n      <th>b<\/th>\n      <th>se<\/th>\n      <th>p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="mr_twosamplemr.html#cb192-1" aria-hidden="true" tabindex="-1"></a>res_loo <span class="ot">&lt;-</span> <span class="fu">mr_leaveoneout</span>(dat)</span>
<span id="cb192-2"><a href="mr_twosamplemr.html#cb192-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(res_loo, <span class="at">caption =</span> <span class="st">&quot;Leave-one-out MR analysis&quot;</span>, <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div id="htmlwidget-9ec2ceb83b1beadfeabb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9ec2ceb83b1beadfeabb">{"x":{"filter":"none","vertical":false,"caption":"<caption>Leave-one-out MR analysis<\/caption>","data":[["Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes","Type 2 diabetes"],["Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534","Diagnoses - main ICD10: I25 Chronic ischaemic heart disease || id:ukb-a-534"],["5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c","5cQr3c"],["ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534","ukb-a-534"],[337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199,337199],["rs10190052","rs10507349","rs10811661","rs10830963","rs10842994","rs1111875","rs11257655","rs11717195","rs12571751","rs12970134","rs1359790","rs1535500","rs1552224","rs1561927","rs163184","rs17106184","rs17168486","rs17791513","rs1801282","rs2028299","rs2075423","rs2261181","rs2284219","rs2796441","rs2812533","rs2943640","rs3132524","rs319598","rs3802177","rs3923113","rs4273712","rs4402960","rs4458523","rs4812829","rs516946","rs5215","rs6813195","rs6937795","rs702634","rs7041847","rs7163757","rs7178572","rs7612463","rs7756992","rs7845219","rs7903146","rs8108269","rs849135","rs9472138","rs9502570","rs9936385","All"],[0.00376172083242628,0.0037660167203777,0.00380312736766689,0.00384274162241619,0.00367452690685115,0.00393608404376505,0.00368008358113336,0.00376891130387615,0.00378127431846496,0.003714925094763,0.0038163694230565,0.00364209594510357,0.00385046586231907,0.00383418945215448,0.00373827754525213,0.00378998503614966,0.00382430092782674,0.00358911663408322,0.00386597342057803,0.00381578780251039,0.00363538284336041,0.00379132130091364,0.00380613133428972,0.0038294942549128,0.00375717034031642,0.00358203869303319,0.00376331507975256,0.00369560151503818,0.00382830306748687,0.00376701420918704,0.00384353599626449,0.00362020100638454,0.00381993268475674,0.00381243217350632,0.00376587608532093,0.00391214212893719,0.00387380414626183,0.00385037246904886,0.00376894448652445,0.00385508232214318,0.0037571986629379,0.00380827732690249,0.00386269646795743,0.00406333184657689,0.00368847651555959,0.00366338020033613,0.00382053831792991,0.00364847230199873,0.00379193270086116,0.00378495359362407,0.00390727124444773,0.00378199956131184],[0.000702088953377578,0.000702225998580468,0.000714235936843267,0.000705702013422949,0.000689779202606288,0.000709702953934797,0.000673957970749978,0.000705355078213438,0.000707120606924602,0.000698883090584553,0.000706214450531574,0.000705982967059635,0.000703960130225676,0.000696769858163737,0.000705536849315832,0.000703314448874264,0.00070683007193577,0.000680844698631746,0.000704258417153003,0.000696278225160686,0.00067995906235545,0.00070724607820546,0.000701110972404967,0.000701576571924835,0.000701436778767489,0.000669706174074065,0.000703175888228055,0.000682908927644537,0.000716485601643177,0.000701075363503568,0.000689907545552192,0.000699524715071698,0.000705683513997356,0.000700606263683615,0.000706190394436204,0.000686359591527734,0.000693262159472234,0.000682450634521928,0.000702926476288295,0.000689087130944056,0.000702529354907943,0.000704090983097692,0.000692639873083728,0.000705425146042348,0.000696845629071189,0.000794426925299824,0.000699850157504442,0.000703491925724867,0.000703010897493294,0.000703113332900059,0.000706454945368161,0.00069381149793921],[8.41958583508748e-08,8.18647638485969e-08,1.01092318626667e-07,5.17253718346334e-08,9.97899506590012e-08,2.92110644249118e-08,4.7504920856936e-08,9.12784393116847e-08,8.92144251528961e-08,1.06354797436605e-07,6.51778770402415e-08,2.48404151279296e-07,4.50743278369544e-08,3.7379337494524e-08,1.16766441397392e-07,7.09498109428752e-08,6.28506004435501e-08,1.35265852684409e-07,4.03245399084157e-08,4.24694032429586e-08,8.96845972939684e-08,8.2908363906478e-08,5.67613837863177e-08,4.80410801169694e-08,8.49002154062932e-08,8.86019251002276e-08,8.70436757946962e-08,6.24786260020047e-08,9.13359482398315e-08,7.73538585570674e-08,2.53152678917022e-08,2.27630888346611e-07,6.19442198315773e-08,5.27985818313837e-08,9.67822992199918e-08,1.1991759455036e-08,2.29976009395779e-08,1.68106148748185e-08,8.24010847176351e-08,2.21287399126985e-08,8.88813628141348e-08,6.34534576815883e-08,2.45018975712936e-08,8.40553190534547e-09,1.20257361888048e-07,4.00063261831971e-06,4.78606853037412e-08,2.14591515843404e-07,6.89652315999458e-08,7.31998259681976e-08,3.1874728943785e-08,5.00739395245472e-08]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>exposure<\/th>\n      <th>outcome<\/th>\n      <th>id.exposure<\/th>\n      <th>id.outcome<\/th>\n      <th>samplesize<\/th>\n      <th>SNP<\/th>\n      <th>b<\/th>\n      <th>se<\/th>\n      <th>p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualisation" class="section level3 hasAnchor" number="15.5.5">
<h3><span class="header-section-number">15.5.5</span> Visualisation<a href="mr_twosamplemr.html#visualisation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is important to present the causal inference graphically.</p>
<p><strong>Scatter plot</strong></p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="mr_twosamplemr.html#cb193-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">mr</span>(dat)</span></code></pre></div>
<pre><code>## Analysing &#39;5cQr3c&#39; on &#39;ukb-a-534&#39;</code></pre>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="mr_twosamplemr.html#cb195-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">mr_scatter_plot</span>(res, dat)</span>
<span id="cb195-2"><a href="mr_twosamplemr.html#cb195-2" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<pre><code>## $`5cQr3c.ukb-a-534`</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/MR_visualisation_Scatter-1.png" width="672" /></p>
<pre><code>## 
## attr(,&quot;split_type&quot;)
## [1] &quot;data.frame&quot;
## attr(,&quot;split_labels&quot;)
##   id.exposure id.outcome
## 1      5cQr3c  ukb-a-534</code></pre>
<p><strong>Single SNP plot</strong></p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="mr_twosamplemr.html#cb198-1" aria-hidden="true" tabindex="-1"></a>res_single <span class="ot">&lt;-</span> <span class="fu">mr_singlesnp</span>(dat)</span>
<span id="cb198-2"><a href="mr_twosamplemr.html#cb198-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">mr_forest_plot</span>(res_single)</span>
<span id="cb198-3"><a href="mr_twosamplemr.html#cb198-3" aria-hidden="true" tabindex="-1"></a>p2</span></code></pre></div>
<pre><code>## $`5cQr3c.ukb-a-534`</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_errorbarh).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/MR_visualisation_Single-1.png" width="672" /></p>
<pre><code>## 
## attr(,&quot;split_type&quot;)
## [1] &quot;data.frame&quot;
## attr(,&quot;split_labels&quot;)
##   id.exposure id.outcome
## 1      5cQr3c  ukb-a-534</code></pre>
<p><strong>Leave-One-Out plot</strong></p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="mr_twosamplemr.html#cb203-1" aria-hidden="true" tabindex="-1"></a>res_loo <span class="ot">&lt;-</span> <span class="fu">mr_leaveoneout</span>(dat)</span>
<span id="cb203-2"><a href="mr_twosamplemr.html#cb203-2" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">mr_leaveoneout_plot</span>(res_loo)</span>
<span id="cb203-3"><a href="mr_twosamplemr.html#cb203-3" aria-hidden="true" tabindex="-1"></a>p3</span></code></pre></div>
<pre><code>## $`5cQr3c.ukb-a-534`</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_errorbarh).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/MR_visualisation_LOO-1.png" width="672" /></p>
<pre><code>## 
## attr(,&quot;split_type&quot;)
## [1] &quot;data.frame&quot;
## attr(,&quot;split_labels&quot;)
##   id.exposure id.outcome
## 1      5cQr3c  ukb-a-534</code></pre>
<p><strong>Funnel plot</strong></p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="mr_twosamplemr.html#cb208-1" aria-hidden="true" tabindex="-1"></a>res_single <span class="ot">&lt;-</span> <span class="fu">mr_singlesnp</span>(dat)</span>
<span id="cb208-2"><a href="mr_twosamplemr.html#cb208-2" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">mr_funnel_plot</span>(res_single)</span>
<span id="cb208-3"><a href="mr_twosamplemr.html#cb208-3" aria-hidden="true" tabindex="-1"></a>p4</span></code></pre></div>
<pre><code>## $`5cQr3c.ukb-a-534`</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/MR_visualisation_Funnel-1.png" width="672" /></p>
<pre><code>## 
## attr(,&quot;split_type&quot;)
## [1] &quot;data.frame&quot;
## attr(,&quot;split_labels&quot;)
##   id.exposure id.outcome
## 1      5cQr3c  ukb-a-534</code></pre>
<!-- ```{r MR_editing} -->
<!-- generate_odds_ratios(res) -->
<!-- res <- mr(dat) -->
<!-- het <- mr_heterogeneity(dat) -->
<!-- plt <- mr_pleiotropy_test(dat) -->
<!-- sin <- mr_singlesnp(dat) -->
<!-- all_res <- combine_all_mrresults(res, het, -->
<!--                                plt, sin, -->
<!--                                ao_slc = TRUE, -->
<!--                                Exp=TRUE, -->
<!--                                split.exposure = FALSE, -->
<!--                                split.outcome = TRUE) -->
<!-- head(all_res[,c("Method","outcome","exposure","nsnp","b","se","pval","intercept","intercept_se","intercept_pval","Q","Q_df","Q_pval","consortium","ncase","ncontrol","pmid","population")]) -->
<!-- DT::datatable(all_res) -->
<!-- ``` -->
</div>
</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-hemani2018" class="csl-entry">
Hemani G., et al. 2018. <span>“Evaluating the Potential Role of Pleiotropy in Mendelian Randomization Studies.”</span> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6061876">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6061876</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="the-end.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="the-end-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics/edit/master/mendelian_randomization_twosamplemr.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["A Practical Primer in Human Complex Genetics.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
