<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>19 Two Sample Mendelian Randomisation | A Practical Primer in Human Complex Genetics for CoCalc</title>
  <meta name="description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="19 Two Sample Mendelian Randomisation | A Practical Primer in Human Complex Genetics for CoCalc" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/img/cover/cover_2024.png" />
  <meta property="og:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="19 Two Sample Mendelian Randomisation | A Practical Primer in Human Complex Genetics for CoCalc" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="twitter:image" content="/img/cover/cover_2024.png" />

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  <link rel="apple-touch-icon-precomposed" sizes="180x180" href="img/cover/iconified/apple-touch-icon-180x180.png" />
  <link rel="shortcut icon" href="img/cover/iconified/favicon.ico" type="image/x-icon" />
<link rel="prev" href="mr_mrbase.html"/>
<link rel="next" href="epilogue.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a></li>
<li class="chapter" data-level="4" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html"><i class="fa fa-check"></i><b>4</b> Getting started - programs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#plink"><i class="fa fa-check"></i><b>4.2</b> PLINK</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#alternatives-to-plink"><i class="fa fa-check"></i><b>4.2.1</b> Alternatives to <code>PLINK</code></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#other-programs"><i class="fa fa-check"></i><b>4.3</b> Other programs</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html"><i class="fa fa-check"></i><b>5</b> Getting started - CoCalc</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#starting-the-course-on-cocalc"><i class="fa fa-check"></i><b>5.1</b> Starting the course on CoCalc</a></li>
<li class="chapter" data-level="5.2" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#but-wait-what-is-cocalc-anyway"><i class="fa fa-check"></i><b>5.2</b> But wait, what is CoCalc anyway?</a></li>
<li class="chapter" data-level="5.3" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#okay-and-whats-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.3</b> Okay, and what’s a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.4" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-does-a-jupyter-notebook-work"><i class="fa fa-check"></i><b>5.4</b> How does a Jupyter Notebook work?</a></li>
<li class="chapter" data-level="5.5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#and-what-is-the-force-behind-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.5</b> And what is the Force behind a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.6" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-to-use-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.6</b> How to use a Jupyter Notebook</a></li>
<li class="chapter" data-level="5.7" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#trouble-shooting-juptyer-notebook"><i class="fa fa-check"></i><b>5.7</b> Trouble shooting Juptyer Notebook</a></li>
<li class="chapter" data-level="5.8" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#cocalc-and-this-practical-primer"><i class="fa fa-check"></i><b>5.8</b> CoCalc and this Practical Primer</a></li>
<li class="chapter" data-level="5.9" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-jupyter-notebook-with-a-r-kernel"><i class="fa fa-check"></i><b>5.9</b> Beginning your own Jupyter Notebook with a <code>R</code> kernel</a></li>
<li class="chapter" data-level="5.10" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-terminal-for-bash-code"><i class="fa fa-check"></i><b>5.10</b> Beginning your own Terminal for <code>bash</code> code</a></li>
<li class="chapter" data-level="5.11" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#this-book-and-cocalc"><i class="fa fa-check"></i><b>5.11</b> This book and CoCalc</a></li>
<li class="chapter" data-level="5.12" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#are-you-ready"><i class="fa fa-check"></i><b>5.12</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html"><i class="fa fa-check"></i><b>6</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#setting-the-stage"><i class="fa fa-check"></i><b>6.1.1</b> Setting the stage</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#converting-the-data"><i class="fa fa-check"></i><b>6.2</b> Converting the data</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#quality-control"><i class="fa fa-check"></i><b>6.3</b> Quality control</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>6.4</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html"><i class="fa fa-check"></i><b>7</b> Sample QC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sex"><i class="fa fa-check"></i><b>7.1</b> Sex</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sample-call-rates"><i class="fa fa-check"></i><b>7.2</b> Sample call rates</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>7.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="7.4" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#relatedness"><i class="fa fa-check"></i><b>7.4</b> Relatedness</a></li>
<li class="chapter" data-level="7.5" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#ancestral-background"><i class="fa fa-check"></i><b>7.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#g-phase-1"><i class="fa fa-check"></i><b>7.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="7.5.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#principal-component-analysis"><i class="fa fa-check"></i><b>7.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="7.5.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>7.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#removing-samples"><i class="fa fa-check"></i><b>7.6</b> Removing samples</a></li>
<li class="chapter" data-level="7.7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#the-next-step"><i class="fa fa-check"></i><b>7.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html"><i class="fa fa-check"></i><b>8</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="8.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#snp-call-rates"><i class="fa fa-check"></i><b>8.1</b> SNP call rates</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>8.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#allele-frequencies"><i class="fa fa-check"></i><b>8.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>8.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>8.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="8.5" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#final-snp-qc"><i class="fa fa-check"></i><b>8.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="8.6" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-milestone"><i class="fa fa-check"></i><b>8.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html"><i class="fa fa-check"></i><b>9</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#exploring-the-data"><i class="fa fa-check"></i><b>9.1</b> Exploring the data</a></li>
<li class="chapter" data-level="9.2" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#genetic-models"><i class="fa fa-check"></i><b>9.2</b> Genetic models</a></li>
<li class="chapter" data-level="9.3" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#logistic-regression"><i class="fa fa-check"></i><b>9.3</b> Logistic regression</a></li>
<li class="chapter" data-level="9.4" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#lets-get-visual"><i class="fa fa-check"></i><b>9.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html"><i class="fa fa-check"></i><b>10</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#qq-plots"><i class="fa fa-check"></i><b>10.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="10.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#manhattan-plots"><i class="fa fa-check"></i><b>10.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="10.2" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#other-plots"><i class="fa fa-check"></i><b>10.2</b> Other plots</a></li>
<li class="chapter" data-level="10.3" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#interactive-plots"><i class="fa fa-check"></i><b>10.3</b> Interactive plots</a></li>
<li class="chapter" data-level="10.4" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#getting-up-close"><i class="fa fa-check"></i><b>10.4</b> Getting up close</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html"><i class="fa fa-check"></i><b>11</b> Regional association plots</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#locuszoom"><i class="fa fa-check"></i><b>11.1</b> LocusZoom</a></li>
<li class="chapter" data-level="11.2" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#racer"><i class="fa fa-check"></i><b>11.2</b> RACER</a></li>
<li class="chapter" data-level="11.3" data-path="regional_plot_cocalc.html"><a href="regional_plot_cocalc.html#stop-playing-around"><i class="fa fa-check"></i><b>11.3</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>12</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="12.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>12.1</b> Genotyping</a></li>
<li class="chapter" data-level="12.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>12.2</b> The data</a></li>
<li class="chapter" data-level="12.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>12.3</b> Assignment</a></li>
<li class="chapter" data-level="12.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>12.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html"><i class="fa fa-check"></i><b>13</b> WTCCC1: quality control</a>
<ul>
<li class="chapter" data-level="13.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#quality-control-1"><i class="fa fa-check"></i><b>13.1</b> Quality control</a></li>
<li class="chapter" data-level="13.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#ancestral-background-1"><i class="fa fa-check"></i><b>13.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#preparing-pca"><i class="fa fa-check"></i><b>13.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="13.2.2" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#running-pca"><i class="fa fa-check"></i><b>13.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="13.2.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#plotting-pca"><i class="fa fa-check"></i><b>13.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="13.2.4" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#removing-samples-1"><i class="fa fa-check"></i><b>13.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="wtccc1-qc-cocalc.html"><a href="wtccc1-qc-cocalc.html#summary"><i class="fa fa-check"></i><b>13.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html"><i class="fa fa-check"></i><b>14</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#association-testing"><i class="fa fa-check"></i><b>14.1</b> Association testing</a></li>
<li class="chapter" data-level="14.2" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#independent-hits"><i class="fa fa-check"></i><b>14.2</b> Independent hits</a></li>
<li class="chapter" data-level="14.3" data-path="wtccc1-assoc-cocalc.html"><a href="wtccc1-assoc-cocalc.html#replication"><i class="fa fa-check"></i><b>14.3</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>15</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="15.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>15.1</b> Clumping</a></li>
<li class="chapter" data-level="15.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>15.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="15.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>15.3</b> Statistical finemapping</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="post-gwas.html"><a href="post-gwas.html#finemap"><i class="fa fa-check"></i><b>15.3.1</b> FINEMAP</a></li>
<li class="chapter" data-level="15.3.2" data-path="post-gwas.html"><a href="post-gwas.html#locuszoom-1"><i class="fa fa-check"></i><b>15.3.2</b> LocusZoom</a></li>
<li class="chapter" data-level="15.3.3" data-path="post-gwas.html"><a href="post-gwas.html#susie"><i class="fa fa-check"></i><b>15.3.3</b> SuSIE</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>15.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="15.5" data-path="post-gwas.html"><a href="post-gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>15.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="15.6" data-path="post-gwas.html"><a href="post-gwas.html#colocalization"><i class="fa fa-check"></i><b>15.6</b> Colocalization</a></li>
<li class="chapter" data-level="15.7" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>15.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="15.8" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>15.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="post-gwas.html"><a href="post-gwas.html#mrbase"><i class="fa fa-check"></i><b>15.8.1</b> MRBase</a></li>
<li class="chapter" data-level="15.8.2" data-path="post-gwas.html"><a href="post-gwas.html#twosamplemr"><i class="fa fa-check"></i><b>15.8.2</b> TwoSampleMR</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>15.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="15.10" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>15.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>16</b> Functional Mapping and Annotation of GWAS</a>
<ul>
<li class="chapter" data-level="16.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>16.1</b> Assignment</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>16.1.1</b> Tutorial</a></li>
<li class="chapter" data-level="16.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>16.1.2</b> Create the input</a></li>
<li class="chapter" data-level="16.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>16.1.3</b> Run FUMA - SNP to gene</a></li>
<li class="chapter" data-level="16.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>16.1.4</b> Run FUMA - Gene to function</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>16.2</b> Some closing thoughts</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>17</b> Phenome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="17.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>17.1</b> What’s next?</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>18</b> MRBase</a>
<ul>
<li class="chapter" data-level="18.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>18.1</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html"><i class="fa fa-check"></i><b>19</b> Two Sample Mendelian Randomisation</a>
<ul>
<li class="chapter" data-level="19.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#loading-the-package"><i class="fa fa-check"></i><b>19.1</b> Loading the package</a></li>
<li class="chapter" data-level="19.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#available-outcome"><i class="fa fa-check"></i><b>19.2</b> Available outcome</a></li>
<li class="chapter" data-level="19.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#instruments"><i class="fa fa-check"></i><b>19.3</b> Instruments</a></li>
<li class="chapter" data-level="19.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#clumping-1"><i class="fa fa-check"></i><b>19.4</b> Clumping</a></li>
<li class="chapter" data-level="19.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#outcomes"><i class="fa fa-check"></i><b>19.5</b> Outcomes</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#harmonisation"><i class="fa fa-check"></i><b>19.5.1</b> Harmonisation</a></li>
<li class="chapter" data-level="19.5.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#mendelian-randomisation"><i class="fa fa-check"></i><b>19.5.2</b> Mendelian randomisation</a></li>
<li class="chapter" data-level="19.5.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#heterogeneity-pleiotropy"><i class="fa fa-check"></i><b>19.5.3</b> Heterogeneity &amp; Pleiotropy</a></li>
<li class="chapter" data-level="19.5.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#leave-one-out"><i class="fa fa-check"></i><b>19.5.4</b> Leave-one-out</a></li>
<li class="chapter" data-level="19.5.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#visualisation"><i class="fa fa-check"></i><b>19.5.5</b> Visualisation</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#the-end-1"><i class="fa fa-check"></i><b>19.6</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>20</b> Epilogue</a>
<ul>
<li class="chapter" data-level="20.1" data-path="epilogue.html"><a href="epilogue.html#course-lecturers"><i class="fa fa-check"></i><b>20.1</b> 2024 Course Lecturers</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="answers.html"><a href="answers.html"><i class="fa fa-check"></i><b>21</b> Answers to the questions</a>
<ul>
<li class="chapter" data-level="21.1" data-path="answers.html"><a href="answers.html#chapter-3.1-gwas-basics"><i class="fa fa-check"></i><b>21.1</b> Chapter 3.1: GWAS Basics</a></li>
<li class="chapter" data-level="21.2" data-path="answers.html"><a href="answers.html#chapter-3.4-gwas-basics---association-testing"><i class="fa fa-check"></i><b>21.2</b> Chapter 3.4: GWAS Basics - Association testing</a></li>
<li class="chapter" data-level="21.3" data-path="answers.html"><a href="answers.html#chapter-3.5-gwas-visualization"><i class="fa fa-check"></i><b>21.3</b> Chapter 3.5: GWAS visualization</a></li>
<li class="chapter" data-level="21.4" data-path="answers.html"><a href="answers.html#chapter-4.2-wtccc1"><i class="fa fa-check"></i><b>21.4</b> Chapter 4.2: WTCCC1</a></li>
<li class="chapter" data-level="21.5" data-path="answers.html"><a href="answers.html#chapter-4.3-wtccc1---association-testing"><i class="fa fa-check"></i><b>21.5</b> Chapter 4.3: WTCCC1 - Association testing</a></li>
<li class="chapter" data-level="21.6" data-path="answers.html"><a href="answers.html#chapter-5.4-fuma"><i class="fa fa-check"></i><b>21.6</b> Chapter 5.4: FUMA</a></li>
<li class="chapter" data-level="21.7" data-path="answers.html"><a href="answers.html#chapter-5.5-phewas"><i class="fa fa-check"></i><b>21.7</b> Chapter 5.5: PheWAS</a></li>
<li class="chapter" data-level="21.8" data-path="answers.html"><a href="answers.html#chapter-5.7-twosample-mr"><i class="fa fa-check"></i><b>21.8</b> Chapter 5.7: TwoSample MR</a></li>
<li class="chapter" data-level="21.9" data-path="answers.html"><a href="answers.html#chapter-8.1-regional-association-plots"><i class="fa fa-check"></i><b>21.9</b> Chapter 8.1: Regional association plots</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>22</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="22.1" data-path="license.html"><a href="license.html#images-and-data-used"><i class="fa fa-check"></i><b>22.1</b> Images and data used</a></li>
<li class="chapter" data-level="22.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>22.2</b> Disclaimer</a></li>
<li class="chapter" data-level="22.3" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>22.3</b> Copyright</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="colophon.html"><a href="colophon.html"><i class="fa fa-check"></i><b>23</b> Colophon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://vanderlaanand.science" target="blank">&copy; 1979-2024. Sander W. van der Laan.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics for CoCalc</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mr_twosamplemr" class="section level1 hasAnchor" number="19">
<h1><span class="header-section-number">19</span> Two Sample Mendelian Randomisation<a href="mr_twosamplemr.html#mr_twosamplemr" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- ![](./img/_headers/banner_man_standing_dna.png){width=100%} -->
<p>We will use the <code>TwoSampleMR</code> package to assess the causal effect of type 2 diabetes on coronary artery disease.</p>
<p>Briefly, the workflow of a Two-Sample MR is as follows (see also diagram below):</p>
<ol style="list-style-type: decimal">
<li><em>Select instruments</em> for the exposure (perform LD clumping if necessary);</li>
<li><em>Extract the instruments</em> from the MR Base GWAS database for the outcomes of interest;</li>
<li><em>Harmonise the effect sizes</em> for the instruments on the exposures and the outcomes to be each for the same reference allele;</li>
<li><em>Perform MR analysis</em>, sensitivity analyses, create plots, compile reports.</li>
</ol>
<p><img src="img/twosamplemr.png" /></p>
<div id="loading-the-package" class="section level2 hasAnchor" number="19.1">
<h2><span class="header-section-number">19.1</span> Loading the package<a href="mr_twosamplemr.html#loading-the-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>TwoSampleMR</code> is free use. It used to require server-authentication using a gmail-account, now it doesn’t anymore. Should run into an issue, please refer to <a href="https://mrcieu.github.io/TwoSampleMR/articles/introduction.html" target="_blank">this page</a>.</p>
<p>First we load the package.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="mr_twosamplemr.html#cb149-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;TwoSampleMR&quot;</span>)</span>
<span id="cb149-2"><a href="mr_twosamplemr.html#cb149-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;MRInstruments&quot;</span>)</span>
<span id="cb149-3"><a href="mr_twosamplemr.html#cb149-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ieugwasr&quot;</span>)</span></code></pre></div>
</div>
<div id="available-outcome" class="section level2 hasAnchor" number="19.2">
<h2><span class="header-section-number">19.2</span> Available outcome<a href="mr_twosamplemr.html#available-outcome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are ready to load in the available outcomes.</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="mr_twosamplemr.html#cb150-1" tabindex="-1"></a>ao_twosamplemr <span class="ot">&lt;-</span> <span class="fu">available_outcomes</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemravout"></span>
<img src="img/twosamplemr/twosamplemr_avout.png" alt="Available data." width="85%" />
<p class="caption">
Figure 19.1: Available data.
</p>
</div>
</div>
<div id="instruments" class="section level2 hasAnchor" number="19.3">
<h2><span class="header-section-number">19.3</span> Instruments<a href="mr_twosamplemr.html#instruments" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>MRInstruments</code>-package, which comes with <code>TwoSampleMR</code>, includes several types of instruments:</p>
<ul>
<li><em>gwas_catalog</em>, which includes data from GWAS collected by the GWAS Catalog</li>
<li><em>metab_qtls</em>, all known quantitative trait loci (meQTLs) associated to circulating metabolites</li>
<li><em>proteomic_qtls</em>, all known protein QTLs (pQTLs) associated to circulating protein levels</li>
<li><em>gtex_eqtl</em>, all known expression QTLs (eQTLs) associated to gene expression in 53 different tissues from GTEx Portal</li>
<li><em>aries_mqtl</em>, all known methylation QTLs (mQTLs) associated to DNA methylation in blood from ARIES</li>
</ul>
<p>Let’s select T2D associated instruments.</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="mr_twosamplemr.html#cb151-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;gwas_catalog&quot;</span>)</span>
<span id="cb151-2"><a href="mr_twosamplemr.html#cb151-2" tabindex="-1"></a>t2d_gwas <span class="ot">&lt;-</span> <span class="fu">subset</span>(gwas_catalog, <span class="fu">grepl</span>(<span class="st">&quot;Mahajan&quot;</span>, Author) <span class="sc">&amp;</span> PubmedID <span class="sc">==</span> <span class="st">&quot;24509480&quot;</span> <span class="sc">&amp;</span> Phenotype <span class="sc">==</span> <span class="st">&quot;Type 2 diabetes&quot;</span>)</span>
<span id="cb151-3"><a href="mr_twosamplemr.html#cb151-3" tabindex="-1"></a>t2d_exp_dat <span class="ot">&lt;-</span> <span class="fu">format_data</span>(t2d_gwas)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrinstruments"></span>
<img src="img/twosamplemr/twosamplemr_instruments.png" alt="Available data." width="85%" />
<p class="caption">
Figure 19.2: Available data.
</p>
</div>
</div>
<div id="clumping-1" class="section level2 hasAnchor" number="19.4">
<h2><span class="header-section-number">19.4</span> Clumping<a href="mr_twosamplemr.html#clumping-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Like with identifying independent GWAS hits, it is important for TwoSampleMR to ensure that the instruments for the exposure are independent. We can use MR Base to perform clumping. The European samples from the 1000 genomes project are used to estimate LD between SNPs. You can provide a list of SNP IDs, the SNPs will be extracted from 1000 genomes data, LD calculated between them, and amongst those SNPs that have LD R-square above the specified threshold only the SNP with the lowest P-value will be retained. To do this, use the following command.</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="mr_twosamplemr.html#cb152-1" tabindex="-1"></a>t2d_exp_dat <span class="ot">&lt;-</span> <span class="fu">clump_data</span>(t2d_exp_dat)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrclump"></span>
<img src="img/twosamplemr/twosamplemr_clump.png" alt="All available data." width="85%" />
<p class="caption">
Figure 19.3: All available data.
</p>
</div>
</div>
<div id="outcomes" class="section level2 hasAnchor" number="19.5">
<h2><span class="header-section-number">19.5</span> Outcomes<a href="mr_twosamplemr.html#outcomes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Right, so now we are ready to select the proper outcome, CAD. There are multiple datasets available in MRBase, the database used by <code>TwoSampleMR</code>. Let’s review them all and select one with the largest overlap with our exposure SNPs.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="mr_twosamplemr.html#cb153-1" tabindex="-1"></a>ao_twosamplemr[<span class="fu">grepl</span>(<span class="st">&quot;heart disease&quot;</span>, ao_twosamplemr<span class="sc">$</span>trait), ]</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrchd"></span>
<img src="img/twosamplemr/twosamplemr_chd.png" alt="Available CHD data." width="85%" />
<p class="caption">
Figure 19.4: Available CHD data.
</p>
</div>
<p>It turns out there are several options listed below.</p>
<table>
<colgroup>
<col width="17%" />
<col width="17%" />
<col width="17%" />
<col width="14%" />
<col width="17%" />
<col width="17%" />
</colgroup>
<thead>
<tr class="header">
<th>Code</th>
<th>Year</th>
<th>Author</th>
<th>Description</th>
<th>Ncases</th>
<th>Ncontrols</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ukb-d-I9_IHD</td>
<td>2018</td>
<td>Neale Labe</td>
<td>Ischaemic heart disease, wide definition</td>
<td>20857</td>
<td>340337</td>
</tr>
<tr class="even">
<td>ukb-d-I9_CHD_NOREV</td>
<td>2018</td>
<td>Neale Labe</td>
<td>Major coronary heart disease event excluding revascularizations 10157 351037</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>ieu-a-7</td>
<td>2015</td>
<td>Nikpay <em>et al.</em></td>
<td>Coronary artery disease</td>
<td>60801 123504</td>
<td></td>
</tr>
<tr class="even">
<td>ieu-a-9</td>
<td>2013</td>
<td>Deloukas <em>et al. </em></td>
<td>Coronary artery disease</td>
<td>63746 130681</td>
<td></td>
</tr>
<tr class="odd">
<td>ieu-a-8</td>
<td>2011</td>
<td>Schunkert <em>et al.</em></td>
<td>Coronary artery disease</td>
<td>22233</td>
<td>64762</td>
</tr>
</tbody>
</table>
<p>We will use the <code>ieu-a-7</code> dataset as this provides the latest and the greatest, but at the same time (when you go into the literature and dig up the publication) you need a good and sharp definition of the outcome you are interested in.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="mr_twosamplemr.html#cb154-1" tabindex="-1"></a>chd_out_dat <span class="ot">&lt;-</span> <span class="fu">extract_outcome_data</span>(</span>
<span id="cb154-2"><a href="mr_twosamplemr.html#cb154-2" tabindex="-1"></a>    <span class="at">snps =</span> t2d_exp_dat<span class="sc">$</span>SNP,</span>
<span id="cb154-3"><a href="mr_twosamplemr.html#cb154-3" tabindex="-1"></a>    <span class="co"># outcomes = &quot;UKB-a:534&quot; # - this one is deprecated, use the one below</span></span>
<span id="cb154-4"><a href="mr_twosamplemr.html#cb154-4" tabindex="-1"></a>    <span class="co"># Deprecated IDs being used? Detected numeric IDs. Trying to fix, but please note the changes below for future.</span></span>
<span id="cb154-5"><a href="mr_twosamplemr.html#cb154-5" tabindex="-1"></a>    <span class="co"># UKB-a:534  -&gt;  ukb-a-534</span></span>
<span id="cb154-6"><a href="mr_twosamplemr.html#cb154-6" tabindex="-1"></a>    <span class="at">outcomes =</span> <span class="st">&quot;ieu-a-7&quot;</span></span>
<span id="cb154-7"><a href="mr_twosamplemr.html#cb154-7" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrchdout"></span>
<img src="img/twosamplemr/twosamplemr_chdout.png" alt="Extracting available top SNPs." width="85%" />
<p class="caption">
Figure 19.5: Extracting available top SNPs.
</p>
</div>
<div id="harmonisation" class="section level3 hasAnchor" number="19.5.1">
<h3><span class="header-section-number">19.5.1</span> Harmonisation<a href="mr_twosamplemr.html#harmonisation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is important to make sure that the effect allele associated to T2D of any given SNP is the same as the effect allele associated CAD. In other words, we need to harmonise the data.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="mr_twosamplemr.html#cb155-1" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">harmonise_data</span>(</span>
<span id="cb155-2"><a href="mr_twosamplemr.html#cb155-2" tabindex="-1"></a>    <span class="at">exposure_dat =</span> t2d_exp_dat,</span>
<span id="cb155-3"><a href="mr_twosamplemr.html#cb155-3" tabindex="-1"></a>    <span class="at">outcome_dat =</span> chd_out_dat</span>
<span id="cb155-4"><a href="mr_twosamplemr.html#cb155-4" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrharmonise"></span>
<img src="img/twosamplemr/twosamplemr_harmonise.png" alt="Harmonising the instruments." width="85%" />
<p class="caption">
Figure 19.6: Harmonising the instruments.
</p>
</div>
<!-- ### Drop duplicate datasets -->
<!-- After data harmonisation, users may find that their dataset contains duplicate exposure-outcome summary sets. This can arise, for example, when a GWAS consortium has released multiple results from separate GWAS analyses for the same trait.  -->
<!-- The best thing to do, is to prune the datasets so that only the exposure-outcome combination with the highested expected power is retained.  -->
<!-- In our case we only have 1 dataset for the exposure and 1 for the outcome. But if you don't, you can use the command below. -->
<!-- ```{r MR_powerprune} -->
<!-- dat <- power_prune(dat,  -->
<!--                    method = 1, -->
<!--                    dist.outcome = "binary") -->
<!-- ``` -->
</div>
<div id="mendelian-randomisation" class="section level3 hasAnchor" number="19.5.2">
<h3><span class="header-section-number">19.5.2</span> Mendelian randomisation<a href="mr_twosamplemr.html#mendelian-randomisation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that the data is harmonised, we are ready to perform a MR and test for a causal relation of T2D with CAD.</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="mr_twosamplemr.html#cb156-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">mr</span>(dat)</span>
<span id="cb156-2"><a href="mr_twosamplemr.html#cb156-2" tabindex="-1"></a></span>
<span id="cb156-3"><a href="mr_twosamplemr.html#cb156-3" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(res, <span class="at">caption =</span> <span class="st">&quot;MR of type 2 diabetes and coronary artery disease&quot;</span>, <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemranalysis"></span>
<img src="img/twosamplemr/twosamplemr_analysis.png" alt="Primary MR analysis." width="85%" />
<p class="caption">
Figure 19.7: Primary MR analysis.
</p>
</div>
<blockquote>
<p>Question: What is the causal effect of T2D on CAD? Can you describe this using the knowledge from the lecture and the results in the table.</p>
</blockquote>
<p>Several tests are applied. You can change the method(s) applied by using the <code>method_list</code> flag. Below a list of available methods.</p>
<table>
<caption>(#tab:MR_methods)Available MR methods</caption>
<colgroup>
<col width="15%" />
<col width="30%" />
<col width="4%" />
<col width="31%" />
<col width="7%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">obj</th>
<th align="left">name</th>
<th align="left">PubmedID</th>
<th align="left">Description</th>
<th align="left">use_by_default</th>
<th align="left">heterogeneity_test</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">mr_wald_ratio</td>
<td align="left">Wald ratio</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_two_sample_ml</td>
<td align="left">Maximum likelihood</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">mr_egger_regression</td>
<td align="left">MR Egger</td>
<td align="left">26050253</td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">mr_egger_regression_bootstrap</td>
<td align="left">MR Egger (bootstrap)</td>
<td align="left">26050253</td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_simple_median</td>
<td align="left">Simple median</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_weighted_median</td>
<td align="left">Weighted median</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_penalised_weighted_median</td>
<td align="left">Penalised weighted median</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_ivw</td>
<td align="left">Inverse variance weighted</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">mr_ivw_radial</td>
<td align="left">IVW radial</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">mr_ivw_mre</td>
<td align="left">Inverse variance weighted (multiplicative random effects)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_ivw_fe</td>
<td align="left">Inverse variance weighted (fixed effects)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_simple_mode</td>
<td align="left">Simple mode</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_weighted_mode</td>
<td align="left">Weighted mode</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_weighted_mode_nome</td>
<td align="left">Weighted mode (NOME)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_simple_mode_nome</td>
<td align="left">Simple mode (NOME)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_raps</td>
<td align="left">Robust adjusted profile score (RAPS)</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">mr_sign</td>
<td align="left">Sign concordance test</td>
<td align="left"></td>
<td align="left">Tests for concordance of signs between exposure and outcome</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">mr_uwr</td>
<td align="left">Unweighted regression</td>
<td align="left"></td>
<td align="left">Doesn’t use any weights</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
</div>
<div id="heterogeneity-pleiotropy" class="section level3 hasAnchor" number="19.5.3">
<h3><span class="header-section-number">19.5.3</span> Heterogeneity &amp; Pleiotropy<a href="mr_twosamplemr.html#heterogeneity-pleiotropy" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Many genetic variants are associated to multiple traits, thus <em>pleiotropy</em> is common and widespread in the human genome. This poses as an issue to tease apart causal relations in Mendelian randomisation.
<em>Vertical pleiotropy</em> arises because a SNP influences one trait, which in turn influences another. We can use MR to estimate the causal influence between the traits.
<em>Horizontal pleiotropy</em> arises due to SNPs influencing two traits through independent pathways<span class="citation">(<a href="#ref-hemani2018">Hemani G. 2018</a>)</span>.</p>
<p>Further, we should test for study heterogeneity.</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="mr_twosamplemr.html#cb157-1" tabindex="-1"></a><span class="fu">mr_heterogeneity</span>(dat)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrheterogeneity"></span>
<img src="img/twosamplemr/twosamplemr_heterogeneity.png" alt="MR heterogeneity." width="85%" />
<p class="caption">
Figure 19.8: MR heterogeneity.
</p>
</div>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="mr_twosamplemr.html#cb158-1" tabindex="-1"></a><span class="fu">mr_pleiotropy_test</span>(dat)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrpleiotropy"></span>
<img src="img/twosamplemr/twosamplemr_pleiotropy.png" alt="MR pleiotropy." width="85%" />
<p class="caption">
Figure 19.9: MR pleiotropy.
</p>
</div>
<blockquote>
<p>Question: What do the results of the heterogeneity and pleiotropy tests tell you?</p>
</blockquote>
</div>
<div id="leave-one-out" class="section level3 hasAnchor" number="19.5.4">
<h3><span class="header-section-number">19.5.4</span> Leave-one-out<a href="mr_twosamplemr.html#leave-one-out" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is also key to inspect the causal effects per SNP and perform a leave-one-out analysis.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="mr_twosamplemr.html#cb159-1" tabindex="-1"></a>res_single <span class="ot">&lt;-</span> <span class="fu">mr_singlesnp</span>(dat)</span>
<span id="cb159-2"><a href="mr_twosamplemr.html#cb159-2" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(res_single, <span class="at">caption =</span> <span class="st">&quot;Single SNP MR analysis&quot;</span>, <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrsingle"></span>
<img src="img/twosamplemr/twosamplemr_singlesnp.png" alt="Single SNP analysis results." width="85%" />
<p class="caption">
Figure 19.10: Single SNP analysis results.
</p>
</div>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="mr_twosamplemr.html#cb160-1" tabindex="-1"></a>res_loo <span class="ot">&lt;-</span> <span class="fu">mr_leaveoneout</span>(dat)</span>
<span id="cb160-2"><a href="mr_twosamplemr.html#cb160-2" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(res_loo, <span class="at">caption =</span> <span class="st">&quot;Leave-one-out MR analysis&quot;</span>, <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrloo"></span>
<img src="img/twosamplemr/twosamplemr_loo.png" alt="Leave-One-Out analysis results." width="85%" />
<p class="caption">
Figure 19.11: Leave-One-Out analysis results.
</p>
</div>
<blockquote>
<p>Question: What do the results of the single SNP and leave-one-out analyses tell you?</p>
</blockquote>
</div>
<div id="visualisation" class="section level3 hasAnchor" number="19.5.5">
<h3><span class="header-section-number">19.5.5</span> Visualisation<a href="mr_twosamplemr.html#visualisation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is important to present the causal inference graphically.</p>
<p><strong>Scatter plot</strong></p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="mr_twosamplemr.html#cb161-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">mr</span>(dat)</span>
<span id="cb161-2"><a href="mr_twosamplemr.html#cb161-2" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">mr_scatter_plot</span>(res, dat)</span>
<span id="cb161-3"><a href="mr_twosamplemr.html#cb161-3" tabindex="-1"></a>p1</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrscatter"></span>
<img src="img/twosamplemr/twosamplemr_scatter.png" alt="Scatter plot." width="85%" />
<p class="caption">
Figure 19.12: Scatter plot.
</p>
</div>
<blockquote>
<p>Question: What do you see in the scatter plot? And what can you conclude from this?</p>
</blockquote>
<p><strong>Single SNP plot</strong></p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="mr_twosamplemr.html#cb162-1" tabindex="-1"></a>res_single <span class="ot">&lt;-</span> <span class="fu">mr_singlesnp</span>(dat)</span>
<span id="cb162-2"><a href="mr_twosamplemr.html#cb162-2" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">mr_forest_plot</span>(res_single)</span>
<span id="cb162-3"><a href="mr_twosamplemr.html#cb162-3" tabindex="-1"></a>p2</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrforest"></span>
<img src="img/twosamplemr/twosamplemr_forest.png" alt="Single SNP plot." width="85%" />
<p class="caption">
Figure 19.13: Single SNP plot.
</p>
</div>
<blockquote>
<p>Question: What do you see in the forest plot? And what can you conclude from this?</p>
</blockquote>
<p><strong>Leave-One-Out plot</strong></p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="mr_twosamplemr.html#cb163-1" tabindex="-1"></a>res_loo <span class="ot">&lt;-</span> <span class="fu">mr_leaveoneout</span>(dat)</span>
<span id="cb163-2"><a href="mr_twosamplemr.html#cb163-2" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">mr_leaveoneout_plot</span>(res_loo)</span>
<span id="cb163-3"><a href="mr_twosamplemr.html#cb163-3" tabindex="-1"></a>p3</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrlooplot"></span>
<img src="img/twosamplemr/twosamplemr_looplot.png" alt="Leave-One-Out plot." width="85%" />
<p class="caption">
Figure 19.14: Leave-One-Out plot.
</p>
</div>
<blockquote>
<p>Question: what does a leave-one-out ploy tell you? And what can you conclude from this?</p>
</blockquote>
<p><strong>Funnel plot</strong></p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="mr_twosamplemr.html#cb164-1" tabindex="-1"></a>res_single <span class="ot">&lt;-</span> <span class="fu">mr_singlesnp</span>(dat)</span>
<span id="cb164-2"><a href="mr_twosamplemr.html#cb164-2" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">mr_funnel_plot</span>(res_single)</span>
<span id="cb164-3"><a href="mr_twosamplemr.html#cb164-3" tabindex="-1"></a>p4</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:twosamplemrfunnelplot"></span>
<img src="img/twosamplemr/twosamplemr_funnelplot.png" alt="Funnel plot." width="85%" />
<p class="caption">
Figure 19.15: Funnel plot.
</p>
</div>
<blockquote>
<p>Question: Why would yoy use a funnel plot? And what can it tell you?</p>
</blockquote>
<!-- ### Editing the results -->
<!-- You can edit the results to summarize them in a nice table.  -->
<!-- ```{r MR_editing, echo = TRUE, eval = TRUE} -->
<!-- generate_odds_ratios(res) -->
<!-- res <- mr(dat) -->
<!-- het <- mr_heterogeneity(dat) -->
<!-- plt <- mr_pleiotropy_test(dat) -->
<!-- sin <- mr_singlesnp(dat) -->
<!-- all_res <- combine_all_mrresults(res, het, -->
<!--                                plt, sin, -->
<!--                                ao_slc = TRUE, -->
<!--                                Exp=TRUE, -->
<!--                                split.exposure = FALSE, -->
<!--                                split.outcome = TRUE) -->
<!-- head(all_res[,c("Method","outcome","exposure","nsnp","b","se","pval","intercept","intercept_se","intercept_pval","Q","Q_df","Q_pval","consortium","ncase","ncontrol","pmid","population")]) -->
<!-- DT::datatable(all_res) -->
<!-- ``` -->
<blockquote>
<p>Question: What are your main conclusions from the MR analysis?</p>
</blockquote>
</div>
</div>
<div id="the-end-1" class="section level2 hasAnchor" number="19.6">
<h2><span class="header-section-number">19.6</span> The end?<a href="mr_twosamplemr.html#the-end-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You are almost at the end. Time to return to the previous chapter on post-GWAS analyses <a href="post-gwas.html#post-gwas">15</a>.</p>
<script>
title=document.getElementById('header');
title.innerHTML = '<img src="img/headers/banner_man_standing_dna.png" alt="TwoSample Mendelian Randomization">' + title.innerHTML
</script>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-hemani2018" class="csl-entry">
Hemani G., et al. 2018. <span>“Evaluating the Potential Role of Pleiotropy in Mendelian Randomization Studies.”</span> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6061876">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6061876</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mr_mrbase.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="epilogue.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans-serif",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"always_allow_html": true,
"split_by": "chapter",
"toolbar": {
"position": "fixed"
},
"info": true,
"code_folding": "show"
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
