<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Prerequisites | A primer in Human Cardiovascular Genetics</title>
  <meta name="description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Prerequisites | A primer in Human Cardiovascular Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Prerequisites | A primer in Human Cardiovascular Genetics" />
  
  <meta name="twitter:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="basics-of-a-genome-wide-association-study-gwas.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A primer in Human Cardiovascular Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.5</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>3</b> Basics of a Genome-Wide Association Study (GWAS)<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="basics-of-a-genome-wide-association-study-gwas.html"><a href="basics-of-a-genome-wide-association-study-gwas.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Let’s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sample-qc.html"><a href="sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="sample-qc.html"><a href="sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="sample-qc.html"><a href="sample-qc.html#sample-callrates"><i class="fa fa-check"></i><b>4.2</b> Sample Callrates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="sample-qc.html"><a href="sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="sample-qc.html"><a href="sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="sample-qc.html"><a href="sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="sample-qc.html"><a href="sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="sample-qc.html"><a href="sample-qc.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.5.2</b> Principal Component Analysis<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="sample-qc.html"><a href="sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.3</b> 1000G phase 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="sample-qc.html"><a href="sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="sample-qc.html"><a href="sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="per-snp-qc.html"><a href="per-snp-qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="per-snp-qc.html"><a href="per-snp-qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="per-snp-qc.html"><a href="per-snp-qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="per-snp-qc.html"><a href="per-snp-qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="per-snp-qc.html"><a href="per-snp-qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="per-snp-qc.html"><a href="per-snp-qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="per-snp-qc.html"><a href="per-snp-qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html"><i class="fa fa-check"></i><b>6</b> Genome-wide association study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study.html"><a href="genome-wide-association-study.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Let’s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#regional-association-plots"><i class="fa fa-check"></i><b>7.4</b> Regional association plots<span></span></a></li>
<li class="chapter" data-level="7.5" data-path="gwas-visualisation.html"><a href="gwas-visualisation.html#stop-playing-around"><i class="fa fa-check"></i><b>7.5</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium (WTCCC)<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="the-welcome-trust-case-control-consortium-wtccc.html"><a href="the-welcome-trust-case-control-consortium-wtccc.html#there-you-go"><i class="fa fa-check"></i><b>8.1</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><a href="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><i class="fa fa-check"></i><b>9</b> WTCCC1: a GWAS on coronary artery disease (CAD)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><a href="wtccc1-a-gwas-on-coronary-artery-disease-cad.html#quality-control-1"><i class="fa fa-check"></i><b>9.1</b> Quality control<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><a href="wtccc1-a-gwas-on-coronary-artery-disease-cad.html#ancestral-background-1"><i class="fa fa-check"></i><b>9.2</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><a href="wtccc1-a-gwas-on-coronary-artery-disease-cad.html#extreme-cleaning"><i class="fa fa-check"></i><b>9.2.1</b> Extreme cleaning<span></span></a></li>
<li class="chapter" data-level="9.2.2" data-path="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><a href="wtccc1-a-gwas-on-coronary-artery-disease-cad.html#removing-samples-1"><i class="fa fa-check"></i><b>9.2.2</b> Removing samples<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><a href="wtccc1-a-gwas-on-coronary-artery-disease-cad.html#association-testing"><i class="fa fa-check"></i><b>9.3</b> Association testing<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1-a-gwas-on-coronary-artery-disease-cad.html"><a href="wtccc1-a-gwas-on-coronary-artery-disease-cad.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="post-gwas-analyses.html"><a href="post-gwas-analyses.html"><i class="fa fa-check"></i><b>10</b> Post-GWAS Analyses<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="post-gwas-analyses.html"><a href="post-gwas-analyses.html#replication-1"><i class="fa fa-check"></i><b>10.1</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="conditional-analysis.html"><a href="conditional-analysis.html"><i class="fa fa-check"></i><b>11</b> Conditional analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="conditional-analysis.html"><a href="conditional-analysis.html#section-1"><i class="fa fa-check"></i><b>11.1</b> Section 1<span></span></a></li>
<li class="chapter" data-level="11.2" data-path="conditional-analysis.html"><a href="conditional-analysis.html#section-2"><i class="fa fa-check"></i><b>11.2</b> Section 2<span></span></a></li>
<li class="chapter" data-level="11.3" data-path="conditional-analysis.html"><a href="conditional-analysis.html#section-3"><i class="fa fa-check"></i><b>11.3</b> Section 3<span></span></a></li>
<li class="chapter" data-level="11.4" data-path="conditional-analysis.html"><a href="conditional-analysis.html#replication-2"><i class="fa fa-check"></i><b>11.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="statistical-finemapping.html"><a href="statistical-finemapping.html"><i class="fa fa-check"></i><b>12</b> Statistical finemapping<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="statistical-finemapping.html"><a href="statistical-finemapping.html#section-1-1"><i class="fa fa-check"></i><b>12.1</b> Section 1<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="statistical-finemapping.html"><a href="statistical-finemapping.html#section-2-1"><i class="fa fa-check"></i><b>12.2</b> Section 2<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="statistical-finemapping.html"><a href="statistical-finemapping.html#section-3-1"><i class="fa fa-check"></i><b>12.3</b> Section 3<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="statistical-finemapping.html"><a href="statistical-finemapping.html#replication-3"><i class="fa fa-check"></i><b>12.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="functional-mapping-and-annotation-of-gwas.html"><a href="functional-mapping-and-annotation-of-gwas.html"><i class="fa fa-check"></i><b>13</b> Functional Mapping and Annotation of GWAS<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="functional-mapping-and-annotation-of-gwas.html"><a href="functional-mapping-and-annotation-of-gwas.html#assignment"><i class="fa fa-check"></i><b>13.1</b> Assignment<span></span></a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="functional-mapping-and-annotation-of-gwas.html"><a href="functional-mapping-and-annotation-of-gwas.html#tutorial"><i class="fa fa-check"></i><b>13.1.1</b> Tutorial<span></span></a></li>
<li class="chapter" data-level="13.1.2" data-path="functional-mapping-and-annotation-of-gwas.html"><a href="functional-mapping-and-annotation-of-gwas.html#create-the-input"><i class="fa fa-check"></i><b>13.1.2</b> Create the input<span></span></a></li>
<li class="chapter" data-level="13.1.3" data-path="functional-mapping-and-annotation-of-gwas.html"><a href="functional-mapping-and-annotation-of-gwas.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>13.1.3</b> Run FUMA - SNP to gene<span></span></a></li>
<li class="chapter" data-level="13.1.4" data-path="functional-mapping-and-annotation-of-gwas.html"><a href="functional-mapping-and-annotation-of-gwas.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>13.1.4</b> Run FUMA - Gene to function<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="functional-mapping-and-annotation-of-gwas.html"><a href="functional-mapping-and-annotation-of-gwas.html#some-closing-thoughts"><i class="fa fa-check"></i><b>13.2</b> Some closing thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phenome-wide-association-study-phewas.html"><a href="phenome-wide-association-study-phewas.html"><i class="fa fa-check"></i><b>14</b> Phenome-Wide Association Study (PheWAS)<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="phenome-wide-association-study-phewas.html"><a href="phenome-wide-association-study-phewas.html#section-1-2"><i class="fa fa-check"></i><b>14.1</b> Section 1<span></span></a></li>
<li class="chapter" data-level="14.2" data-path="phenome-wide-association-study-phewas.html"><a href="phenome-wide-association-study-phewas.html#section-2-2"><i class="fa fa-check"></i><b>14.2</b> Section 2<span></span></a></li>
<li class="chapter" data-level="14.3" data-path="phenome-wide-association-study-phewas.html"><a href="phenome-wide-association-study-phewas.html#section-3-2"><i class="fa fa-check"></i><b>14.3</b> Section 3<span></span></a></li>
<li class="chapter" data-level="14.4" data-path="phenome-wide-association-study-phewas.html"><a href="phenome-wide-association-study-phewas.html#replication-4"><i class="fa fa-check"></i><b>14.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html"><i class="fa fa-check"></i><b>15</b> Mendelian Randomization (MR)<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-1-3"><i class="fa fa-check"></i><b>15.1</b> Section 1<span></span></a></li>
<li class="chapter" data-level="15.2" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-2-3"><i class="fa fa-check"></i><b>15.2</b> Section 2<span></span></a></li>
<li class="chapter" data-level="15.3" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#section-3-3"><i class="fa fa-check"></i><b>15.3</b> Section 3<span></span></a></li>
<li class="chapter" data-level="15.4" data-path="mendelian-randomization-mr.html"><a href="mendelian-randomization-mr.html#replication-5"><i class="fa fa-check"></i><b>15.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="mendelian-randomization-mr-1.html"><a href="mendelian-randomization-mr-1.html"><i class="fa fa-check"></i><b>16</b> Mendelian Randomization (MR)<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="mendelian-randomization-mr-1.html"><a href="mendelian-randomization-mr-1.html#section-1-4"><i class="fa fa-check"></i><b>16.1</b> Section 1<span></span></a></li>
<li class="chapter" data-level="16.2" data-path="mendelian-randomization-mr-1.html"><a href="mendelian-randomization-mr-1.html#section-2-4"><i class="fa fa-check"></i><b>16.2</b> Section 2<span></span></a></li>
<li class="chapter" data-level="16.3" data-path="mendelian-randomization-mr-1.html"><a href="mendelian-randomization-mr-1.html#section-3-4"><i class="fa fa-check"></i><b>16.3</b> Section 3<span></span></a></li>
<li class="chapter" data-level="16.4" data-path="mendelian-randomization-mr-1.html"><a href="mendelian-randomization-mr-1.html#replication-6"><i class="fa fa-check"></i><b>16.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="license-your-gitbook.html"><a href="license-your-gitbook.html"><i class="fa fa-check"></i><b>17</b> License your GitBook<span></span></a></li>
<li class="chapter" data-level="18" data-path="license-your-gitbook-1.html"><a href="license-your-gitbook-1.html"><i class="fa fa-check"></i><b>18</b> License your GitBook<span></span></a></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A primer in Human Cardiovascular Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="prerequisites" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Prerequisites<a href="prerequisites.html#prerequisites" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="img/using_gitbook.jpeg" /></p>
<div id="linux-macos-and-windows" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Linux, macOS, and Windows<a href="prerequisites.html#linux-macos-and-windows" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Most programs made to execute genetic epidemiology studies are developed for the Unix environment, for example Linux and macOS. So, they may not work as intended in a Windows environment. Windows does allow users to install a linux subsystem within Windows 10 and you can find the detail <a href="https://docs.microsoft.com/en-us/windows/wsl/about" target="_blank">guide</a> here.</p>
<p>However, I highly recommend to 1) either install a linux subsystem on your Windows computer (for example <a href="https://blog.storagecraft.com/the-dead-simple-guide-to-installing-a-linux-virtual-machine-on-windows/" target="_blank">a virtual machine with Ubuntu could work</a>), or 2) switch to macOS in combination with <a href="https://brew.sh" target="_blank">homebrew</a>. This will give you all the flexibility to use Unix-based programs for your genetic epidemiology work and at the same time you’ll keep the advantage of a powerful computer with a user-friendly interface (either Windows or macOS).</p>
<blockquote>
<p>For this practical we use a Windows laptop with Ubuntu on a VirtualMachine. Therefore every command is intended for Linux/macOS, in other words Unix-systems.</p>
</blockquote>
</div>
<div id="programs-you-need" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Programs you need<a href="prerequisites.html#programs-you-need" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<!-- Made this table with https://www.tablesgenerator.com/markdown_tables -->
<p>You need few programs for this practical, or for your (future) genetic epidemiology work for that matter (<strong>Table 1</strong>).</p>
<table>
<colgroup>
<col width="11%" />
<col width="33%" />
<col width="54%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Program</strong></th>
<th><strong>Link</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><em>PLINK</em></td>
<td><a href="https://www.cog-genomics.org/plink2/" class="uri">https://www.cog-genomics.org/plink2/</a></td>
<td>PLINK is a free, open-source genetic analysis tool set, designed to perform a range of basic data parsing and quality control, as well as basic and large-scale analyses in a computationally efficient manner.</td>
</tr>
<tr class="even">
<td><em>R</em></td>
<td><a href="https://cran.r-project.org/" class="uri">https://cran.r-project.org/</a></td>
<td>A program to perform statistical analysis and visualizations.</td>
</tr>
<tr class="odd">
<td><em>RStudio</em></td>
<td><a href="https://www.rstudio.com" class="uri">https://www.rstudio.com</a></td>
<td>A user-friendly R-wrap-around for code editing, debugging, analyses, and visualization.</td>
</tr>
<tr class="even">
<td><em>Homebrew</em></td>
<td><a href="https://brew.sh" class="uri">https://brew.sh</a></td>
<td>A great extension for Mac-users to install really useful programs that Apple didn’t.</td>
</tr>
</tbody>
</table>
<p><strong>Table 1:</strong> Programs needed for genetic epidemiology.</p>
<p>All genetic analyses can be done in PLINK, even on your laptop, but with large datasets, for example <a href="https://www.ukbiobank.ac.uk">UK Biobank</a> size, it is better to switch to a <a href="https://en.wikipedia.org/wiki/High-performance_computing" target="_blank">high-performance computing cluster</a> like we have available at the <a href="https://wiki.bioinformatics.umcutrecht.nl/bin/view/HPC/WebHome" target="_blank">Utrecht Science Park</a>.</p>
<p>Nowadays, a lot of people also use programs like <a href="snptest" target="_blank">SNPTEST</a>, <a href="https://data.broadinstitute.org/alkesgroup/BOLT-LMM/" target="_blank">BOLT-LMM</a>, <a href="http://cnsgenomics.com/software/gcta/#Overview" target="_blank">GCTA</a>, or <a href="https://rgcgithub.github.io/regenie/" target="_blank">regenie</a> as alternatives to execute GWAS and downstream analyses, for example heritability estimation, Fst-calculation, and so on.</p>
<p>Mendelian randomization can be done either with the <a href="http://cnsgenomics.com/software/smr/#Overview" target="_blank">SMR</a> or <a href="http://cnsgenomics.com/software/gsmr/" target="_blank">GSMR</a> function from GCTA, or with R-packages, like <a href="https://mrcieu.github.io/TwoSampleMR/" target="_blank"><code>TwoSampleMR</code></a>.</p>
</div>
<div id="the-terminal" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> The Terminal<a href="prerequisites.html#the-terminal" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For all the above programs, except RStudio, you will need the <code>Terminal</code>. This comes with every major operating system; on Windows it is called ‘PowerShell’, but let’s not go there. And regardless, you will (have to start to) make your own scripts. The benefit of using scripts is that each step in your workflow is clearly stipulated and annotated, and it allows for greater reproducibility, easier troubleshooting, and scaling up to high-performance computer clusters.</p>
<p>Open the terminal, it should be on the left in the toolbar as a little black computer-monitor-like icon.
Mac users can type <code>command + space</code> and type <code>terminal</code>, a terminal screen should open.</p>
<blockquote>
<p>From now on we will use little code blocks like the example to indicate a code you should type/copy-paste and hit enter. If a code is followed by a comment, it is indicated by a # - you don’t need to copy-paste and execute this.</p>
</blockquote>
<pre><code>CODE BLOCK

CODE BLOCK # some comment here
</code></pre>
<div id="download-the-data" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Download the data<a href="prerequisites.html#download-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, let’s start by downloading the data you need for this course to your Desktop: <a href="">LINK</a>.</p>
<p>Alternatively, you could do this through this command. This will create a directory on your Desktop with the command <code>mkdir</code>. The <code>-v</code> flag indicates the program should be <em>verbose</em>, meaning it should tell you what it is doing.</p>
<pre><code>mkdir -v ~/Desktop/practical/</code></pre>
<pre><code>wget &quot;https://www.dropbox.com/sh/kumfwm7drt2flhp/AAB5n0OcUvJixI9pNiymx6-La?dl=0&quot; -P ~/Desktop/practical/
</code></pre>
</div>
<div id="navigating-the-terminal" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Navigating the Terminal<a href="prerequisites.html#navigating-the-terminal" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You can navigate around the computer through the terminal by typing <code>cd &lt;path&gt;</code>; <code>cd</code> stands for “change directory” and <path> means “some_file_directory_you_want_to_go_to”.</p>
<pre><code># For Linux/macOS Users
cd ~ # will bring you to your home directory
cd ../ # will bring you to the parent directory (up one level) 
cd XXX # will bring you to the XXX directory</code></pre>
<p>Let’s navigate to the folder you just downloaded.</p>
<pre><code>cd ~/Desktop/practical</code></pre>
<p>Let’s check out what is inside the directory, by listing (<code>ls</code>) its contents.</p>
<pre><code>ls -lh

# For Linux/macOS Users
ls -l # shows files as list
ls -lh # shows files as list with human readable format 
ls -lt # shows the files as list sorted by time edited
ls -lS # shows the files as list sorted by size</code></pre>
<p>Adding the flags <code>-lh</code> will get you the contents of a directory in a list (<code>-l</code>) and make the size ‘human-readable’ (<code>-h</code>).</p>
<p>You can also count the number of files.</p>
<pre><code>ls | wc -l</code></pre>
</div>
</div>
<div id="installing-some-r-packages" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Installing some R packages<a href="prerequisites.html#installing-some-r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I tested this VirtualMachine and everything should be fine, except some libraries weren’t there. We need to install them.</p>
<p>To be able to install certain <code>r</code>-packages, we need to install some Linux (Ubuntu) software. Type the following:</p>
<pre><code>sudo apt-get install libcurl4 libcurl4-openssl-dev -y

sudo apt-get install libssl-dev</code></pre>
<p>Now close the terminal window - really making sure that the terminal-program has quit.</p>
<p>Open a new terminal window and open <code>r</code> by simply typing <code>R</code> or <code>r</code>. You should install the following packages, and then you’re good to go!</p>
<pre><code>install.packages(c(&quot;httr&quot;, &quot;usethis&quot;, &quot;data.table&quot;, &quot;devtools&quot;, &quot;qqman&quot;, &quot;CMplot&quot;, &quot;tibble&quot;, &quot;plotly&quot;, &quot;dplyr&quot;))
devtools::install_github(&quot;kassambara/ggpubr&quot;)</code></pre>
<p>You should load these packages too.</p>
<pre><code>library(&quot;ggpubr&quot;)
library(&quot;httr&quot;)
library(&quot;usethis&quot;)
library(&quot;data.table&quot;)
library(&quot;devtools&quot;)
library(&quot;qqman&quot;)
library(&quot;CMplot&quot;)
library(&quot;tibble&quot;)
library(&quot;plotly&quot;)
library(&quot;dplyr&quot;)</code></pre>
<p>All in all this may take some time, good moment to relax, review your notes, stretch your legs, or take a coffee.</p>
</div>
<div id="are-you-ready" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Are you ready?<a href="prerequisites.html#are-you-ready" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Are you ready? Did you bring coffee and a good dose of energy? Let’s start! We first cover some basics in (Chapter @ref(gwas_basics)).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="basics-of-a-genome-wide-association-study-gwas.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics/edit/master/prerequisites.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["A primer in Human Cardiovascular Genetics.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
