<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Prerequisites | A primer in Human Cardiovascular Genetics</title>
  <meta name="description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Prerequisites | A primer in Human Cardiovascular Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Prerequisites | A primer in Human Cardiovascular Genetics" />
  
  <meta name="twitter:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan | swvanderlaan" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="getgitbook.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A GitBook for Teaching</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getgitbook.html"><a href="getgitbook.html"><i class="fa fa-check"></i><b>3</b> Get your GitBook<span></span></a></li>
<li class="chapter" data-level="4" data-path="editing-the-book.html"><a href="editing-the-book.html"><i class="fa fa-check"></i><b>4</b> Editing the book<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="editing-the-book.html"><a href="editing-the-book.html#creating-new-chapters"><i class="fa fa-check"></i><b>4.1</b> Creating new chapters<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="editing-the-book.html"><a href="editing-the-book.html#linking-across-chapters"><i class="fa fa-check"></i><b>4.2</b> Linking across chapters<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="editing-the-book.html"><a href="editing-the-book.html#advanced-editing"><i class="fa fa-check"></i><b>4.3</b> Advanced editing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figtab.html"><a href="figtab.html"><i class="fa fa-check"></i><b>5</b> Figures and tables<span></span></a></li>
<li class="chapter" data-level="6" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>6</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="examples.html"><a href="examples.html#doing-meta-analysis-in-r"><i class="fa fa-check"></i><b>6.1</b> Doing Meta-Analysis in R<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="examples.html"><a href="examples.html#theory-construction-and-statistical-modeling"><i class="fa fa-check"></i><b>6.2</b> Theory Construction and Statistical Modeling<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="open-educational-resources.html"><a href="open-educational-resources.html"><i class="fa fa-check"></i><b>7</b> Open Educational Resources<span></span></a></li>
<li class="chapter" data-level="8" data-path="compatibility-with-existing-systems.html"><a href="compatibility-with-existing-systems.html"><i class="fa fa-check"></i><b>8</b> Compatibility with existing systems<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="compatibility-with-existing-systems.html"><a href="compatibility-with-existing-systems.html#add-a-hyperlink"><i class="fa fa-check"></i><b>8.1</b> Add a hyperlink<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="compatibility-with-existing-systems.html"><a href="compatibility-with-existing-systems.html#embed-the-whole-book"><i class="fa fa-check"></i><b>8.2</b> Embed the whole book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="license-your-gitbook.html"><a href="license-your-gitbook.html"><i class="fa fa-check"></i><b>9</b> License your GitBook<span></span></a></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A primer in Human Cardiovascular Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="prerequisites" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Prerequisites<a href="prerequisites.html#prerequisites" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="linux-macos-and-windows" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Linux, macOS, and Windows<a href="prerequisites.html#linux-macos-and-windows" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Most programs made to execute genetic studies are developed for the Unix environment, <em>e.g.</em> Linux and macOS. Therefore, they may not work as intended on window’s environment. Fortunately, Windows now allow users to install a linux subsystem within Windows 10 and you can find the detail <a href="https://docs.microsoft.com/en-us/windows/wsl/about" target="_blank">guide</a> here.<br />
However, we highly recommend to 1) either install a linux subsystem on your Windows computer (<em>e.g.</em> <a href="https://blog.storagecraft.com/the-dead-simple-guide-to-installing-a-linux-virtual-machine-on-windows/" target="_blank">a virtual machine with Ubuntu could work</a>), or 2) switch to macOS in combination with <a href="https://brew.sh" target="_blank">homebrew</a> as this will give you advantage of a powerful computer with a userfriendly interface and all the flexibility to use Unix-based programs for your work.</p>
<blockquote>
<p>For this practical we use a Windows laptop with Ubuntu on a VirtualMachine. Therefore every command is intended for Linux/macOS.</p>
</blockquote>
</div>
<div id="programs-you-need" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Programs you need<a href="prerequisites.html#programs-you-need" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You need few programs for this practical or for your (future) genetic epidemiology work for that matter.</p>
<table style="width:85%;">
<colgroup>
<col width="16%" />
<col width="11%" />
<col width="56%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Program</th>
<th align="left">Link</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PLINK</td>
<td align="left">[<a href="https:/" class="uri">https:/</a></td>
<td align="left">/www.cog-genomics.org/plink2/](<a href="https://www.cog-genomics.org/plink2/" class="uri">https://www.cog-genomics.org/plink2/</a>){target=“_blank”} PLINK is a free, open-source genetic analysis toolset, designed to perform a range of basic data parsing and quality control, as well as basic and large-scale analyses in a computationally efficient manner.</td>
</tr>
<tr class="even">
<td align="left">R</td>
<td align="left">[<a href="https:/" class="uri">https:/</a></td>
<td align="left">/cran.r-project.org](<a href="https://cran.r-project.org/" class="uri">https://cran.r-project.org/</a>){target=“_blank”} Progam to perform statistical analysis and visualisations.</td>
</tr>
<tr class="odd">
<td align="left">RStudio</td>
<td align="left">[<a href="https:/" class="uri">https:/</a></td>
<td align="left">/www.rstudio.com](<a href="https://www.rstudio.com" class="uri">https://www.rstudio.com</a>){target=“_blank”} A userfriendly R-wrap-around for code editing, debugging, analyses, and visualisation</td>
</tr>
</tbody>
</table>
<p>Table 1: Programs needed for genetic epidemiology.</p>
<p>All genetic analyses can be done in PLINK, even on your laptop, but with large datasets, e.g. UK-biobank size, it’s better to switch to a <a href="https://wiki.bioinformatics.umcutrecht.nl/bin/view/HPC/WebHome" target="_blank">high-performance computing cluster</a> like we have at the Uithof.</p>
<p>Nowadays, a lot of people also use programs like <a href="snptest" target="_blank">SNPTEST</a>, <a href="https://data.broadinstitute.org/alkesgroup/BOLT-LMM/" target="_blank">BOLT-LMM</a>, and <a href="http://cnsgenomics.com/software/gcta/#Overview" target="_blank">GCTA</a> as alternatives to run GWAS and downstream analyses, <em>e.g.</em> heritability estimation, Fst-calculation, etc.</p>
<p>Mendelian randomisation can be done either with the <a href="http://cnsgenomics.com/software/smr/#Overview" target="_blank">SMR</a> or <a href="http://cnsgenomics.com/software/gsmr/" target="_blank">GSMR</a> function from GCTA or with R-packages, like <code>TwoSampleMR</code>.</p>
</div>
<div id="the-terminal" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> The Terminal<a href="prerequisites.html#the-terminal" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For all the above programs, except RStudio, you will need the <code>Terminal</code>. This comes with every major operating system. And you will (start to) make your own scripts. The benefit of using scripts is that each step is clearly layed out and easily replicable, thus allows for greater reproducibility, easier troubleshooting and can be run on high-performance computer clusters.</p>
<p>Download the data you need to your Desktop: LINK.</p>
<p>Now open the terminal, it should be on the left in the toolbar as a little black computer-monitor-like icon.
Mac uses can type <code>command + space</code> and type <code>terminal</code>, a terminal screen should open.</p>
<blockquote>
<p>From now on we will use little code blocks like the example to indicate a code you should type/copy-paste and hit enter. If a code is followed by a comment, it is indicated by a # - you don’t need to copy-paste and execute this.</p>
</blockquote>
<pre><code>CODE BLOCK

CODE BLOCK # some comment here

</code></pre>
<p>You can navigate around the computer through the terminal by typing <code>cd &lt;path&gt;</code>; <code>cd</code> stands for “change directory” and <path> means “some_file_directory_you_want_to_go_to”.</p>
<pre><code># For Linux/macOS Users
cd ~ # will bring you to your home directory
cd ../ # will bring you to the parent directory (up one level) 
cd XXX # will bring you to the XXX directory</code></pre>
<p>Let’s navigate to the folder you just downloaded.</p>
<pre><code>cd ~/Desktop/practical</code></pre>
<p>Let’s check out what is inside the directory, by listing (<code>ls</code>) its contents.</p>
<pre><code>ls -lh

# For Linux/macOS Users
ls -l # shows files as list
ls -lh # shows files as list with human readable format 
ls -lt # shows the files as list sorted by time edited
ls -lS # shows the files as list sorted by size</code></pre>
<p>Adding the flags <code>-lh</code> will get you the contents of a directory in a list (<code>-l</code>) and make the size ‘human-readable’ (<code>-h</code>).</p>
<p>You can also count the number of files.</p>
<pre><code>ls | wc -l</code></pre>
</div>
<div id="installing-some-r-packages" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Installing some R packages<a href="prerequisites.html#installing-some-r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I tested this VirtualMachine and everything should be fine, except some libraries weren’t there. We need to install them.</p>
<p>To be able to install certain <code>r</code>-packages, we need to install some Linux (Ubuntu) software. Type the following:</p>
<pre><code>sudo apt-get install libcurl4 libcurl4-openssl-dev -y

sudo apt-get install libssl-dev</code></pre>
<p>Now close the terminal window - really making sure that the terminal-program has quit.</p>
<p>Open a new terminal window and open <code>r</code> by simply typing <code>R</code>. You should install the following packages, and then you’re good to go!</p>
<pre><code>install.packages(c(&quot;httr&quot;, &quot;usethis&quot;, &quot;data.table&quot;, &quot;devtools&quot;, &quot;qqman&quot;, &quot;CMplot&quot;, &quot;tibble&quot;, &quot;plotly&quot;, &quot;dplyr&quot;))

library(c(&quot;httr&quot;, &quot;usethis&quot;, &quot;data.table&quot;, &quot;devtools&quot;, &quot;qqman&quot;, &quot;CMplot&quot;, &quot;tibble&quot;, &quot;plotly&quot;, &quot;dplyr&quot;))
devtools::install_github(&quot;kassambara/ggpubr&quot;)
library(&quot;ggpubr&quot;)
</code></pre>
<p>All in all this may take some time, good moment to relax, review your notes, stretch your legs, or take a coffee.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="getgitbook.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/cjvanlissa/gitbook-demo/edit/master/prerequisites.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["A primer in Human Cardiovascular Genetics.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
