<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Sample QC | A primer in Human Cardiovascular Genetics</title>
  <meta name="description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Sample QC | A primer in Human Cardiovascular Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Sample QC | A primer in Human Cardiovascular Genetics" />
  
  <meta name="twitter:description" content="This is a primer in Human Cardiovascular Genetics. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="prerequisites.html"/>
<link rel="next" href="summary.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A primer in Human Cardiovascular Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#eigensoft"><i class="fa fa-check"></i><b>2.5</b> EIGENSOFT<span></span></a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="prerequisites.html"><a href="prerequisites.html#install-homebrew"><i class="fa fa-check"></i><b>2.5.1</b> Install homebrew<span></span></a></li>
<li class="chapter" data-level="2.5.2" data-path="prerequisites.html"><a href="prerequisites.html#install-missing-packages"><i class="fa fa-check"></i><b>2.5.2</b> Install missing packages<span></span></a></li>
<li class="chapter" data-level="2.5.3" data-path="prerequisites.html"><a href="prerequisites.html#installing-eigensoft"><i class="fa fa-check"></i><b>2.5.3</b> Installing EIGENSOFT<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.6</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sample-qc.html"><a href="sample-qc.html"><i class="fa fa-check"></i><b>3</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="sample-qc.html"><a href="sample-qc.html#sex"><i class="fa fa-check"></i><b>3.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="sample-qc.html"><a href="sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>3.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="sample-qc.html"><a href="sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>3.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="sample-qc.html"><a href="sample-qc.html#relatedness"><i class="fa fa-check"></i><b>3.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="3.5" data-path="sample-qc.html"><a href="sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>3.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="sample-qc.html"><a href="sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>3.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="3.5.2" data-path="sample-qc.html"><a href="sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>3.5.2</b> 1000G phase 1<span></span></a></li>
<li class="chapter" data-level="3.5.3" data-path="sample-qc.html"><a href="sample-qc.html#running-pca"><i class="fa fa-check"></i><b>3.5.3</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="3.5.4" data-path="sample-qc.html"><a href="sample-qc.html#pca-plotting"><i class="fa fa-check"></i><b>3.5.4</b> PCA plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="sample-qc.html"><a href="sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>3.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="3.7" data-path="sample-qc.html"><a href="sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>3.7</b> The next step<span></span></a></li>
</ul></li>
<li><a href="summary.html#summary">Summary<span></span></a>
<ul>
<li class="chapter" data-level="3.8" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>3.8</b> Section 1<span></span></a></li>
<li class="chapter" data-level="3.9" data-path="summary.html"><a href="summary.html#section-2"><i class="fa fa-check"></i><b>3.9</b> Section 2<span></span></a></li>
<li class="chapter" data-level="3.10" data-path="summary.html"><a href="summary.html#section-3"><i class="fa fa-check"></i><b>3.10</b> Section 3<span></span></a></li>
<li class="chapter" data-level="3.11" data-path="summary.html"><a href="summary.html#replication"><i class="fa fa-check"></i><b>3.11</b> Replication!<span></span></a></li>
</ul></li>
<li><a href="epilogue.html#epilogue">Epilogue<span></span></a></li>
<li><a href="licenses-and-disclaimers.html#licenses-and-disclaimers">Licenses and disclaimers<span></span></a></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A primer in Human Cardiovascular Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sample-qc" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Sample QC<a href="sample-qc.html#sample-qc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="img/gwas_sample_qc.png" style="width:70.0%" /></p>
<p>Let’s start with the per-sample quality control.</p>
<div id="sex" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Sex<a href="sample-qc.html#sex" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We need to identify of individuals with discordant sex information comparing phenotypic and genotypic data. Let’s calculate the mean homozygosity rate across X-chromosome markers for each individual in the study.</p>
<pre><code>plink --bfile rawdata/rawdata --check-sex --out rawdata/rawdata
</code></pre>
<p>This produces a file with the following columns:</p>
<ul>
<li><em>FID</em> Family ID</li>
<li><em>IID</em> Within-family ID</li>
<li><em>PEDSEX</em> Sex code in input file</li>
<li><em>SNPSEX</em> Imputed sex code (1 = male, 2 = female, 0 = unknown)</li>
<li><em>STATUS</em> ‘OK’ if PEDSEX and SNPSEX match and are nonzero, ‘PROBLEM’ otherwise</li>
<li><em>F</em> Inbreeding coefficient, considering only X chromosome. Not present with ‘y-only’.</li>
<li><em>YCOUNT</em> Number of nonmissing genotype calls on Y chromosome. Requires ‘ycount’/‘y-only’.</li>
</ul>
<p>We need to get a list of individuals with discordant sex data.</p>
<pre><code>cat rawdata/rawdata.sexcheck | awk &#39;$5 ==&quot;STATUS&quot; || $5 ==&quot;PROBLEM&quot;&#39;  &gt; rawdata/rawdata.sexprobs.txt</code></pre>
<p>Let’s have a look at the results.</p>
<pre><code>cat rawdata/rawdata.sexprobs.txt</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="sample-qc.html#cb19-1" aria-hidden="true" tabindex="-1"></a>sexissues <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc,<span class="st">&quot;/rawdata/rawdata.sexprobs.txt&quot;</span>))</span></code></pre></div>
<table>
<caption><span id="tab:sex-issues">Table 3.1: </span>Sex issues</caption>
<thead>
<tr class="header">
<th align="right">FID</th>
<th align="right">IID</th>
<th align="right">PEDSEX</th>
<th align="right">SNPSEX</th>
<th align="left">STATUS</th>
<th align="right">F</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">772</td>
<td align="right">772</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">PROBLEM</td>
<td align="right">0.3084</td>
</tr>
<tr class="even">
<td align="right">853</td>
<td align="right">853</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">PROBLEM</td>
<td align="right">0.3666</td>
</tr>
<tr class="odd">
<td align="right">1920</td>
<td align="right">1920</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">PROBLEM</td>
<td align="right">0.4066</td>
</tr>
</tbody>
</table>
<p>When the homozygosity rate (<em>F</em>) is more than 0.2, but less than 0.8, the genotype data are inconclusive regarding the sex of an individual and these are marked in column <em>SNPSEX</em> with a 0, and the column <em>STATUS</em> “PROBLEM”.</p>
<p>Report the IDs of individuals with discordant sex information (Table <a href="sample-qc.html#tab:sex-issues">3.1</a>) to those who conducted sex phenotyping. In situations in which discrepancy cannot be resolved, add the family ID (FID) and individual ID (IID) of the samples to a file named “fail-sexcheck-qc.txt” (one individual per line, tab delimited).</p>
<pre><code>grep &quot;PROBLEM&quot; rawdata/rawdata.sexcheck | awk &#39;{ print $1, $2}&#39;  &gt; rawdata/fail-sexcheck-qc.txt
</code></pre>
</div>
<div id="sample-call-rates" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Sample call rates<a href="sample-qc.html#sample-call-rates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s get an overview of the missing data per sample and per SNP.</p>
<pre><code>plink --bfile rawdata/rawdata --missing --out rawdata/rawdata
</code></pre>
<p>This produces two files, <code>rawdata/rawdata.imiss</code> and <code>rawdata/rawdata.lmiss</code>. In the .imiss file the <em>N_MISS</em> column denotes the number of missing SNPs, and the <em>F_MISS</em> column denotes the proportion of missing SNPs per individual.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="sample-qc.html#cb22-1" aria-hidden="true" tabindex="-1"></a>raw_IMISS <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/rawdata/rawdata.imiss&quot;</span>))</span>
<span id="cb22-2"><a href="sample-qc.html#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="sample-qc.html#cb22-3" aria-hidden="true" tabindex="-1"></a>raw_IMISS<span class="sc">$</span>callrate <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> raw_IMISS<span class="sc">$</span>F_MISS</span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="sample-qc.html#cb23-1" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(raw_IMISS, <span class="at">x =</span> <span class="st">&quot;callrate&quot;</span>,</span>
<span id="cb23-2"><a href="sample-qc.html#cb23-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb23-3"><a href="sample-qc.html#cb23-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>, <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb23-4"><a href="sample-qc.html#cb23-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb23-5"><a href="sample-qc.html#cb23-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;per sample call rate&quot;</span>) <span class="sc">+</span></span>
<span id="cb23-6"><a href="sample-qc.html#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb23-7"><a href="sample-qc.html#cb23-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:sample-callrate"></span>
<img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/sample-callrate-1.png" alt="Call rate" width="80%" />
<p class="caption">
Figure 3.1: Call rate
</p>
</div>
<p>The grey dashed line in Figure <a href="sample-qc.html#fig:sample-callrate">3.1</a> indicates the mean call rate, while the red dashed line indicates the threshold we had determined above.</p>
</div>
<div id="heterozygosity-rate" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Heterozygosity rate<a href="sample-qc.html#heterozygosity-rate" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To properly calculate heterozygosity rate and relatedness (identity-by-descent [IBD]) we need to do four things:</p>
<ol style="list-style-type: decimal">
<li>pre-clean the data to get a high-quality set,</li>
<li>of independent SNPs,</li>
<li>exclude long-range linkage disequilibrium (LD) blocks that bias with these calculations, and</li>
<li>exclude A/T and C/G SNPs as these may be ambivalent in interpretation when frequencies between cases and controls are close (MAF ± 0.45),</li>
<li>remove all non-autosomal SNPs.</li>
</ol>
<p>We will use the following settings:</p>
<ul>
<li>remove A/T and C/G SNPs with the flag <code>--exclude rawdata/all.atcg.variants.txt</code>,</li>
<li>call rate &lt;1% with the flag <code>--geno 0.10</code>,</li>
<li>Hardy-Weinberg Equilibrium (HWE) p-value &gt; 1x10-3 with the flag <code>--hwe 1e-3</code>,</li>
<li>and MAF&gt;10% with the flag <code>--maf 0.10</code>,</li>
<li>prune the data to only select independent SNPs (with low LD r^2) of one pair each with <code>r^2 = 0.2</code> with the flags <code>--indep-pairwise 100 10 0.2</code> and <code>--extract rawdata/raw-GWA-data.prune.in</code>,</li>
<li>SNPs in long-range LD regions (for example: MHC chr 6 25.8-36Mb, chr 8 inversion 6-16Mb, chr17 40-45Mb, and a few more) with the flag <code>--exclude range support/exclude_problematic_range.txt</code>,</li>
<li>remove non-autosomal SNPs with the flag <code>--allow-no-sex --autosome</code>.</li>
</ul>
<p>First, get a list of A/T and C/G SNPs.</p>
<pre><code>cat rawdata/rawdata.bim | \
awk &#39;($5 == &quot;A&quot; &amp;&amp; $6 == &quot;T&quot;) || ($5 == &quot;T&quot; &amp;&amp; $6 == &quot;A&quot;) || ($5 == &quot;C&quot; &amp;&amp; $6 == &quot;G&quot;) || ($5 == &quot;G&quot; &amp;&amp; $6 == &quot;C&quot;)&#39; | awk &#39;{ print $2, $1, $4, $3, $5, $6 }&#39; \
&gt; rawdata/all.atcg.variants.txt</code></pre>
<p>Second, clean the data and get a list of independent SNPs.</p>
<!-- *NEXT YEAR: further explain the exact commands  and how they relate with the above.* -->
<pre><code>plink --bfile rawdata/rawdata \
--allow-no-sex --autosome \
--maf 0.10 --geno 0.10 --hwe 1e-3 \
--indep-pairwise 100 10 0.2 \
--exclude range support/exclude_problematic_range.txt \
--make-bed --out rawdata/rawdata.clean.temp</code></pre>
<blockquote>
<p>Please note, we have create a dataset without taking into account LD structure. Thus the flag <code>--indep-pairwise 100 10 0.2</code> doesn’t actually work. However, with real-data you can use it to prune out unwanted SNPs in high LD.</p>
</blockquote>
<p>Third, exclude the pruned SNPs. Note, how we include a file to exclude high-LD for the purpose of the practical.</p>
<pre><code>plink --bfile rawdata/rawdata.clean.temp \
--extract rawdata/raw-GWA-data.prune.in \
--make-bed --out rawdata/rawdata.clean.ultraclean.temp
</code></pre>
<p>Fourth, remove the A/T and C/G SNPs.</p>
<pre><code>plink --bfile rawdata/rawdata.clean.ultraclean.temp \
--exclude rawdata/all.atcg.variants.txt \
--make-bed --out rawdata/rawdata.clean.ultraclean</code></pre>
<blockquote>
<p>Please note, this dataset doesn’t actually include this type of SNP, hence <code>rawdata/all.atcg.variants.txt</code> is empty! Again, you can use this command in real-data to exclude A/T and C/G SNPs.</p>
</blockquote>
<p>Lastly, remove the temporary files.</p>
<pre><code>rm -v rawdata/*.temp*
</code></pre>
<p>Finally, we can calculate the heterozygosity rate.</p>
<pre><code>plink --bfile rawdata/rawdata.clean.ultraclean --het --out rawdata/rawdata.clean.ultraclean</code></pre>
<p>This creates the file <code>rawdata/rawdata.clean.ultraclean.het</code>, in which the third column denotes the observed number of homozygous genotypes, O(Hom), and the fifth column denotes the number of nonmissing genotypes, N(NM), per individual. We can now calculate the observed heterozygosity rate per individual using the formula (N(NM) - O(Hom))/N(NM).</p>
<p>Often there is a correlation between heterozygosity rate and missing data. Thus, we should plot the observed heterozygosity rate per individual on the x-axis and the proportion of missing SNP, that is the ‘SNP call rate’, per individuals on the y-axis (Figure <a href="sample-qc.html#fig:show-heterozygosity">3.2</a>).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="sample-qc.html#cb30-1" aria-hidden="true" tabindex="-1"></a>raw_HET <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/rawdata/rawdata.clean.ultraclean.het&quot;</span>))</span>
<span id="cb30-2"><a href="sample-qc.html#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="sample-qc.html#cb30-3" aria-hidden="true" tabindex="-1"></a>raw_IMISS<span class="sc">$</span>logF_MISS <span class="ot">=</span> <span class="fu">log10</span>(raw_IMISS<span class="sc">$</span>F_MISS)</span>
<span id="cb30-4"><a href="sample-qc.html#cb30-4" aria-hidden="true" tabindex="-1"></a>prop_miss <span class="ot">=</span> <span class="sc">-</span><span class="fl">1.522879</span></span>
<span id="cb30-5"><a href="sample-qc.html#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="sample-qc.html#cb30-6" aria-hidden="true" tabindex="-1"></a>raw_HET<span class="sc">$</span>meanHet <span class="ot">=</span> (raw_HET<span class="sc">$</span><span class="st">`</span><span class="at">N(NM)</span><span class="st">`</span> <span class="sc">-</span> raw_HET<span class="sc">$</span><span class="st">`</span><span class="at">O(HOM)</span><span class="st">`</span>)<span class="sc">/</span>raw_HET<span class="sc">$</span><span class="st">`</span><span class="at">N(NM)</span><span class="st">`</span></span>
<span id="cb30-7"><a href="sample-qc.html#cb30-7" aria-hidden="true" tabindex="-1"></a>lower_meanHet <span class="ot">=</span> <span class="fu">mean</span>(raw_HET<span class="sc">$</span>meanHet) <span class="sc">-</span> (<span class="dv">2</span><span class="sc">*</span><span class="fu">sd</span>(raw_HET<span class="sc">$</span>meanHet))</span>
<span id="cb30-8"><a href="sample-qc.html#cb30-8" aria-hidden="true" tabindex="-1"></a>upper_meanHet <span class="ot">=</span> <span class="fu">mean</span>(raw_HET<span class="sc">$</span>meanHet) <span class="sc">+</span> (<span class="dv">2</span><span class="sc">*</span><span class="fu">sd</span>(raw_HET<span class="sc">$</span>meanHet))</span>
<span id="cb30-9"><a href="sample-qc.html#cb30-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-10"><a href="sample-qc.html#cb30-10" aria-hidden="true" tabindex="-1"></a>raw_IMISSHET <span class="ot">=</span> <span class="fu">merge</span>(raw_IMISS, raw_HET, <span class="at">by =</span> <span class="st">&quot;IID&quot;</span>)</span>
<span id="cb30-11"><a href="sample-qc.html#cb30-11" aria-hidden="true" tabindex="-1"></a>raw_IMISSHET<span class="sc">$</span>FID.y <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb30-12"><a href="sample-qc.html#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_IMISSHET)[<span class="fu">colnames</span>(raw_IMISSHET)<span class="sc">==</span><span class="st">&quot;FID.x&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;FID&quot;</span></span>
<span id="cb30-13"><a href="sample-qc.html#cb30-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-14"><a href="sample-qc.html#cb30-14" aria-hidden="true" tabindex="-1"></a>colors  <span class="ot">&lt;-</span> <span class="fu">densCols</span>(raw_IMISSHET<span class="sc">$</span>logF_MISS, raw_IMISSHET<span class="sc">$</span>meanHet)</span></code></pre></div>
<pre><code>## Warning in KernSmooth::bkde2D(x, bandwidth = bandwidth, gridsize = nbin, :
## Binning grid too coarse for current (small) bandwidth: consider increasing
## &#39;gridsize&#39;</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="sample-qc.html#cb32-1" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggscatter</span>(raw_IMISSHET, <span class="at">x =</span> <span class="st">&quot;logF_MISS&quot;</span>, <span class="at">y =</span> <span class="st">&quot;meanHet&quot;</span>,</span>
<span id="cb32-2"><a href="sample-qc.html#cb32-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> colors,</span>
<span id="cb32-3"><a href="sample-qc.html#cb32-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">xlab =</span> <span class="st">&quot;Proportion of missing genotypes&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Heterozygosity rate&quot;</span>) <span class="sc">+</span></span>
<span id="cb32-4"><a href="sample-qc.html#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;-3&quot;</span> <span class="ot">=</span> <span class="st">&quot;0.001&quot;</span>, <span class="st">&quot;-2&quot;</span> <span class="ot">=</span> <span class="st">&quot;0.01&quot;</span>,</span>
<span id="cb32-5"><a href="sample-qc.html#cb32-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;-1&quot;</span> <span class="ot">=</span> <span class="st">&quot;0.1&quot;</span>, <span class="st">&quot;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb32-6"><a href="sample-qc.html#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> lower_meanHet, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb32-7"><a href="sample-qc.html#cb32-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb32-8"><a href="sample-qc.html#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> upper_meanHet, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb32-9"><a href="sample-qc.html#cb32-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb32-10"><a href="sample-qc.html#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> prop_miss, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb32-11"><a href="sample-qc.html#cb32-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-heterozygosity"></span>
<img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/show-heterozygosity-1.png" alt="Heterozygosity rate as a function of missing genotypes" width="80%" />
<p class="caption">
Figure 3.2: Heterozygosity rate as a function of missing genotypes
</p>
</div>
<p>Examine the plot to decide reasonable thresholds at which to exclude individuals based on elevated missing or extreme heterozygosity. We chose to exclude all individuals with a genotype failure rate &gt;= 0.03 (vertical dashed line) and/or a heterozygosity rate ± 3 s.d. from the mean (horizontal dashed lines). Add the FID and IID of the samples failing this QC to the file named <code>fail-imisshet-qc.txt</code>.</p>
<blockquote>
<p>How would you create this file?</p>
</blockquote>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="sample-qc.html#cb33-1" aria-hidden="true" tabindex="-1"></a>raw_IMISSHETsub <span class="ot">=</span> <span class="fu">subset</span>(raw_IMISSHET, logF_MISS <span class="sc">&gt;</span> prop_miss <span class="sc">|</span> (meanHet <span class="sc">&lt;</span> lower_meanHet <span class="sc">|</span> meanHet <span class="sc">&gt;</span> upper_meanHet),</span>
<span id="cb33-2"><a href="sample-qc.html#cb33-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;FID&quot;</span>, <span class="st">&quot;IID&quot;</span>))</span>
<span id="cb33-3"><a href="sample-qc.html#cb33-3" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">fwrite</span>(raw_IMISSHETsub, <span class="fu">paste0</span>(COURSE_loc,<span class="st">&quot;/rawdata/fail-raw_IMISSHETsub.txt&quot;</span>), <span class="at">sep =</span><span class="st">&quot; &quot;</span>)</span></code></pre></div>
</div>
<div id="relatedness" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Relatedness<a href="sample-qc.html#relatedness" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We calculate Identity-by-Descent (IBS), to identify duplicated and related samples (Table <a href="sample-qc.html#tab:show-relatedness">3.2</a>). IBS is measured by calculating pi-hat, which is in essence the proportion of the DNA that a pair of samples share. To calculate this, we needed this ultraclean dataset, without low-quality SNPs and without high-LD regions. Now we are ready to calculate the IBS.</p>
<table>
<caption><span id="tab:show-relatedness">Table 3.2: </span>Familial relations and % DNA shared.</caption>
<thead>
<tr class="header">
<th align="left">Relatedness</th>
<th align="left">%.DNA.sharing</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Monozygotic twins</td>
<td align="left">±100%</td>
</tr>
<tr class="even">
<td align="left">Parents/child</td>
<td align="left">±50%</td>
</tr>
<tr class="odd">
<td align="left">Sibling</td>
<td align="left">±50%</td>
</tr>
<tr class="even">
<td align="left">Fraternal twins</td>
<td align="left">±50%</td>
</tr>
<tr class="odd">
<td align="left">Grandparent/grandchild</td>
<td align="left">±25%</td>
</tr>
<tr class="even">
<td align="left">Aunt/Uncle/Niece/Nephew</td>
<td align="left">±25%</td>
</tr>
<tr class="odd">
<td align="left">Half-sibling</td>
<td align="left">±25%</td>
</tr>
<tr class="even">
<td align="left">First-cousin</td>
<td align="left">±12.5%</td>
</tr>
<tr class="odd">
<td align="left">Half first-cousin</td>
<td align="left">±6.25%</td>
</tr>
<tr class="even">
<td align="left">First-cousin once removed</td>
<td align="left">±6.25%</td>
</tr>
<tr class="odd">
<td align="left">Second-cousin</td>
<td align="left">±3.13%</td>
</tr>
<tr class="even">
<td align="left">Second-cousin once removed</td>
<td align="left">±1.56%</td>
</tr>
</tbody>
</table>
<pre><code>plink --bfile rawdata/rawdata.clean.ultraclean --genome --out rawdata/rawdata.clean.ultraclean</code></pre>
<p>We can now identify all pairs of individuals with an IBD &gt; 0.185. The code looks at the individual call rates stored in rawdata.imiss and outputs the IDs of the individual with the lowest call rate to ‘fail-IBD-QC.txt’ for subsequent removal (Table <a href="sample-qc.html#tab:show-ibdcallissues">3.3</a>).</p>
<pre><code>cd rawdata

perl ../scripts/run-IBD-QC.pl rawdata rawdata.clean.ultraclean

cd ..
</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="sample-qc.html#cb36-1" aria-hidden="true" tabindex="-1"></a>ibdcallissues <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc,<span class="st">&quot;/rawdata/fail-IBD-QC.txt&quot;</span>))</span></code></pre></div>
<table>
<caption><span id="tab:show-ibdcallissues">Table 3.3: </span>Failed IBD and callrate.</caption>
<thead>
<tr class="header">
<th align="right">V1</th>
<th align="right">V2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1952</td>
<td align="right">1952</td>
</tr>
<tr class="even">
<td align="right">1953</td>
<td align="right">1953</td>
</tr>
<tr class="odd">
<td align="right">1954</td>
<td align="right">1954</td>
</tr>
<tr class="even">
<td align="right">1955</td>
<td align="right">1955</td>
</tr>
<tr class="odd">
<td align="right">1957</td>
<td align="right">1957</td>
</tr>
<tr class="even">
<td align="right">1959</td>
<td align="right">1959</td>
</tr>
<tr class="odd">
<td align="right">1961</td>
<td align="right">1961</td>
</tr>
<tr class="even">
<td align="right">1963</td>
<td align="right">1963</td>
</tr>
<tr class="odd">
<td align="right">1965</td>
<td align="right">1965</td>
</tr>
<tr class="even">
<td align="right">1967</td>
<td align="right">1967</td>
</tr>
<tr class="odd">
<td align="right">1969</td>
<td align="right">1969</td>
</tr>
<tr class="even">
<td align="right">1971</td>
<td align="right">1971</td>
</tr>
<tr class="odd">
<td align="right">1973</td>
<td align="right">1973</td>
</tr>
<tr class="even">
<td align="right">1975</td>
<td align="right">1975</td>
</tr>
</tbody>
</table>
</div>
<div id="ancestral-background" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Ancestral background<a href="sample-qc.html#ancestral-background" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using a <strong>Principal Component Analysis (PCA)</strong> we can reduce the dimensions of the data, and project the “ancestral distances”. In other words, the principal component 1 (the first dimension) and principal component 2 (the second dimension) which will capture most of the variation in the data and represent how much each sample is alike the next. And when compared to a reference, you can deduce the ancestral background of each sample in your dataset. Of course this is relative: we will only know that a given sample is very much a like samples from a given population that exists today.</p>
<p>Nowadays we run such PCA against a large and diverse dataset containing many different populations. Old-school GWAS would compare a dataset against HapMap 3, nowadays we prefer at a minimum the 1000G populations. The thing is, the preferred software to run a PCA is tricky to install (see Chapter <a href="prerequisites.html#prerequisites">2</a>). However, for you convenience I ran the PCA for mapping of this dummy dataset against HapMap 3 and 1000G already and shared the files in the <code>ref_pca</code>-folder. This means you can skip the <a href="sample-qc.html#hapmap-3">HapMap 3</a> and <a href="sample-qc.html#g-phase-1">1000G phase 1</a> sections and jump straight to <a href="sample-qc.html#pca-plotting">PCA plotting</a>.</p>
<div id="hapmap-3" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> HapMap 3<a href="sample-qc.html#hapmap-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will project our data to a reference, in this example HapMap Phase II (HapMap3), which includes individuals from four distinct global populations, but it could also be 1000G phase 1. Or any other reference depending on the dataset.</p>
<p>To this end we will merge our data with HapMap3. The alleles at each marker must be aligned to the same DNA strand to allow our data to merge correctly. Because not all SNPs are required for this analysis, A-&gt;T and C-&gt;G SNPs, which are more difficult to align, can be omitted.</p>
<div id="filter-the-hapmap-3-data" class="section level4 hasAnchor" number="3.5.1.1">
<h4><span class="header-section-number">3.5.1.1</span> Filter the HapMap 3 data<a href="sample-qc.html#filter-the-hapmap-3-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s start by creating a new BED file, excluding from the GWA data those SNPs that do not feature in the genotype data of the four original HapMap3 populations.</p>
<pre><code>plink --bfile rawdata/rawdata --extract ref_hapmap3/hapmap3r2_CEU.CHB.JPT.YRI.no-at-cg-snps.txt --make-bed --out rawdata/rawdata.hm3</code></pre>
</div>
<div id="merging-datasets" class="section level4 hasAnchor" number="3.5.1.2">
<h4><span class="header-section-number">3.5.1.2</span> Merging datasets<a href="sample-qc.html#merging-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now, let’s try to merge <code>rawdata/rawdata.hm3</code> with the HapMap data and extract the pruned SNP set from above.</p>
<pre><code>plink --bfile rawdata/rawdata.hm3 --bmerge ref_hapmap3/hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps --extract rawdata/raw-GWA-data.prune.in --make-bed --out rawdata/rawdata.hapmap3r2.pruned</code></pre>
<p>You probably get an error like below:</p>
<pre><code>Error: 59 variants with 3+ alleles present.
* If you believe this is due to strand inconsistency, try --flip with
  rawdata/rawdata.hapmap3r2.pruned-merge.missnp.
  (Warning: if this seems to work, strand errors involving SNPs with A/T or C/G
  alleles probably remain in your data.  If LD between nearby SNPs is high,
  --flip-scan should detect them.)
* If you are dealing with genuine multiallelic variants, we recommend exporting
  that subset of the data to VCF (via e.g. &#39;--recode vcf&#39;), merging with
  another tool/script, and then importing the result; PLINK is not yet suited
  to handling them.</code></pre>
<p>Because all A-&gt;T and C-&gt;G SNPs have been removed before undertaking this analysis, all other SNPs that are discordant for DNA strands between the two data sets are listed in the <code>rawdata.hapmap3r2.pruned-merge.missnp</code> file. To align the strands across the data sets and successfully complete the merge, we can do the following:</p>
<pre><code>plink --bfile rawdata/rawdata --extract ref_hapmap3/hapmap3r2_CEU.CHB.JPT.YRI.no-at-cg-snps.txt --flip rawdata/rawdata.hapmap3r2.pruned-merge.missnp --make-bed --out rawdata/rawdata.hm3</code></pre>
<p>And repeat this:</p>
<pre><code>plink --bfile rawdata/rawdata.hm3 --bmerge ref_hapmap3/hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps --extract rawdata/raw-GWA-data.prune.in --make-bed --out rawdata/rawdata.hapmap3r2.pruned</code></pre>
<p>Let’s not be lazy and clean this dataset too.</p>
<pre><code>plink --bfile rawdata/rawdata.hapmap3r2.pruned \
--allow-no-sex --autosome \
--maf 0.10 --geno 0.10 --hwe 1e-3 \
--indep-pairwise 100 10 0.2 \
--exclude range support/exclude_problematic_range.txt \
--make-bed --out rawdata/rawdata.hapmap3r2.pruned.clean</code></pre>
<p>Now we have prepared our dataset to include high-quality SNPs with few missing data, high-frequent SNPs, excluding problematic ranges, and merged to the HapMap 3 reference dataset.</p>
</div>
</div>
<div id="g-phase-1" class="section level3 hasAnchor" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> 1000G phase 1<a href="sample-qc.html#g-phase-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You could also merge your data and project it against the 1000G phase 1 populations. The 1000G phase 1 data is provided seperately, as it is rather big to share if you’re not planning on using it.</p>
<div id="download-1000g-phase-1" class="section level4 hasAnchor" number="3.5.2.1">
<h4><span class="header-section-number">3.5.2.1</span> Download 1000G phase 1<a href="sample-qc.html#download-1000g-phase-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We start by creating the necessary folder to save the data to.</p>
<pre><code>mkdir -v ~/Desktop/practical/ref_1kg_phase1_all</code></pre>
<p>Next, we’ll download the reference.</p>
<pre><code>wget &quot;https://www.dropbox.com/sh/kumfwm7drt2flhp/AAB5n0OcUvJixI9pNiymx6-La?dl=0&quot; -P ~/Desktop/practical/ref_1kg_phase1_all
</code></pre>
</div>
<div id="filter-the-1000g-data" class="section level4 hasAnchor" number="3.5.2.2">
<h4><span class="header-section-number">3.5.2.2</span> Filter the 1000G data<a href="sample-qc.html#filter-the-1000g-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First, we should get a list of relevant variants from our <code>rawdata</code>-dataset. We don’t need the other variants present in the 1000G dataset, right?</p>
<pre><code>cat rawdata/rawdata.bim | grep &quot;rs&quot; &gt; rawdata/all.variants.txt</code></pre>
<p>Extract those from the 1000G phase 1 data.</p>
<pre><code>plink --bfile ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_all --extract rawdata/all.variants.txt --make-bed --out ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_raw</code></pre>
</div>
<div id="filter-at-cg-snps" class="section level4 hasAnchor" number="3.5.2.3">
<h4><span class="header-section-number">3.5.2.3</span> Filter A/T &amp; C/G SNPs<a href="sample-qc.html#filter-at-cg-snps" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>As explained, the A/T and C/G SNPs are problematic, we want to exclude these too. So let’s get a list of A/T and C/G variants from 1000G to exclude.</p>
<pre><code>cat ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_raw.bim | \
awk &#39;($5 == &quot;A&quot; &amp;&amp; $6 == &quot;T&quot;) || ($5 == &quot;T&quot; &amp;&amp; $6 == &quot;A&quot;) || ($5 == &quot;C&quot; &amp;&amp; $6 == &quot;G&quot;) || ($5 == &quot;G&quot; &amp;&amp; $6 == &quot;C&quot;)&#39; | awk &#39;{ print $2, $1, $4, $3, $5, $6 }&#39; \
&gt; ref_1kg_phase1_all/1kg_phase1_all/all.1kg.atcg.variants.txt</code></pre>
<p>Exclude those A/T and C/G variants in both datasets.</p>
<pre><code>plink --bfile ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_raw --exclude ref_1kg_phase1_all/1kg_phase1_all/all.1kg.atcg.variants.txt --make-bed --out ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_raw_no_atcg

plink --bfile rawdata/rawdata --exclude ref_1kg_phase1_all/1kg_phase1_all/all.1kg.atcg.variants.txt --make-bed --out rawdata/rawdata_1kg_phase1_raw_no_atcg</code></pre>
</div>
<div id="merging-datasets-1" class="section level4 hasAnchor" number="3.5.2.4">
<h4><span class="header-section-number">3.5.2.4</span> Merging datasets<a href="sample-qc.html#merging-datasets-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Try and merge the data while extracting the pruned SNP-set.</p>
<pre><code>plink --bfile rawdata/rawdata_1kg_phase1_raw_no_atcg --bmerge ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_raw_no_atcg --extract rawdata/raw-GWA-data.prune.in --make-bed --out rawdata/rawdata.1kg_phase1.pruned</code></pre>
<p>There probably is an error …</p>
<pre><code>Error: 72 variants with 3+ alleles present.
* If you believe this is due to strand inconsistency, try --flip with
  rawdata/rawdata.1kg_phase1.pruned-merge.missnp.
  (Warning: if this seems to work, strand errors involving SNPs with A/T or C/G
  alleles probably remain in your data.  If LD between nearby SNPs is high,
  --flip-scan should detect them.)
* If you are dealing with genuine multiallelic variants, we recommend exporting
  that subset of the data to VCF (via e.g. &#39;--recode vcf&#39;), merging with
  another tool/script, and then importing the result; PLINK is not yet suited
  to handling them.
See https://www.cog-genomics.org/plink/1.9/data#merge3 for more discussion.</code></pre>
<p>So let’s flip some variants.</p>
<pre><code>plink --bfile rawdata/rawdata --exclude ref_1kg_phase1_all/1kg_phase1_all/all.1kg.atcg.variants.txt --flip rawdata/rawdata.1kg_phase1.pruned-merge.missnp --make-bed --out rawdata/rawdata_1kg_phase1_raw_no_atcg</code></pre>
<p>Let’s try and merge the data while extracting the pruned SNP-set.</p>
<pre><code>plink --bfile rawdata/rawdata_1kg_phase1_raw_no_atcg --bmerge ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_raw_no_atcg --extract rawdata/raw-GWA-data.prune.in --make-bed --out rawdata/rawdata.1kg_phase1.pruned</code></pre>
<p>There still is an error – there are multi-allelic variants present which PLINK can’t handle.</p>
<pre><code>Error: 14 variants with 3+ alleles present.
* If you believe this is due to strand inconsistency, try --flip with
  rawdata/rawdata.1kg_phase1.pruned-merge.missnp.
  (Warning: if this seems to work, strand errors involving SNPs with A/T or C/G
  alleles probably remain in your data.  If LD between nearby SNPs is high,
  --flip-scan should detect them.)
* If you are dealing with genuine multiallelic variants, we recommend exporting
  that subset of the data to VCF (via e.g. &#39;--recode vcf&#39;), merging with
  another tool/script, and then importing the result; PLINK is not yet suited
  to handling them.
See https://www.cog-genomics.org/plink/1.9/data#merge3 for more discussion.</code></pre>
<p>Let’s just remove these multi-allelic variants.</p>
<pre><code>plink --bfile rawdata/rawdata_1kg_phase1_raw_no_atcg --exclude rawdata/rawdata.1kg_phase1.pruned-merge.missnp --make-bed --out rawdata/rawdata_1kg_phase1_raw_no_atcg_bi</code></pre>
<p>Now we should be able to merge the data…</p>
<pre><code>plink --bfile rawdata/rawdata_1kg_phase1_raw_no_atcg_bi --bmerge ref_1kg_phase1_all/1kg_phase1_all/1kg_phase1_raw_no_atcg --extract rawdata/raw-GWA-data.prune.in --make-bed --out rawdata/rawdata.1kg_phase1.pruned</code></pre>
<p>Now we have prepared our dataset to include high-quality SNPs with few missing data, high-frequent SNPs, excluding problematic ranges, and merged to the 1000G phase 1 reference dataset.</p>
</div>
</div>
<div id="running-pca" class="section level3 hasAnchor" number="3.5.3">
<h3><span class="header-section-number">3.5.3</span> Running PCA<a href="sample-qc.html#running-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now you either prepared a dataset to project against HapMap 3 or 1000G phase 1. In the next two section I show you how to run <code>EIGENSOFT</code> to calculate principal components. It’s a bit funny (and confusing perhaps), but to run <code>EIGENSOFT</code> you execute a <code>perl</code>-script:</p>
<pre><code>perl ~/git/EIG/bin/smartpca.perl</code></pre>
<p>Well now, that isn’t named <code>EIGENSOFT</code> but rather <code>smartpca</code>. And that’s right: this <code>perl</code>-script is actually ‘calling’ a program named <code>smartpca</code>. It needs a couple of ‘flags’ whhich are explained below.</p>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Command</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>-i example.geno</code></td>
<td>genotype file in any format (see also ../CONVERTF/README)</td>
</tr>
<tr class="even">
<td><code>-a example.snp</code></td>
<td>snp file in any format (see also ../CONVERTF/README)</td>
</tr>
<tr class="odd">
<td><code>-b example.ind</code></td>
<td>indiv file in any format (see also ../CONVERTF/README)</td>
</tr>
<tr class="even">
<td><code>-k k</code></td>
<td>output file of principal components. Individuals removed</td>
</tr>
<tr class="odd">
<td><code>-o example.pca</code></td>
<td>(Default is 10) number of principal components to output as outliers will have all values set to 0.0 in this file.</td>
</tr>
<tr class="even">
<td><code>-p example.plot</code></td>
<td>prefix of output plot files of top 2 principal components (labeling individuals according to labels in indiv file).</td>
</tr>
<tr class="odd">
<td><code>-e example.eval</code></td>
<td>output file of all eigenvalues</td>
</tr>
<tr class="even">
<td><code>-l example.log</code></td>
<td>output logfile</td>
</tr>
<tr class="odd">
<td><code>-m maxiter</code></td>
<td>(Default is 5) maximum number of outlier removal iterations. To turn off outlier removal, set -m 0.</td>
</tr>
<tr class="even">
<td><code>-t topk</code></td>
<td>(Default is 10) number of principal components along which to remove outliers during each outlier removal</td>
</tr>
<tr class="odd">
<td><code>-s sigma</code></td>
<td>(Default is 6.0) number of standard deviations which an individual must exceed, along one of topk top principal components, in order to be removed as an outlier.</td>
</tr>
</tbody>
</table>
<p>OPTIONAL FLAGS</p>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Command</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>-w poplist</code></td>
<td>compute eigenvectors using populations in poplist only, where poplist is an ASCII file with one population per line</td>
</tr>
<tr class="even">
<td><code>-y plotlist</code></td>
<td>output plot will include populations in plotlist only, where plotlist is an ASCII file with one population per line</td>
</tr>
<tr class="odd">
<td><code>-z badsnpname</code></td>
<td>list of SNPs which should be excluded from the analysis</td>
</tr>
<tr class="even">
<td><code>-q YES/NO</code></td>
<td>If set to YES, assume that there is a single population and the population field contains real-valued phenotypes. (Corresponds to qtmode parameter in smartpca program.) The default value for this parameter is NO.</td>
</tr>
</tbody>
</table>
<p>As I mentioned before, you might not have been able to make <code>EIGENSOFT</code> work, no worries. You can skip right ahead to <a href="sample-qc.html#pca-plotting">PCA Plotting</a>.</p>
<div id="hapmap-3-vs-data" class="section level4 hasAnchor" number="3.5.3.1">
<h4><span class="header-section-number">3.5.3.1</span> HapMap 3 vs data<a href="sample-qc.html#hapmap-3-vs-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To run the PCA against HapMap 3, we first need to make a copy of the BIM and FAM-files.</p>
<pre><code>cp -v rawdata/rawdata.hapmap3r2.pruned.bim rawdata/rawdata.hapmap3r2.pruned.pedsnp
cp -v rawdata/rawdata.hapmap3r2.pruned.fam rawdata/rawdata.hapmap3r2.pruned.pedind</code></pre>
<p>Should you run this on your personal laptop, be aware it will take a few minutes to do so - perfect moment for a cup of coffee or to stretch your legs.</p>
<pre><code>perl ~/git/EIG/bin/smartpca.perl \
-i rawdata/rawdata.hapmap3r2.pruned.bed \
-a rawdata/rawdata.hapmap3r2.pruned.pedsnp \
-b rawdata/rawdata.hapmap3r2.pruned.pedind \
-k 10 \
-o rawdata/rawdata.hapmap3r2.pruned.pca \
-p rawdata/rawdata.hapmap3r2.pruned.plot \
-e rawdata/rawdata.hapmap3r2.pruned.eval \
-l rawdata/rawdata.hapmap3r2.pruned.log \
-m 5 \
-t 10 \
-s 6.0 \
-w ref_hapmap3/hapmap3r2_CEU.CHB.JPT.YRI-pca-populations.txt</code></pre>
</div>
<div id="g-phase-1-vs-data" class="section level4 hasAnchor" number="3.5.3.2">
<h4><span class="header-section-number">3.5.3.2</span> 1000G phase 1 vs data<a href="sample-qc.html#g-phase-1-vs-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To run the PCA against 1000G phase 1, we first need to make a copy of the BIM and FAM-files.</p>
<pre><code>cp -v rawdata/rawdata.1kg_phase1.pruned.bim rawdata/rawdata.1kg_phase1.pruned.pedsnp
cp -v rawdata/rawdata.1kg_phase1.pruned.fam rawdata/rawdata.1kg_phase1.pruned.pedind</code></pre>
<pre><code>perl ~/git/EIG/bin/smartpca.perl \
-i rawdata/rawdata.1kg_phase1.pruned.bed \
-a rawdata/rawdata.1kg_phase1.pruned.pedsnp \
-b rawdata/rawdata.1kg_phase1.pruned.pedind \
-k 10 \
-o rawdata/rawdata.1kg_phase1.pruned.pca \
-p rawdata/rawdata.1kg_phase1.pruned.plot \
-e rawdata/rawdata.1kg_phase1.pruned.eval \
-l rawdata/rawdata.1kg_phase1.pruned.log \
-m 5 \
-t 10 \
-s 6.0 \
-w ref_1kg_phase1_all/1kg_phase1_all/1kg-pca-populations.txt</code></pre>
</div>
</div>
<div id="pca-plotting" class="section level3 hasAnchor" number="3.5.4">
<h3><span class="header-section-number">3.5.4</span> PCA plotting<a href="sample-qc.html#pca-plotting" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If all is peachy, you ran PCA against either HapMap 3 or 1000G phase 1. Using <code>smartpca</code> (you know, <code>EIGENSOFT</code>) we have calculated principal components (PCs) and we can now start plotting them. Let’s create a scatter diagram of the first two principal components, including all individuals in the file <code>rawdata.hapmap3r2.pruned.pca.evec</code> (the first and second principal components are columns 2 and 3, respectively). Use the data in column 4 to color the points according to sample origin.</p>
<p>A R script for creating this plot (<code>scripts/plot-pca-results.Rscript</code>) is also provided (although any standard graphing software can be used), but below you’ll find some fancy codes too.</p>
<blockquote>
<p>Please note! You may have been able to make <code>EIGENSOFT</code> to work. So you may have to change “/ref_pca/rawdata.[ref].pruned.pca.evec” to “/rawdata/rawdata.[ref].pruned.pca.evec” in the command below.</p>
</blockquote>
<div id="hapmap-3-vs-data-1" class="section level4 hasAnchor" number="3.5.4.1">
<h4><span class="header-section-number">3.5.4.1</span> HapMap 3 vs data<a href="sample-qc.html#hapmap-3-vs-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We could plot our data against HapMap 3.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="sample-qc.html#cb61-1" aria-hidden="true" tabindex="-1"></a>PCA <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc,<span class="st">&quot;/ref_pca/rawdata.hapmap3r2.pruned.pca.evec&quot;</span>), <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="sample-qc.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Case/Control -&gt; black, pch = &quot;+&quot;</span></span>
<span id="cb62-2"><a href="sample-qc.html#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="co"># CEU = 3 -&gt; red, pch = 20</span></span>
<span id="cb62-3"><a href="sample-qc.html#cb62-3" aria-hidden="true" tabindex="-1"></a><span class="co"># CHB = 4 -&gt; pink, pch = 20</span></span>
<span id="cb62-4"><a href="sample-qc.html#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co"># JPT = 5 -&gt; purple, pch = 20</span></span>
<span id="cb62-5"><a href="sample-qc.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="co"># YRI = 6 -&gt; green, pch = 20</span></span>
<span id="cb62-6"><a href="sample-qc.html#cb62-6" aria-hidden="true" tabindex="-1"></a>PCA<span class="sc">$</span>V12[PCA<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;Case&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Case&quot;</span> <span class="co">#595A5C</span></span>
<span id="cb62-7"><a href="sample-qc.html#cb62-7" aria-hidden="true" tabindex="-1"></a>PCA<span class="sc">$</span>V12[PCA<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;Control&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Control&quot;</span> <span class="co">#595A5C</span></span>
<span id="cb62-8"><a href="sample-qc.html#cb62-8" aria-hidden="true" tabindex="-1"></a>PCA<span class="sc">$</span>V12[PCA<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;3&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CEU&quot;</span> <span class="co">#E55738</span></span>
<span id="cb62-9"><a href="sample-qc.html#cb62-9" aria-hidden="true" tabindex="-1"></a>PCA<span class="sc">$</span>V12[PCA<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;4&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CHB&quot;</span> <span class="co">#D5267B</span></span>
<span id="cb62-10"><a href="sample-qc.html#cb62-10" aria-hidden="true" tabindex="-1"></a>PCA<span class="sc">$</span>V12[PCA<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;5&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;JPT&quot;</span> <span class="co">#9A3480</span></span>
<span id="cb62-11"><a href="sample-qc.html#cb62-11" aria-hidden="true" tabindex="-1"></a>PCA<span class="sc">$</span>V12[PCA<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;6&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;YRI&quot;</span> <span class="co">#49A01D</span></span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="sample-qc.html#cb63-1" aria-hidden="true" tabindex="-1"></a>PCAplot <span class="ot">&lt;-</span> ggpubr<span class="sc">::</span><span class="fu">ggscatter</span>(PCA, <span class="at">x =</span> <span class="st">&quot;V2&quot;</span>, <span class="at">y =</span> <span class="st">&quot;V3&quot;</span>,</span>
<span id="cb63-2"><a href="sample-qc.html#cb63-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">color =</span> <span class="st">&quot;V12&quot;</span>,</span>
<span id="cb63-3"><a href="sample-qc.html#cb63-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#595A5C&quot;</span>, <span class="st">&quot;#595A5C&quot;</span>, <span class="st">&quot;#E55738&quot;</span>, <span class="st">&quot;#D5267B&quot;</span>, <span class="st">&quot;#9A3480&quot;</span>, <span class="st">&quot;#49A01D&quot;</span>),</span>
<span id="cb63-4"><a href="sample-qc.html#cb63-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">shape =</span> <span class="st">&quot;V12&quot;</span>,</span>
<span id="cb63-5"><a href="sample-qc.html#cb63-5" aria-hidden="true" tabindex="-1"></a>                             <span class="at">xlab =</span> <span class="st">&quot;principal component 1&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;principal component 2&quot;</span>) <span class="sc">+</span></span>
<span id="cb63-6"><a href="sample-qc.html#cb63-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.075</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb63-7"><a href="sample-qc.html#cb63-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#A2A3A4&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb63-8"><a href="sample-qc.html#cb63-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-9"><a href="sample-qc.html#cb63-9" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">ggpar</span>(PCAplot,</span>
<span id="cb63-10"><a href="sample-qc.html#cb63-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;Principal Component Analysis&quot;</span>,</span>
<span id="cb63-11"><a href="sample-qc.html#cb63-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="st">&quot;Reference population: HapMap 3&quot;</span>,</span>
<span id="cb63-12"><a href="sample-qc.html#cb63-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.title =</span> <span class="st">&quot;Populations&quot;</span>, <span class="at">legend =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:pca-hapmap3"></span>
<img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/pca-hapmap3-1.png" alt="PCA - Your data vs. HapMap 3" width="80%" />
<p class="caption">
Figure 3.3: PCA - Your data vs. HapMap 3
</p>
</div>
<p>Derive PC1 and PC2 thresholds so that only individuals who match the given ancestral population are included. For populations of European descent, this will be either the CEU or TSI HapMap3 individuals (Figure <a href="sample-qc.html#fig:pca-hapmap3">3.3</a>). Here, we chose to exclude all individuals with a second principal component score less than 0.075.</p>
<p>Write the FID and IID of these individuals to a file called <code>fail-ancestry-QC.txt</code>.</p>
<pre><code>cat rawdata/rawdata.hapmap3r2.pruned.pca.evec | tail -n +2 | \
awk &#39;$3 &lt; 0.075&#39; | awk &#39;{ print $1 }&#39; | awk -F&quot;:&quot; &#39;{ print $1, $2 }&#39; &gt; rawdata/fail-ancestry-QC.txt</code></pre>
<p>Choosing which thresholds to apply (and thus which individuals to remove) is not a straightforward process. The key is to remove those individuals with greatly divergent ancestry, as these samples introduce the most bias to the study. Identification of more fine-scale ancestry can be conducted by using less divergent reference samples (<em>e.g.</em>, within Europe, stratification could be identified using the CEU, TSI (Italian), GBR (British), FIN (Finnish) and IBS (Iberian) samples from the 1,000 Genomes Project (<a href="http://www.1000genomes.org/" class="uri">http://www.1000genomes.org/</a>)). Robust identification of fine-scale population structure often requires the construction of many (2–10) principal components.</p>
</div>
<div id="g-phase-1-vs-data-1" class="section level4 hasAnchor" number="3.5.4.2">
<h4><span class="header-section-number">3.5.4.2</span> 1000G phase 1 vs data<a href="sample-qc.html#g-phase-1-vs-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We could plot our data against 1000G phase 1.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="sample-qc.html#cb65-1" aria-hidden="true" tabindex="-1"></a>PCA_1kG <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc,<span class="st">&quot;/ref_pca/rawdata.1kg_phase1.pruned.pca.evec&quot;</span>), <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="sample-qc.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Population    Description Super population    Code    Counts</span></span>
<span id="cb66-2"><a href="sample-qc.html#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ASW   African Ancestry in Southwest US                              AFR   4     #49A01D</span></span>
<span id="cb66-3"><a href="sample-qc.html#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="co"># CEU   Utah residents with Northern and Western European ancestry  EUR 7     #E55738</span></span>
<span id="cb66-4"><a href="sample-qc.html#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="co"># CHB   Han Chinese in Bejing, China                                  EAS   8     #9A3480</span></span>
<span id="cb66-5"><a href="sample-qc.html#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="co"># CHS   Southern Han Chinese, China                                 EAS 9     #705296</span></span>
<span id="cb66-6"><a href="sample-qc.html#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="co"># CLM   Colombian in Medellin, Colombia                             MR  10  #8D5B9A</span></span>
<span id="cb66-7"><a href="sample-qc.html#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="co"># FIN   Finnish in Finland                                          EUR 12  #2F8BC9</span></span>
<span id="cb66-8"><a href="sample-qc.html#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="co"># GBR   British in England and Scotland                             EUR 13  #1290D9</span></span>
<span id="cb66-9"><a href="sample-qc.html#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="co"># IBS   Iberian populations in Spain                                  EUR   16  #1396D8</span></span>
<span id="cb66-10"><a href="sample-qc.html#cb66-10" aria-hidden="true" tabindex="-1"></a><span class="co"># JPT   Japanese in Tokyo, Japan                                      EAS   18  #D5267B</span></span>
<span id="cb66-11"><a href="sample-qc.html#cb66-11" aria-hidden="true" tabindex="-1"></a><span class="co"># LWK   Luhya in Webuye, Kenya                                      AFR 20  #78B113</span></span>
<span id="cb66-12"><a href="sample-qc.html#cb66-12" aria-hidden="true" tabindex="-1"></a><span class="co"># MXL   Mexican Ancestry in Los Angeles, California                 AMR 22  #F59D10</span></span>
<span id="cb66-13"><a href="sample-qc.html#cb66-13" aria-hidden="true" tabindex="-1"></a><span class="co"># PUR   Puerto Rican in Puerto Rico                                 AMR 25  #FBB820</span></span>
<span id="cb66-14"><a href="sample-qc.html#cb66-14" aria-hidden="true" tabindex="-1"></a><span class="co"># TSI   Toscani in Italy                                              EUR   27  #4C81BF</span></span>
<span id="cb66-15"><a href="sample-qc.html#cb66-15" aria-hidden="true" tabindex="-1"></a><span class="co"># YRI   Yoruba in Ibadan, Nigeria                                     AFR   28  #C5D220</span></span>
<span id="cb66-16"><a href="sample-qc.html#cb66-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-17"><a href="sample-qc.html#cb66-17" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;Case&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Case&quot;</span></span>
<span id="cb66-18"><a href="sample-qc.html#cb66-18" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;Control&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Control&quot;</span></span>
<span id="cb66-19"><a href="sample-qc.html#cb66-19" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;4&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;ASW&quot;</span></span>
<span id="cb66-20"><a href="sample-qc.html#cb66-20" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;7&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CEU&quot;</span></span>
<span id="cb66-21"><a href="sample-qc.html#cb66-21" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;8&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CHB&quot;</span></span>
<span id="cb66-22"><a href="sample-qc.html#cb66-22" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;9&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CHS&quot;</span></span>
<span id="cb66-23"><a href="sample-qc.html#cb66-23" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;10&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CLM&quot;</span></span>
<span id="cb66-24"><a href="sample-qc.html#cb66-24" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;12&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;FIN&quot;</span></span>
<span id="cb66-25"><a href="sample-qc.html#cb66-25" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;13&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;GBR&quot;</span></span>
<span id="cb66-26"><a href="sample-qc.html#cb66-26" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;16&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;IBS&quot;</span></span>
<span id="cb66-27"><a href="sample-qc.html#cb66-27" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;18&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;JPT&quot;</span></span>
<span id="cb66-28"><a href="sample-qc.html#cb66-28" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;20&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;LWK&quot;</span></span>
<span id="cb66-29"><a href="sample-qc.html#cb66-29" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;22&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;MXL&quot;</span></span>
<span id="cb66-30"><a href="sample-qc.html#cb66-30" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;25&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;PUR&quot;</span></span>
<span id="cb66-31"><a href="sample-qc.html#cb66-31" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;27&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;TSI&quot;</span></span>
<span id="cb66-32"><a href="sample-qc.html#cb66-32" aria-hidden="true" tabindex="-1"></a>PCA_1kG<span class="sc">$</span>V12[PCA_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;28&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;YRI&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="sample-qc.html#cb67-1" aria-hidden="true" tabindex="-1"></a>PCA_1kGplot <span class="ot">&lt;-</span> ggpubr<span class="sc">::</span><span class="fu">ggscatter</span>(PCA_1kG, <span class="at">x =</span> <span class="st">&quot;V2&quot;</span>, <span class="at">y =</span> <span class="st">&quot;V3&quot;</span>,</span>
<span id="cb67-2"><a href="sample-qc.html#cb67-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">color =</span> <span class="st">&quot;V12&quot;</span>,</span>
<span id="cb67-3"><a href="sample-qc.html#cb67-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#595A5C&quot;</span>, <span class="st">&quot;#49A01D&quot;</span>, <span class="st">&quot;#E55738&quot;</span>, <span class="st">&quot;#9A3480&quot;</span>, <span class="st">&quot;#705296&quot;</span>, </span>
<span id="cb67-4"><a href="sample-qc.html#cb67-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">&quot;#595A5C&quot;</span>, <span class="st">&quot;#8D5B9A&quot;</span>, <span class="st">&quot;#2F8BC9&quot;</span>, <span class="st">&quot;#1290D9&quot;</span>, <span class="st">&quot;#1396D8&quot;</span>, </span>
<span id="cb67-5"><a href="sample-qc.html#cb67-5" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">&quot;#D5267B&quot;</span>, <span class="st">&quot;#78B113&quot;</span>, <span class="st">&quot;#F59D10&quot;</span>, <span class="st">&quot;#FBB820&quot;</span>, <span class="st">&quot;#4C81BF&quot;</span>, <span class="st">&quot;#C5D220&quot;</span>),</span>
<span id="cb67-6"><a href="sample-qc.html#cb67-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">xlab =</span> <span class="st">&quot;principal component 1&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;principal component 2&quot;</span>) <span class="sc">+</span></span>
<span id="cb67-7"><a href="sample-qc.html#cb67-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.023</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb67-8"><a href="sample-qc.html#cb67-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb67-9"><a href="sample-qc.html#cb67-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-10"><a href="sample-qc.html#cb67-10" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">ggpar</span>(PCA_1kGplot,</span>
<span id="cb67-11"><a href="sample-qc.html#cb67-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;Principal Component Analysis&quot;</span>,</span>
<span id="cb67-12"><a href="sample-qc.html#cb67-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="st">&quot;Reference population: 1000 G, phase 1&quot;</span>,</span>
<span id="cb67-13"><a href="sample-qc.html#cb67-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.title =</span> <span class="st">&quot;Populations&quot;</span>, <span class="at">legend =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:pca-1000g"></span>
<img src="A-primer-in-Human-Cardiovascular-Genetics_files/figure-html/pca-1000g-1.png" alt="PCA - Your data vs. 1000G" width="80%" />
<p class="caption">
Figure 3.4: PCA - Your data vs. 1000G
</p>
</div>
<p>In a similar fashion as in the above with the HapMap3 reference, you could <strong>remove the samples below the threshold</strong> based on this PCA (Figure <a href="sample-qc.html#fig:pca-1000g">3.4</a>).</p>
</div>
</div>
</div>
<div id="removing-samples" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Removing samples<a href="sample-qc.html#removing-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally! We have a list of samples of poor quality or divergent ancestry, and duplicated or related samples. We should remove these. Let’s collect all IDs from our <code>fail-*</code>-files into a single file.</p>
<pre><code>cat rawdata/fail-* | sort -k1 | uniq &gt; rawdata/fail-qc-inds.txt</code></pre>
<p>This new file should now contain a list of unique individuals failing the previous QC steps which we want to remove.</p>
<pre><code>plink --bfile rawdata/rawdata --remove rawdata/fail-qc-inds.txt --make-bed --out rawdata/clean_inds_data</code></pre>
</div>
<div id="the-next-step" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> The next step<a href="sample-qc.html#the-next-step" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that you filtered samples, we should turn our attention to step 2 of the QC for GWAS: identify SNPs of poor quality in Chapter @ref(gwas_basics_snp_qc).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="prerequisites.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="summary.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_primer_in_Human_Cardiovascular_Genetics/edit/master/gwas_basics_sample_qc.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["A primer in Human Cardiovascular Genetics.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
