<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Steps in a Genome-Wide Association Study {#gwas-basics} | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="Steps in a Genome-Wide Association Study {#gwas-basics} | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Steps in a Genome-Wide Association Study {#gwas-basics} | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="getting-started.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#your-computer"><i class="fa fa-check"></i><b>3.1</b> Your computer</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#cocalc-vs.-standalone"><i class="fa fa-check"></i><b>3.2</b> CoCalc vs. Standalone</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#the-programs-we-use"><i class="fa fa-check"></i><b>3.3</b> The programs we use</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="getting-started.html"><a href="getting-started.html#rstudio"><i class="fa fa-check"></i><b>3.3.1</b> RStudio</a></li>
<li class="chapter" data-level="3.3.2" data-path="getting-started.html"><a href="getting-started.html#plink"><i class="fa fa-check"></i><b>3.3.2</b> PLINK</a></li>
<li class="chapter" data-level="3.3.3" data-path="getting-started.html"><a href="getting-started.html#alternatives-to-plink"><i class="fa fa-check"></i><b>3.3.3</b> Alternatives to <code>PLINK</code></a></li>
<li class="chapter" data-level="3.3.4" data-path="getting-started.html"><a href="getting-started.html#other-programs"><i class="fa fa-check"></i><b>3.3.4</b> Other programs</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="getting-started.html"><a href="getting-started.html#cocalc"><i class="fa fa-check"></i><b>3.4</b> CoCalc</a></li>
<li class="chapter" data-level="3.5" data-path="getting-started.html"><a href="getting-started.html#standalone"><i class="fa fa-check"></i><b>3.5</b> Standalone</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="getting-started.html"><a href="getting-started.html#the-data-you-need"><i class="fa fa-check"></i><b>3.5.1</b> The data you need</a></li>
<li class="chapter" data-level="3.5.2" data-path="getting-started.html"><a href="getting-started.html#terminal"><i class="fa fa-check"></i><b>3.5.2</b> Terminal</a></li>
<li class="chapter" data-level="3.5.3" data-path="getting-started.html"><a href="getting-started.html#navigating-the-terminal"><i class="fa fa-check"></i><b>3.5.3</b> Navigating the Terminal</a></li>
<li class="chapter" data-level="3.5.4" data-path="getting-started.html"><a href="getting-started.html#installing-the-software"><i class="fa fa-check"></i><b>3.5.4</b> Installing the software</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="getting-started.html"><a href="getting-started.html#are-you-ready"><i class="fa fa-check"></i><b>3.6</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="steps-in-a-genome-wide-association-study-gwas-basics.html"><a href="steps-in-a-genome-wide-association-study-gwas-basics.html"><i class="fa fa-check"></i>Steps in a Genome-Wide Association Study {#gwas-basics}</a>
<ul>
<li class="chapter" data-level="3.7" data-path="steps-in-a-genome-wide-association-study-gwas-basics.html"><a href="steps-in-a-genome-wide-association-study-gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>3.7</b> Converting datasets</a></li>
<li class="chapter" data-level="3.8" data-path="steps-in-a-genome-wide-association-study-gwas-basics.html"><a href="steps-in-a-genome-wide-association-study-gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>3.8</b> Quality control</a></li>
<li class="chapter" data-level="3.9" data-path="steps-in-a-genome-wide-association-study-gwas-basics.html"><a href="steps-in-a-genome-wide-association-study-gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.9</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="steps-in-a-genome-wide-association-study-gwas-basics" class="section level1 unnumbered hasAnchor">
<h1>Steps in a Genome-Wide Association Study {#gwas-basics}<a href="steps-in-a-genome-wide-association-study-gwas-basics.html#steps-in-a-genome-wide-association-study-gwas-basics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- ![](./img/_headers/woman_working_on_code.png){width=100%} -->
<p>Now that you understand a bit of the navigation in Unix-systems, you’re ready to start this practical primer. We will make use of a dummy dataset containing cases and controls. We will explain and execute the following steps:</p>
<ol style="list-style-type: decimal">
<li>convert raw data to a more memory-efficient format</li>
<li>apply extensive quality control on samples and SNPs</li>
<li>assess the ancestral background of your study population</li>
<li>perform association testing</li>
<li>visualize association results</li>
</ol>
<div id="converting-datasets" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Converting datasets<a href="steps-in-a-genome-wide-association-study-gwas-basics.html#converting-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The format in which genotype data are returned to investigators varies among genome-wide SNP platforms and genotyping centers. Usually genotypes have been called by a genotyping center and returned in the standard <code>PED</code> and <code>MAP</code> file formats designed for <code>PLINK</code>.</p>
<p>A <code>PED</code> file is a white space (space or tab)-delimited file in which each line represents one individual and the first six columns are mandatory and in the following order:</p>
<ul>
<li>‘Family ID’,</li>
<li>‘Individual ID’,</li>
<li>‘Paternal ID’,</li>
<li>‘Maternal ID’,</li>
<li>‘Sex (1=male, 2=female, 0=missing)’, and</li>
<li>‘Phenotype (1=unaffected, 2=affected, 0=missing)’.</li>
</ul>
<p>The subsequent columns denote genotypes that can be any character (e.g., 1, 2, 3, 4 or A, C, G, T). Zero denotes a missing genotype. Each SNP must have two alleles (i.e., both alleles are either present or absent).
The order of SNPs in the PED file is given in the MAP file, in which each line denotes a single marker and the four white-space–separated columns are chromosome (1–22, X, Y or 0 for unplaced), marker name (typically an rs number), genetic distance in Morgans (this can be fixed to 0) and base-pair position (bp units).</p>
<p>Let’s start by using <code>PLINK</code> to converting the datasets to a lighter, binary form (a <code>.bed</code>-file). This file saves data in a more memory- and time-efficient manner (in a ‘binary’-format) to facilitate the analysis of large-scale data sets <span class="citation">(<a href="#ref-purcell2007">Purcell S. 2007</a>)</span>. The marker-information is stored in the <code>.bim</code>-file and the family information in the <code>.fam</code>-file. <code>PLINK</code> creates a <code>.log</code> file (named <code>raw-GWA-data.log</code>) that details (among other information) the implemented commands, the number of cases and controls in the input files, any excluded data and the genotyping rate in the remaining data. This file is very useful for checking whether the software is successfully completing commands.</p>
<p>Make sure you are in the right directory. Do you remember how to get there?</p>
<pre><code>cd ~/Desktop/practical</code></pre>
<p>Next, we’ll make a project directory.</p>
<pre><code>mkdir -v ~/Desktop/practical/dummy_project</code></pre>
<p>Now, we’ll convert the <code>.ped</code>/<code>.map</code> files to the binary-format.</p>
<pre><code>plink --file rawdata/raw-GWA-data --make-bed --out dummy_project/rawdata</code></pre>
<p>Let’s review the <code>.log</code>-file for a bit. It should look something like this:</p>
<pre><code>PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to dummy_project/rawdata.log.
Options in effect:
  --file rawdata/raw-GWA-data
  --make-bed
  --out dummy_project/rawdata

16384 MB RAM detected; reserving 8192 MB for main workspace.
.ped scan complete (for binary autoconversion).
Performing single-pass .bed write (317503 variants, 2000 people).
--file: dummy_project/rawdata-temporary.bed +
dummy_project/rawdata-temporary.bim + dummy_project/rawdata-temporary.fam
written.
317503 variants loaded from .bim file.
2000 people (997 males, 1003 females) loaded from .fam.
2000 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 2000 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Warning: 11440 het. haploid genotypes present (see dummy_project/rawdata.hh );
many commands treat these as missing.
Total genotyping rate is 0.985682.
317503 variants and 2000 people pass filters and QC.
Among remaining phenotypes, 1023 are cases and 977 are controls.
--make-bed to dummy_project/rawdata.bed + dummy_project/rawdata.bim +
dummy_project/rawdata.fam ... done.</code></pre>
<p>So, there are 317,503 variants included for 2,000 people, 997 males and 1,003 females. All of these individuals are <a href="https://www.cog-genomics.org/plink/1.9/filter#nonfounders" target="_blank">‘founders’</a>. There are 1,023 cases and 977 controls. The genotyping rate is about 98.6% which is pretty good. Lastly, there are 11,440 heterozygous haploid genotypes present.</p>
<blockquote>
<p>Question: Can you think off what the ‘11,440 heterozygous haploid genotypes present’ represent?</p>
</blockquote>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Quality control<a href="steps-in-a-genome-wide-association-study-gwas-basics.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are ready for some quality control and quality assurance, heavily inspired by Anderson <em>et al.</em> <span class="citation">(<a href="#ref-anderson2010">Anderson C. A. 2010</a>)</span> and Laurie <em>et al.</em> <span class="citation">(<a href="#ref-laurie2010">Laurie C. C. 2010</a>)</span>. In general, we should check out a couple of things regarding the data quality on two levels:</p>
<ol style="list-style-type: decimal">
<li>samples</li>
<li>variants</li>
</ol>
<p>So, we will investigate the following:</p>
<ul>
<li>Are the <em>sexes</em> based on genetic data matching the ones given by the phenotype file?</li>
<li>Identify individuals that are outliers in terms of missing data (<em>call rate</em>) or heterozygosity rates. This could indicate a genotyping error or sample swap.</li>
<li>Identify duplicated or related individuals.</li>
<li>Identify individuals with divergent ancestry.</li>
<li>What are the allele frequencies?</li>
<li>What is the per-SNP call rate?</li>
<li>In the case of a case-control study (which is the case here), we need to check differential missingness between cases and controls.</li>
</ul>
<blockquote>
<p>Question: Can you think of other scenarios in which you may want to extend the check on differential missingness beyond a check between cases and controls?</p>
</blockquote>
</div>
<div id="lets-get-our-hands-dirty" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> Let’s get our hands dirty<a href="steps-in-a-genome-wide-association-study-gwas-basics.html#lets-get-our-hands-dirty" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>All clear? Let’s start the work. On to step 1 of the QC for GWAS: filter samples of poor quality in Chapter <a href="#gwas-basics-sample-qc"><strong>??</strong></a>.</p>
<!-- ```{js, echo = FALSE} -->
<!-- title=document.getElementById('header'); -->
<!-- title.innerHTML = '<img src="img/_headers/woman_working_on_code.png" alt="GWAS basics">' + title.innerHTML -->
<!-- ``` -->

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-anderson2010" class="csl-entry">
Anderson C. A., et al. 2010. <span>“Data QC in Genetic Case-Control Association Studies.”</span> <a href="https://www.ncbi.nlm.nih.gov/pubmed/21085122">https://www.ncbi.nlm.nih.gov/pubmed/21085122</a>.
</div>
<div id="ref-laurie2010" class="csl-entry">
Laurie C. C., et al. 2010. <span>“Quality Control and Quality Assurance in Genotypic Data for Genome-Wide Association Studies.”</span> <a href="https://www.ncbi.nlm.nih.gov/pubmed/20718045">https://www.ncbi.nlm.nih.gov/pubmed/20718045</a>.
</div>
<div id="ref-purcell2007" class="csl-entry">
Purcell S., et al. 2007. <span>“PLINK, a Tool Set for Whole-Genome Association and Population-Based Linkage Analyses.”</span> <a href="https://www.ncbi.nlm.nih.gov/pubmed/17701901">https://www.ncbi.nlm.nih.gov/pubmed/17701901</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="getting-started.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics03_1_gwas_basics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"info": true
});
});
</script>

</body>

</html>
