<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 WTCCC1: a GWAS on coronary artery disease (CAD) | A Practical Primer in Human Complex Genetics</title>
  <meta name="description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 WTCCC1: a GWAS on coronary artery disease (CAD) | A Practical Primer in Human Complex Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 WTCCC1: a GWAS on coronary artery disease (CAD) | A Practical Primer in Human Complex Genetics" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics with a use-case in cardiovascular disease. The output format for this primer is bookdown::gitbook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="wtccc1-intro.html"/>
<link rel="next" href="post-gwas.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#background-reading"><i class="fa fa-check"></i><b>1.2</b> Background reading<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#meet-the-team"><i class="fa fa-check"></i><b>1.3</b> Meet the Team<span></span></a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#final-thoughts"><i class="fa fa-check"></i><b>1.4</b> Final thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#linux-macos-and-windows"><i class="fa fa-check"></i><b>2.1</b> Linux, macOS, and Windows<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#programs-you-need"><i class="fa fa-check"></i><b>2.2</b> Programs you need<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="prerequisites.html"><a href="prerequisites.html#the-terminal"><i class="fa fa-check"></i><b>2.3</b> The Terminal<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="prerequisites.html"><a href="prerequisites.html#download-the-data"><i class="fa fa-check"></i><b>2.3.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="prerequisites.html"><a href="prerequisites.html#navigating-the-terminal"><i class="fa fa-check"></i><b>2.3.2</b> Navigating the Terminal<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="prerequisites.html"><a href="prerequisites.html#installing-some-r-packages"><i class="fa fa-check"></i><b>2.4</b> Installing some R packages<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="prerequisites.html"><a href="prerequisites.html#are-you-ready"><i class="fa fa-check"></i><b>2.5</b> Are you ready?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-basics.html"><a href="gwas-basics.html"><i class="fa fa-check"></i><b>3</b> Steps in a Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-basics.html"><a href="gwas-basics.html#converting-datasets"><i class="fa fa-check"></i><b>3.1</b> Converting datasets<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="gwas-basics.html"><a href="gwas-basics.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality control<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="gwas-basics.html"><a href="gwas-basics.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>3.3</b> Let’s get our hands dirty<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html"><i class="fa fa-check"></i><b>4</b> Sample QC<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sex"><i class="fa fa-check"></i><b>4.1</b> Sex<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#sample-call-rates"><i class="fa fa-check"></i><b>4.2</b> Sample call rates<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>4.3</b> Heterozygosity rate<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#relatedness"><i class="fa fa-check"></i><b>4.4</b> Relatedness<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#ancestral-background"><i class="fa fa-check"></i><b>4.5</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#hapmap-3"><i class="fa fa-check"></i><b>4.5.1</b> HapMap 3<span></span></a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#g-phase-1"><i class="fa fa-check"></i><b>4.5.2</b> 1000G phase 1<span></span></a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#running-pca"><i class="fa fa-check"></i><b>4.5.3</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="4.5.4" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#pca-plotting"><i class="fa fa-check"></i><b>4.5.4</b> PCA plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#removing-samples"><i class="fa fa-check"></i><b>4.6</b> Removing samples<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="gwas-basics-sample-qc.html"><a href="gwas-basics-sample-qc.html#the-next-step"><i class="fa fa-check"></i><b>4.7</b> The next step<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html"><i class="fa fa-check"></i><b>5</b> Per-SNP QC<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#snp-call-rates"><i class="fa fa-check"></i><b>5.1</b> SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>5.2</b> Differential SNP call rates<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#allele-frequencies"><i class="fa fa-check"></i><b>5.3</b> Allele frequencies<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>5.3.1</b> A note on allele coding<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>5.4</b> Hardy-Weinberg Equilibrium<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#final-snp-qc"><i class="fa fa-check"></i><b>5.5</b> Final SNP QC<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="gwas_basics_snp_qc.html"><a href="gwas_basics_snp_qc.html#a-milestone"><i class="fa fa-check"></i><b>5.6</b> A Milestone<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-testing.html"><a href="gwas-testing.html"><i class="fa fa-check"></i><b>6</b> Genome-Wide Association Study<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-testing.html"><a href="gwas-testing.html#exploring-the-data"><i class="fa fa-check"></i><b>6.1</b> Exploring the data<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="gwas-testing.html"><a href="gwas-testing.html#genetic-models"><i class="fa fa-check"></i><b>6.2</b> Genetic models<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="gwas-testing.html"><a href="gwas-testing.html#logistic-regression"><i class="fa fa-check"></i><b>6.3</b> Logistic regression<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="gwas-testing.html"><a href="gwas-testing.html#lets-get-visual"><i class="fa fa-check"></i><b>6.4</b> Let’s get visual<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-visuals.html"><a href="gwas-visuals.html"><i class="fa fa-check"></i><b>7</b> GWAS visualisation<span></span></a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#qq-plots"><i class="fa fa-check"></i><b>7.0.1</b> QQ plots<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="gwas-visuals.html"><a href="gwas-visuals.html#manhattan-plots"><i class="fa fa-check"></i><b>7.1</b> Manhattan plots<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="gwas-visuals.html"><a href="gwas-visuals.html#other-plots"><i class="fa fa-check"></i><b>7.2</b> Other plots<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="gwas-visuals.html"><a href="gwas-visuals.html#interactive-plots"><i class="fa fa-check"></i><b>7.3</b> Interactive plots<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="gwas-visuals.html"><a href="gwas-visuals.html#stop-playing-around"><i class="fa fa-check"></i><b>7.4</b> Stop playing around<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html"><i class="fa fa-check"></i><b>8</b> The Welcome Trust Case-Control Consortium<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#genotyping"><i class="fa fa-check"></i><b>8.1</b> Genotyping<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#the-data"><i class="fa fa-check"></i><b>8.2</b> The data<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#assignment"><i class="fa fa-check"></i><b>8.3</b> Assignment<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="wtccc1-intro.html"><a href="wtccc1-intro.html#there-you-go"><i class="fa fa-check"></i><b>8.4</b> There you go<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>9</b> WTCCC1: a GWAS on coronary artery disease (CAD)<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>9.1</b> Quality control<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>9.2</b> Ancestral background<span></span></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>9.2.1</b> Preparing PCA<span></span></a></li>
<li class="chapter" data-level="9.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca-1"><i class="fa fa-check"></i><b>9.2.2</b> Running PCA<span></span></a></li>
<li class="chapter" data-level="9.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>9.2.3</b> Plotting PCA<span></span></a></li>
<li class="chapter" data-level="9.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>9.2.4</b> Removing samples<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>9.3</b> Association testing<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>9.4</b> Replication!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="post-gwas.html"><a href="post-gwas.html"><i class="fa fa-check"></i><b>10</b> Post-GWAS Analyses<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="post-gwas.html"><a href="post-gwas.html#clumping"><i class="fa fa-check"></i><b>10.1</b> Clumping<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="post-gwas.html"><a href="post-gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>10.2</b> Conditional analyses<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="post-gwas.html"><a href="post-gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>10.3</b> Statistical finemapping<span></span></a></li>
<li class="chapter" data-level="10.4" data-path="post-gwas.html"><a href="post-gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>10.4</b> FUMA: FUnctional Mapping and Annotation of GWAS<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>10.5</b> PheWAS<span></span></a></li>
<li class="chapter" data-level="10.6" data-path="post-gwas.html"><a href="post-gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>10.6</b> Genetic correlation<span></span></a></li>
<li class="chapter" data-level="10.7" data-path="post-gwas.html"><a href="post-gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>10.7</b> Causal inference: Mendelian Randomization<span></span></a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="post-gwas.html"><a href="post-gwas.html#your-choice"><i class="fa fa-check"></i><b>10.7.1</b> Your choice<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="post-gwas.html"><a href="post-gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>10.8</b> Polygenic scores<span></span></a></li>
<li class="chapter" data-level="10.9" data-path="post-gwas.html"><a href="post-gwas.html#what-comes-next"><i class="fa fa-check"></i><b>10.9</b> What comes next<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>11</b> Functional Mapping and Annotation of GWAS<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>11.1</b> Assignment<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>11.1.1</b> Tutorial<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>11.1.2</b> Create the input<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>11.1.3</b> Run FUMA - SNP to gene<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>11.1.4</b> Run FUMA - Gene to function<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>11.2</b> Some closing thoughts<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="post-gwas.html"><a href="post-gwas.html#phewas"><i class="fa fa-check"></i><b>12</b> Phenome-Wide Association Study (PheWAS)<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>12.1</b> What’s next?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>13</b> MRBase<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>13.1</b> The end?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>14</b> Epilogue<span></span></a></li>
<li class="chapter" data-level="15" data-path="add_chapter_regional_plot.html"><a href="add_chapter_regional_plot.html"><i class="fa fa-check"></i><b>15</b> Additional: regional association plots<span></span></a></li>
<li class="chapter" data-level="16" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>16</b> Additional: EIGENSOFT<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>16.1</b> Install homebrew<span></span></a></li>
<li class="chapter" data-level="16.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>16.2</b> Install missing packages<span></span></a></li>
<li class="chapter" data-level="16.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>16.3</b> Installing EIGENSOFT<span></span></a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>17</b> Licenses and disclaimers<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>17.1</b> Copyright<span></span></a></li>
<li class="chapter" data-level="17.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>17.2</b> Disclaimer<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="wtccc1" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> WTCCC1: a GWAS on coronary artery disease (CAD)<a href="wtccc1.html#wtccc1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>As usual, we start by exploring the data in hand.</p>
<pre><code>plink --bfile wtccc1/CADn1871_500Kb37fwd --bmerge wtccc1/UKBSn1397_500Kb37fwd --make-bed --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --freq --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --hardy --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --missing --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --test-missing --out wtccc1/wtccc1

cat wtccc1/wtccc1.missing | awk &#39;$5 &lt; 0.00001&#39; | awk &#39;{ print $2 }&#39; &gt; wtccc1/wtccc1-fail-diffmiss-qc.txt</code></pre>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="wtccc1.html#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb126-2"><a href="wtccc1.html#cb126-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-3"><a href="wtccc1.html#cb126-3" aria-hidden="true" tabindex="-1"></a>COURSE_loc <span class="ot">=</span> <span class="st">&quot;~/Desktop/practical&quot;</span> <span class="co"># getwd()</span></span>
<span id="cb126-4"><a href="wtccc1.html#cb126-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-5"><a href="wtccc1.html#cb126-5" aria-hidden="true" tabindex="-1"></a>wtccc1_HWE <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/wtccc1/wtccc1.hwe&quot;</span>))</span>
<span id="cb126-6"><a href="wtccc1.html#cb126-6" aria-hidden="true" tabindex="-1"></a>wtccc1_FRQ <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/wtccc1/wtccc1.frq&quot;</span>))</span>
<span id="cb126-7"><a href="wtccc1.html#cb126-7" aria-hidden="true" tabindex="-1"></a>wtccc1_IMISS <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/wtccc1/wtccc1.imiss&quot;</span>))</span>
<span id="cb126-8"><a href="wtccc1.html#cb126-8" aria-hidden="true" tabindex="-1"></a>wtccc1_LMISS <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/wtccc1/wtccc1.lmiss&quot;</span>))</span>
<span id="cb126-9"><a href="wtccc1.html#cb126-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-10"><a href="wtccc1.html#cb126-10" aria-hidden="true" tabindex="-1"></a>wtccc1_HWE<span class="sc">$</span>logP <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(wtccc1_HWE<span class="sc">$</span>P)</span></code></pre></div>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="wtccc1.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb127-2"><a href="wtccc1.html#cb127-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-3"><a href="wtccc1.html#cb127-3" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(wtccc1_HWE, <span class="at">x =</span> <span class="st">&quot;logP&quot;</span>,</span>
<span id="cb127-4"><a href="wtccc1.html#cb127-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>,</span>
<span id="cb127-5"><a href="wtccc1.html#cb127-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb127-6"><a href="wtccc1.html#cb127-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb127-7"><a href="wtccc1.html#cb127-7" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># color = &quot;#1290D9&quot;, fill = &quot;#1290D9&quot;,</span></span>
<span id="cb127-8"><a href="wtccc1.html#cb127-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;TEST&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;TEST&quot;</span>,</span>
<span id="cb127-9"><a href="wtccc1.html#cb127-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">palette =</span> <span class="st">&quot;lancet&quot;</span>,</span>
<span id="cb127-10"><a href="wtccc1.html#cb127-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">facet.by =</span> <span class="st">&quot;TEST&quot;</span>,</span>
<span id="cb127-11"><a href="wtccc1.html#cb127-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb127-12"><a href="wtccc1.html#cb127-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;HWE -log10(P)&quot;</span>) <span class="sc">+</span></span>
<span id="cb127-13"><a href="wtccc1.html#cb127-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">5</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb127-14"><a href="wtccc1.html#cb127-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1%20HWE-1.png" width="672" /></p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="wtccc1.html#cb128-1" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(wtccc1_FRQ, <span class="at">x =</span> <span class="st">&quot;MAF&quot;</span>,</span>
<span id="cb128-2"><a href="wtccc1.html#cb128-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb128-3"><a href="wtccc1.html#cb128-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb128-4"><a href="wtccc1.html#cb128-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb128-5"><a href="wtccc1.html#cb128-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;minor allele frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb128-6"><a href="wtccc1.html#cb128-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.05</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb128-7"><a href="wtccc1.html#cb128-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1%20FREQ-1.png" width="672" /></p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="wtccc1.html#cb129-1" aria-hidden="true" tabindex="-1"></a>wtccc1_IMISS<span class="sc">$</span>callrate <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> wtccc1_IMISS<span class="sc">$</span>F_MISS</span>
<span id="cb129-2"><a href="wtccc1.html#cb129-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-3"><a href="wtccc1.html#cb129-3" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(wtccc1_IMISS, <span class="at">x =</span> <span class="st">&quot;callrate&quot;</span>,</span>
<span id="cb129-4"><a href="wtccc1.html#cb129-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb129-5"><a href="wtccc1.html#cb129-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>, <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb129-6"><a href="wtccc1.html#cb129-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb129-7"><a href="wtccc1.html#cb129-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;per sample call rate&quot;</span>) <span class="sc">+</span></span>
<span id="cb129-8"><a href="wtccc1.html#cb129-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb129-9"><a href="wtccc1.html#cb129-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Warning: geom_vline(): Ignoring `mapping` because `xintercept` was provided.</code></pre>
<pre><code>## Warning: geom_vline(): Ignoring `data` because `xintercept` was provided.</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1%20callrate%20samples-1.png" width="672" /></p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="wtccc1.html#cb132-1" aria-hidden="true" tabindex="-1"></a>wtccc1_LMISS<span class="sc">$</span>callrate <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> wtccc1_LMISS<span class="sc">$</span>F_MISS</span>
<span id="cb132-2"><a href="wtccc1.html#cb132-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-3"><a href="wtccc1.html#cb132-3" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">gghistogram</span>(wtccc1_LMISS, <span class="at">x =</span> <span class="st">&quot;callrate&quot;</span>,</span>
<span id="cb132-4"><a href="wtccc1.html#cb132-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">add =</span> <span class="st">&quot;mean&quot;</span>, <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>),</span>
<span id="cb132-5"><a href="wtccc1.html#cb132-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rug =</span> <span class="cn">TRUE</span>, <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb132-6"><a href="wtccc1.html#cb132-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;#1290D9&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#1290D9&quot;</span>,</span>
<span id="cb132-7"><a href="wtccc1.html#cb132-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xlab =</span> <span class="st">&quot;per SNP call rate&quot;</span>) <span class="sc">+</span></span>
<span id="cb132-8"><a href="wtccc1.html#cb132-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb132-9"><a href="wtccc1.html#cb132-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#E55738&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Warning: geom_vline(): Ignoring `mapping` because `xintercept` was provided.</code></pre>
<pre><code>## Warning: geom_vline(): Ignoring `data` because `xintercept` was provided.</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1%20callrate%20SNPs-1.png" width="672" /></p>
<div id="quality-control-1" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Quality control<a href="wtccc1.html#quality-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have handle on the data, we can filter it.</p>
<blockquote>
<p>Do you have any thoughts on that? Do you agree with the filters I set below? How would you do it differently and why?</p>
</blockquote>
<pre><code>plink --bfile wtccc1/wtccc1 --exclude wtccc1/wtccc1-fail-diffmiss-qc.txt --maf 0.01 --geno 0.05 --hwe 0.00001 --make-bed --out wtccc1/wtccc1_clean</code></pre>
</div>
<div id="ancestral-background-1" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Ancestral background<a href="wtccc1.html#ancestral-background-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If these individuals are all from the United Kingdom, we are certain there will be admixture from other populations given UK’s history. Let’s project the WTCCC1 data on 1000G phase 1 populations.</p>
<p>We will face the same issue as before with our dummy dataset with respect to <code>EIGENSOFT</code>. So I created the data for you to skip to the <a href="wtccc1.html#plotting-pca">Plotting PCA</a> section immediately. Regardless, in the <a href="wtccc1.html#preparing-pca">Preparing PCA</a> and <a href="wtccc1.html#running-pca-1">Running PCA</a> sections I show you how to get there.</p>
<div id="preparing-pca" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Preparing PCA<a href="wtccc1.html#preparing-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Filtering WTCCC1</strong></p>
<p>For PCA we need to perform extreme clean.</p>
<pre><code>plink --bfile wtccc1/wtccc1_clean --maf 0.1 --geno 0.1 --indep-pairwise 100 50 0.2 --exclude support/exclude_problematic_range.txt --make-bed --out wtccc1/wtccc1_temp

plink --bfile wtccc1/wtccc1_temp --exclude wtccc1/wtccc1_temp.prune.out --make-bed --out wtccc1/wtccc1_extrclean

rm -v wtccc1/wtccc1_temp*

cat wtccc1/wtccc1_extrclean.bim | awk &#39;{ print $2 }&#39; &gt; wtccc1/wtccc1_extrclean.variants.txt

cat wtccc1/wtccc1.bim | grep &quot;rs&quot; &gt; wtccc1/all.variants.txt</code></pre>
<p><strong>Download 1000G phase 1</strong></p>
<p>Next, we are going to have to download the 1000G phase 1 data. If you haven’t done that already, here’s the how.</p>
<p>We start by creating the necessary folder to save the data to.</p>
<pre><code>mkdir -v ~/Desktop/practical/ref_1kg_phase1_all</code></pre>
<p>Next, we’ll download the reference.</p>
<pre><code>wget &quot;https://www.dropbox.com/sh/kumfwm7drt2flhp/AAB5n0OcUvJixI9pNiymx6-La?dl=0&quot; -P ~/Desktop/practical/ref_1kg_phase1_all
</code></pre>
<p><strong>Merging WTCCC1 with 1000G phase 1</strong></p>
<p>Now we are ready to extract the WTCCC1 variants from the 1000G phase 1 reference</p>
<pre><code>plink --bfile ref_1kg_phase1_all/1kg_phase1_all --extract wtccc1/all.variants.txt --make-bed --out ref_1kg_phase1_all/1kg_phase1_wtccc1</code></pre>
<p>Extracting the A/T and C/G SNPs as well.</p>
<pre><code>cat ref_1kg_phase1_all/1kg_phase1_wtccc1.bim | \
awk &#39;($5 == &quot;A&quot; &amp;&amp; $6 == &quot;T&quot;) || ($5 == &quot;T&quot; &amp;&amp; $6 == &quot;A&quot;) || ($5 == &quot;C&quot; &amp;&amp; $6 == &quot;G&quot;) || ($5 == &quot;G&quot; &amp;&amp; $6 == &quot;C&quot;)&#39; | awk &#39;{ print $2, $1, $4, $3, $5, $6 }&#39; \
&gt; ref_1kg_phase1_all/all.1kg_wtccc1.atcg.variants.txt</code></pre>
<pre><code>plink --bfile ref_1kg_phase1_all/1kg_phase1_wtccc1 --exclude ref_1kg_phase1_all/all.1kg_wtccc1.atcg.variants.txt --make-bed --out ref_1kg_phase1_all/1kg_phase1_wtccc1_no_atcg

plink --bfile ref_1kg_phase1_all/1kg_phase1_wtccc1_no_atcg --extract wtccc1/wtccc1_extrclean.variants.txt --make-bed --out ref_1kg_phase1_all/1kg_phase1_raw_no_atcg_wtccc1</code></pre>
<p>Finally we will merge the datasets.</p>
<pre><code>plink --bfile wtccc1/wtccc1_extrclean --bmerge ref_1kg_phase1_all/1kg_phase1_raw_no_atcg_wtccc1 --maf 0.1 --geno 0.1 --exclude support/exclude_problematic_range.txt --make-bed --out wtccc1/wtccc1_extrclean_1kg</code></pre>
</div>
<div id="running-pca-1" class="section level3 hasAnchor" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Running PCA<a href="wtccc1.html#running-pca-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>cp -v wtccc1/wtccc1_extrclean_1kg.bim wtccc1/wtccc1_extrclean_1kg.pedsnp
cp -v wtccc1/wtccc1_extrclean_1kg.fam wtccc1/wtccc1_extrclean_1kg.pedind</code></pre>
<p>Now that the cleaning is done, we can execute the actual PCA.</p>
<pre><code>perl ~/git/EIG/bin/smartpca.perl \
-i wtccc1/wtccc1_extrclean_1kg.bed \
-a wtccc1/wtccc1_extrclean_1kg.pedsnp \
-b wtccc1/wtccc1_extrclean_1kg.pedind \
-k 10 \
-o wtccc1/wtccc1_extrclean_1kg.pca \
-p wtccc1/wtccc1_extrclean_1kg.plot \
-e wtccc1/wtccc1_extrclean_1kg.eval \
-l wtccc1/wtccc1_extrclean_1kg.log \
-m 5 \
-t 10 \
-s 6.0 \
-w ref_1kg_phase1_all/1kg-pca-populations.txt</code></pre>
</div>
<div id="plotting-pca" class="section level3 hasAnchor" number="9.2.3">
<h3><span class="header-section-number">9.2.3</span> Plotting PCA<a href="wtccc1.html#plotting-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If all is peachy, you were able to run the PCA for the WTCCC1 data against 1000G phase 1. Using <code>smartpca</code> (you know, <code>EIGENSOFT</code>) we have calculated principal components (PCs) and we can now start plotting them. Let’s create a scatter diagram of the first two principal components, including all individuals in the file <code>wtccc1_extrclean_1kg.pca.evec</code> (the first and second principal components are columns 2 and 3, respectively). Use the data in column 4 to color the points according to sample origin.</p>
<blockquote>
<p>Please note! You may have been able to make <code>EIGENSOFT</code> to work. So you may have to change “/ref_pca_wtccc1/wtccc1_extrclean_1kg.pca.evec” to “/wtccc1/wtccc1_extrclean_1kg.pca.evec” in the command below.</p>
</blockquote>
<p>And we should visualize the PCA results: are these individuals really all from European (UK) ancestry?</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="wtccc1.html#cb145-1" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc,<span class="st">&quot;/ref_pca_wtccc1/wtccc1_extrclean_1kg.pca.evec&quot;</span>), <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="wtccc1.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Population    Description Super population    Code    Counts</span></span>
<span id="cb146-2"><a href="wtccc1.html#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ASW   African Ancestry in Southwest US                              AFR   4     #49A01D</span></span>
<span id="cb146-3"><a href="wtccc1.html#cb146-3" aria-hidden="true" tabindex="-1"></a><span class="co"># CEU   Utah residents with Northern and Western European ancestry  EUR 7     #E55738</span></span>
<span id="cb146-4"><a href="wtccc1.html#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="co"># CHB   Han Chinese in Bejing, China                                  EAS   8     #9A3480</span></span>
<span id="cb146-5"><a href="wtccc1.html#cb146-5" aria-hidden="true" tabindex="-1"></a><span class="co"># CHS   Southern Han Chinese, China                                 EAS 9     #705296</span></span>
<span id="cb146-6"><a href="wtccc1.html#cb146-6" aria-hidden="true" tabindex="-1"></a><span class="co"># CLM   Colombian in Medellin, Colombia                             MR  10    #8D5B9A</span></span>
<span id="cb146-7"><a href="wtccc1.html#cb146-7" aria-hidden="true" tabindex="-1"></a><span class="co"># FIN   Finnish in Finland                                          EUR 12  #2F8BC9</span></span>
<span id="cb146-8"><a href="wtccc1.html#cb146-8" aria-hidden="true" tabindex="-1"></a><span class="co"># GBR   British in England and Scotland                             EUR 13  #1290D9</span></span>
<span id="cb146-9"><a href="wtccc1.html#cb146-9" aria-hidden="true" tabindex="-1"></a><span class="co"># IBS   Iberian populations in Spain                                  EUR   16  #1396D8</span></span>
<span id="cb146-10"><a href="wtccc1.html#cb146-10" aria-hidden="true" tabindex="-1"></a><span class="co"># JPT   Japanese in Tokyo, Japan                                      EAS   18  #D5267B</span></span>
<span id="cb146-11"><a href="wtccc1.html#cb146-11" aria-hidden="true" tabindex="-1"></a><span class="co"># LWK   Luhya in Webuye, Kenya                                      AFR 20  #78B113</span></span>
<span id="cb146-12"><a href="wtccc1.html#cb146-12" aria-hidden="true" tabindex="-1"></a><span class="co"># MXL   Mexican Ancestry in Los Angeles, California                 AMR 22  #F59D10</span></span>
<span id="cb146-13"><a href="wtccc1.html#cb146-13" aria-hidden="true" tabindex="-1"></a><span class="co"># PUR   Puerto Rican in Puerto Rico                                 AMR 25  #FBB820</span></span>
<span id="cb146-14"><a href="wtccc1.html#cb146-14" aria-hidden="true" tabindex="-1"></a><span class="co"># TSI   Toscani in Italy                                              EUR   27  #4C81BF</span></span>
<span id="cb146-15"><a href="wtccc1.html#cb146-15" aria-hidden="true" tabindex="-1"></a><span class="co"># YRI   Yoruba in Ibadan, Nigeria                                     AFR   28  #C5D220</span></span>
<span id="cb146-16"><a href="wtccc1.html#cb146-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-17"><a href="wtccc1.html#cb146-17" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;2&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CAD&quot;</span></span>
<span id="cb146-18"><a href="wtccc1.html#cb146-18" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;1&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;UKBS&quot;</span></span>
<span id="cb146-19"><a href="wtccc1.html#cb146-19" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;4&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;ASW&quot;</span></span>
<span id="cb146-20"><a href="wtccc1.html#cb146-20" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;7&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CEU&quot;</span></span>
<span id="cb146-21"><a href="wtccc1.html#cb146-21" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;8&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CHB&quot;</span></span>
<span id="cb146-22"><a href="wtccc1.html#cb146-22" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;9&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CHS&quot;</span></span>
<span id="cb146-23"><a href="wtccc1.html#cb146-23" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;10&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CLM&quot;</span></span>
<span id="cb146-24"><a href="wtccc1.html#cb146-24" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;12&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;FIN&quot;</span></span>
<span id="cb146-25"><a href="wtccc1.html#cb146-25" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;13&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;GBR&quot;</span></span>
<span id="cb146-26"><a href="wtccc1.html#cb146-26" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;16&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;IBS&quot;</span></span>
<span id="cb146-27"><a href="wtccc1.html#cb146-27" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;18&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;JPT&quot;</span></span>
<span id="cb146-28"><a href="wtccc1.html#cb146-28" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;20&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;LWK&quot;</span></span>
<span id="cb146-29"><a href="wtccc1.html#cb146-29" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;22&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;MXL&quot;</span></span>
<span id="cb146-30"><a href="wtccc1.html#cb146-30" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;25&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;PUR&quot;</span></span>
<span id="cb146-31"><a href="wtccc1.html#cb146-31" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;27&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;TSI&quot;</span></span>
<span id="cb146-32"><a href="wtccc1.html#cb146-32" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kG<span class="sc">$</span>V12[PCA_WTCCC1_1kG<span class="sc">$</span>V12 <span class="sc">==</span> <span class="st">&quot;28&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;YRI&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="wtccc1.html#cb147-1" aria-hidden="true" tabindex="-1"></a>PCA_WTCCC1_1kGplot <span class="ot">&lt;-</span> ggpubr<span class="sc">::</span><span class="fu">ggscatter</span>(PCA_WTCCC1_1kG, <span class="at">x =</span> <span class="st">&quot;V2&quot;</span>, <span class="at">y =</span> <span class="st">&quot;V3&quot;</span>,</span>
<span id="cb147-2"><a href="wtccc1.html#cb147-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">color =</span> <span class="st">&quot;V12&quot;</span>,</span>
<span id="cb147-3"><a href="wtccc1.html#cb147-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#49A01D&quot;</span>, <span class="st">&quot;#595A5C&quot;</span>, <span class="st">&quot;#E55738&quot;</span>, <span class="st">&quot;#9A3480&quot;</span>, <span class="st">&quot;#705296&quot;</span>, </span>
<span id="cb147-4"><a href="wtccc1.html#cb147-4" aria-hidden="true" tabindex="-1"></a>                                                    <span class="st">&quot;#8D5B9A&quot;</span>, <span class="st">&quot;#2F8BC9&quot;</span>, <span class="st">&quot;#1290D9&quot;</span>, <span class="st">&quot;#1396D8&quot;</span>, <span class="st">&quot;#D5267B&quot;</span>, </span>
<span id="cb147-5"><a href="wtccc1.html#cb147-5" aria-hidden="true" tabindex="-1"></a>                                                    <span class="st">&quot;#78B113&quot;</span>, <span class="st">&quot;#F59D10&quot;</span>, <span class="st">&quot;#FBB820&quot;</span>, <span class="st">&quot;#4C81BF&quot;</span>, <span class="st">&quot;#595A5C&quot;</span>, <span class="st">&quot;#C5D220&quot;</span>),</span>
<span id="cb147-6"><a href="wtccc1.html#cb147-6" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">xlab =</span> <span class="st">&quot;principal component 1&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;principal component 2&quot;</span>) <span class="sc">+</span></span>
<span id="cb147-7"><a href="wtccc1.html#cb147-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.023</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb147-8"><a href="wtccc1.html#cb147-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;#595A5C&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb147-9"><a href="wtccc1.html#cb147-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-10"><a href="wtccc1.html#cb147-10" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">ggpar</span>(PCA_WTCCC1_1kGplot,</span>
<span id="cb147-11"><a href="wtccc1.html#cb147-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="st">&quot;Principal Component Analysis&quot;</span>,</span>
<span id="cb147-12"><a href="wtccc1.html#cb147-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">subtitle =</span> <span class="st">&quot;Reference population: 1000 G, phase 1&quot;</span>,</span>
<span id="cb147-13"><a href="wtccc1.html#cb147-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.title =</span> <span class="st">&quot;Populations&quot;</span>, <span class="at">legend =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:pca-1000g-wtccc"></span>
<img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/pca-1000g-wtccc-1.png" alt="PCA - WTCCC1 vs. 1000G" width="80%" />
<p class="caption">
Figure 9.1: PCA - WTCCC1 vs. 1000G
</p>
</div>
</div>
<div id="removing-samples-1" class="section level3 hasAnchor" number="9.2.4">
<h3><span class="header-section-number">9.2.4</span> Removing samples<a href="wtccc1.html#removing-samples-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In a similar fashion as in the example <em>gwas</em> and <em>rawdata</em> datasets, you should consider to <strong>remove the samples below the threshold</strong> based on this PCA (Figure <a href="wtccc1.html#fig:pca-1000g-wtccc">9.1</a>).</p>
<blockquote>
<p>Go ahead, try that.</p>
</blockquote>
<p>You’re code would be something like below:</p>
<pre><code>cat wtccc1/wtccc1_extrclean_1kg.pca.evec | tail -n +2 | \
awk &#39;$3 &lt; 0.023&#39; | awk &#39;{ print $1 }&#39; | awk -F&quot;:&quot; &#39;{ print $1, $2 }&#39; &gt; wtccc1/fail-ancestry-QC.txt</code></pre>
<p>Next we filter these samples and get a final fully QC’d dataset.</p>
<pre><code>plink --bfile wtccc1/wtccc1_clean --exclude wtccc1/fail-ancestry-QC.txt --make-bed --out wtccc1/wtccc1_qc</code></pre>
</div>
</div>
<div id="association-testing" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Association testing<a href="wtccc1.html#association-testing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have explored the data, we are ready for some simple association testing. However, it would be great to have some PCs to correct for. We can use PLINK for that too.</p>
<pre><code>plink --bfile wtccc1/wtccc1_extrclean --exclude wtccc1/fail-ancestry-QC.txt --pca --out wtccc1/wtccc1_extrclean</code></pre>
<p>Let’s add those PCs to the covariates-file.</p>
<pre><code>echo &quot;IID PC1 PC2 PC3 PC4 PC5 PC6 PC7 PC8 PC9 PC10 PC11 PC12 PC13 PC14 PC15 PC16 PC17 PC18 PC19 PC20&quot; &gt; wtccc1/wtccc1_qc.pca

cat wtccc1/wtccc1_extrclean.eigenvec | awk &#39;{ print $2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20,$21,$22}&#39; &gt;&gt; wtccc1/wtccc1_qc.pca

perl scripts/mergeTables.pl --file1 wtccc1/wtccc1_qc.pca --file2 wtccc1/wtccc1.covar --index IID --format NORM &gt; wtccc1/wtccc1_qc.covar_pca</code></pre>
<p>Exciting, now we are ready to perform a GWAS on CAD in WTCCC1.</p>
<pre><code>plink --bfile wtccc1/wtccc1_qc --logistic sex --covar wtccc1/wtccc1_qc.covar_pca --out wtccc1/wtccc1_qc_log_covar_pca</code></pre>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="wtccc1.html#cb153-1" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(COURSE_loc, <span class="st">&quot;/wtccc1/wtccc1_qc_log_covar_pca.assoc.logistic&quot;</span>))</span>
<span id="cb153-2"><a href="wtccc1.html#cb153-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-3"><a href="wtccc1.html#cb153-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(wtccc1_assoc)</span></code></pre></div>
<pre><code>## [1] 8726361       9</code></pre>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="wtccc1.html#cb155-1" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>]</span></code></pre></div>
<pre><code>##    CHR       SNP     BP A1            TEST NMISS        OR     STAT          P
## 1:   1 rs3094315 752566  G             ADD  3256 1.036e+00  0.37270  7.094e-01
## 2:   1 rs3094315 752566  G age_recruitment  3256 4.351e+00 25.96000 1.471e-148
## 3:   1 rs3094315 752566  G             PC1  3256 1.909e+01  1.11300  2.657e-01
## 4:   1 rs3094315 752566  G             PC2  3256 1.009e-01 -0.87060  3.840e-01
## 5:   1 rs3094315 752566  G             PC3  3256 1.842e-01 -0.62560  5.316e-01
## 6:   1 rs3094315 752566  G             PC4  3256 7.391e-01 -0.11110  9.116e-01
## 7:   1 rs3094315 752566  G             PC5  3256 1.817e-01 -0.63400  5.261e-01
## 8:   1 rs3094315 752566  G             PC6  3256 9.170e-01 -0.03229  9.742e-01
## 9:   1 rs3094315 752566  G             PC7  3256 1.839e+04  3.61000  3.057e-04</code></pre>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="wtccc1.html#cb157-1" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(wtccc1_assoc, TEST <span class="sc">==</span> <span class="st">&quot;ADD&quot;</span>)</span>
<span id="cb157-2"><a href="wtccc1.html#cb157-2" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_sub<span class="sc">$</span>TEST <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb157-3"><a href="wtccc1.html#cb157-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-4"><a href="wtccc1.html#cb157-4" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">subset</span>(wtccc1_FRQ, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;A2&quot;</span>, <span class="st">&quot;MAF&quot;</span>, <span class="st">&quot;NCHROBS&quot;</span>))</span>
<span id="cb157-5"><a href="wtccc1.html#cb157-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-6"><a href="wtccc1.html#cb157-6" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_subfrq <span class="ot">&lt;-</span> <span class="fu">merge</span>(wtccc1_assoc_sub, temp, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>)</span>
<span id="cb157-7"><a href="wtccc1.html#cb157-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-8"><a href="wtccc1.html#cb157-8" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">subset</span>(wtccc1_LMISS, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;callrate&quot;</span>))</span>
<span id="cb157-9"><a href="wtccc1.html#cb157-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-10"><a href="wtccc1.html#cb157-10" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_subfrqlmiss <span class="ot">&lt;-</span> <span class="fu">merge</span>(wtccc1_assoc_subfrq, temp, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>)</span>
<span id="cb157-11"><a href="wtccc1.html#cb157-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wtccc1_assoc_subfrqlmiss)</span></code></pre></div>
<pre><code>##           SNP CHR        BP A1 NMISS     OR     STAT      P A2    MAF NCHROBS
## 1: rs10000012   4   1357325  G  3265 1.0480  0.48180 0.6299  C 0.1345    6530
## 2:  rs1000002   3 183635768  T  3264 1.1100  1.55000 0.1212  C 0.4827    6528
## 3:  rs1000003   3  98342907  G  3232 1.0080  0.08926 0.9289  A 0.1454    6464
## 4: rs10000033   4 139599898  C  3266 1.0070  0.10840 0.9137  T 0.4545    6532
## 5: rs10000037   4  38924330  A  3267 1.1100  1.35900 0.1741  G 0.2557    6534
## 6: rs10000062   4   5254744  C  3266 0.8589 -1.59000 0.1119  G 0.1408    6532
##    callrate
## 1: 0.999082
## 2: 0.998776
## 3: 0.988980
## 4: 0.999388
## 5: 0.999694
## 6: 0.999388</code></pre>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="wtccc1.html#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remember:</span></span>
<span id="cb159-2"><a href="wtccc1.html#cb159-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - that z = beta/se</span></span>
<span id="cb159-3"><a href="wtccc1.html#cb159-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - beta = log(OR), because log is the natural log in r</span></span>
<span id="cb159-4"><a href="wtccc1.html#cb159-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-5"><a href="wtccc1.html#cb159-5" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_subfrqlmiss<span class="sc">$</span>BETA <span class="ot">=</span> <span class="fu">log</span>(wtccc1_assoc_subfrqlmiss<span class="sc">$</span>OR)</span>
<span id="cb159-6"><a href="wtccc1.html#cb159-6" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_subfrqlmiss<span class="sc">$</span>SE <span class="ot">=</span> wtccc1_assoc_subfrqlmiss<span class="sc">$</span>BETA<span class="sc">/</span>wtccc1_assoc_subfrqlmiss<span class="sc">$</span>STAT</span>
<span id="cb159-7"><a href="wtccc1.html#cb159-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-8"><a href="wtccc1.html#cb159-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-9"><a href="wtccc1.html#cb159-9" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_subfrqlmiss_tib <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">as_tibble</span>(wtccc1_assoc_subfrqlmiss)</span>
<span id="cb159-10"><a href="wtccc1.html#cb159-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-11"><a href="wtccc1.html#cb159-11" aria-hidden="true" tabindex="-1"></a>col_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb159-12"><a href="wtccc1.html#cb159-12" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;A1&quot;</span>, <span class="st">&quot;A2&quot;</span>, <span class="st">&quot;MAF&quot;</span>, <span class="st">&quot;callrate&quot;</span>, <span class="st">&quot;NMISS&quot;</span>, <span class="st">&quot;NCHROBS&quot;</span>,</span>
<span id="cb159-13"><a href="wtccc1.html#cb159-13" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;BETA&quot;</span>, <span class="st">&quot;SE&quot;</span>, <span class="st">&quot;OR&quot;</span>, <span class="st">&quot;STAT&quot;</span>, <span class="st">&quot;P&quot;</span>)</span>
<span id="cb159-14"><a href="wtccc1.html#cb159-14" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_compl <span class="ot">&lt;-</span> wtccc1_assoc_subfrqlmiss_tib[, col_order]</span>
<span id="cb159-15"><a href="wtccc1.html#cb159-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-16"><a href="wtccc1.html#cb159-16" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(wtccc1_assoc_compl)</span></code></pre></div>
<pre><code>## [1] 379407     14</code></pre>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="wtccc1.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wtccc1_assoc_compl)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 14
##   SNP        CHR     BP A1    A2      MAF callrate NMISS NCHROBS     BETA     SE
##   &lt;chr&gt;    &lt;int&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;int&gt;   &lt;int&gt;    &lt;dbl&gt;  &lt;dbl&gt;
## 1 rs10000…     4 1.36e6 G     C     0.134    0.999  3265    6530  0.0469  0.0973
## 2 rs10000…     3 1.84e8 T     C     0.483    0.999  3264    6528  0.104   0.0673
## 3 rs10000…     3 9.83e7 G     A     0.145    0.989  3232    6464  0.00797 0.0893
## 4 rs10000…     4 1.40e8 C     T     0.454    0.999  3266    6532  0.00698 0.0644
## 5 rs10000…     4 3.89e7 A     G     0.256    1.00   3267    6534  0.104   0.0768
## 6 rs10000…     4 5.25e6 C     G     0.141    0.999  3266    6532 -0.152   0.0957
## # … with 3 more variables: OR &lt;dbl&gt;, STAT &lt;dbl&gt;, P &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="wtccc1.html#cb163-1" aria-hidden="true" tabindex="-1"></a>wtccc1_assoc_complsub <span class="ot">&lt;-</span> <span class="fu">subset</span>(wtccc1_assoc_compl, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;BP&quot;</span>, <span class="st">&quot;P&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="wtccc1.html#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;CMplot&quot;</span>)</span>
<span id="cb164-2"><a href="wtccc1.html#cb164-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-3"><a href="wtccc1.html#cb164-3" aria-hidden="true" tabindex="-1"></a><span class="fu">CMplot</span>(wtccc1_assoc_complsub,</span>
<span id="cb164-4"><a href="wtccc1.html#cb164-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot.type =</span> <span class="st">&quot;b&quot;</span>, <span class="at">LOG10 =</span> <span class="cn">TRUE</span>, <span class="at">ylim =</span> <span class="cn">NULL</span>,</span>
<span id="cb164-5"><a href="wtccc1.html#cb164-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">threshold =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>,<span class="fl">1e-4</span>), <span class="at">threshold.lty =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">threshold.lwd =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">threshold.col =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb164-6"><a href="wtccc1.html#cb164-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">amplify =</span> <span class="cn">TRUE</span>,</span>
<span id="cb164-7"><a href="wtccc1.html#cb164-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">bin.size =</span> <span class="fl">1e6</span>, <span class="at">chr.den.col =</span> <span class="fu">c</span>(<span class="st">&quot;darkgreen&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb164-8"><a href="wtccc1.html#cb164-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">signal.col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>), <span class="at">signal.cex =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">signal.pch =</span> <span class="fu">c</span>(<span class="dv">19</span>,<span class="dv">19</span>),</span>
<span id="cb164-9"><a href="wtccc1.html#cb164-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">file =</span> <span class="st">&quot;jpg&quot;</span>, <span class="at">memo =</span> <span class="st">&quot;&quot;</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">file.output =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##  SNP-Density Plotting.</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1ResultsManhattan-1.png" width="672" /><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1ResultsManhattan-2.png" width="672" /></p>
<pre><code>##  Circular-Manhattan Plotting P.
##  Rectangular-Manhattan Plotting P.</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1ResultsManhattan-3.png" width="672" /></p>
<pre><code>##  QQ Plotting P.</code></pre>
<p><img src="A-Practical-Primer-in-Human-Complex-Genetics_files/figure-html/WTCCC1ResultsManhattan-4.png" width="672" /></p>
</div>
<div id="replication" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Replication!<a href="wtccc1.html#replication" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You reached an important milestone.</p>
<p><strong>You recreated the work by the whole WTCCC1-team that took them years in just one afternoon.</strong></p>
<p>Wow. Take a pause. And realize what you’ve done.</p>
<p>Back then there wasn’t much on analyses <em>after</em> a GWAS, nowadays there are many post-GWAS analyses methods. We will cover them in the next Chapter <a href="post-gwas.html#post-gwas">10</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="wtccc1-intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="post-gwas.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["twitter", "linkedin", "whatsapp"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics/edit/master/wtccc1.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
