<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12 WTCCC1: a GWAS on coronary artery disease (CAD) | A Practical Primer in Human Complex Genetics for CoCalc</title>
  <meta name="description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="generator" content="bookdown 0.38.1 and GitBook 2.6.7" />

  <meta property="og:title" content="12 WTCCC1: a GWAS on coronary artery disease (CAD) | A Practical Primer in Human Complex Genetics for CoCalc" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  <meta name="github-repo" content="swvanderlaan/A_Practical_Primer_in_Human_Complex_Genetics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12 WTCCC1: a GWAS on coronary artery disease (CAD) | A Practical Primer in Human Complex Genetics for CoCalc" />
  
  <meta name="twitter:description" content="This is a practical primer in human complex genetics for CoCalc with a use-case in cardiovascular disease. The output format for this primer is a GitBook." />
  

<meta name="author" content="dr. Sander W. van der Laan  " />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="wtccc1_intro.html"/>
<link rel="next" href="post_gwas.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/htmltools-fill-0.5.8/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Practical Primer in Human Complex Genetics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this primer</a></li>
<li class="chapter" data-level="2" data-path="somebackgroundreading.html"><a href="somebackgroundreading.html"><i class="fa fa-check"></i><b>2</b> Some background reading</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a></li>
<li class="chapter" data-level="4" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html"><i class="fa fa-check"></i><b>4</b> Getting started - programs</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#plink"><i class="fa fa-check"></i><b>4.2</b> PLINK</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#alternatives-to-plink"><i class="fa fa-check"></i><b>4.2.1</b> Alternatives to <code>PLINK</code></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started-programs-cocalc.html"><a href="getting-started-programs-cocalc.html#other-programs"><i class="fa fa-check"></i><b>4.3</b> Other programs</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html"><i class="fa fa-check"></i><b>5</b> Getting started - CoCalc</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#starting-the-course-on-cocalc"><i class="fa fa-check"></i><b>5.1</b> Starting the course on CoCalc</a></li>
<li class="chapter" data-level="5.2" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#but-wait-what-is-cocalc-anyway"><i class="fa fa-check"></i><b>5.2</b> But wait, what is CoCalc anyway?</a></li>
<li class="chapter" data-level="5.3" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#okay-and-whats-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.3</b> Okay, and what’s a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.4" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-does-a-jupyter-notebook-work"><i class="fa fa-check"></i><b>5.4</b> How does a Jupyter Notebook work?</a></li>
<li class="chapter" data-level="5.5" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#and-what-is-the-force-behind-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.5</b> And what is the Force behind a Jupyter Notebook?</a></li>
<li class="chapter" data-level="5.6" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#how-to-use-a-jupyter-notebook"><i class="fa fa-check"></i><b>5.6</b> How to use a Jupyter Notebook</a></li>
<li class="chapter" data-level="5.7" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#trouble-shooting-juptyer-notebook"><i class="fa fa-check"></i><b>5.7</b> Trouble shooting Juptyer Notebook</a></li>
<li class="chapter" data-level="5.8" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#cocalc-and-this-practical-primer"><i class="fa fa-check"></i><b>5.8</b> CoCalc and this Practical Primer</a></li>
<li class="chapter" data-level="5.9" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-jupyter-notebook-with-a-r-kernel"><i class="fa fa-check"></i><b>5.9</b> Beginning your own Jupyter Notebook with a <code>R</code> kernel</a></li>
<li class="chapter" data-level="5.10" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#beginning-your-own-terminal-for-bash-code"><i class="fa fa-check"></i><b>5.10</b> Beginning your own Terminal for <code>bash</code> code</a></li>
<li class="chapter" data-level="5.11" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#this-book-and-cocalc"><i class="fa fa-check"></i><b>5.11</b> This book and CoCalc</a></li>
<li class="chapter" data-level="5.12" data-path="getting-started-cocalc.html"><a href="getting-started-cocalc.html#are-you-ready"><i class="fa fa-check"></i><b>5.12</b> Are you ready?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html"><i class="fa fa-check"></i><b>6</b> Steps in a Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#setting-the-stage"><i class="fa fa-check"></i><b>6.1.1</b> Setting the stage</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#converting-the-data"><i class="fa fa-check"></i><b>6.2</b> Converting the data</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#quality-control"><i class="fa fa-check"></i><b>6.3</b> Quality control</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-basics-cocalc.html"><a href="gwas-basics-cocalc.html#lets-get-our-hands-dirty"><i class="fa fa-check"></i><b>6.4</b> Let’s get our hands dirty</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html"><i class="fa fa-check"></i><b>7</b> Sample QC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sex"><i class="fa fa-check"></i><b>7.1</b> Sex</a></li>
<li class="chapter" data-level="7.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#sample-call-rates"><i class="fa fa-check"></i><b>7.2</b> Sample call rates</a></li>
<li class="chapter" data-level="7.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#heterozygosity-rate"><i class="fa fa-check"></i><b>7.3</b> Heterozygosity rate</a></li>
<li class="chapter" data-level="7.4" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#relatedness"><i class="fa fa-check"></i><b>7.4</b> Relatedness</a></li>
<li class="chapter" data-level="7.5" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#ancestral-background"><i class="fa fa-check"></i><b>7.5</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#g-phase-1"><i class="fa fa-check"></i><b>7.5.1</b> 1000G phase 1</a></li>
<li class="chapter" data-level="7.5.2" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#principal-component-analysis"><i class="fa fa-check"></i><b>7.5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="7.5.3" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#plotting-the-pca-results"><i class="fa fa-check"></i><b>7.5.3</b> Plotting the PCA results</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#removing-samples"><i class="fa fa-check"></i><b>7.6</b> Removing samples</a></li>
<li class="chapter" data-level="7.7" data-path="gwas-basics-sample-qc-cocalc.html"><a href="gwas-basics-sample-qc-cocalc.html#the-next-step"><i class="fa fa-check"></i><b>7.7</b> The next step</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html"><i class="fa fa-check"></i><b>8</b> Per-SNP QC</a>
<ul>
<li class="chapter" data-level="8.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#snp-call-rates"><i class="fa fa-check"></i><b>8.1</b> SNP call rates</a></li>
<li class="chapter" data-level="8.2" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#differential-snp-call-rates"><i class="fa fa-check"></i><b>8.2</b> Differential SNP call rates</a></li>
<li class="chapter" data-level="8.3" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#allele-frequencies"><i class="fa fa-check"></i><b>8.3</b> Allele frequencies</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-note-on-allele-coding"><i class="fa fa-check"></i><b>8.3.1</b> A note on allele coding</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>8.4</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="8.5" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#final-snp-qc"><i class="fa fa-check"></i><b>8.5</b> Final SNP QC</a></li>
<li class="chapter" data-level="8.6" data-path="gwas-basics-snp-qc-cocalc.html"><a href="gwas-basics-snp-qc-cocalc.html#a-milestone"><i class="fa fa-check"></i><b>8.6</b> A Milestone</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html"><i class="fa fa-check"></i><b>9</b> Genome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#exploring-the-data"><i class="fa fa-check"></i><b>9.1</b> Exploring the data</a></li>
<li class="chapter" data-level="9.2" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#genetic-models"><i class="fa fa-check"></i><b>9.2</b> Genetic models</a></li>
<li class="chapter" data-level="9.3" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#logistic-regression"><i class="fa fa-check"></i><b>9.3</b> Logistic regression</a></li>
<li class="chapter" data-level="9.4" data-path="gwas-testing-cocalc.html"><a href="gwas-testing-cocalc.html#lets-get-visual"><i class="fa fa-check"></i><b>9.4</b> Let’s get visual</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html"><i class="fa fa-check"></i><b>10</b> GWAS visualisation</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#qq-plots"><i class="fa fa-check"></i><b>10.0.1</b> QQ plots</a></li>
<li class="chapter" data-level="10.1" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#manhattan-plots"><i class="fa fa-check"></i><b>10.1</b> Manhattan plots</a></li>
<li class="chapter" data-level="10.2" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#other-plots"><i class="fa fa-check"></i><b>10.2</b> Other plots</a></li>
<li class="chapter" data-level="10.3" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#interactive-plots"><i class="fa fa-check"></i><b>10.3</b> Interactive plots</a></li>
<li class="chapter" data-level="10.4" data-path="gwas-visuals-cocalc.html"><a href="gwas-visuals-cocalc.html#stop-playing-around"><i class="fa fa-check"></i><b>10.4</b> Stop playing around</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html"><i class="fa fa-check"></i><b>11</b> The Welcome Trust Case-Control Consortium</a>
<ul>
<li class="chapter" data-level="11.1" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#genotyping"><i class="fa fa-check"></i><b>11.1</b> Genotyping</a></li>
<li class="chapter" data-level="11.2" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#the-data"><i class="fa fa-check"></i><b>11.2</b> The data</a></li>
<li class="chapter" data-level="11.3" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#assignment"><i class="fa fa-check"></i><b>11.3</b> Assignment</a></li>
<li class="chapter" data-level="11.4" data-path="wtccc1_intro.html"><a href="wtccc1_intro.html#there-you-go"><i class="fa fa-check"></i><b>11.4</b> There you go</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="wtccc1.html"><a href="wtccc1.html"><i class="fa fa-check"></i><b>12</b> WTCCC1: a GWAS on coronary artery disease (CAD)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="wtccc1.html"><a href="wtccc1.html#quality-control-1"><i class="fa fa-check"></i><b>12.1</b> Quality control</a></li>
<li class="chapter" data-level="12.2" data-path="wtccc1.html"><a href="wtccc1.html#ancestral-background-1"><i class="fa fa-check"></i><b>12.2</b> Ancestral background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="wtccc1.html"><a href="wtccc1.html#preparing-pca"><i class="fa fa-check"></i><b>12.2.1</b> Preparing PCA</a></li>
<li class="chapter" data-level="12.2.2" data-path="wtccc1.html"><a href="wtccc1.html#running-pca"><i class="fa fa-check"></i><b>12.2.2</b> Running PCA</a></li>
<li class="chapter" data-level="12.2.3" data-path="wtccc1.html"><a href="wtccc1.html#plotting-pca"><i class="fa fa-check"></i><b>12.2.3</b> Plotting PCA</a></li>
<li class="chapter" data-level="12.2.4" data-path="wtccc1.html"><a href="wtccc1.html#removing-samples-1"><i class="fa fa-check"></i><b>12.2.4</b> Removing samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="wtccc1.html"><a href="wtccc1.html#association-testing"><i class="fa fa-check"></i><b>12.3</b> Association testing</a></li>
<li class="chapter" data-level="12.4" data-path="wtccc1.html"><a href="wtccc1.html#replication"><i class="fa fa-check"></i><b>12.4</b> Replication!</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="post_gwas.html"><a href="post_gwas.html"><i class="fa fa-check"></i><b>13</b> Post-GWAS Analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="post_gwas.html"><a href="post_gwas.html#clumping"><i class="fa fa-check"></i><b>13.1</b> Clumping</a></li>
<li class="chapter" data-level="13.2" data-path="post_gwas.html"><a href="post_gwas.html#conditional-analyses"><i class="fa fa-check"></i><b>13.2</b> Conditional analyses</a></li>
<li class="chapter" data-level="13.3" data-path="post_gwas.html"><a href="post_gwas.html#statistical-finemapping"><i class="fa fa-check"></i><b>13.3</b> Statistical finemapping</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="post_gwas.html"><a href="post_gwas.html#finemap"><i class="fa fa-check"></i><b>13.3.1</b> FINEMAP</a></li>
<li class="chapter" data-level="13.3.2" data-path="post_gwas.html"><a href="post_gwas.html#locuszoom"><i class="fa fa-check"></i><b>13.3.2</b> LocusZoom</a></li>
<li class="chapter" data-level="13.3.3" data-path="post_gwas.html"><a href="post_gwas.html#susie"><i class="fa fa-check"></i><b>13.3.3</b> SuSIE</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="post_gwas.html"><a href="post_gwas.html#fuma-functional-mapping-and-annotation-of-gwas"><i class="fa fa-check"></i><b>13.4</b> FUMA: FUnctional Mapping and Annotation of GWAS</a></li>
<li class="chapter" data-level="13.5" data-path="post_gwas.html"><a href="post_gwas.html#phenome-wide-association-study"><i class="fa fa-check"></i><b>13.5</b> Phenome-wide association study</a></li>
<li class="chapter" data-level="13.6" data-path="post_gwas.html"><a href="post_gwas.html#colocalization"><i class="fa fa-check"></i><b>13.6</b> Colocalization</a></li>
<li class="chapter" data-level="13.7" data-path="post_gwas.html"><a href="post_gwas.html#genetic-correlation"><i class="fa fa-check"></i><b>13.7</b> Genetic correlation</a></li>
<li class="chapter" data-level="13.8" data-path="post_gwas.html"><a href="post_gwas.html#causal-inference-mendelian-randomization"><i class="fa fa-check"></i><b>13.8</b> Causal inference: Mendelian Randomization</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="post_gwas.html"><a href="post_gwas.html#mrbase"><i class="fa fa-check"></i><b>13.8.1</b> MRBase</a></li>
<li class="chapter" data-level="13.8.2" data-path="post_gwas.html"><a href="post_gwas.html#twosamplemr"><i class="fa fa-check"></i><b>13.8.2</b> TwoSampleMR</a></li>
<li class="chapter" data-level="13.8.3" data-path="post_gwas.html"><a href="post_gwas.html#your-choice"><i class="fa fa-check"></i><b>13.8.3</b> Your choice</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="post_gwas.html"><a href="post_gwas.html#polygenic-scores"><i class="fa fa-check"></i><b>13.9</b> Polygenic scores</a></li>
<li class="chapter" data-level="13.10" data-path="post_gwas.html"><a href="post_gwas.html#what-comes-next"><i class="fa fa-check"></i><b>13.10</b> What comes next</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="fuma.html"><a href="fuma.html"><i class="fa fa-check"></i><b>14</b> Functional Mapping and Annotation of GWAS</a>
<ul>
<li class="chapter" data-level="14.1" data-path="fuma.html"><a href="fuma.html#assignment-1"><i class="fa fa-check"></i><b>14.1</b> Assignment</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="fuma.html"><a href="fuma.html#tutorial"><i class="fa fa-check"></i><b>14.1.1</b> Tutorial</a></li>
<li class="chapter" data-level="14.1.2" data-path="fuma.html"><a href="fuma.html#create-the-input"><i class="fa fa-check"></i><b>14.1.2</b> Create the input</a></li>
<li class="chapter" data-level="14.1.3" data-path="fuma.html"><a href="fuma.html#run-fuma---snp-to-gene"><i class="fa fa-check"></i><b>14.1.3</b> Run FUMA - SNP to gene</a></li>
<li class="chapter" data-level="14.1.4" data-path="fuma.html"><a href="fuma.html#run-fuma---gene-to-function"><i class="fa fa-check"></i><b>14.1.4</b> Run FUMA - Gene to function</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="fuma.html"><a href="fuma.html#some-closing-thoughts"><i class="fa fa-check"></i><b>14.2</b> Some closing thoughts</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="phewas.html"><a href="phewas.html"><i class="fa fa-check"></i><b>15</b> Phenome-Wide Association Study</a>
<ul>
<li class="chapter" data-level="15.1" data-path="phewas.html"><a href="phewas.html#whats-next"><i class="fa fa-check"></i><b>15.1</b> What’s next?</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="mr_mrbase.html"><a href="mr_mrbase.html"><i class="fa fa-check"></i><b>16</b> MRBase</a>
<ul>
<li class="chapter" data-level="16.1" data-path="mr_mrbase.html"><a href="mr_mrbase.html#the-end"><i class="fa fa-check"></i><b>16.1</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html"><i class="fa fa-check"></i><b>17</b> Two Sample Mendelian Randomisation</a>
<ul>
<li class="chapter" data-level="17.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#loading-the-package"><i class="fa fa-check"></i><b>17.1</b> Loading the package</a></li>
<li class="chapter" data-level="17.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#available-outcome"><i class="fa fa-check"></i><b>17.2</b> Available outcome</a></li>
<li class="chapter" data-level="17.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#instruments"><i class="fa fa-check"></i><b>17.3</b> Instruments</a></li>
<li class="chapter" data-level="17.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#clumping-1"><i class="fa fa-check"></i><b>17.4</b> Clumping</a></li>
<li class="chapter" data-level="17.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#outcomes"><i class="fa fa-check"></i><b>17.5</b> Outcomes</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#harmonisation"><i class="fa fa-check"></i><b>17.5.1</b> Harmonisation</a></li>
<li class="chapter" data-level="17.5.2" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#mendelian-randomisation"><i class="fa fa-check"></i><b>17.5.2</b> Mendelian randomisation</a></li>
<li class="chapter" data-level="17.5.3" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#heterogeneity-pleiotropy"><i class="fa fa-check"></i><b>17.5.3</b> Heterogeneity &amp; Pleiotropy</a></li>
<li class="chapter" data-level="17.5.4" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#leave-one-out"><i class="fa fa-check"></i><b>17.5.4</b> Leave-one-out</a></li>
<li class="chapter" data-level="17.5.5" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#visualisation"><i class="fa fa-check"></i><b>17.5.5</b> Visualisation</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="mr_twosamplemr.html"><a href="mr_twosamplemr.html#the-end-1"><i class="fa fa-check"></i><b>17.6</b> The end?</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="epilogue.html"><a href="epilogue.html"><i class="fa fa-check"></i><b>18</b> Epilogue</a>
<ul>
<li class="chapter" data-level="18.1" data-path="epilogue.html"><a href="epilogue.html#course-lecturers"><i class="fa fa-check"></i><b>18.1</b> 2024 Course Lecturers</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="regional_plot.html"><a href="regional_plot.html"><i class="fa fa-check"></i><b>19</b> Regional association plots</a>
<ul>
<li class="chapter" data-level="19.1" data-path="regional_plot.html"><a href="regional_plot.html#locuszoom-1"><i class="fa fa-check"></i><b>19.1</b> LocusZoom</a></li>
<li class="chapter" data-level="19.2" data-path="regional_plot.html"><a href="regional_plot.html#racer"><i class="fa fa-check"></i><b>19.2</b> RACER</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="eigensoft.html"><a href="eigensoft.html"><i class="fa fa-check"></i><b>20</b> EIGENSOFT</a>
<ul>
<li class="chapter" data-level="20.1" data-path="eigensoft.html"><a href="eigensoft.html#install-homebrew"><i class="fa fa-check"></i><b>20.1</b> Install homebrew</a></li>
<li class="chapter" data-level="20.2" data-path="eigensoft.html"><a href="eigensoft.html#install-missing-packages"><i class="fa fa-check"></i><b>20.2</b> Install missing packages</a></li>
<li class="chapter" data-level="20.3" data-path="eigensoft.html"><a href="eigensoft.html#installing-eigensoft"><i class="fa fa-check"></i><b>20.3</b> Installing EIGENSOFT</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="answers.html"><a href="answers.html"><i class="fa fa-check"></i><b>21</b> Answers to the questions</a>
<ul>
<li class="chapter" data-level="21.1" data-path="answers.html"><a href="answers.html#chapter-3.1-gwas-basics"><i class="fa fa-check"></i><b>21.1</b> Chapter 3.1: GWAS Basics</a></li>
<li class="chapter" data-level="21.2" data-path="answers.html"><a href="answers.html#chapter-3.4-gwas-basics---association-testing"><i class="fa fa-check"></i><b>21.2</b> Chapter 3.4: GWAS Basics - Association testing</a></li>
<li class="chapter" data-level="21.3" data-path="answers.html"><a href="answers.html#chapter-3.5-gwas-visualization"><i class="fa fa-check"></i><b>21.3</b> Chapter 3.5: GWAS visualization</a></li>
<li class="chapter" data-level="21.4" data-path="answers.html"><a href="answers.html#chapter-4.2-wtccc1"><i class="fa fa-check"></i><b>21.4</b> Chapter 4.2: WTCCC1</a></li>
<li class="chapter" data-level="21.5" data-path="answers.html"><a href="answers.html#chapter-5.4-fuma"><i class="fa fa-check"></i><b>21.5</b> Chapter 5.4: FUMA</a></li>
<li class="chapter" data-level="21.6" data-path="answers.html"><a href="answers.html#chapter-5.7-twosample-mr"><i class="fa fa-check"></i><b>21.6</b> Chapter 5.7: TwoSample MR</a></li>
<li class="chapter" data-level="21.7" data-path="answers.html"><a href="answers.html#chapter-8.1-regional-association-plots"><i class="fa fa-check"></i><b>21.7</b> Chapter 8.1: Regional association plots</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>22</b> Licenses and disclaimers</a>
<ul>
<li class="chapter" data-level="22.1" data-path="license.html"><a href="license.html#images-and-data-used"><i class="fa fa-check"></i><b>22.1</b> Images and data used</a></li>
<li class="chapter" data-level="22.2" data-path="license.html"><a href="license.html#disclaimer"><i class="fa fa-check"></i><b>22.2</b> Disclaimer</a></li>
<li class="chapter" data-level="22.3" data-path="license.html"><a href="license.html#copyright"><i class="fa fa-check"></i><b>22.3</b> Copyright</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="colophon.html"><a href="colophon.html"><i class="fa fa-check"></i><b>23</b> Colophon</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://vanderlaanand.science" target="blank">&copy; 1979-2024. Sander W. van der Laan.</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Practical Primer in Human Complex Genetics for CoCalc</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="wtccc1" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">12</span> WTCCC1: a GWAS on coronary artery disease (CAD)<a href="wtccc1.html#wtccc1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- ![](./img/_headers/interactive_plot.png){width=100%} -->
<p>As usual, we start by exploring the data in hand.</p>
<p>First, let’s make a clean slate and create a working directory for the WTCCC1 data.</p>
<pre><code>clear
mkdir -v ~/wtccc1</code></pre>
<p>Next, we’ll run a few <code>plink</code> commands</p>
<pre><code>plink --bfile ~/data/shared/wtccc1/CADn1871_500Kb37fwd --bmerge ~/data/shared/wtccc1/UKBSn1397_500Kb37fwd --make-bed --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --freq --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --hardy --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --missing --out wtccc1/wtccc1 &amp;&amp; \
plink --bfile wtccc1/wtccc1 --test-missing --out wtccc1/wtccc1

cat wtccc1/wtccc1.missing | awk &#39;$5 &lt; 0.00001&#39; | awk &#39;{ print $2 }&#39; &gt; wtccc1/wtccc1-fail-diffmiss-qc.txt</code></pre>
<p>Let’s investigate the HWE p-value in the whole cohort, and per stratum (cases and controls) with the code below.</p>
<p>This will result in Figure <a href="wtccc1.html#fig:show-wtccc1-hwe">12.1</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-hwe"></span>
<img src="img/gwas_wtccc/WTCCC1-HWE.png" alt="Stratified HWE p-values." width="85%" />
<p class="caption">
Figure 12.1: Stratified HWE p-values.
</p>
</div>
<p>We should also inspect the allele frequencies. Note that <em>by default</em> PLINK (whether v0.7, v1.9, or v2.0) stores the alleles as minor (A1) and major (A2), and therefore <code>--maf</code> <em>always</em> calculates the frequency of the minor allele (A1).</p>
<p>This will result in Figure <a href="wtccc1.html#fig:show-wtccc1-freq">12.2</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-freq"></span>
<img src="img/gwas_wtccc/WTCCC1-FREQ.png" alt="Minor allele frequencies." width="85%" />
<p class="caption">
Figure 12.2: Minor allele frequencies.
</p>
</div>
<p>There could be sample with very poor overall call rate, where for many SNPs there is no data. We will want to identify these samples and exclude them.</p>
<p>This will result in Figure <a href="wtccc1.html#fig:show-wtccc1-callratesample">12.3</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-callratesample"></span>
<img src="img/gwas_wtccc/WTCCC1-callrate-samples.png" alt="Per sample call rate." width="85%" />
<p class="caption">
Figure 12.3: Per sample call rate.
</p>
</div>
<blockquote>
<p>Question: What do you notice in the ‘per sample call rate’ graph? Can you think of a reason why this is? And how would you deal with this?</p>
</blockquote>
<p>Lastly, we must inspect the per SNP call rate; we need to know if there are SNPs that have no data for many samples. We will want to identify such SNPs and exclude these.</p>
<p>This will result in Figure <a href="wtccc1.html#fig:show-wtccc1-callratesnp">12.4</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-callratesnp"></span>
<img src="img/gwas_wtccc/WTCCC1-callrate-SNPs.png" alt="Per SNP call rate." width="85%" />
<p class="caption">
Figure 12.4: Per SNP call rate.
</p>
</div>
<div id="quality-control-1" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> Quality control<a href="wtccc1.html#quality-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have handle on the data, we can filter it.</p>
<blockquote>
<p>Question: Do you have any thoughts on that? Do you agree with the filters I set below? How would you do it differently and why?</p>
</blockquote>
<pre><code>plink --bfile wtccc1/wtccc1 --exclude wtccc1/wtccc1-fail-diffmiss-qc.txt --maf 0.01 --geno 0.05 --hwe 0.00001 --make-bed --out wtccc1/wtccc1_clean</code></pre>
</div>
<div id="ancestral-background-1" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Ancestral background<a href="wtccc1.html#ancestral-background-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If these individuals are all from the United Kingdom, we are certain there will be admixture from other populations given UK’s history. Let’s project the WTCCC1 data on 1000G phase 1 populations.</p>
<p>We will face the same issue as before with our dummy dataset with respect to <code>EIGENSOFT</code>. So I created the data for you to skip to the <a href="wtccc1.html#plotting-pca">Plotting PCA</a> section immediately. Regardless, in the <a href="wtccc1.html#preparing-pca">Preparing PCA</a> and <a href="wtccc1.html#running-pca">Running PCA</a> sections I show you how to get there.</p>
<div id="preparing-pca" class="section level3 hasAnchor" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Preparing PCA<a href="wtccc1.html#preparing-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Filtering WTCCC1</strong></p>
<p>For PCA we need to perform extreme clean.</p>
<pre><code>plink --bfile wtccc1/wtccc1_clean --maf 0.1 --geno 0.1 --indep-pairwise 100 50 0.2 --exclude ~/data/shared/support/exclude_problematic_range.txt --make-bed --out wtccc1/wtccc1_temp

plink --bfile wtccc1/wtccc1_temp --exclude wtccc1/wtccc1_temp.prune.out --make-bed --out wtccc1/wtccc1_extrclean

rm -fv wtccc1/wtccc1_temp*

cat wtccc1/wtccc1_extrclean.bim | awk &#39;{ print $2 }&#39; &gt; wtccc1/wtccc1_extrclean.variants.txt

cat wtccc1/wtccc1.bim | grep &quot;rs&quot; &gt; wtccc1/all.variants.txt</code></pre>
<p>Notice that you are using real world data: there are thousands of variants ‘pruned’ due to the <code>--indep-pairwise 100 50 0.2</code>-flag.</p>
<p><strong>Merging WTCCC1 with 1000G phase 1</strong></p>
<p>Now we are ready to extract the WTCCC1 variants from the 1000G phase 1 reference</p>
<pre><code>plink --bfile ~/data/shared/ref_1kg_phase1_all/1kg_phase1_all --extract wtccc1/all.variants.txt --make-bed --out wtccc1/1kg_phase1_wtccc1</code></pre>
<p>Extracting the A/T and C/G SNPs as well.</p>
<pre><code>cat wtccc1/1kg_phase1_wtccc1.bim | \
awk &#39;($5 == &quot;A&quot; &amp;&amp; $6 == &quot;T&quot;) || ($5 == &quot;T&quot; &amp;&amp; $6 == &quot;A&quot;) || ($5 == &quot;C&quot; &amp;&amp; $6 == &quot;G&quot;) || ($5 == &quot;G&quot; &amp;&amp; $6 == &quot;C&quot;)&#39; | awk &#39;{ print $2, $1, $4, $3, $5, $6 }&#39; \
&gt; wtccc1/all.1kg_wtccc1.atcg.variants.txt</code></pre>
<pre><code>plink --bfile wtccc1/1kg_phase1_wtccc1 --exclude wtccc1/all.1kg_wtccc1.atcg.variants.txt --make-bed --out wtccc1/1kg_phase1_wtccc1_no_atcg

plink --bfile wtccc1/1kg_phase1_wtccc1_no_atcg --extract wtccc1/wtccc1_extrclean.variants.txt --make-bed --out wtccc1/1kg_phase1_raw_no_atcg_wtccc1</code></pre>
<p>Finally we will merge the datasets.</p>
<pre><code>plink --bfile wtccc1/wtccc1_extrclean --bmerge wtccc1/1kg_phase1_raw_no_atcg_wtccc1 --maf 0.1 --geno 0.1 --exclude ~/data/shared/support/exclude_problematic_range.txt --make-bed --out wtccc1/wtccc1_extrclean_1kg</code></pre>
</div>
<div id="running-pca" class="section level3 hasAnchor" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> Running PCA<a href="wtccc1.html#running-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Great, we’ve prepared our dummy project data and merged this with 1000G phase 1. Let’s execute the PCA using <code>--pca</code> in <code>PLINK</code>.</p>
<pre><code>plink --bfile wtccc1/wtccc1_extrclean_1kg --pca --out wtccc1/wtccc1_extrclean_1kg</code></pre>
</div>
<div id="plotting-pca" class="section level3 hasAnchor" number="12.2.3">
<h3><span class="header-section-number">12.2.3</span> Plotting PCA<a href="wtccc1.html#plotting-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If all is peachy, you were able to run the PCA for the WTCCC1 data against 1000G phase 1. Using <code>--pca</code> in <code>plink</code> we have calculated principal components (PCs) and we can now start plotting them. Let’s create a scatter diagram of the first two principal components just like we did with the dummy data.</p>
<p>And we should visualize the PCA results: are these individuals really all from European (UK) ancestry?</p>
<p>We expect most individuals from the WTCCC to be 100% British, but a substantial group will have a different ancestral background as shown in the Figure <a href="wtccc1.html#fig:showwtccc1pca">12.5</a> you just made.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:showwtccc1pca"></span>
<img src="img/gwas_wtccc/wtccc1-qc-pca-1000g.png" alt="PCA - WTCCC1 vs. 1000G" width="85%" />
<p class="caption">
Figure 12.5: PCA - WTCCC1 vs. 1000G
</p>
</div>
</div>
<div id="removing-samples-1" class="section level3 hasAnchor" number="12.2.4">
<h3><span class="header-section-number">12.2.4</span> Removing samples<a href="wtccc1.html#removing-samples-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In a similar fashion as in the example <em>gwas</em> and <em>rawdata</em> datasets, you should consider to <strong>remove the samples below the threshold</strong> based on this PCA (Figure <a href="wtccc1.html#fig:showwtccc1pca">12.5</a>).</p>
<blockquote>
<p>Go ahead, try that.</p>
</blockquote>
<p>You’re code would be something like below:</p>
<pre><code>cat wtccc1/wtccc1_extrclean_1kg.eigenvec | \
awk &#39;$3 &lt; 0.0023&#39; | awk &#39;{ print $1, $2 }&#39; &gt; wtccc1/fail-ancestry-QC.txt</code></pre>
<p>Next we filter these samples and get a final fully QC’d dataset.</p>
<pre><code>plink --bfile wtccc1/wtccc1_clean --exclude wtccc1/fail-ancestry-QC.txt --make-bed --out wtccc1/wtccc1_qc</code></pre>
</div>
</div>
<div id="association-testing" class="section level2 hasAnchor" number="12.3">
<h2><span class="header-section-number">12.3</span> Association testing<a href="wtccc1.html#association-testing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have explored the data, we are ready for some simple association testing. However, it would be great to have some PCs to correct for. We can use PLINK for that too.</p>
<pre><code>plink --bfile wtccc1/wtccc1_extrclean --exclude wtccc1/fail-ancestry-QC.txt --pca --out wtccc1/wtccc1_extrclean</code></pre>
<p>Let’s add those PCs to the covariates-file.</p>
<pre><code>echo &quot;IID PC1 PC2 PC3 PC4 PC5 PC6 PC7 PC8 PC9 PC10 PC11 PC12 PC13 PC14 PC15 PC16 PC17 PC18 PC19 PC20&quot; &gt; wtccc1/wtccc1_qc.pca

cat wtccc1/wtccc1_extrclean.eigenvec | awk &#39;{ print $2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20,$21,$22}&#39; &gt;&gt; wtccc1/wtccc1_qc.pca

perl ~/data/shared/scripts/mergeTables.pl --file1 wtccc1/wtccc1_qc.pca --file2 ~/data/shared/wtccc1/wtccc1.covar --index IID --format NORM &gt; wtccc1/wtccc1_qc.covar_pca</code></pre>
<p>Exciting, now we are ready to perform a GWAS on CAD in WTCCC1. This may take some time (when I tested it, it took at least 5 minutes), so it’s a perfect moment to take a break, stretch your legs and get a coffee.</p>
<pre><code>plink --bfile wtccc1/wtccc1_qc --logistic sex --covar wtccc1/wtccc1_qc.covar_pca --out wtccc1/wtccc1_qc_log_covar_pca</code></pre>
<p>After you ran the association analysis, you’re ready to process the data and take a first look at the results. First, we prepare the raw output.</p>
<p>You could visualize these results with the code below.</p>
This would lead to the following graphs.
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-graphs-density"></span>
<img src="img/gwas_wtccc/WTCCC1ResultsManhattan-density.png" alt="SNP density of the association results." width="85%" />
<p class="caption">
Figure 12.6: SNP density of the association results.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-graphs-qq"></span>
<img src="img/gwas_wtccc/WTCCC1ResultsManhattan-qq.png" alt="A QQ plot including a 95% confidence interval (blue area) and genome-wide significant hits (red)." width="85%" />
<p class="caption">
Figure 12.7: A QQ plot including a 95% confidence interval (blue area) and genome-wide significant hits (red).
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-graphs-manhattan"></span>
<img src="img/gwas_wtccc/WTCCC1ResultsManhattan-manhattan.png" alt="A regular manhattan plot. Colored by chromosome, suggestive hits are green, genome-wide hits are red. The bottom graph shows the per-chromosome SNP density." width="85%" />
<p class="caption">
Figure 12.8: A regular manhattan plot. Colored by chromosome, suggestive hits are green, genome-wide hits are red. The bottom graph shows the per-chromosome SNP density.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:show-wtccc1-graphs-circular"></span>
<img src="img/gwas_wtccc/WTCCC1ResultsManhattan-circular.png" alt="A circular manhattan." width="85%" />
<p class="caption">
Figure 12.9: A circular manhattan.
</p>
</div>
</div>
<div id="replication" class="section level2 hasAnchor" number="12.4">
<h2><span class="header-section-number">12.4</span> Replication!<a href="wtccc1.html#replication" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You reached an important milestone.</p>
<p><strong>You recreated the work by the whole WTCCC1-team that took them years in just one afternoon!</strong></p>
<p>Wow. Take a pause. And realize what you’ve done.</p>
<p>Back then there wasn’t much on analyses <em>after</em> a GWAS, nowadays there are many post-GWAS analyses methods. We will cover them in the next Chapter @ref(post_gwas).</p>
<script>
title=document.getElementById('header');
title.innerHTML = '<img src="./img/headers/interactive_plot.png" alt="WTCCC1: coronary artery disease">' + title.innerHTML
</script>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="wtccc1_intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="post_gwas.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": true,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["A-Practical-Primer-in-Human-Complex-Genetics.pdf", "PDF"], ["A-Practical-Primer-in-Human-Complex-Genetics.epub", "EPUB"]],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"info": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
